---
title: "African_Elephant_Microbiome_ALL_CODE"
author: "Joe Gunn"
date: "11/4/2019"
output: html_document
---
# Libraries
```{r}
library(readxl)
library(cowplot)
library(tidyverse)
library(qiime2R)
library(vegan)
library(devtools)
library(DescTools)
library(factoextra)
library(ade4)
library(lme4)
library(nlme)
library(ggrepel)
library(GGally)
library(lattice)
library(epiDisplay)
library(multcomp)
library(taRifx)
library(ape)
library(openxlsx)
library(multcomp)
library(phyloseq)
library(pscl)
library(ggpubr)
library(car)
```

# METADATA
```{r Metadata}
##Read in the metadata, which includes all labels and covariates associated with each individual in the data set. Metadata includes individual sample ID (index), barcode sequence, linker primer sequence, species (africana or cyclotis), diet (crop raider or non-crop raider), habitat (forest or savanna), age, sex, and group (a descriptor for each unique combination of species-diet-habitat)

metadata <- read_tsv("METADATA.tsv") #metadata read in

metadata <- metadata %>% as.data.frame() %>% mutate(BarcodeSequence = factor(BarcodeSequence), LinkerPrimerSequence = factor(LinkerPrimerSequence), Species = factor(Species), Raider = factor(Raider), Habitat = factor(Habitat), Age = factor(Age), Sex = factor(Sex), Elephant = factor(Elephant), Group = factor(Group), Description = factor(Description)) #convert table to data frame and convert all characters, except for sample id, into factors
colnames(metadata)[colnames(metadata) == "Sample-id"] <- "sample_id" 
```
 
# TAXONOMIC CONFIDENCE ASSESSMENT

## Taxonomy Table (OTUs present in the total data set) and Confidence Calculations
```{r All OTUs - no abundance data}

##Read in Taxonomy Table. This table contains all OTUs present in the total data set, but does not provide the relative abundance of each taxon per sample. It only contains the taxonomic level of each OTU present and the confidence at which that taxonomic unit was classified.

tax_all <- read_qza("taxonomy.qza") #read in table

tax_all<- tax_all$data %>% as.tibble() %>% separate(Taxon, sep = "; ", c("Kingdom", "Phylum", "Class", "Order", "Family", "Genus", "Species")) #clean data and label columns by taxonomic level
tax_all_df <- as.data.frame(tax_all)
colnames(tax_all_df) <- c("otu_names", "Kingdom", "Phylum", "Class", "Order", "Family", "Genus", "Species", "conf")

##Calculate mean and standard deviation of confidence

mean_conf <- mean(tax_all$Confidence) #mean confidence across all OTUs (mean = 0.9615)
sd_conf <- sd(tax_all$Confidence) #standard deviation of confidence across all OTUs (sd = 0.07256)

#change all unclassified taxa that aren't already called "NA" to "NA"

tax_all$Species[tax_all$Species == "s__"] <- NA
tax_all$Genus[tax_all$Genus == "g__"] <- NA
tax_all$Family[tax_all$Family == "f__"] <- NA
tax_all$Order[tax_all$Order == "o__"] <- NA
tax_all$Class[tax_all$Class == "c__"] <- NA
tax_all$Phylum[tax_all$Phylum == "p__"] <- NA
tax_all$Kingdom[tax_all$Kingdom == "k__"] <- NA

phy_order <- tax_all[,-c(1:2,4,6:9 )]

tax_all <- as.data.frame(tax_all) #change table to data frame 

#Create separate data frames for each taxon and confidence 

kingdom <- as.data.frame(cbind(tax_all$Kingdom, tax_all$Confidence)) #kingdom
colnames(kingdom) <- c("Kingdom", "Confidence") #add column names

phylum <- as.data.frame(cbind(tax_all$Phylum, tax_all$Confidence)) #phylum
colnames(phylum) <- c("Phylum", "Confidence")
  
class <- as.data.frame(cbind(tax_all$Class, tax_all$Confidence)) #class
colnames(class) <- c("Class", "Confidence")

order <- as.data.frame(cbind(tax_all$Order, tax_all$Confidence)) #order
colnames(order) <- c("Order", "Confidence")

family <- as.data.frame(cbind(tax_all$Family, tax_all$Confidence)) #family
colnames(family) <- c("Family", "Confidence")

genus <- as.data.frame(cbind(tax_all$Genus, tax_all$Confidence)) #genus
colnames(genus) <- c("Genus", "Confidence")

species <- as.data.frame(cbind(tax_all$Species, tax_all$Confidence)) #species
colnames(species) <- c("Species", "Confidence")

##Create data frames for ONLY the individual OTUs that were successfully classified at each level. Kingdom had many individual OTUs successfully classified, while Species had substantially fewer OTUs successfully classified
  
known_species <- as.data.frame(species[complete.cases(species),]) #species
known_genus <-  as.data.frame(genus[complete.cases(genus),]) #genus
known_family <- as.data.frame(family[complete.cases(family),]) #family
known_order <- as.data.frame(order[complete.cases(order),]) #order 
known_class <- as.data.frame(class[complete.cases(class),]) #class 
known_phylum <- as.data.frame(phylum[complete.cases(phylum),]) #phylum
known_kingdom <- as.data.frame(kingdom[complete.cases(kingdom),]) #kingdom

##Raw number of unique OTUs per taxonomic level (classified successfully)
#nrow(known_species) #271
#nrow(known_genus) #2279
#nrow(known_family) #6138
#nrow(known_order) #8706
#nrow(known_class) #8840
#nrow(known_phylum) #8878
#nrow(known_kingdom) #9066

otu_prop_vector <- c(271/9066, 2279/9066, 6138/9066, 8706/9066, 8840/9066, 8878/9066, 9066/9066)
otu_prop_vector <- as.data.frame(otu_prop_vector)

##mean confidence at each taxonomic level

known_species <- known_species %>% mutate(Confidence = as.character(Confidence)) #need to change the confidence variable from a factor to a numeric. NOTE: you MUST first convert from factor to character, then from character to factor. Not sure why, but otherwise it doesn't work right

  known_species <- known_species %>% mutate(Confidence = as.numeric(Confidence))
  known_genus <- known_genus %>% mutate(Confidence = as.character(Confidence))
  known_genus <-  known_genus %>% mutate(Confidence = as.numeric(Confidence))
  known_family <- known_family %>% mutate(Confidence = as.character(Confidence))
  known_family <-  known_family %>% mutate(Confidence = as.numeric(Confidence))
  known_order <- known_order %>% mutate(Confidence = as.character(Confidence))
  known_order <-  known_order %>% mutate(Confidence = as.numeric(Confidence))
  known_class <- known_class %>% mutate(Confidence = as.character(Confidence))
  known_class <-  known_class %>% mutate(Confidence = as.numeric(Confidence))
  known_phylum <- known_phylum %>% mutate(Confidence = as.character(Confidence))
  known_phylum <-  known_phylum %>% mutate(Confidence = as.numeric(Confidence))
  known_kingdom <- known_kingdom%>% mutate(Confidence = as.character(Confidence))
  known_kingdom <-  known_kingdom %>% mutate(Confidence = as.numeric(Confidence))

##Species Confidence
ms <- mean(known_species$Confidence) #0.9107
ss <- sd(known_species$Confidence) #0.0976

##Genus Confidence
mg <- mean(known_genus$Confidence) #0.9577
sg <- sd(known_genus$Confidence) #0.0772

##Family Confidence
mf <- mean(known_family$Confidence) #0.9605
sf <- sd(known_family$Confidence) #0.0734

##Order Confidence
mo <- mean(known_order$Confidence) #0.9631
so <- sd(known_order$Confidence) #0.0708

##Class confidence
mc <- mean(known_class$Confidence) #0.9623
sc <- sd(known_class$Confidence) #0.0715

##Phylum confidence
mp <- mean(known_phylum$Confidence) #0.9619
sp <- sd(known_phylum$Confidence) #0.0721

##Kingdom confidence
mk <- mean(known_kingdom$Confidence) #0.9615
sk <- sd(known_kingdom$Confidence) #0.0725

##Plot relative confidence of each taxonomic group classificaiton

conf_by_tax <- read_excel("confidence.xlsx") #made a new dataframe using mean and sd data calculated above. Reading in this data frame and cleaning the data below

conf_by_tax <- conf_by_tax %>% mutate(taxon = factor(taxon))
conf_by_tax$taxon <- factor(conf_by_tax$taxon, levels = c("kingdom", "phylum", "class", "order", "family", "genus", "species")) #reorder levels - otherwise it will automatically list aplphabetically
conf_by_tax <- cbind(conf_by_tax, otu_prop_vector)


#Visualize

jpeg("/Users/joegunn/Desktop/Grad_School_Stuff/Projects/Elephant_Microbiome/NEW_STATS/AEMB_FINAL_STATS/African_Elephant_Microbiome/tax_confidence.jpeg", width=5000, height=4000, units='px', res=900) 

ggplot(conf_by_tax, aes(x = taxon, y = mean)) + geom_point(size = 2) +
     geom_errorbar(aes(ymin=mean-sd, ymax=mean+sd), width=.3) + scale_y_continuous(name = "Mean Confidence of Classification", sec.axis = sec_axis(~. /0.97, name = "Proportion of Classified OTUs")) + geom_line(mapping = aes(x = conf_by_tax$taxon, y = conf_by_tax$otu_prop_vector, group = 1), size = 2, color = "blue", alpha = 0.6) + theme(axis.text.x = element_text(angle = 60, hjust = 1)) + labs(x = "Taxonomic Group")

dev.off()
```

# OTU ABUNDANCE - ORDER LEVEL

## Rarefied OTU abundance data 
```{r, echo = FALSE}
options(scipen = 999) #gets rid of scientific notatation, useful for reading p-values

#read in taxonomic designations and OTU abundance
aem_physeq <- qza_to_phyloseq(features = "table.qza", taxonomy = "taxonomy.qza",  tree = "rooted-tree.qza", metadata = "METADATA.tsv") 

#omit sample OB182 due to low read count. We don't want to rarefy to such a low number of reads across the whole dataset
aem_physeq <- subset_samples(aem_physeq, Elephant != "OB182") 

#Rarefy and clean the data set for use in downstream analysis
aem_physeq <- rarefy_even_depth(aem_physeq, rngsee = 5) #must set seed in order to maintain the same rarefied number of otus

    #Number of OTUS: 8248
    #Number of reads per sample: 11460 
      
#Extract data of all types (meta data, all taxonomy data)
aem_meta <- as.data.frame(sample_data(aem_physeq))
aem_tax <- as.data.frame(tax_table(aem_physeq))
aem_tax <- aem_tax %>% rownames_to_column("otu")
aem_data <- as.data.frame(otu_table(aem_physeq))
aem_data <- aem_data %>% rownames_to_column("otu")
aem_tax_data <- merge(aem_tax, aem_data, by = "otu")
```

## Most Abundant Phyla
```{r}
#Subset the data by the three most abundant phyla: Bacteroidetes, firmicutes, proteobacteria
aem_bacteroidetes <- aem_tax_data %>% filter(Phylum == "p__Bacteroidetes")
aem_firmicutes <- aem_tax_data %>% filter(Phylum == "p__Firmicutes")
aem_proteobacteria <- aem_tax_data %>% filter(Phylum == "p__Proteobacteria")
```

## Phylum and Order Proportions per African Elephant Group
```{r, echo = FALSE}

#Proportions of each order contributing to each african elephant group
aem_order_phylum_data <- aem_tax_data[,-c(1:2,4,6:8)]
aem_order_phylum_data <- as.tibble(aem_order_phylum_data)
aem_order_phylum_data <- aem_order_phylum_data %>% group_by(Phylum, Order) %>% summarize_all(funs(sum))
```

## Core Microbiome - Order Level
```{r}
#Summarize Rarefied OTU abundance by ALL Orders to figure out the core microbiome 
aem_order_data <- aem_tax_data[,-c(1:4,6:8)]
aem_order_data <- as.data.frame(aem_order_data)
aem_order_data <- aem_order_data %>% group_by(Order) %>% summarize_all(funs(sum))
aem_order_data <- aem_order_data %>% drop_na()
aem_order_data <- column_to_rownames(aem_order_data, "Order")
aem_order_data_t <- t(aem_order_data)
aem_order_data_t <- as.data.frame(aem_order_data_t)
aem_order_data_t <- aem_order_data_t %>% rownames_to_column("sample_id")
aem_order <- merge(metadata, aem_order_data_t, by = "sample_id")

###IMPORTANT NOTE: At this point, aem_orders, which provides abundance across all orders for all samples, was copied and entered into excel in order to determine the core microbiome and any other shared orders among african elephant groups. Figrue 1 in the manuscript is produced from the data within this excel file.
```

## Order Abundance within Phylum Bacteroidetes, Firmicutes, and Proteobacteria
```{r}
#Summarize Rarefied OTU abundance by Orders WITHIN BACTERIOIDETES
aem_bact_orders <- aem_bacteroidetes[,-c(1:4,6:8)]
aem_bact_orders <- as.data.frame(aem_bact_orders)
aem_bact_orders <- aem_bact_orders %>% group_by(Order) %>% summarize_all(funs(sum))
aem_bact_orders <- aem_bact_orders %>% drop_na()
aem_bact_orders <- column_to_rownames(aem_bact_orders, "Order")
aem_bact_orders_t <- t(aem_bact_orders)
aem_bact_orders_t <- as.data.frame(aem_bact_orders_t)
aem_bact_orders_t <- aem_bact_orders_t %>% rownames_to_column("sample_id")
aem_bact_orders_sums <- merge(metadata, aem_bact_orders_t, by = "sample_id")

#Summarize Rarefied OTU abundance by Orders WITHIN FIRMICUTES
aem_firm_orders <- aem_firmicutes[,-c(1:4,6:8)]
aem_firm_orders <- as.data.frame(aem_firm_orders)
aem_firm_orders <- aem_firm_orders %>% group_by(Order) %>% summarize_all(funs(sum))
aem_firm_orders <- aem_firm_orders %>% drop_na()
aem_firm_orders <- column_to_rownames(aem_firm_orders, "Order")
aem_firm_orders_t <- t(aem_firm_orders)
aem_firm_orders_t <- as.data.frame(aem_firm_orders_t)
aem_firm_orders_t <- aem_firm_orders_t %>% rownames_to_column("sample_id")
aem_firm_orders_sums <- merge(metadata, aem_firm_orders_t, by = "sample_id")

#Summarize Rarefied OTU abundance by Orders WITHIN PROTEOBACTERIA
aem_proteo_orders <- aem_proteobacteria[,-c(1:4,6:8)]
aem_proteo_orders <- as.data.frame(aem_proteo_orders)
aem_proteo_orders <- aem_proteo_orders %>% group_by(Order) %>% summarize_all(funs(sum))
aem_proteo_orders <- aem_proteo_orders %>% drop_na()
aem_proteo_orders <- column_to_rownames(aem_proteo_orders, "Order")
aem_proteo_orders_t <- t(aem_proteo_orders)
aem_proteo_orders_t <- as.data.frame(aem_proteo_orders_t)
aem_proteo_orders_t <- aem_proteo_orders_t %>% rownames_to_column("sample_id")
aem_proteo_orders_sums <- merge(metadata, aem_proteo_orders_t, by = "sample_id")

#Compile sums for most abundant phyla and subset the data for downstream analyses. One dataset for Africana (n = 35), one dataset for Forest non-crop-raiders (n = 19)

#Africana - diet and habitat
aem_bact_orders_sums_africana <- aem_bact_orders_sums %>% filter(Species == "africana")
aem_firm_orders_sums_africana <- aem_firm_orders_sums %>% filter(Species == "africana")
aem_proteo_orders_sums_africana <- aem_proteo_orders_sums %>% filter(Species == "africana")

#Forest, non-crop-raiders - species
aem_bact_orders_sums_forest_ncr <- aem_bact_orders_sums %>% filter(Habitat == "Forest") %>% filter(Raider == "No")
aem_firm_orders_sums_forest_ncr <- aem_firm_orders_sums %>% filter(Habitat == "Forest") %>% filter(Raider == "No")
aem_proteo_orders_sums_forest_ncr <- aem_proteo_orders_sums %>% filter(Habitat == "Forest") %>% filter(Raider == "No")


#Analyze OTU sum data by African elephant species
aem_bact_species <- aem_bact_orders_sums_forest_ncr[,-c(1:3,5:6,9:11)]
colnames(aem_bact_species) <- c("Species", "Age", "Sex", "Saprospirales", "Bacteroidales", "Cytophagales", "Flavobacteriales", "Sphingobacteriales")

aem_firm_species <- aem_firm_orders_sums_forest_ncr[,-c(1:3,5:6,9:11)]
colnames(aem_firm_species) <- c("Species", "Age", "Sex", "Bacillales", "Clostridiales", "Erysipelotrichales", "Lactobacillales")


aem_proteo_species <- aem_proteo_orders_sums_forest_ncr[,-c(1:3,5:6,9:11)]
colnames(aem_proteo_species) <- c("Species", "Age", "Sex", "Aeromonadales", "Alteromonadales", "Bdellovibrionales", "Burkholderiales", "Caulobacterales", "Desulfovibrionales", "Enterobacteriales", "gmd14h09", "Kiloniellales", "Neisseriales", "Oceanospirillales", "Pasteurellales", "phoshd29", "Pseudomonadales", "rf32", "Rhizobiales", "Rhodobacterales", "Rickettsiales", "Sphingomonadales", "Spirobacillales", "Tremblayales", "Xanthomonadales")

#Analyze OTU sum data by African elephant Groups
aem_bact_groups <- aem_bact_orders_sums_africana[,-c(1:6,9,11)]
colnames(aem_bact_groups) <- c("Age", "Sex", "Group" , "Saprospirales", "Bacteroidales", "Cytophagales", "Flavobacteriales", "Sphingobacteriales")

aem_firm_groups <- aem_firm_orders_sums_africana[,-c(1:6,9,11)]
colnames(aem_firm_groups) <- c("Age", "Sex", "Group", "Bacillales", "Clostridiales", "Erysipelotrichales", "Lactobacillales")

aem_proteo_groups <- aem_proteo_orders_sums_africana[,-c(1:6,9,11)]
colnames(aem_proteo_groups) <- c("Age", "Sex", "Group","Aeromonadales", "Alteromonadales", "Bdellovibrionales", "Burkholderiales", "Caulobacterales", "Desulfovibrionales", "Enterobacteriales", "gmd14h09", "Kiloniellales", "Neisseriales", "Oceanospirillales", "Pasteurellales", "phoshd29", "Pseudomonadales", "rf32", "Rhizobiales", "Rhodobacterales", "Rickettsiales", "Sphingomonadales", "Spirobacillales", "Tremblayales", "Xanthomonadales")
```

## Testing for Normality in all orders
```{r}

bact_norm1 <- ggqqplot(aem_bact_species$Bacteroidales)
      sb1 <- shapiro.test(aem_bact_species$Bacteroidales) #NORMAL
bact_norm2 <- ggqqplot(aem_bact_species$Cytophagales)
      sb2 <- shapiro.test(aem_bact_species$Cytophagales) #NON-NORMAL
bact_norm3 <- ggqqplot(aem_bact_species$Saprospirales)
      sb3 <- shapiro.test(aem_bact_species$Saprospirales) #NON-NORMAL
bact_norm4 <- ggqqplot(aem_bact_species$Flavobacteriales)
      sb4 <- shapiro.test(aem_bact_species$Flavobacteriales) #NON-NORMAL
bact_norm5 <- ggqqplot(aem_bact_species$Sphingobacteriales)
      sb5 <- shapiro.test(aem_bact_species$Sphingobacteriales) #NON-NORMAL


firm_norm1 <- ggqqplot(aem_firm_species$Bacillales)
      sf1 <- shapiro.test(aem_firm_species$Bacillales) #NON-NORMAL
firm_norm2 <- ggqqplot(aem_firm_species$Clostridiales)
      sf2 <- shapiro.test(aem_firm_species$Clostridiales) #NORMAL
firm_norm3 <- ggqqplot(aem_firm_species$Erysipelotrichales)
      sf3 <- shapiro.test(aem_firm_species$Erysipelotrichales) #NON-NORMAL
firm_norm4 <- ggqqplot(aem_firm_species$Lactobacillales)
      sf4 <- shapiro.test(aem_firm_species$Lactobacillales) #NON-NORMAL
      
#ALL NOT NORMAL EXCEPT PSUEDOMONADALES
proteo_norm1 <- ggqqplot(aem_proteo_species$Aeromonadales)
      sp1 <- shapiro.test(aem_proteo_species$Aeromonadales)
proteo_norm2 <- ggqqplot(aem_proteo_species$Alteromonadales)
      sp2 <- shapiro.test(aem_proteo_species$Alteromonadales)
proteo_norm3 <- ggqqplot(aem_proteo_species$Bdellovibrionales)
      sp3 <- shapiro.test(aem_proteo_species$Bdellovibrionales)
proteo_norm4 <- ggqqplot(aem_proteo_species$Burkholderiales)
      sp4 <- shapiro.test(aem_proteo_species$Burkholderiales)
proteo_norm5 <- ggqqplot(aem_proteo_species$Caulobacterales)
      sp5 <- shapiro.test(aem_proteo_species$Caulobacterales)
proteo_norm6 <- ggqqplot(aem_proteo_species$Desulfovibrionales)
      sp6 <- shapiro.test(aem_proteo_species$Desulfovibrionales)
proteo_norm7 <- ggqqplot(aem_proteo_species$Enterobacteriales)
      sp7 <- shapiro.test(aem_proteo_species$Enterobacteriales)
proteo_norm8 <- ggqqplot(aem_proteo_species$gmd14h09)
      sp8 <- shapiro.test(aem_proteo_species$gmd14h09)
proteo_norm9 <- ggqqplot(aem_proteo_species$Kiloniellales)
      sp9 <- shapiro.test(aem_proteo_species$Kiloniellales)
proteo_norm10 <- ggqqplot(aem_proteo_species$Neisseriales)
      sp10 <- shapiro.test(aem_proteo_species$Neisseriales)
proteo_norm11 <- ggqqplot(aem_proteo_species$Oceanospirillales)
      sp11 <- shapiro.test(aem_proteo_species$Oceanospirillales)
proteo_norm12 <- ggqqplot(aem_proteo_species$Pasteurellales)
      sp12 <- shapiro.test(aem_proteo_species$Pasteurellales)
proteo_norm13 <- ggqqplot(aem_proteo_species$phoshd29)
      sp13 <- shapiro.test(aem_proteo_species$phoshd29)
proteo_norm14 <- ggqqplot(aem_proteo_species$Pseudomonadales) 
      sp14 <- shapiro.test(aem_proteo_species$Pseudomonadales) #NORMAL
proteo_norm15 <- ggqqplot(aem_proteo_species$rf32)
      sp15 <- shapiro.test(aem_proteo_species$rf32)
proteo_norm16 <- ggqqplot(aem_proteo_species$Rhizobiales)
      sp16 <- shapiro.test(aem_proteo_species$Rhizobiales)
proteo_norm17 <- ggqqplot(aem_proteo_species$Rhodobacterales)
      sp17 <- shapiro.test(aem_proteo_species$Rhodobacterales)
proteo_norm18 <- ggqqplot(aem_proteo_species$Rickettsiales)
      sp18 <- shapiro.test(aem_proteo_species$Rickettsiales)
proteo_norm19 <- ggqqplot(aem_proteo_species$Sphingomonadales)
      sp19 <- shapiro.test(aem_proteo_species$Sphingomonadales)
proteo_norm20 <- ggqqplot(aem_proteo_species$Tremblayales)
      sp20 <- shapiro.test(aem_proteo_species$Tremblayales)
proteo_norm21 <- ggqqplot(aem_proteo_species$Xanthomonadales)
      sp21 <- shapiro.test(aem_proteo_species$Xanthomonadales)
```

## Statistical Significance of Bacterial Orders By African Elephant Species
```{r}
aem_bact_species_rel0.1 <- aem_bact_species[,-c(4,6,8)]
aem_firm_species_rel0.1 <- aem_firm_species[,-c(6,7)]
aem_proteo_species_rel0.1 <- aem_proteo_species[,-c(4:9,11:16,18:25)]

#PHYLUM Bacteroidetes
p_vals_bact_species <- data.frame(numeric(length = 2))

    for (ii in 4:ncol(aem_bact_species_rel0.1)) {
  
      col = aem_bact_species_rel0.1[,ii] #this tells the for loop that it should iterate through columns in the data set
      
      ff <- as.formula(paste0(colnames(aem_bact_species_rel0.1)[ii]," ~ Species + (1|Sex)"))
      
      glmer_temp <- glmer.nb(ff,  aem_bact_species_rel0.1) #tells the for loop to run linear model  for each column           specified above
      p_value <- summary(glmer_temp)$coefficients[2,4] #extracts p value
      p_vals_bact_species[ii-3,] <- p_value
    }

#PHYLUM Firmicutes
p_vals_firm_species <- data.frame(numeric(length = 2))

    for (ii in 4:ncol(aem_firm_species_rel0.1)) {
  
      col = aem_firm_species_rel0.1[,ii] #this tells the for loop that it should iterate through columns in the data set
      ff <- as.formula(paste0(colnames(aem_firm_species_rel0.1)[ii]," ~ Species + (1|Sex)"))
      
      glmer_temp <- glmer.nb(ff, data = aem_firm_species_rel0.1) #tells the for loop to run linear model  for each column           specified above
      p_value <- summary(glmer_temp)$coefficients[2,4] #extracts p value
      p_vals_firm_species[ii-3,] <- p_value
    }

#PHYLUM Proteobacteria
aem_proteo_species <- aem_proteo_species[,-c(23)] #SPIROBACILLES REMOVED BECAUSE IT WAS ALL ZEROS
p_vals_proteo_species <- data.frame(numeric(length = 2))

    for (ii in 4:ncol(aem_proteo_species_rel0.1)) {
  
      col = aem_proteo_species_rel0.1[,ii] #this tells the for loop that it should iterate through columns in the data set
      ff <- as.formula(paste0(colnames(aem_proteo_species_rel0.1)[ii]," ~ Species + (1|Sex)"))
      
      glmer_temp <- glmer.nb(ff, data = aem_proteo_species_rel0.1) #tells the for loop to run linear model  for each column         specified above
      p_value <- summary(glmer_temp)$coefficients[2,4] #extracts p value
      p_vals_proteo_species[ii-3,] <- p_value
    }

#Convert dataframes with p_values for each order within each phylum into tables we can work with
    
#PHYLUM Bacteroidetes orders p-values
p_vals_bact_species <- p_vals_bact_species %>% drop_na()
colnames(p_vals_bact_species) <- c("pvals")
p_vals_bact_species <- as.data.frame(p_vals_bact_species)

#PHYLUM Firmicutes orders p-values
p_vals_firm_species <- p_vals_firm_species %>% drop_na()
colnames(p_vals_firm_species) <- c("pvals")
p_vals_firm_species <- as.data.frame(p_vals_firm_species)

#PHYLUM Proteobacteria orders p-values
p_vals_proteo_species <- p_vals_proteo_species %>% drop_na()
colnames(p_vals_proteo_species) <- c("pvals")
p_vals_proteo_species <- as.data.frame(p_vals_proteo_species)

    
#Correct p-values for multiple tests using false discovery rate
p_vals_bact_species_fdr <- p.adjust(p_vals_bact_species$pvals, method = "fdr")
p_vals_firm_species_fdr <- p.adjust(p_vals_firm_species$pvals, method = "fdr")
p_vals_proteo_species_fdr <- p.adjust(p_vals_proteo_species$pvals, method = "fdr")

#p_vals_bact_species_fdr #Nothing is significant
#p_vals_firm_species_fdr #Bacillales and Lactobacillales are significant
#p_vals_proteo_species_fdr #nothing significant
```

## Make Box Plots for ALL ORDERS
```{r}
#Log-transform y axis so that all values will show up

#PHYLUM Bacteroidetes
species_bacteroidetes_order_abun <- read_excel("species_bacteroidetes_order_abun.xlsx")
species_bacteroidetes_order_abun <- species_bacteroidetes_order_abun %>% mutate(log_abundance = log10(abundance), sqrt_abundance = sqrt(abundance))

#PHYLUM Firmicutes
species_firmicutes_order_abun <- read_excel("species_firmicutes_order_abun.xlsx")
species_firmicutes_order_abun <- species_firmicutes_order_abun %>% mutate(log_abundance = log10(abundance), sqrt_abundance = sqrt(abundance))

#PHYLUM Proteobacteria
species_proteobacteria_order_abun <- read_excel("species_proteobacteria_order_abun.xlsx")
species_proteobacteria_order_abun <- species_proteobacteria_order_abun %>% mutate(log_abundance = log10(abundance), sqrt_abundance = sqrt(abundance))
 
#Make box-plots of Orders for African Elephant Species

#Bacteroidetes
species_bacteroidetes_box_plots <- ggplot(species_bacteroidetes_order_abun, aes(x = order, y = sqrt_abundance, fill = sex)) + geom_boxplot(show.legend = T, outlier.shape = NA) + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + labs(fill = "Elephant Sex", title = "Orders by Elephant Species", y = "Bacteroidetes Abundance") + scale_fill_manual(values = c("red", "blue")) + theme(axis.title.x = element_blank()) + theme(legend.position = c(0.65,0.7)) + theme(legend.text = element_text(face = "italic")) + theme(axis.text.x = element_text(face = "italic")) 

#Firmicutes
species_firmicutes_box_plots <- ggplot(species_firmicutes_order_abun, aes(x = order, y = sqrt_abundance, fill = species)) + geom_boxplot(show.legend = F, outlier.shape = NA) + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + labs(y = "Firmicutes Abundance") + scale_fill_manual(values = c("red", "blue")) + theme(axis.title.x = element_blank()) + theme(axis.text.x = element_text(face = "italic"))


#Proteobacteria
species_proteobacteria_box_plots <- ggplot(species_proteobacteria_order_abun, aes(x = order, y = sqrt_abundance, fill = Species)) + geom_boxplot(show.legend = F, outlier.shape = NA) + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + labs(y = "Proteobacteria Abundance") + scale_fill_manual(values = c("red", "blue")) + theme(axis.title.x = element_blank())+ theme(axis.text.x = element_text(face = "italic"))
    

jpeg("/Users/joegunn/Desktop/Grad_School_Stuff/Projects/Elephant_Microbiome/NEW_STATS/AEMB_FINAL_STATS/African_Elephant_Microbiome/phylum_species_plots.jpeg", width=2000, height=3000, units='px', res=250) 

plot_grid(species_bacteroidetes_box_plots, species_firmicutes_box_plots, species_proteobacteria_box_plots, nrow = 3, labels = c("A", "B", "C"))

dev.off()


#For combining figures
phy_species_plot <- plot_grid(species_bacteroidetes_box_plots, species_firmicutes_box_plots, species_proteobacteria_box_plots, nrow = 3, labels = c("A", "B", "C"))
```

## Make Box Plots for FILTERED ORDERS (AVERAGE RELATIVE ABUNDANCE = 0.1) by Species
```{r}
#Log-transform y axis so that all values will show up

#PHYLUM Bacteroidetes
species_bacteroidetes_order_abun_rel0.1 <- read_excel("species_bacteroidetes_order_abun_rel0.1.xlsx")
species_bacteroidetes_order_abun_rel0.1 <- species_bacteroidetes_order_abun_rel0.1 %>% mutate(log_abundance = log10(abundance), sqrt_abundance = sqrt(abundance))

#PHYLUM Firmicutes
species_firmicutes_order_abun_rel0.1 <- read_excel("species_firmicutes_order_abun_rel0.1.xlsx")
species_firmicutes_order_abun_rel0.1 <- species_firmicutes_order_abun_rel0.1 %>% mutate(log_abundance = log10(abundance), sqrt_abundance = sqrt(abundance))

#PHYLUM Proteobacteria
species_proteobacteria_order_abun_rel0.1 <- read_excel("species_proteobacteria_order_abun_rel0.1.xlsx")
species_proteobacteria_order_abun_rel0.1 <- species_proteobacteria_order_abun_rel0.1 %>% mutate(log_abundance = log10(abundance), sqrt_abundance = sqrt(abundance))
 
#Make box-plots of Orders for African Elephant Species

#Bacteroidetes
species_bacteroidetes_box_plots_rel0.1 <- ggplot(species_bacteroidetes_order_abun_rel0.1, aes(x = order, y = sqrt_abundance, fill = species)) + geom_boxplot(show.legend = T, outlier.shape = NA) + theme(axis.text = element_text(size = 15)) + labs(fill = "Elephant Species", title = "Orders by Elephant Species", y = "Bacteroidetes Abundance") + theme(plot.title = element_text(size = 20, hjust = 0.5, face = "bold")) + scale_fill_manual(values = c("red", "blue")) + theme(axis.title.x = element_blank()) + theme(legend.position = c(0.85,0.8)) + theme(axis.title.y = element_text(size = 20)) + theme(legend.text = element_text(size = 20, face = "italic")) + theme(axis.text.x = element_text(face = "italic", size = 20)) + theme(plot.background = element_blank()) + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),panel.background = element_blank(), axis.line = element_line(colour = "black")) + theme(legend.title = element_text(size = 20))

#Firmicutes
species_firmicutes_box_plots_rel0.1 <- ggplot(species_firmicutes_order_abun_rel0.1, aes(x = order, y = sqrt_abundance, fill = species)) + geom_boxplot(show.legend = F, outlier.shape = NA) + theme(axis.text = element_text(size = 15)) + labs(y = "Firmicutes Abundance") + scale_fill_manual(values = c("red", "blue")) + theme(axis.title.x = element_blank()) + theme(axis.text.x = element_text(face = "italic", size = 20))  + theme(plot.background = element_blank())  + theme(axis.title.y = element_text(size = 20)) + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),panel.background = element_blank(), axis.line = element_line(colour = "black")) + theme(legend.box.background = element_blank())

#Proteobacteria
species_proteobacteria_box_plots_rel0.1 <- ggplot(species_proteobacteria_order_abun_rel0.1, aes(x = order, y = sqrt_abundance, fill = Species)) + geom_boxplot(show.legend = F, outlier.shape = NA) + theme(axis.text = element_text(size = 15)) + labs(y = "Proteobacteria Abundance") + scale_fill_manual(values = c("red", "blue")) + theme(axis.title.x = element_blank())+ theme(axis.text.x = element_text(face = "italic", size = 20))  + theme(axis.title.y = element_text(size = 20)) + theme(plot.background = element_blank())  + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),panel.background = element_blank(), axis.line = element_line(colour = "black")) + theme(legend.box.background = element_blank())
    

jpeg("/Users/joegunn/Desktop/Grad_School_Stuff/Projects/Elephant_Microbiome/NEW_STATS/AEMB_FINAL_STATS/African_Elephant_Microbiome/phylum_species_plots_rel0.1.jpeg", width=2000, height=3000, units='px', res=250) 

plot_grid(species_bacteroidetes_box_plots_rel0.1, species_firmicutes_box_plots_rel0.1, species_proteobacteria_box_plots_rel0.1, nrow = 3, labels = c("A", "B", "C"), label_size = 20)

dev.off()


#For combining figures
phy_species_plot_rel0.1 <- plot_grid(species_bacteroidetes_box_plots_rel0.1, species_firmicutes_box_plots_rel0.1, species_proteobacteria_box_plots_rel0.1, nrow = 3, labels = c("A", "B", "C"), label_size = 20)
```

## Statistical Signifiance of Orders By African Elephant Groups
```{r}
aem_bact_groups_rel0.1 <- aem_bact_groups[,-c(4,6,8)]
aem_firm_groups_rel0.1 <- aem_firm_groups[,-c(6,7)]
aem_proteo_groups_rel0.1 <- aem_proteo_groups[,-c(4:9,11:16,18:25)]

#PHYLUM Bacteroidetes
gbact <- glmer.nb(Bacteroidales ~ Group + (1|Sex), data = aem_bact_groups_rel0.1)
gbact_p <- Anova(gbact)$`Pr(>Chisq)` #P = 0.1889
gflav <- glmer.nb(Flavobacteriales ~ Group + (1|Sex), data = aem_bact_groups_rel0.1, control=glmerControl(optimizer="bobyqa", tolPwrss=1e-3, optCtrl = list(maxfun = 100000)))
gflav_p <- Anova(gflav)$`Pr(>Chisq)` #P = 0.001998

p_vals_bact_groups <- as.data.frame(c(gbact_p, gflav_p))
colnames(p_vals_bact_groups) <- c("pval")

#Post-hoc Tukey-test
post_flavo <- glht(gflav, mcp(Group = "Tukey"))
summary(post_flavo)

#PHYLUM Firmicutes
gbaci <- glmer.nb(Bacillales ~ Group + (1|Sex), data = aem_firm_groups_rel0.1)
gbaci_p <- Anova(gbaci)$`Pr(>Chisq)`
gclos <- glmer.nb(Clostridiales ~ Group + (1|Sex), data = aem_firm_groups_rel0.1)
gclos_p <- Anova(gclos)$`Pr(>Chisq)`

p_vals_firm_groups <- as.data.frame(c(gbaci_p, gclos_p))
colnames(p_vals_firm_groups) <- c("pval")

#Post-hoc Tukey-test
post_bacillales <- glht(gbaci, mcp(Group="Tukey"))
summary(post_bacillales)

#PHYLUM Proteobacteria
genter <- glmer.nb(Enterobacteriales ~ Group + (1|Sex), data = aem_proteo_groups_rel0.1)
genter_p <- Anova(genter)$`Pr(>Chisq)`
gpseudo <- glmer.nb(Pseudomonadales ~ Group + (1|Sex), data = aem_proteo_groups_rel0.1)
gpseudo_p <- Anova(gpseudo)$`Pr(>Chisq)`

p_vals_proteo_groups <- as.data.frame(c(genter_p, gpseudo_p))
colnames(p_vals_proteo_groups) <- c("pval")

#Post-hoc Tukey-Test
post_pseudo <- glht(gpseudo, mcp(Group = "Tukey"))
summary(post_pseudo)


#For loops for ALL ORDERS

#Bacteroides
p_vals_bact_groups <- data.frame(numeric(length = 2)) #This for loop had trouble knitting to html, so I included hashtags to force it to finish. WE DID USE THIS CODE TO PRODUCE RESULTS FOR THE MANUSCRIPT

   # for (ii in 4:ncol(aem_bact_groups_rel0.1)) {
      
    #  ff <- as.formula(paste0(colnames(aem_bact_groups_rel0.1)[ii]," ~ Group + (1|Sex)"))
    #  glmer_temp <- glmer.nb(ff, data = aem_bact_groups_rel0.1, control=glmerControl(optimizer="bobyqa", tolPwrss=1e-3, optCtrl = list(maxfun = 100000))) #tells the for loop to run linear model  for each column specified above
    #  p_value <- Anova(glmer_temp)$`Pr(>Chisq)` #extracts f statistics
    #  p_vals_bact_groups[ii-3,] <- p_value
    #}

#PHYLUM Firmicutes
p_vals_firm_groups <- data.frame(numeric(length = 2))

    for (ii in 4:ncol(aem_firm_groups_rel0.1)) {
  
      col = aem_firm_groups_rel0.1[,ii] #this tells the for loop that it should iterate through columns in the data set
      lm_temp <- lm(col ~ Group, data = aem_firm_groups_rel0.1) #tells the for loop to run linear model for each column specified above
      fstat <- summary(lm_temp)$fstatistic
      p_value <- pf(fstat[1], fstat[2], fstat[3], lower.tail = F) #extracts p value
      p_vals_firm_groups[ii-3,] <- p_value
    }

#PHYLUM Proteobacteria
p_vals_proteo_groups <- data.frame(numeric(length = 2))

    for (ii in 1:ncol(aem_proteo_groups_rel0.1)) {
  
      col = aem_proteo_groups_rel0.1[,ii] #this tells the for loop that it should iterate through columns in the data set
      lm_temp <- lm(col ~ Group, data = aem_proteo_groups_rel0.1) #tells the for loop to run linear model  for each column specified above
      fstat <- summary(lm_temp)$fstatistic
      p_value <- pf(fstat[1], fstat[2], fstat[3], lower.tail = F) #extracts p value
      p_vals_proteo_groups[ii,] <- p_value
    }

#Convert dataframes with p_values into tables we can work with

#PHYLUM Bacterioidetes
p_vals_bact_groups <- p_vals_bact_groups %>% drop_na()
colnames(p_vals_bact_groups) <- c("pval")
p_vals_bact_groups <- as.data.frame(p_vals_bact_groups)

#PHYLUM Firmicutes
p_vals_firm_groups <- p_vals_firm_groups %>% drop_na()
colnames(p_vals_firm_groups) <- c("pval")
p_vals_firm_groups <- as.data.frame(p_vals_firm_groups)

#PHYLUM Proteobacteria
p_vals_proteo_groups <- p_vals_proteo_groups %>% drop_na()
colnames(p_vals_proteo_groups) <- c("pval")
p_vals_proteo_groups <- as.data.frame(p_vals_proteo_groups)

#Correct p-values for multiple tests using false discovery rate
p_vals_bact_groups_fdr <- p.adjust(p_vals_bact_groups$pval, method = "fdr")
p_vals_firm_groups_fdr <- p.adjust(p_vals_firm_groups$pval, method = "fdr")
p_vals_proteo_groups_fdr <- p.adjust(p_vals_proteo_groups$pval, method = "fdr")

#p_vals_bact_groups_fdr
#p_vals_firm_groups_fdr
#p_vals_proteo_groups_fdr
```

## Make Box Plots for ALL ORDERS
```{r}
#Log-transform y axis so that all values will show up
groups_bacteroidetes_order_abun <- read_excel("groups_bacteroidetes_order_abun.xlsx")
groups_bacteroidetes_order_abun <- groups_bacteroidetes_order_abun %>% mutate(log_abundance = log10(abundance), sqrt_abundance = sqrt(abundance))

groups_firmicutes_order_abun <- read_excel("groups_firmicutes_order_abun.xlsx")
groups_firmicutes_order_abun <- groups_firmicutes_order_abun %>% mutate(log_abundance = log10(abundance), sqrt_abundance = sqrt(abundance))

groups_proteobacteria_order_abun <- read_excel("groups_proteobacteria_order_abun.xlsx")
groups_proteobacteria_order_abun  <- groups_proteobacteria_order_abun %>% mutate(log_abundance = log10(abundance), sqrt_abundance = sqrt(abundance))


#PHYLUM Bacterioidetes
bact_group_plots <- ggplot(groups_bacteroidetes_order_abun, aes(x = order, y = sqrt_abundance, fill = group)) + geom_boxplot(outlier.shape = NA, show.legend = T) + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + labs(title = "Orders by Elephant Group", fill = "Elephant Group") + scale_fill_manual(values = c("green", "purple", "red", "goldenrod")) + theme(axis.title.x = element_blank()) + theme(axis.title.y = element_blank()) + theme(legend.position = c(0.65,0.7)) + theme(axis.text.x = element_text(face = "italic"))  


#PHYLUM Firmicutes
firm_group_plots <- ggplot(groups_firmicutes_order_abun, aes(x = order, y = sqrt_abundance, fill = group)) + geom_boxplot(outlier.shape = NA, show.legend = F) + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + scale_fill_manual(values = c("green", "purple", "red", "goldenrod")) + theme(axis.title.x = element_blank()) + theme(axis.title.y = element_blank()) + theme(axis.text.x = element_text(face = "italic"))

#PHYLUM Proteobacteria
proteo_group_plots <- ggplot(groups_proteobacteria_order_abun, aes(x = order, y = sqrt_abundance, fill = group)) + geom_boxplot(outlier.shape = NA, show.legend = F) + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + scale_fill_manual(values = c("green", "purple", "red", "goldenrod")) + theme(axis.title.x = element_blank()) + theme(axis.title.y = element_blank()) + theme(axis.text.x = element_text(face = "italic"))

jpeg("/Users/joegunn/Desktop/Grad_School_Stuff/Projects/Elephant_Microbiome/NEW_STATS/AEMB_FINAL_STATS/African_Elephant_Microbiome/phylum_group_plots.jpeg", width=2000, height=3000, units='px', res=250) 

plot_grid(bact_group_plots, firm_group_plots, proteo_group_plots, nrow = 3, labels = c("A", "B", "C"))

dev.off()

#For combining figures

phy_group_plot <- plot_grid(bact_group_plots, firm_group_plots, proteo_group_plots, nrow = 3, labels = c("D", "E", "F"))

##WE did not look for genera within Saprospirales, because it is in such low abundance (only found in n = 8 samples)


###COMBINE PHYLUM PLOTS
pdf("/Users/joegunn/Desktop/Grad_School_Stuff/Projects/Elephant_Microbiome/NEW_STATS/AEMB_FINAL_STATS/African_Elephant_Microbiome/phy_combined_plots.pdf", width=15, height=20) 

plot_grid(phy_species_plot, phy_group_plot, ncol = 2)

dev.off()
```

## Make Box Plots for FILTERED ORDERS (AVERAGE RELATIVE ABUNDANCE = 0.1 or more)
```{r}
#Log-transform y axis so that all values will show up
groups_bacteroidetes_order_abun_rel0.1 <- read_excel("groups_bacteroidetes_order_abun_rel0.1.xlsx")
groups_bacteroidetes_order_abun_rel0.1 <- groups_bacteroidetes_order_abun_rel0.1 %>% mutate(log_abundance = log10(abundance), sqrt_abundance = sqrt(abundance))

groups_firmicutes_order_abun_rel0.1 <- read_excel("groups_firmicutes_order_abun_rel0.1.xlsx")
groups_firmicutes_order_abun_rel0.1 <- groups_firmicutes_order_abun %>% mutate(log_abundance = log10(abundance), sqrt_abundance = sqrt(abundance))

groups_proteobacteria_order_abun_rel0.1 <- read_excel("groups_proteobacteria_order_abun_rel0.1.xlsx")
groups_proteobacteria_order_abun_rel0.1  <- groups_proteobacteria_order_abun_rel0.1 %>% mutate(log_abundance = log10(abundance), sqrt_abundance = sqrt(abundance))


#PHYLUM Bacterioidetes
bact_group_plots_rel0.1 <- ggplot(groups_bacteroidetes_order_abun_rel0.1, aes(x = order, y = sqrt_abundance, fill = group)) + geom_boxplot(outlier.shape = NA, show.legend = T) + theme(plot.title = element_text(size = 20, hjust = 0.5, face = "bold")) + theme(axis.text = element_text(size = 20)) + labs(title = "Orders by Elephant Group", fill = "Elephant Group") + scale_fill_manual(values = c("green", "purple", "red", "goldenrod")) + theme(axis.title.x = element_blank()) + theme(axis.title.y = element_blank()) + theme(legend.position = c(0.85,0.8)) + theme(legend.text = element_text(size = 20)) + theme(axis.text.x = element_text(face = "italic", size = 20)) + theme(plot.background = element_blank())  + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),panel.background = element_blank(), axis.line = element_line(colour = "black")) + theme(legend.title = element_text(size = 20))


#PHYLUM Firmicutes
firm_group_plots_rel0.1 <- ggplot(groups_firmicutes_order_abun_rel0.1, aes(x = order, y = sqrt_abundance, fill = group)) + geom_boxplot(outlier.shape = NA, show.legend = F) + theme(axis.text = element_text(size = 15)) + scale_fill_manual(values = c("green", "purple", "red", "goldenrod")) + theme(axis.title.x = element_blank()) + theme(axis.title.y = element_blank()) + theme(axis.text.x = element_text(face = "italic", size = 20)) + theme(plot.background = element_blank())  + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), panel.background = element_blank(), axis.line = element_line(colour = "black")) + theme(legend.title = element_text(size = 20))

#PHYLUM Proteobacteria
proteo_group_plots_rel0.1 <- ggplot(groups_proteobacteria_order_abun_rel0.1, aes(x = order, y = sqrt_abundance, fill = group)) + geom_boxplot(outlier.shape = NA, show.legend = F) + theme(axis.text = element_text(size = 15)) + scale_fill_manual(values = c("green", "purple", "red", "goldenrod")) + theme(axis.title.x = element_blank()) + theme(axis.title.y = element_blank()) + theme(axis.text.x = element_text(face = "italic", size = 20)) + theme(plot.background = element_blank())  + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),panel.background = element_blank(), axis.line = element_line(colour = "black")) + theme(legend.title = element_text(size = 20))

jpeg("/Users/joegunn/Desktop/Grad_School_Stuff/Projects/Elephant_Microbiome/NEW_STATS/AEMB_FINAL_STATS/African_Elephant_Microbiome/phylum_group_plots_rel0.1.jpeg", width=2000, height=3000, units='px', res=250) 

plot_grid(bact_group_plots_rel0.1, firm_group_plots_rel0.1, proteo_group_plots_rel0.1, nrow = 3, labels = c("A", "B", "C"))

dev.off()


#For combining figures

phy_group_plot_rel0.1 <- plot_grid(bact_group_plots_rel0.1, firm_group_plots_rel0.1, proteo_group_plots_rel0.1, nrow = 3, labels = c("D", "E", "F"), label_size = 20)


##WE did not look for genera within Saprospirales, because it is in such low abundance (only found in n = 8 samples)


###COMBINE PHYLUM PLOTS
pdf("/Users/joegunn/Desktop/Grad_School_Stuff/Projects/Elephant_Microbiome/NEW_STATS/AEMB_FINAL_STATS/African_Elephant_Microbiome/phy_combined_plots_rel0.1.pdf", width=15, height=20)

plot_grid(phy_species_plot_rel0.1, phy_group_plot_rel0.1, ncol = 2)

dev.off()
```

# OTU ABUNDANCE - GENUS LEVEL

## Statistical Signifiance of Genera within Orders Bacillales and Lactobacillales By African Elephant Species
```{r}
#Now we want to find the genera within each of the significant orders (Bacillales and Lactobacillales within Firmicutes) and see which are significantly different between africana and cyclotis species
    
#Data frames with genera only within the significant orders for african elephant SPECIES
aem_firm_bacillales <- aem_firmicutes %>% filter(Order == "o__Bacillales")
aem_firm_lactobacillales <- aem_firmicutes %>% filter(Order == "o__Lactobacillales")

#Data frames with genera only within the significant orders for african elephant GROUPS
aem_bact_flavobacteriales <- aem_bacteroidetes %>% filter(Order == "o__Flavobacteriales")
aem_proteo_pseudomonadales <- aem_proteobacteria %>% filter(Order == "o__Pseudomonadales")

#Extract Genera from Bacillales and Lactobacillales within Firmicutes (SPECIES)

    #Bacillales
    aem_firm_bacillales_genera <- aem_firm_bacillales[,-c(1:6,8)]
    aem_firm_bacillales_genera <- as.data.frame(aem_firm_bacillales_genera)
    aem_firm_bacillales_genera <- aem_firm_bacillales_genera %>% group_by(Genus) %>% summarize_all(funs(sum))
    aem_firm_bacillales_genera <- aem_firm_bacillales_genera %>% drop_na()
    aem_firm_bacillales_genera <- column_to_rownames(aem_firm_bacillales_genera, "Genus")
    aem_firm_bacillales_genera_t <- t(aem_firm_bacillales_genera)
    aem_firm_bacillales_genera_t <- as.data.frame(aem_firm_bacillales_genera_t)
    aem_firm_bacillales_genera_t <- aem_firm_bacillales_genera_t %>% rownames_to_column("sample_id")
    aem_firm_bacillales_genera_sums <- merge(metadata, aem_firm_bacillales_genera_t, by = "sample_id")
    
    #Lactobacillales
    aem_firm_lactobacillales_genera <- aem_firm_lactobacillales[,-c(1:6,8)]
    aem_firm_lactobacillales_genera <- as.data.frame(aem_firm_lactobacillales_genera)
    aem_firm_lactobacillales_genera <- aem_firm_lactobacillales_genera %>% group_by(Genus) %>% summarize_all(funs(sum))
    aem_firm_lactobacillales_genera <- aem_firm_lactobacillales_genera %>% drop_na()
    aem_firm_lactobacillales_genera <- column_to_rownames(aem_firm_lactobacillales_genera, "Genus")
    aem_firm_lactobacillales_genera_t <- t(aem_firm_lactobacillales_genera)
    aem_firm_lactobacillales_genera_t <- as.data.frame(aem_firm_lactobacillales_genera_t)
    aem_firm_lactobacillales_genera_t <- aem_firm_lactobacillales_genera_t %>% rownames_to_column("sample_id")
    aem_firm_lactobacillales_genera_sums <- merge(metadata, aem_firm_lactobacillales_genera_t, by = "sample_id")


#Extract Genera from Flavobacteriales within Bacteroidetes (GROUPS)

    #Flavobacteriales
    aem_bact_flavobacteriales_genera <- aem_bact_flavobacteriales[,-c(1:6,8)]
    aem_bact_flavobacteriales_genera <- as.data.frame(aem_bact_flavobacteriales_genera)
    aem_bact_flavobacteriales_genera <- aem_bact_flavobacteriales_genera %>% group_by(Genus) %>% summarize_all(funs(sum))
    aem_bact_flavobacteriales_genera <- aem_bact_flavobacteriales_genera %>% drop_na()
    aem_bact_flavobacteriales_genera <- column_to_rownames(aem_bact_flavobacteriales_genera, "Genus")
    aem_bact_flavobacteriales_genera_t <- t(aem_bact_flavobacteriales_genera)
    aem_bact_flavobacteriales_genera_t <- as.data.frame(aem_bact_flavobacteriales_genera_t)
    aem_bact_flavobacteriales_genera_t <- aem_bact_flavobacteriales_genera_t %>% rownames_to_column("sample_id")
    aem_bact_flavobacteriales_genera_sums <- merge(metadata, aem_bact_flavobacteriales_genera_t, by = "sample_id")

    
#Extract Genera from Pseudomonadales within Proteobacteria (GROUPS)

    #Pseudomonadales
    aem_proteo_pseudomonadales_genera <- aem_proteo_pseudomonadales[,-c(1:6,8)]
    aem_proteo_pseudomonadales_genera <- as.data.frame(aem_proteo_pseudomonadales_genera)
    aem_proteo_pseudomonadales_genera <- aem_proteo_pseudomonadales_genera %>% group_by(Genus) %>% summarize_all(funs(sum))
    aem_proteo_pseudomonadales_genera <- aem_proteo_pseudomonadales_genera %>% drop_na()
    aem_proteo_pseudomonadales_genera <- column_to_rownames(aem_proteo_pseudomonadales_genera, "Genus")
    aem_proteo_pseudomonadales_genera_t <- t(aem_proteo_pseudomonadales_genera)
    aem_proteo_pseudomonadales_genera_t <- as.data.frame(aem_proteo_pseudomonadales_genera_t)
    aem_proteo_pseudomonadales_genera_t <- aem_proteo_pseudomonadales_genera_t %>% rownames_to_column("sample_id")
    aem_proteo_pseudomonadales_genera_sums <- merge(metadata, aem_proteo_pseudomonadales_genera_t, by = "sample_id")
    
  
#Separate genera into subsetted datasets
aem_firm_lactobacillales_genera_sums_forest_ncr <- aem_firm_lactobacillales_genera_sums %>% filter(Habitat == "Forest") %>% filter(Raider == "No")
aem_firm_bacillales_genera_sums_forest_ncr <- aem_firm_bacillales_genera_sums %>% filter(Habitat == "Forest") %>% filter(Raider == "No")

aem_bacillales_genera_by_species <- aem_firm_bacillales_genera_sums_forest_ncr[,-c(1:3,5,6,9:11)]
aem_lactobacillales_genera_by_species <- aem_firm_lactobacillales_genera_sums_forest_ncr[,-c(1:3,5,6,9:11)]


aem_bact_flavobacteriales_genera_sums_africana <- aem_bact_flavobacteriales_genera_sums %>% filter(Species == "africana")
aem_proteo_pseudomonadales_genera_sums_africana <- aem_proteo_pseudomonadales_genera_sums %>% filter(Species == "africana")

aem_flavobacteriales_genera_by_group <- aem_bact_flavobacteriales_genera_sums_africana[,-c(1:6,9,11)]
aem_pseudomonadales_genera_by_group <- aem_proteo_pseudomonadales_genera_sums_africana[,-c(1:6,9,11)]


colnames(aem_flavobacteriales_genera_by_group) <- c("Age", "Sex", "Group", "Chryseobacterium", "Elizabethkingia", "Flavobacterium", "Fluviicola", "Wautersiella")
colnames(aem_pseudomonadales_genera_by_group) <- c("Age", "Sex", "Group", "Acinetobacter", "Perlucidibaca", "Pseudomonas", "Serpens" )

#Calculate p-values for genera at group level and at species level

gwaut <- glmer.nb(Wautersiella ~ Group + (1|Sex), data = aem_flavobacteriales_genera_by_group)
gwaut_p <- Anova(gwaut)$`Pr(>Chisq)`

#Post-hoc Tukey_test
post_gwaut <- glht(gwaut, mcp(Group = "Tukey"))
summary(post_gwaut)

gacin <- glmer.nb(Acinetobacter ~ Group + (1|Sex), data = aem_pseudomonadales_genera_by_group)
gacin_p <- Anova(gacin)$`Pr(>Chisq)`

#Post-hoc Tukey_test
post_gacin <- glht(gacin, mcp(Group = "Tukey"))
summary(post_gacin)

gbacillus <- glmer.nb(g__Bacillus ~ Species + (1|Sex), data = aem_bacillales_genera_by_species)
gbacillus_p <- Anova(gbacillus)$`Pr(>Chisq)`

gsolibac <- glmer.nb(g__Solibacillus ~ Species + (1|Sex), data = aem_bacillales_genera_by_species)
gsolibac_p <- Anova(gsolibac)$`Pr(>Chisq)`

p_vals_bacillales_species <- as.data.frame(c(gbacillus_p, gsolibac_p))
colnames(p_vals_bacillales_species) <- c("pval")
p_vals_bacillales_species_fdr <- p.adjust(p_vals_bacillales_species$pval, method = "fdr")
```

## calculate p values for ALL GENERA
```{r}
#P-values

#Bacillales
p_vals_bacillales_genera_by_species <- data.frame(numeric(length = ncol(aem_bacillales_genera_by_species)))

    for (ii in 1:ncol(aem_bacillales_genera_by_species)) {
  
      col = aem_bacillales_genera_by_species[,ii] #this tells the for loop that it should iterate through columns in the data set
      lm_temp <- lm(col ~ Species, data = aem_bacillales_genera_by_species) #tells the for loop to run linear model  for each column           specified above
      p_value <- summary(lm_temp)$coefficients[2,4] #extracts p value
      p_vals_bacillales_genera_by_species[ii,] <- p_value
    }

#Lactobacillales
p_vals_lactobacillales_genera_by_species <- data.frame(numeric(length = ncol(aem_lactobacillales_genera_by_species)))

    for (ii in 1:ncol(aem_lactobacillales_genera_by_species)) {
  
      col = aem_lactobacillales_genera_by_species[,ii] #this tells the for loop that it should iterate through columns in the data set
      lm_temp <- lm(col ~ Species, data = aem_lactobacillales_genera_by_species) #tells the for loop to run linear model  for each            column specified above
      p_value <- summary(lm_temp)$coefficients[2,4] #extracts p value
      p_vals_lactobacillales_genera_by_species[ii,] <- p_value
    }

#Correct p-values for multiple tests using false discovery rate
p_vals_bacillales_genera_by_species <- p_vals_bacillales_genera_by_species %>% drop_na() %>% as.data.frame()
colnames(p_vals_bacillales_genera_by_species) <- c("pvals")

p_vals_bacillales_genera_by_species_fdr <- p.adjust(p_vals_bacillales_genera_by_species$pvals, method = "fdr")

p_vals_lactobacillales_genera_by_species <- p_vals_lactobacillales_genera_by_species %>% drop_na() %>% as.data.frame()
colnames(p_vals_lactobacillales_genera_by_species) <- c("pvals")
p_vals_lactobacillales_genera_by_species_fdr <- p.adjust(p_vals_lactobacillales_genera_by_species$pvals, method ="fdr")

```

## Make box plots for ALL GENERA
```{r}
#square root transform the y-axis
species_bacillales_genera_abun <- read_excel("species_bacillales_genera_abun.xlsx")
species_bacillales_genera_abun <- species_bacillales_genera_abun %>% mutate(log_abundance = log10(abundance), sqrt_abundance = sqrt(abundance))

species_lactobacillales_genera_abun <- read_excel("species_lactobacillales_genera_abun.xlsx")
species_lactobacillales_genera_abun <- species_lactobacillales_genera_abun %>% mutate(log_abundance = log10(abundance), sqrt_abundance = sqrt(abundance))


#Plots of genera within Firmicutes Orders by African Elephant Species
bacillales_plot <- ggplot(species_bacillales_genera_abun, aes(x = order, y = sqrt_abundance, fill = species)) + geom_boxplot(outlier.shape = NA, show.legend = T) + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + labs(title = "Genera of Order Bacillales", fill = "Elephant Species", x = "Genus", y = "Mean Abundance") + scale_fill_manual(values = c("red", "blue")) + theme(legend.text = element_text(face = "italic")) + theme(axis.text.x = element_text(face = "italic")) + theme(legend.position = c(0.05, 0.7))

lacto_plot <- ggplot(species_lactobacillales_genera_abun, aes(x = order, y = sqrt_abundance, fill = species)) + geom_boxplot(outlier.shape = NA, show.legend = F) + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + labs(title = "Genera of Order Lactobacillales", x = "Genus", y = "Mean Abundance") + scale_fill_manual(values = c("red", "blue")) + theme(axis.text.x = element_text(face = "italic"))


jpeg("/Users/joegunn/Desktop/Grad_School_Stuff/Projects/Elephant_Microbiome/NEW_STATS/AEMB_FINAL_STATS/African_Elephant_Microbiome/firmicutes_genera_byspecies.jpeg", width=3700, height=5000, units='px', res=600) 

plot_grid(bacillales_plot, lacto_plot, nrow = 2, labels = c("A", "B"))

dev.off()
```

## Make Box Plots for FILTERED GENERA (AVERAGE RELATIVE ABUNDANCE = 0.1 OR MORE)
```{r}
#BY SPECIES

#square root transform the y-axis
bacillales_genera_abun_rel0.1 <- read_excel("bacillales_genera_abun_rel0.1.xlsx")
bacillales_genera_abun_rel0.1 <- bacillales_genera_abun_rel0.1 %>% mutate(log_abundance = log10(abundance), sqrt_abundance = sqrt(abundance))

#Plots of genera within Firmicutes Orders by African Elephant Species
bacillales_plot_rel0.1 <- ggplot(bacillales_genera_abun_rel0.1, aes(x = Genus, y = sqrt_abundance, fill = Species)) + geom_boxplot(outlier.shape = NA, show.legend = T) + theme(axis.text = element_text(size = 15)) + theme(axis.text.x = element_text(size = 20)) + labs(title = "Genera of Phylum Firmicutes: Order Bacillales", fill = "Elephant Species", x = "Genus", y = "Mean Abundance") + theme(plot.title = element_text(size = 20, hjust = 0.5, face = "bold")) + scale_fill_manual(values = c("red", "blue")) + theme(legend.text = element_text(face = "italic")) + theme(axis.text.x = element_text(face = "italic")) + theme(legend.position = c(0.2, 0.7)) + theme(plot.background = element_blank())  + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),panel.background = element_blank(), axis.line = element_line(colour = "black")) + theme(legend.title = element_text(size = 20)) + theme(legend.text = element_text(size = 20)) + theme(axis.title = element_text(size = 20))


pdf("/Users/joegunn/Desktop/Grad_School_Stuff/Projects/Elephant_Microbiome/NEW_STATS/AEMB_FINAL_STATS/African_Elephant_Microbiome/firmicutes_genera_byspecies.pdf", width=12, height=10) 

bacillales_plot_rel0.1

dev.off()


#BY GROUPS


flavobacteriales_genera_abun_rel0.1 <- read_excel("flavobacteriales_genera_abun_rel0.1.xlsx")
flavobacteriales_genera_abun_rel0.1 <- flavobacteriales_genera_abun_rel0.1 %>% mutate(log_abundance = log10(abundance), sqrt_abundance = sqrt(abundance))

pseudomonadales_genera_abun_rel0.1 <- read_excel("pseudomonadales_genera_abun_rel0.1.xlsx")
pseudomonadales_genera_abun_rel0.1 <- pseudomonadales_genera_abun_rel0.1 %>% mutate(log_abundance = log10(abundance), sqrt_abundance = sqrt(abundance))

#Flavobacteriales Plot

flavobacteriales_plot_rel0.1 <- ggplot(flavobacteriales_genera_abun_rel0.1, aes(x = Genus, y = sqrt_abundance, fill = Group)) + geom_boxplot(outlier.shape = NA, show.legend = T) + theme(axis.text = element_text(size = 15)) + theme(axis.text.x = element_text(size = 20)) + labs(title = "Genus of Phylum Bacteroidetes: Order Flavobacteriales", fill = "Elephant Group", x = "Genus", y = "Mean Abundance") + theme(plot.title = element_text(size = 20, hjust = 0.5, face = "bold")) + scale_fill_manual(values = c("green", "purple", "red", "goldenrod")) + theme(legend.text = element_text(face = "italic")) + theme(axis.text.x = element_text(face = "italic")) + theme(legend.position = c(0.8, 0.8)) + theme(plot.background = element_blank())  + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),panel.background = element_blank(), axis.line = element_line(colour = "black")) + theme(legend.title = element_text(size = 20)) + theme(legend.text = element_text(size = 20)) + theme(axis.title = element_text(size = 20))

#Pseudomonadales Plot

pseudomonadales_plot_rel0.1 <- ggplot(pseudomonadales_genera_abun_rel0.1, aes(x = Genus, y = sqrt_abundance, fill = Group)) + geom_boxplot(outlier.shape = NA, show.legend = T) + theme(axis.text = element_text(size = 15)) + theme(axis.text.x = element_text(size = 20)) + labs(title = "Genus of Phylum Proteobacteria: Order Pseudomonadales", fill = "Elephant Group", x = "Genus", y = "Mean Abundance") + theme(plot.title = element_text(size = 20, hjust = 0.5, face = "bold")) + scale_fill_manual(values = c("green", "purple", "red", "goldenrod")) + theme(legend.text = element_text(face = "italic")) + theme(axis.text.x = element_text(face = "italic")) + theme(legend.position = c(0.8, 0.8)) + theme(plot.background = element_blank())  + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),panel.background = element_blank(), axis.line = element_line(colour = "black")) + theme(legend.title = element_text(size = 20)) + theme(legend.text = element_text(size = 20)) + theme(axis.title = element_text(size = 20))

#Combined Plot

pdf("/Users/joegunn/Desktop/Grad_School_Stuff/Projects/Elephant_Microbiome/NEW_STATS/AEMB_FINAL_STATS/African_Elephant_Microbiome/genera_combined_bygroup.pdf", width=10, height=15) 

plot_grid(flavobacteriales_plot_rel0.1, pseudomonadales_plot_rel0.1, nrow = 2, labels = c("A", "B"), label_size = 20)

dev.off()

```

# BETA DIVERSITY

## Prepare and clean data for two datasets
```{r}
#Prepare data at the OTU level for analysis with both datasets (just forest non-cropraiders from both species, and just africana)
aem_physeq <- as.data.frame(otu_table(aem_physeq))
aem_physeq_t <- t(aem_physeq)
aem_physeq <- as.data.frame(aem_physeq_t)
aem_physeq <- aem_physeq %>% rownames_to_column("sample_id")
aem_physeq <- merge(metadata, aem_physeq, by = "sample_id")
aem_physeq_forest_ncr <- aem_physeq %>% filter(Habitat == "Forest") %>% filter(Raider == "No")
aem_physeq_africana <- aem_physeq %>% filter(Species == "africana")
```

## Run PERMANOVA analysis for African elephant Species, Diet, and Habitat
```{r}
###Prepare data for PERMANOVA FOR SPECIES
aem_veg_data <- aem_physeq_forest_ncr[,-c(2:11)]
aem_veg_data <- column_to_rownames(aem_veg_data, "sample_id")
###Analysis##################################
aem_veg <- decostand(aem_veg_data, "total") #standardize abundance data by dividing each number by the total number of samples (individuals in the dataset)
aem_veg <- vegdist(aem_veg, "bray") #calculate Bray-curtis distances between samples
aem_pcoa <- pcoa(aem_veg) #calculates principal components
#############################################

#Prepare PCOA (unused in manuscript)
aem_pcs <- aem_pcoa$vectors
aem_pcs <- as.data.frame(aem_pcs)
aem_pcs <- aem_pcs %>% rownames_to_column("sample_id")
aem_pcs <- merge(metadata, aem_pcs, by = "sample_id")
aem_physeq_ado1 <- column_to_rownames(aem_physeq_forest_ncr, "sample_id")

#Beta Dispersion Analysis for African Elephant species
species_beta <- factor(aem_physeq_ado1[,3])
beta_analysis <- betadisper(aem_veg, species_beta)
beta_vectors <- beta_analysis$vectors
beta_vectors <- as.data.frame(beta_vectors)
beta_vectors <- beta_vectors %>% rownames_to_column("sample_id")
beta_vectors <- merge(metadata, beta_vectors, by = "sample_id")
beta_graph <- ggplot(beta_vectors, aes(x = PCoA1, y = PCoA2, color = Species)) + geom_point()

#####################################################################################

###Prepare data for PERMANOVA FOR DIET
aem_veg2_data <- aem_physeq_africana[,-c(2:11)]
aem_veg2_data <- column_to_rownames(aem_veg2_data, "sample_id")

##Analysis###################################
aem_veg2 <- decostand(aem_veg2_data, "total")
aem_veg2 <- vegdist(aem_veg2, "bray")
aem_pcoa2 <- pcoa(aem_veg2)
#############################################

#Prepare PCOA
aem_pcs2 <- aem_pcoa2$vectors
aem_pcs2 <- as.data.frame(aem_pcs2)
aem_pcs2 <- aem_pcs2 %>% rownames_to_column("sample_id")
aem_pcs2 <- merge(metadata, aem_pcs2, by = "sample_id")

aem_physeq_ado2 <- column_to_rownames(aem_physeq_africana, "sample_id")

#Beta Dispersion Analysis for Diet and Habitat
diet_beta <- factor(aem_physeq_ado2[,4])
habitat_beta <- factor(aem_physeq_ado2[,5])
beta_analysis2 <- betadisper(aem_veg2, diet_beta)
#plot(beta_analysis2, axes = c(1,2), cex = 0.7, col = NULL, lty = "solid", lwd = 1, hull = TRUE)
beta_analysis3 <- betadisper(aem_veg2, habitat_beta)
#plot(beta_analysis3, axes = c(1,2), cex = 0.7, col = NULL, lty = "solid", lwd = 1, hull = TRUE)

#Run Adonis code: Permanova. We ran this code for many different linear combinations and for all variables.
ado1 <- adonis(aem_veg ~ Species, aem_physeq_ado1, distance = "bray", permutations = 9999) #Significant
ado2 <- adonis(aem_veg2 ~ Raider, aem_physeq_ado2, distance = "bray", permutations = 9999)
#When we analyze only by crop raider, permanova is significant; when we analyze with habitat as a stratum, it is still significant.
ado3 <- adonis(aem_veg2 ~ Habitat, aem_physeq_ado2, distance = "bray", permutations = 9999) #VERY significant if just modeled by habitat alone. Also very significant if raider is a stratum
ado4 <- adonis(aem_veg2 ~ Age, aem_physeq_ado2, distance = "bray", permutations = 9999) #NOT SIGNIFICANT
ado5 <- adonis(aem_veg2 ~ Sex, aem_physeq_ado2, distance = "bray", permutations = 9999) #NOT SIGNIFICANT
ado6 <- adonis(aem_veg2 ~ Raider*Habitat, aem_physeq_ado2, distance = "bray", permutations = 9999)
ado7 <- adonis(aem_veg2 ~ Raider*Sex, aem_physeq_ado2, distance = "bray", permutations = 9999)
ado8 <- adonis(aem_veg2 ~ Raider*Age, aem_physeq_ado2, distance = "bray", permutations = 9999)
ado9 <- adonis(aem_veg2 ~ Habitat*Sex, aem_physeq_ado2, distance = "bray", permutations = 9999)
ado10 <- adonis(aem_veg2 ~ Habitat*Age, aem_physeq_ado2, distance = "bray", permutations = 9999)
ado11 <- adonis(aem_veg2 ~ Raider, aem_physeq_ado2, Strata = Habitat, distance = "bray", permutations = 9999)
ado12 <- adonis(aem_veg2 ~ Habitat, aem_physeq_ado2, Strata = Raider, distance = "bray", permutations = 9999)
ado13 <- adonis(aem_veg2 ~ Raider, aem_physeq_ado2, Strata = Sex, distance = "bray", permutations = 9999)
ado14 <- adonis(aem_veg2 ~ Raider, aem_physeq_ado2, Strata = Age, distance = "bray", permutations = 9999)
ado15 <- adonis(aem_veg2 ~ Habitat, aem_physeq_ado2, Strata = Sex, distance = "bray", permutations = 9999)
ado16 <- adonis(aem_veg2 ~ Habitat, aem_physeq_ado2, Strata = Age, distance = "bray", permutations = 9999)

ado17 <- adonis(aem_veg ~ Sex, aem_physeq_ado1, distance = "bray", permutations = 9999)
ado18 <- adonis(aem_veg ~ Age, aem_physeq_ado1, distance = "bray", permutations = 9999)
ado19 <- adonis(aem_veg ~ Species, aem_physeq_ado1, Strata = Age, distance = "bray", permutations = 9999)

ado20 <- adonis(aem_veg ~ Age, aem_physeq_ado1, Strata = Species, distance = "bray", permutations = 9999)
ado21 <- adonis(aem_veg ~ Species*Age, aem_physeq_ado1, distance = "bray", permutations = 9999)

#ado4#p=0.1257

#one-way PERMANOVA models for L.africana and L.cyclotis (only random effects of age and sex, no interaction)
##################################################################################################################

ado_species_age <- adonis(aem_veg ~ Species, aem_physeq_ado1, Strata = Age, distance = "bray", permutations = 9999)

ado_species_sex <- adonis(aem_veg ~ Species, aem_physeq_ado1, Strata = Sex, distance = "bray", permutations = 9999)

###################################################################################################################

#two-way interaction models for L. africana, including random effects of age and sex
ado_diet_hab_age <- adonis(aem_veg2 ~ Raider + Habitat, aem_physeq_ado2, Strata = Age, distance = "bray", permutations = 9999)
    #diet = 0.007
    #habitat = 0.0001
    #interaction = 0.095

ado_diet_hab_sex <- adonis(aem_veg2 ~ Raider + Habitat, aem_physeq_ado2, Strata = Sex, distance = "bray", permutations = 9999)
    #diet = 0.007
    #habitat = 0.0001
    #interaction = 0.099


#PERMANOVA INDIVIDUAL RESULTS

#ado1 #Species
#ado2 #Diet
#ado3 #Habitat
#ado4 #Age within L. africana
#ado5 #Sex within L. africana
#ado6 #Diet by habitat interaction within L.africana
#ado7 #Diet by Sex interaction within L. africana
#ado8 #Diet by Age interaction within L. africana
#ado9 #Habitat by Sex interaction within L. africana
#ado10 #Habitat by Age interaction within L. africana
#ado11 #Diet with Habitat as a random effect within L. africana
#ado12 #Habitat with Diet as a random effect within L. africana
#ado13 #Diet with Sex as a random effect within L. africana
#ado14 #Diet with Age as a random effect within L. africana
#ado15 #Habitat with Sex as a random effect within L. africana
#ado16 #Habitat with Age as a random effect within L. africana

#ado17 #Sex between Species
#ado18 #Age between Species
#ado19 #Species with Age as a random effect within species
#ado20 #Age with Species as a random effect within species
#ado21 #Species by Age interaction


#Put individual PERMANOVA results into a single table
ado_p_vals <- data.frame(numeric(16))
ado_p_vals[1,1] <- ado1$aov.tab[1,6]
ado_p_vals[2,1] <- ado2$aov.tab[1,6]
ado_p_vals[3,1] <- ado3$aov.tab[1,6]
ado_p_vals[4,1] <- ado4$aov.tab[1,6]
ado_p_vals[5,1] <- ado5$aov.tab[1,6]
ado_p_vals[6,1] <- ado6$aov.tab[1,6]
ado_p_vals[7,1] <- ado7$aov.tab[1,6]
ado_p_vals[8,1] <- ado8$aov.tab[1,6]
ado_p_vals[9,1] <- ado9$aov.tab[1,6]
ado_p_vals[10,1] <- ado10$aov.tab[1,6]
ado_p_vals[11,1] <- ado11$aov.tab[1,6]
ado_p_vals[12,1] <- ado12$aov.tab[1,6]
ado_p_vals[13,1] <- ado13$aov.tab[1,6]
ado_p_vals[14,1] <- ado14$aov.tab[1,6]
ado_p_vals[15,1] <- ado15$aov.tab[1,6]
ado_p_vals[16,1] <- ado16$aov.tab[1,6]
ado_p_vals <- as.data.frame(ado_p_vals)
ado_p_vals <- ado_p_vals %>% rownames_to_column("model")
colnames(ado_p_vals) <- c("model", "p_value")

```

# NMDS for Abundance data

##Visualize Nonlinear Multi-Dimensional Scaling (NMDS)
```{r}
#####Run NMDS######################################################
vare.mds0 <- isoMDS(aem_veg, trace = 0)
vare.mds <- metaMDS(aem_veg_data, trace = FALSE, autotransform = F)
score <- scores(vare.mds)
score <- as.data.frame(score)
score <- score %>% rownames_to_column("sample_id")
merge <- merge(metadata, score, by = "sample_id")
##################################################################

levels(merge$Species) <- c("L. africana", "L. cyclotis")
centroids <- aggregate(cbind(NMDS1,NMDS2)~Species,merge,mean)
gg <- merge(merge,aggregate(cbind(mean.x=NMDS1,mean.y=NMDS2)~Species,merge,mean),by="Species")

species_perm <- ggplot(gg, aes(NMDS1, NMDS2, color = Species)) + geom_point(size=1) + geom_point(aes(x=mean.x,y=mean.y),size=1) + geom_segment(aes(x=mean.x, y=mean.y, xend=NMDS1, yend=NMDS2)) + theme(legend.position=c(0.01,0.85)) + scale_color_manual(values = c("red2", "blue2")) + labs(color = "Elephant Species", title = "Beta Diversity: OTU Abundance", y = "NMDS 2") + theme(axis.title.x = element_blank()) + theme(legend.text = element_text(face = "italic"))


#PLOT NMDS BY DIET

#####Run NMDS######################################################
vare.mds1 <- isoMDS(aem_veg2, trace = 0)
vare.mds2 <- metaMDS(aem_veg2_data, trace = FALSE, autotransform = F)
score_diet <- scores(vare.mds2)
score_diet <- as.data.frame(score_diet)
score_diet <- score_diet %>% rownames_to_column("sample_id")
merge_diet <- merge(metadata, score_diet, by = "sample_id")
##################################################################

#Plot
levels(merge_diet$Raider) <- c("Non-Crop-Raider", "Crop-Raider")
centroids_diet <- aggregate(cbind(NMDS1,NMDS2)~Raider,merge_diet,mean)

gg_diet <- merge(merge_diet,aggregate(cbind(mean.x=NMDS1,mean.y=NMDS2)~Raider,merge_diet,mean),by="Raider")

diet_perm <- ggplot(gg_diet, aes(NMDS1, NMDS2, color = Raider)) + geom_point(size=1)+
  geom_point(aes(x=mean.x,y=mean.y),size=1)+
  geom_segment(aes(x=mean.x, y=mean.y, xend=NMDS1, yend=NMDS2)) + theme(legend.position = c(0.01,0.85)) + scale_color_manual(values = c("midnightblue", "deeppink1")) + theme(axis.title.x = element_blank()) + labs(y = "NMDS 2", color = "Elephant Diet")


#######################################################################################################
#PLOT NMDS BY HABITAT

#####Run NMDS######################################################
vare.mds_hab <- isoMDS(aem_veg2, trace = 0)
vare.mds_hab2 <- metaMDS(aem_veg2_data, trace = FALSE, autotransform = F)
score_hab <- scores(vare.mds_hab2)
score_hab <- as.data.frame(score_hab)
score_hab <- score_hab %>% rownames_to_column("sample_id")
merge_hab <- merge(metadata, score_hab, by = "sample_id")
##################################################################

#Plot
centroids_hab <- aggregate(cbind(NMDS1,NMDS2)~Habitat,merge_hab,mean)

gg_hab <- merge(merge_hab,aggregate(cbind(mean.x=NMDS1,mean.y=NMDS2)~Habitat,merge_hab,mean),by="Habitat")

habitat_perm <- ggplot(gg_hab, aes(NMDS1, NMDS2, color = Habitat)) + geom_point(size=1)+
  geom_point(aes(x=mean.x,y=mean.y),size=1)+
  geom_segment(aes(x=mean.x, y=mean.y, xend=NMDS1, yend=NMDS2)) + theme(legend.position = c(0.01,0.85)) + scale_color_manual(values = c("forestgreen", "coral1")) + labs(color = "Elephant Habitat", x = "NMDS 1", y = "NMDS 2")


#######################################################################################
#PLOT EVERYTHING TOGETHER

jpeg("/Users/joegunn/Desktop/Grad_School_Stuff/Projects/Elephant_Microbiome/NEW_STATS/AEMB_FINAL_STATS/African_Elephant_Microbiome/beta_diversity.jpeg", width=1800, height=3700, units='px', res=400) 

plot_grid(species_perm, diet_perm, habitat_perm, nrow = 3, labels = c("A", "B", "C"))

dev.off()
```

# ALPHA DIVERSITY

## Alpha Diversity - Shannon Diversity Index
```{r}


#####Analysis##########################################
shan_div_species <- diversity(aem_veg_data, "shannon")
#######################################################
shan_div_diet <- diversity(aem_veg2_data, "shannon")
#######################################################

#Prepare data for linear models - SPECIES
shan_div_species <- as.data.frame(shan_div_species)
shan_div_species <- shan_div_species %>% rownames_to_column("sample_id")
colnames(shan_div_species) <- c("sample_id", "shannon")
shan_div_species <- merge(metadata, shan_div_species, by = "sample_id")

#Prepare data for linear models - DIET
shan_div_diet <- as.data.frame(shan_div_diet)
shan_div_diet <- shan_div_diet %>% rownames_to_column("sample_id")
colnames(shan_div_diet) <- c("sample_id", "shannon")
shan_div_diet <- merge(metadata, shan_div_diet, by = "sample_id")

#Averages and standard deviations of Shannon Diversity for species, diet, and habitat
shan_div_diet_aves <- shan_div_diet %>% group_by(Raider) %>% summarize(mean = mean(shannon))
shan_div_diet_sd <- shan_div_diet %>% group_by(Raider) %>% summarize(sd = sd(shannon))
shan_div_habitat_aves <- shan_div_diet %>% group_by(Habitat) %>% summarize(mean = mean(shannon))
shan_div_habitat_sd <- shan_div_diet %>% group_by(Habitat) %>% summarize(sd = sd(shannon))
shan_div_species_aves <- shan_div_species %>% group_by(Species) %>% summarize(mean = mean(shannon))
shan_div_species_sd <- shan_div_species %>% group_by(Species) %>% summarize(sd = sd(shannon))


###SPECIES 

#Linear Models of alpha diversity with Species data set
lm1 <- lm(shannon ~ Species, data = shan_div_species) #linear model by species alone
lm2 <- lm(shannon ~ Species*Sex, data = shan_div_species) #linear model with sex and species interaction (two-way ANOVA)
lm3 <- lm(shannon ~ Species*Age, data = shan_div_species) #linear model with age and sex interaction
lm_sex_species <- lm(shannon ~ Sex, data = shan_div_species)  #p = 0.821
lm_age_species <- lm(shannon ~ Age, data = shan_div_species)  #p = 0.588


###DIET

#Linear models of alpha diversity with diet and habitat data
lmA <- lm(shannon ~ Raider, data = shan_div_diet) #NOT SIGNIFICANT 
lmB <- lm(shannon ~ Habitat, data = shan_div_diet) #NOT SIGNIFICANT
lmC <- lm(shannon ~ Raider*Habitat, data = shan_div_diet)

#Various linear mixed effect models with different random effects just for fun
lme_species_age <- lme(shannon ~ Species, random = ~1|Age, data = shan_div_species)
lme_species_sex <- lme(shannon ~ Species, random = ~1|Sex, data = shan_div_species)

lme_int_diet_hab_age <- lme(shannon ~ Raider*Habitat, random = ~1|Age, data = shan_div_diet)
lme_int_diet_hab_sex <- lme(shannon ~ Raider*Habitat, random = ~1|Sex, data = shan_div_diet)
  
lme1 <- lme(shannon ~ Raider, random = ~1|Habitat, data = shan_div_diet) #$tTable
lme2 <- lme(shannon ~ Habitat, random = ~1|Raider, data = shan_div_diet)
lme3 <- lme(shannon ~ Sex, random = ~1|Raider, data = shan_div_diet)
lme4 <- lme(shannon ~ Sex, random = ~1|Habitat, data = shan_div_diet)

lm_sex <- lm(shannon ~ Sex, data = shan_div_diet)
lm_age <- lm(shannon ~ Age, data = shan_div_diet)

#summary(lm_age) # p = 0.885

```

## Visualize Alpha Diversity. We can add a lot of graphs here, and we can compare all kinds of groups within groups.
```{r}

levels(shan_div_species$Species) <- c("L. africana", "L cyclotis")

alpha_species <- ggplot(shan_div_species, aes(x = Species, y = shannon, fill = Species)) + geom_boxplot(outlier.shape = NA, show.legend = F) + labs(x = "Species", y = "Shannon Diversity") + scale_fill_manual(values = c("red", "blue")) + theme(axis.text.x = element_text(face = "italic"))

levels(shan_div_diet$Raider) <- c("Non-Crop-Raider", "Crop-Raider")

alpha_diet <- ggplot(shan_div_diet, aes(x = Raider, y = shannon, fill = Raider)) + geom_boxplot(outlier.shape = NA, show.legend = F) + labs(x = "Diet", y = "Shannon Diversity") + scale_fill_manual(values = c("midnightblue", "deeppink1"))

alpha_habitat <- ggplot(shan_div_diet, aes(x = Habitat, y = shannon, fill = Habitat)) + geom_boxplot(outlier.shape = NA, show.legend = F) + labs(x = "Habitat", y = "Shannon Diversity") + scale_fill_manual(values = c("forestgreen", "coral1"))
#Print the Figure

jpeg("/Users/joegunn/Desktop/Grad_School_Stuff/Projects/Elephant_Microbiome/NEW_STATS/AEMB_FINAL_STATS/African_Elephant_Microbiome/alpha_diversity.jpeg", width=1800, height=3700, units='px', res=400) 

plot_grid(alpha_species, alpha_diet, alpha_habitat, nrow = 3, labels = c("A", "B", "C"))

dev.off()


#Include Sex in Habitat and Diet Alpha diversity graphs, becasue sex is significant in some of the variables. This will be a supplementary figure.

#Make the Figures

alpha_diet_sex <- ggplot(shan_div_diet, aes(x = Raider, y = shannon, color = Sex)) + geom_boxplot(outlier.shape = NA) + labs(x = "Diet", y = "Shannon Diversity") + scale_color_manual(values = c("magenta", "mediumseagreen"))

alpha_habitat_sex <- ggplot(shan_div_diet, aes(x = Habitat, y = shannon, color = Sex)) + geom_boxplot(outlier.shape = NA) + labs(x = "Habitat", y = "Shannon Diversity") + scale_color_manual(values = c("magenta", "mediumseagreen"))

#Print the Figures

jpeg("/Users/joegunn/Desktop/Grad_School_Stuff/Projects/Elephant_Microbiome/NEW_STATS/AEMB_FINAL_STATS/African_Elephant_Microbiome/alpha_by_sex.jpeg", width=3700, height=3700, units='px', res=500) 

plot_grid(alpha_diet_sex, alpha_habitat_sex, nrow = 2, labels = c("A", "B"))

dev.off()

```

# METABOLIC ANALYSIS

## KEGG PATHWAYS
```{r}

#Metadata by species (kegg_1_species) and by habitat and range (kegg_1_africana) for KEGG Pathways at level 1 metabolism 
kegg_1_all <- read_excel("kegg_1.xlsx")

kegg_1_meta <- kegg_1_all[,-1]
kegg_1_meta <- kegg_1_meta %>% group_by(Metabolism_1) %>% summarize_all(funs(sum))
kegg_1_meta <- as.data.frame(kegg_1_meta)
kegg_1_meta <- column_to_rownames(kegg_1_meta, "Metabolism_1")
kegg_1_meta <- t(kegg_1_meta)
kegg_1_meta <- as.data.frame(kegg_1_meta)
kegg_1_meta <- rownames_to_column(kegg_1_meta, "sample_id")
kegg_1_meta <- merge(metadata, kegg_1_meta, by = "sample_id")
kegg_1_species <- kegg_1_meta %>% filter(Habitat == "Forest") %>% filter(Raider == "No")
kegg_1_africana <- kegg_1_meta %>% filter(Species == "africana")
kegg_1_species <- column_to_rownames(kegg_1_species, "sample_id")

#Preparing Vegan matrix for both datasets
kegg_1 <- kegg_1_all[,-1] 
kegg_1 <- kegg_1 %>% group_by(Metabolism_1) %>% summarize_all(funs(sum))
kegg_1 <- as.data.frame(kegg_1)
kegg_1 <- column_to_rownames(kegg_1, "Metabolism_1")
kegg_1 <- t(kegg_1)
kegg_1 <- as.data.frame(kegg_1)
kegg_1 <- rownames_to_column(kegg_1, "sample_id")
kegg_1 <- merge(metadata, kegg_1, by = "sample_id")
kegg_1_by_species <- kegg_1 %>% filter(Habitat == "Forest") %>% filter(Raider == "No")
kegg_1_by_species <- kegg_1_by_species[,-c(2:11)]
kegg_1_by_species <- column_to_rownames(kegg_1_by_species, "sample_id")
kegg_1_by_africana <- kegg_1 %>% filter(Species == "africana")
kegg_1_by_africana <- kegg_1_by_africana[,-c(2:11)]
kegg_1_by_africana <- column_to_rownames(kegg_1_by_africana, "sample_id")

#####BETA DIVERSITY########################################################

#Run Vegan
kegg_1_veg_species_stand <- decostand(kegg_1_by_species, "total")
kegg_1_veg_species <- vegdist(kegg_1_veg_species_stand, "bray")

kegg_1_veg_africana_stand <- decostand(kegg_1_by_africana, "total")
kegg_1_veg_africana <- vegdist(kegg_1_veg_africana_stand, "bray")

#Beta Diversity with PERMANOVA
ado_kegg_species_age <- adonis(kegg_1_veg_species ~ Species, kegg_1_species, Strata = Age, distance = "bray", permutations = 9999)
ado_kegg_species_sex <- adonis(kegg_1_veg_species ~ Species, kegg_1_species, Strata = Sex, distance = "bray", permutations = 9999)
ado_kegg_species_age
ado_kegg_diet_hab_age <- adonis(kegg_1_veg_africana ~ Raider*Habitat, kegg_1_africana, Strata = Age, distance = "bray", permutations = 9999)
ado_kegg_diet_hab_sex <- adonis(kegg_1_veg_africana ~ Raider*Habitat, kegg_1_africana, Strata = Sex, distance = "bray", permutations = 9999)


kegg_1_props_species <- read_excel("kegg_1_props_species.xlsx")
kegg_1_props_species <- as.data.frame(kegg_1_props_species)
kegg_1_props_species <- kegg_1_props_species %>% mutate(Species = factor(Species), Raider = factor(Raider), Habitat = factor(Habitat), Age = factor(Age), Sex = factor(Sex), Elephant = factor(Elephant), Group = factor(Group), Description = factor(Description))

kegg_1_props_africana <- read_excel("kegg_1_props_africana.xlsx")
kegg_1_props_africana <- kegg_1_props_africana %>% as.data.frame() %>% mutate(Species = factor(Species), Raider = factor(Raider), Habitat = factor(Habitat), Age = factor(Age), Sex = factor(Sex), Elephant = factor(Elephant), Group = factor(Group), Description = factor(Description))

#By Species
p_vals_species <- data.frame("A" = numeric(1),  "B" = numeric(1),  "C" = numeric(1), 
                              "D" = numeric(1),  "E" = numeric(1),  "F" = numeric(1),
                              "G" = numeric(1),  "H" = numeric(1),  "I" = numeric(1), 
                              "J" = numeric(1),  "K" = numeric(1))

#iterate by column index of the data frame

options(scipen = 999) #this removes scientific notation from p-values, makes them easier to read

    for (ii in 12:ncol(kegg_1_props_species)) {
  
      col = kegg_1_props_species[,ii] #this tells the for loop that it should iterate through       columns in the data set
  
      lm_temp <- lm(col ~ Species, data = kegg_1_props_species) #tells the for loop to run         linear model  for each column specified above
  
      p_value <- summary(lm_temp)$coefficients[2,4] #extracts p value
  
      p_vals_species[,ii-11] <- p_value #stores p value in data frame constructed above. Need to       subtract 11 from the iterator so that the loop            doesn't fill starting at row 12
    }


#By habitat
p_vals_habitat <- data.frame("A" = numeric(1),  "B" = numeric(1),  "C" = numeric(1), 
                              "D" = numeric(1),  "E" = numeric(1),  "F" = numeric(1),
                              "G" = numeric(1),  "H" = numeric(1),  "I" = numeric(1), 
                              "J" = numeric(1),  "K" = numeric(1))

#iterate by column index of the data frame

options(scipen = 999) #this removes scientific notation from p-values, makes them easier to read

    for (ii in 12:ncol(kegg_1_props_africana)) {
  
      col = kegg_1_props_africana[,ii] #this tells the for loop that it should iterate through       columns in the data set
  
      lm_temp <- lm(col ~ Habitat, data = kegg_1_props_africana) #tells the for loop to run         linear model  for each column specified above
  
      p_value <- summary(lm_temp)$coefficients[2,4] #extracts p value
  
      p_vals_habitat[,ii-11] <- p_value #stores p value in data frame constructed above. Need to       subtract 11 from the iterator so that the loop            doesn't fill starting at row 12
    }

p_vals_habitat_fdr <- p.adjust(p_vals_habitat, method = "fdr")


#By Diet
p_vals_diet <- data.frame("A" = numeric(1),  "B" = numeric(1),  "C" = numeric(1), 
                              "D" = numeric(1),  "E" = numeric(1),  "F" = numeric(1),
                              "G" = numeric(1),  "H" = numeric(1),  "I" = numeric(1), 
                              "J" = numeric(1),  "K" = numeric(1))

options(scipen = 999) #this removes scientific notation from p-values, makes them easier to read

    for (ii in 12:ncol(kegg_1_props_africana)) {
  
      col = kegg_1_props_africana[,ii] #this tells the for loop that it should iterate through       columns in the data set
  
      lm_temp <- lm(col ~ Raider, data = kegg_1_props_africana) #tells the for loop to run         linear model  for each column specified above
  
      p_value <- summary(lm_temp)$coefficients[2,4] #extracts p value
  
      p_vals_diet[,ii-11] <- p_value #stores p value in data frame constructed above. Need to       subtract 11 from the iterator so that the loop            doesn't fill starting at row 12
    }

p_vals_diet_fdr <- p.adjust(p_vals_diet, method = "fdr")

##Species was the only factor that contained any signficant pathways, so we chose to only plot the metabolic pathways for this analysis. 

p_vals_species_fdr <- p.adjust(p_vals_species, method ="fdr")
p_vals_species_fdr <- as.data.frame(p_vals_species_fdr)
p_vals_species_fdr <- t(p_vals_species_fdr)
p_vals_species_fdr <- t(p_vals_species_fdr)
p_vals_species_fdr <- as.data.frame(p_vals_species_fdr)
p_vals_species_fdr <- rownames_to_column(p_vals_species_fdr, "code")

kegg_codes <- read_excel("kegg_codes.xlsx")
p_vals_species_fdr <- merge(kegg_codes, p_vals_species_fdr, by = "code")
```

## Plot metabolic pathways at KEGG Level 1 for species
```{r}

#NOTE: After we compiled data individual metabolic proportions across species, we exported the data into Excel and prepared a file organizing proportion by species and by metabolic pathway to save time (and to avoid long R code). We then used this re-organized data to caclucate averages and standard deviations between species for all pathways in R, and then exported this data again to prepare it for plotting in R.

kegg_species_met1 <- read_excel("kegg_1_species_prop_graph.xlsx")
kegg_species_met1 <- kegg_species_met1 %>% mutate(species = factor(species))
levels(kegg_species_met1$species) <- c("L. africana", "L. cyclotis")

jpeg("/Users/joegunn/Desktop/Grad_School_Stuff/Projects/Elephant_Microbiome/NEW_STATS/AEMB_FINAL_STATS/African_Elephant_Microbiome/kegg_level1_species.jpeg", width=6000, height=4000, units='px', res=300) 


ggplot(kegg_species_met1, aes(x = pathway, y = mean, fill = species)) + geom_bar(position=position_dodge(), stat="identity", show.legend = T) + geom_errorbar(aes(ymin=mean-sd, ymax=mean+sd), width=.4, position=position_dodge(.9)) + scale_fill_manual(values = c("red2", "blue2")) + theme(axis.text.x = element_text(angle = 0, hjust = 1)) + theme(axis.text = element_text(size = 20)) + theme(axis.title = element_text(size = 30)) + theme(legend.position = c(0.6,0.9)) + labs(fill = "Elephant Species", x = "Metabolic Pathway", y = "Mean Metabolic Contribution") + theme(legend.text = element_text(size = 30)) + theme(legend.title = element_text(size = 30)) + theme(legend.text = element_text(size = 30)) + theme(legend.text = element_text(face = "italic")) + coord_flip() + theme(axis.text.y = element_text(size = 30)) + theme(axis.title = element_text(size = 30))

dev.off()
```

