---
title: "elephants_PHYLUM_ABUNDANCE"
author: "Joe Gunn"
date: "8/12/2019"
output: html_document
---

# AFRICAN ELEPHANT MICROBIOME - PHYLUM ABUNDANCE

Purpose: Here we are determining the phyla that are most abundant among African Elephant individuals across the entire dataset. We are then considering the three most abundant phyla and determing their respective abundances (raw sums) across all samples.   

Data used:

      TABLE output from QIIME: all OTUs detected across African Elephant sampels, taxonomic level, and raw abundance for each individual <- 8,248 OTUs (after rarefaction) 

## Libraries needed for analysis
```{r libraries, echo = FALSE}

library(readxl)
library(cowplot)
library(tidyverse)
library(qiime2R)
library(vegan)
library(devtools)
library(DescTools)
library(factoextra)
library(ade4)
library(lme4)
library(nlme)
library(ggrepel)
library(GGally)
library(lattice)
library(epiDisplay)
library(multcomp)
library(taRifx)
library(ape)
library(openxlsx)
library(multcomp)
library(phyloseq)
```

## Metadata
```{r Metadata, include = FALSE}
##Read in the metadata, which includes all labels and covariates associated with each individual in the data set. Metadata includes individual sample ID (index), barcode sequence, linker primer sequence, species (africana or cyclotis), diet (crop raider or non-crop raider), habitat (forest or savanna), age, sex, and group (a descriptor for each unique combination of species-diet-habitat)

metadata <- read_tsv("../data/metadata/METADATA.tsv") #metadata read in

metadata <- metadata %>% 
  as.data.frame() %>% 
  mutate(BarcodeSequence = factor(BarcodeSequence), LinkerPrimerSequence = factor(LinkerPrimerSequence), Species = factor(Species), Raider = factor(Raider), Habitat = factor(Habitat), Age = factor(Age), Sex = factor(Sex), Elephant = factor(Elephant), Group = factor(Group), Description = factor(Description)) #convert table to data frame and convert all characters, except for sample id, into factors

colnames(metadata)[colnames(metadata) == "Sample-id"] <- "sample_id" 
```

## Get Rarefied OTU abundance data 
```{r}
options(scipen = 999) #gets rid of scientific notatation, useful for reading p-values

#read in taxonomic designations and OTU abundance
aem_physeq <- qza_to_phyloseq(features = "../data/qiime_data/table.qza", taxonomy = "../data/qiime_data/taxonomy.qza",  tree = "../data/qiime_data/rooted-tree.qza", metadata = "../data/metadata/METADATA.tsv") 

#omit sample OB182 due to low read count. We don't want to rarefy to such a low number of reads across the whole dataset
aem_physeq <- subset_samples(aem_physeq, Elephant != "OB182") 

#Rarefy and clean the data set for use in downstream analysis
aem_physeq <- rarefy_even_depth(aem_physeq, rngsee = 5) #must set seed in order to maintain the same rarefied number of otus

    #Number of OTUS: 8248
    #Number of reads per sample: 11460 
      
#Extract data of all types (meta data, all taxonomy data)
aem_meta <- as.data.frame(sample_data(aem_physeq))
aem_tax <- as.data.frame(tax_table(aem_physeq))
aem_tax <- aem_tax %>% 
  rownames_to_column("otu")

aem_data <- as.data.frame(otu_table(aem_physeq))
aem_data <- aem_data %>% 
  rownames_to_column("otu")

aem_tax_data <- merge(aem_tax, aem_data, by = "otu")

```

## Get Most Abundant Phyla
```{r}
#Subset the data by the three most abundant phyla: Bacteroidetes, firmicutes, proteobacteria
aem_bacteroidetes <- aem_tax_data %>% 
  filter(Phylum == "p__Bacteroidetes")

aem_firmicutes <- aem_tax_data %>% 
  filter(Phylum == "p__Firmicutes")

aem_proteobacteria <- aem_tax_data %>% 
  filter(Phylum == "p__Proteobacteria")
```

## Clean the phylum-level dataset
```{r}

#Summarize Rarefied OTU abundance by ALL Phyla
aem_phylum_data <- aem_tax_data[,-c(1:2,4:8)]
aem_phylum_data <- as.data.frame(aem_phylum_data)
aem_phylum_data <- aem_phylum_data %>% 
  group_by(Phylum) %>% 
  summarize_all(funs(sum))

aem_phylum_data <- aem_phylum_data %>% 
  drop_na()

aem_phylum_data <- column_to_rownames(aem_phylum_data, "Phylum")

aem_phylum_data_t <- t(aem_phylum_data)
aem_phylum_data_t <- as.data.frame(aem_phylum_data_t)
aem_phylum_data_t <- aem_phylum_data_t %>% 
  rownames_to_column("sample_id")

aem_phylum <- merge(metadata, aem_phylum_data_t, by = "sample_id")
```

## Phylum and Order Proportions per African Elephant Group
```{r}
#Proportions of each order contributing to each african elephant group
aem_order_phylum_data <- aem_tax_data[,-c(1:2,4,6:8)]
aem_order_phylum_data <- as.tibble(aem_order_phylum_data)
aem_order_phylum_data <- aem_order_phylum_data %>% 
  group_by(Phylum, Order) %>% 
  summarize_all(funs(sum))
```
