---
title: "elephants_GENERA_ABUNDANCE"
author: "Joe Gunn"
date: "8/13/2019"
output: html_document
---

# AFRICAN ELEPHANT MICROBIOME - GENERA ABUNDANCE

Purpose: Here we determine OTU abundance among African Elephant individuals across the entire dataset at the microbial GENUS level. We then assess differences in abundance among African Elephant species and "groups" using general linear models (with a negative binomial distribution on zero-inflated count data). Groups include all combinations of diet and habitat so that all combinations can be compared.

Data used:

      TABLE output from QIIME: all OTUs detected across African Elephant sampels, taxonomic level, and raw abundance for each individual <- 8,248 OTUs (after rarefaction) 

## Libraries
```{r libraries, echo = FALSE}

library(readxl)
library(cowplot)
library(tidyverse)
library(qiime2R)
library(vegan)
library(devtools)
library(DescTools)
library(factoextra)
library(ade4)
library(lme4)
library(nlme)
library(ggrepel)
library(GGally)
library(lattice)
library(epiDisplay)
library(multcomp)
library(taRifx)
library(ape)
library(openxlsx)
library(multcomp)
library(phyloseq)
library(lmerTest)
library(car)
```

## Metadata
```{r Metadata}
##Read in the metadata, which includes all labels and covariates associated with each individual in the data set. Metadata includes individual sample ID (index), barcode sequence, linker primer sequence, species (africana or cyclotis), diet (crop raider or non-crop raider), habitat (forest or savanna), age, sex, and group (a descriptor for each unique combination of species-diet-habitat)

metadata <- read_tsv("../data/metadata/METADATA.tsv") #metadata read in

metadata <- metadata %>% 
  as.data.frame() %>% 
  mutate(BarcodeSequence = factor(BarcodeSequence), LinkerPrimerSequence = factor(LinkerPrimerSequence), Species = factor(Species), Raider = factor(Raider), Habitat = factor(Habitat), Age = factor(Age), Sex = factor(Sex), Elephant = factor(Elephant), Group = factor(Group), Description = factor(Description)) #convert table to data frame and convert all characters, except for sample id, into factors
colnames(metadata)[colnames(metadata) == "Sample-id"] <- "sample_id" 
```

## Rarefied OTU abundance data 
```{r, echo = FALSE}
options(scipen = 999) #gets rid of scientific notatation, useful for reading p-values

#read in taxonomic designations and OTU abundance
aem_physeq <- qza_to_phyloseq(features = "../data/qiime_data/table.qza", taxonomy = "../data/qiime_data/taxonomy.qza",  tree = "../data/qiime_data/rooted-tree.qza", metadata = "../data/metadata/METADATA.tsv") 

#omit sample OB182 due to low read count. We don't want to rarefy to such a low number of reads across the whole dataset
aem_physeq <- subset_samples(aem_physeq, Elephant != "OB182") 

#Rarefy and clean the data set for use in downstream analysis
aem_physeq <- rarefy_even_depth(aem_physeq, rngsee = 5) #must set seed in order to maintain the same rarefied number of otus

    #Number of OTUS: 8248
    #Number of reads per sample: 11460 
      
#Extract data of all types (meta data, all taxonomy data)
aem_meta <- as.data.frame(sample_data(aem_physeq))
aem_tax <- as.data.frame(tax_table(aem_physeq))
aem_tax <- aem_tax %>% 
  rownames_to_column("otu")

aem_data <- as.data.frame(otu_table(aem_physeq))
aem_data <- aem_data %>% 
  rownames_to_column("otu")

aem_tax_data <- merge(aem_tax, aem_data, by = "otu")
```

## Subset the data by the three most abundant phyla: Bacteroidetes, firmicutes, proteobacteria
```{r}
aem_bacteroidetes <- aem_tax_data %>% 
  filter(Phylum == "p__Bacteroidetes")

aem_firmicutes <- aem_tax_data %>% 
  filter(Phylum == "p__Firmicutes")

aem_proteobacteria <- aem_tax_data %>% 
  filter(Phylum == "p__Proteobacteria")
```

## Statistical Signifiance of Genera within Orders Bacillales and Lactobacillales By African Elephant Species
```{r}
#Now we want to find the genera within each of the significant orders (Bacillales and Lactobacillales within Firmicutes) and see which are significantly different between africana and cyclotis species
    
#Data frames with genera only within the significant orders for african elephant SPECIES
aem_firm_bacillales <- aem_firmicutes %>% 
  filter(Order == "o__Bacillales")

aem_firm_lactobacillales <- aem_firmicutes %>% 
  filter(Order == "o__Lactobacillales")

#Data frames with genera only within the significant orders for african elephant GROUPS
aem_bact_flavobacteriales <- aem_bacteroidetes %>% 
  filter(Order == "o__Flavobacteriales")

aem_proteo_pseudomonadales <- aem_proteobacteria %>% 
  filter(Order == "o__Pseudomonadales")

#Extract Genera from Bacillales and Lactobacillales within Firmicutes (SPECIES)

    #Bacillales
aem_firm_bacillales_genera <- aem_firm_bacillales[,-c(1:6,8)]
aem_firm_bacillales_genera <- as.data.frame(aem_firm_bacillales_genera)
aem_firm_bacillales_genera <- aem_firm_bacillales_genera %>% 
    group_by(Genus) %>% 
    summarize_all(funs(sum))
    
aem_firm_bacillales_genera <- aem_firm_bacillales_genera %>% 
    drop_na()
aem_firm_bacillales_genera <- column_to_rownames(aem_firm_bacillales_genera, "Genus")
aem_firm_bacillales_genera_t <- t(aem_firm_bacillales_genera)
aem_firm_bacillales_genera_t <- as.data.frame(aem_firm_bacillales_genera_t)
aem_firm_bacillales_genera_t <- aem_firm_bacillales_genera_t %>% 
    rownames_to_column("sample_id")
    
aem_firm_bacillales_genera_sums <- merge(metadata, aem_firm_bacillales_genera_t, by = "sample_id")
    
    #Lactobacillales
aem_firm_lactobacillales_genera <- aem_firm_lactobacillales[,-c(1:6,8)]
aem_firm_lactobacillales_genera <- as.data.frame(aem_firm_lactobacillales_genera)
aem_firm_lactobacillales_genera <- aem_firm_lactobacillales_genera %>% 
    group_by(Genus) %>% 
    summarize_all(funs(sum))
    
aem_firm_lactobacillales_genera <- aem_firm_lactobacillales_genera %>% 
    drop_na()
aem_firm_lactobacillales_genera <- column_to_rownames(aem_firm_lactobacillales_genera, "Genus")
aem_firm_lactobacillales_genera_t <- t(aem_firm_lactobacillales_genera)
aem_firm_lactobacillales_genera_t <- as.data.frame(aem_firm_lactobacillales_genera_t)
aem_firm_lactobacillales_genera_t <- aem_firm_lactobacillales_genera_t %>% 
    rownames_to_column("sample_id")
    
aem_firm_lactobacillales_genera_sums <- merge(metadata, aem_firm_lactobacillales_genera_t, by = "sample_id")

#Extract Genera from Flavobacteriales within Bacteroidetes (GROUPS)
#Flavobacteriales
aem_bact_flavobacteriales_genera <- aem_bact_flavobacteriales[,-c(1:6,8)]
aem_bact_flavobacteriales_genera <- as.data.frame(aem_bact_flavobacteriales_genera)
aem_bact_flavobacteriales_genera <- aem_bact_flavobacteriales_genera %>% 
    group_by(Genus) %>% 
    summarize_all(funs(sum))
    
aem_bact_flavobacteriales_genera <- aem_bact_flavobacteriales_genera %>% 
    drop_na()
    
aem_bact_flavobacteriales_genera <- column_to_rownames(aem_bact_flavobacteriales_genera, "Genus")
aem_bact_flavobacteriales_genera_t <- t(aem_bact_flavobacteriales_genera)
aem_bact_flavobacteriales_genera_t <- as.data.frame(aem_bact_flavobacteriales_genera_t)
aem_bact_flavobacteriales_genera_t <- aem_bact_flavobacteriales_genera_t %>% 
    rownames_to_column("sample_id")
    
aem_bact_flavobacteriales_genera_sums <- merge(metadata, aem_bact_flavobacteriales_genera_t, by = "sample_id")

    
#Extract Genera from Pseudomonadales within Proteobacteria (GROUPS)
#Pseudomonadales
aem_proteo_pseudomonadales_genera <- aem_proteo_pseudomonadales[,-c(1:6,8)]
aem_proteo_pseudomonadales_genera <- as.data.frame(aem_proteo_pseudomonadales_genera)
aem_proteo_pseudomonadales_genera <- aem_proteo_pseudomonadales_genera %>% 
    group_by(Genus) %>% 
    summarize_all(funs(sum))
    
aem_proteo_pseudomonadales_genera <- aem_proteo_pseudomonadales_genera %>% 
    drop_na()
aem_proteo_pseudomonadales_genera <- column_to_rownames(aem_proteo_pseudomonadales_genera, "Genus")
aem_proteo_pseudomonadales_genera_t <- t(aem_proteo_pseudomonadales_genera)
aem_proteo_pseudomonadales_genera_t <- as.data.frame(aem_proteo_pseudomonadales_genera_t)
aem_proteo_pseudomonadales_genera_t <- aem_proteo_pseudomonadales_genera_t %>% 
    rownames_to_column("sample_id")
    
aem_proteo_pseudomonadales_genera_sums <- merge(metadata, aem_proteo_pseudomonadales_genera_t, by = "sample_id")
    
#Separate genera into subsetted datasets
aem_firm_lactobacillales_genera_sums_forest_ncr <- aem_firm_lactobacillales_genera_sums %>% 
  filter(Habitat == "Forest") %>% 
  filter(Raider == "No")

aem_firm_bacillales_genera_sums_forest_ncr <- aem_firm_bacillales_genera_sums %>% 
  filter(Habitat == "Forest") %>% 
  filter(Raider == "No")

aem_bacillales_genera_by_species <- aem_firm_bacillales_genera_sums_forest_ncr[,-c(1:3,5,6,9:11)]
aem_lactobacillales_genera_by_species <- aem_firm_lactobacillales_genera_sums_forest_ncr[,-c(1:3,5,6,9:11)]


aem_bact_flavobacteriales_genera_sums_africana <- aem_bact_flavobacteriales_genera_sums %>% 
  filter(Species == "africana")

aem_proteo_pseudomonadales_genera_sums_africana <- aem_proteo_pseudomonadales_genera_sums %>% 
  filter(Species == "africana")

aem_flavobacteriales_genera_by_group <- aem_bact_flavobacteriales_genera_sums_africana[,-c(1:6,9,11)]
aem_pseudomonadales_genera_by_group <- aem_proteo_pseudomonadales_genera_sums_africana[,-c(1:6,9,11)]


colnames(aem_flavobacteriales_genera_by_group) <- c("Age", "Sex", "Group", "Chryseobacterium", "Elizabethkingia", "Flavobacterium", "Fluviicola", "Wautersiella")

colnames(aem_pseudomonadales_genera_by_group) <- c("Age", "Sex", "Group", "Acinetobacter", "Perlucidibaca", "Pseudomonas", "Serpens" )

#Calculate p-values for genera at group level and at species level
gwaut <- glmer.nb(Wautersiella ~ Group + (1|Sex), data = aem_flavobacteriales_genera_by_group)
gwaut_p <- Anova(gwaut)$`Pr(>Chisq)`

anova(gwaut)

#Post-hoc Tukey_test
post_gwaut <- glht(gwaut, mcp(Group = "Tukey"))
summary(post_gwaut)

gacin <- glmer.nb(Acinetobacter ~ Group + (1|Sex), data = aem_pseudomonadales_genera_by_group)
gacin_p <- Anova(gacin)$`Pr(>Chisq)`

anova(gacin)

#Post-hoc Tukey_test
post_gacin <- glht(gacin, mcp(Group = "Tukey"))
summary(post_gacin)

gbacillus <- glmer.nb(g__Bacillus ~ Species + (1|Sex), data = aem_bacillales_genera_by_species)
gbacillus_p <- Anova(gbacillus)$`Pr(>Chisq)`

gsolibac <- glmer.nb(g__Solibacillus ~ Species + (1|Sex), data = aem_bacillales_genera_by_species)
gsolibac_p <- Anova(gsolibac)$`Pr(>Chisq)`

anova(gsolibac)

p_vals_bacillales_species <- as.data.frame(c(gbacillus_p, gsolibac_p))
colnames(p_vals_bacillales_species) <- c("pval")
p_vals_bacillales_species_fdr <- p.adjust(p_vals_bacillales_species$pval, method = "fdr")
p_vals_bacillales_species_fdr
```

## calculate p-values for ALL GENERA
```{r}
#Bacillales
p_vals_bacillales_genera_by_species <- data.frame(numeric(length = ncol(aem_bacillales_genera_by_species)))

    for (ii in 1:ncol(aem_bacillales_genera_by_species)) {
  
      col = aem_bacillales_genera_by_species[,ii] #this tells the for loop that it should iterate through columns in the data set
      lm_temp <- lm(col ~ Species, data = aem_bacillales_genera_by_species) #tells the for loop to run linear model  for each column           specified above
      p_value <- summary(lm_temp)$coefficients[2,4] #extracts p value
      p_vals_bacillales_genera_by_species[ii,] <- p_value
    }

#Lactobacillales
p_vals_lactobacillales_genera_by_species <- data.frame(numeric(length = ncol(aem_lactobacillales_genera_by_species)))

    for (ii in 1:ncol(aem_lactobacillales_genera_by_species)) {
  
      col = aem_lactobacillales_genera_by_species[,ii] #this tells the for loop that it should iterate through columns in the data set
      lm_temp <- lm(col ~ Species, data = aem_lactobacillales_genera_by_species) #tells the for loop to run linear model  for each            column specified above
      p_value <- summary(lm_temp)$coefficients[2,4] #extracts p value
      p_vals_lactobacillales_genera_by_species[ii,] <- p_value
    }

#Correct p-values for multiple tests using false discovery rate
p_vals_bacillales_genera_by_species <- p_vals_bacillales_genera_by_species %>% 
  drop_na() %>% 
  as.data.frame()
colnames(p_vals_bacillales_genera_by_species) <- c("pvals")
p_vals_bacillales_genera_by_species
p_vals_bacillales_genera_by_species_fdr <- p.adjust(p_vals_bacillales_genera_by_species$pvals, method = "fdr")

p_vals_lactobacillales_genera_by_species <- p_vals_lactobacillales_genera_by_species %>% 
  drop_na() %>% 
  as.data.frame()
colnames(p_vals_lactobacillales_genera_by_species) <- c("pvals")
p_vals_lactobacillales_genera_by_species_fdr <- p.adjust(p_vals_lactobacillales_genera_by_species$pvals, method ="fdr")
```

## Make Box Plots for FILTERED GENERA (AVERAGE RELATIVE ABUNDANCE = 0.1 OR MORE)
## BY SPECIES
```{r}
#square root transform the y-axis
bacillales_genera_abun_rel0.1 <- read_excel("../data/excel_data/abundance/bacillales_genera_abun_rel0.1.xlsx")
bacillales_genera_abun_rel0.1 <- bacillales_genera_abun_rel0.1 %>% 
  mutate(log_abundance = log10(abundance), sqrt_abundance = sqrt(abundance))

#Plots of genera within Firmicutes Orders by African Elephant Species
bacillales_plot_rel0.1 <- ggplot(bacillales_genera_abun_rel0.1, aes(x = Genus, y = sqrt_abundance, fill = Species)) + 
  geom_boxplot(outlier.shape = NA, show.legend = T, alpha = 0.8) + 
  geom_point(position = position_dodge(width = 0.75), show.legend = F, size = 3) +
  theme(axis.text = element_text(size = 15)) + 
  theme(axis.text.x = element_text(size = 20)) + 
  labs(title = "Genera of Phylum Firmicutes: Order Bacillales", fill = "Elephant Species", x = "Genus", y = "Abundance") + 
  theme(plot.title = element_text(size = 20, hjust = 0.5, face = "bold")) + 
  scale_fill_manual(values = c("red", "blue")) + 
  theme(legend.text = element_text(face = "italic")) + 
  theme(axis.text.x = element_text(face = "italic")) + 
  theme(legend.position = c(0.2, 0.7)) + 
  theme(plot.background = element_blank())  + 
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), panel.background = element_blank(), axis.line = element_line(colour = "black")) + 
  theme(legend.title = element_text(size = 20)) + 
  theme(legend.text = element_text(size = 20)) + 
  theme(axis.title = element_text(size = 20))

#Combine plots for firmicutes
pdf("/Users/joegunn/Desktop/Grad_School_Stuff/Research/Projects/Elephant_Microbiome/Attempt_2/visualization/genera_abundance_figures/firmicutes_genera_byspecies.pdf", width=12, height=10) 

bacillales_plot_rel0.1

dev.off()
```

## Make Box Plots for FILTERED GENERA (AVERAGE RELATIVE ABUNDANCE = 0.1 OR MORE)
## BY GROUPS
```{r}
#BY GROUPS
flavobacteriales_genera_abun_rel0.1 <- read_excel("../data/excel_data/abundance/flavobacteriales_genera_abun_rel0.1.xlsx")
flavobacteriales_genera_abun_rel0.1 <- flavobacteriales_genera_abun_rel0.1 %>% mutate(log_abundance = log10(abundance), sqrt_abundance = sqrt(abundance))

pseudomonadales_genera_abun_rel0.1 <- read_excel("../data/excel_data/abundance/pseudomonadales_genera_abun_rel0.1.xlsx")
pseudomonadales_genera_abun_rel0.1 <- pseudomonadales_genera_abun_rel0.1 %>% mutate(log_abundance = log10(abundance), sqrt_abundance = sqrt(abundance))

#Flavobacteriales Plot
flavobacteriales_plot_rel0.1 <- ggplot(flavobacteriales_genera_abun_rel0.1, aes(x = Group, y = sqrt_abundance, fill = Group)) + 
  geom_boxplot(outlier.shape = NA, show.legend = F, alpha = 0.8) + 
  geom_point(position = position_dodge(width = 0.75), show.legend = F, size = 3) +
  theme(axis.text = element_text(size = 15)) + 
  theme(axis.text.x = element_text(size = 20)) + 
  labs(title = "Order Flavobacteriales: Genus Wautersiella", fill = "Elephant Group", x = "Elephant Group", y = "Abundance") + 
  theme(plot.title = element_text(size = 20, hjust = 0.5, face = "bold")) + 
  scale_fill_manual(values = c("green", "purple", "red", "goldenrod")) + 
  theme(legend.text = element_text(face = "italic")) + 
  theme(axis.text.x = element_text(face = "italic")) + 
  theme(legend.position = c(0.75, 0.8)) + 
  theme(axis.title.x = element_blank()) +
  theme(axis.text.x = element_blank()) +
  theme(plot.background = element_blank()) + 
  theme(panel.grid.major = element_blank()) +
  theme(panel.grid.minor = element_blank()) + 
  theme(panel.background = element_blank()) +
  theme(axis.line = element_line(colour = "black")) +
  theme(legend.title = element_text(size = 20)) + 
  theme(legend.text = element_text(size = 20)) + 
  theme(axis.title = element_text(size = 20))

#Pseudomonadales Plot
pseudomonadales_plot_rel0.1 <- ggplot(pseudomonadales_genera_abun_rel0.1, aes(x = Group, y = sqrt_abundance, fill = Group)) + 
  geom_boxplot(outlier.shape = NA, show.legend = F, alpha = 0.8) + 
  geom_point(position = position_dodge(width = 0.75), show.legend = F, size = 3) +
  theme(axis.text = element_text(size = 15)) + 
  theme(axis.text.x = element_text(size = 20)) + 
  labs(title = "Order Pseudomonadales: Genus Acinetobacter", fill = "Elephant Group", x = "Elephant Group", y = "Abundance") + 
  theme(plot.title = element_text(size = 20, hjust = 0.5, face = "bold")) + 
  scale_fill_manual(values = c("green", "purple", "red", "goldenrod")) + 
  theme(legend.text = element_text(face = "italic")) + 
  theme(legend.position = c(0.8, 0.8)) + 
  theme(plot.background = element_blank()) + 
  theme(panel.grid.major = element_blank()) +
  theme(panel.grid.minor = element_blank()) + 
  theme(panel.background = element_blank()) + 
  theme(axis.line = element_line(colour = "black")) + 
  theme(legend.title = element_text(size = 20)) + 
  theme(legend.text = element_text(size = 20)) + 
  theme(axis.title = element_text(size = 20))

#Combined Plot
pdf("/Users/joegunn/Desktop/Grad_School_Stuff/Research/Projects/Elephant_Microbiome/Attempt_2/visualization/genera_abundance_figures/genera_combined_bygroup.pdf", width=10, height=15) 

plot_grid(flavobacteriales_plot_rel0.1, pseudomonadales_plot_rel0.1, nrow = 2, labels = c("A", "B"), label_size = 20)

dev.off()
```