# Reproduce analyses for Budd & Gunn et al. 2020
<font size="+1">Follow the steps listed below in the <b><i>Analyses</i></b> section to reproduce analyses for Budd & Gunn et al. (2020). Each step below gives a summary of the analysis and directs you to a general code file which then works through the analysis step-by-step. This general file will usually point you to other Rmd code, bash shell scripts, or python scripts. Each analysis is contained within sub-directories of the same name in the main R project directory.</font>

<b>Find the article here</b>: <a href="url">https://onlinelibrary.wiley.com/doi/10.1002/ece3.6305</a>

<b>Citation here</b>: Budd K, Gunn JC, Finch T, Klymus K, Sitati N, Eggert LS. Effects of diet, habitat, and phylogeny on the fecal microbiome of wild African savanna (<i>Loxodonta africana</i>) and forest elephants (<i>L. cyclotis</i>). Ecol Evol. 2020;10:5637â€“5650.

## Project: Assessing effects of phylogeny, diet, and habitat on gut microbial composition in African elephants 
We characterized the Afican elephant gut microbiome (aka, A.E.G.M.), specifically of African Savanna elephants (<i>Loxodonta africana</i>) and African Forest elephants (<i>Loxodonta cyclotis</i>). We assessed the relationship of gut microbial composition, including analyses of alpha and beta diversity, with host phylogeny (i.e., species) and habitat type (i.e., forest or savanna) for both species. We also assessed the relationship between microbial composition and diet (i.e., crop-raiding vs. non-crop-raiding) within <i>L. africana</i>. 

## General information on repository structure
This is a publicly visible GitHub repository storing code (and a small amount of data, although we have done our best to avoid uploading large amounts of data due to the limited storage in GitHub) for Budd & Gunn et al. (2020). In the home directory of the repository (Elephant_Microbiome), you will find a README.md file (the source script for this information), the R Project file (Elephant_Microbiome.Rproj), a .gitignore file, and five different "analysis" directories, each of which corresponds with a specific analysis conducted in our study:

1) 01_bioinformatic_analysis
2) 02_data_summary_analysis
3) 03_core_microbiome_analysis
4) 04_microbiome_diversity_analysis
5) 05_functional_microbiome_analysis

Within each analysis directory, you will find an R markdown script (.Rmd) with the name of the analysis, which contains all of the code needed to run the full analysis. Additionally, you will find one:

1) code

The code directory will store all source code, shell scripts, lists of bash commands, and software packages needed for analysis. 

Once you have downloaded the repository and located the code directory, you should create two additional sub-directories within each analysis (on the same level as the code directory):

2) data
3) figures

The data directory will store all raw data, processed data, and metadata needed for analysis. The figures folder will contain any raw figures generated in ggplot for each analysis. Ideally, the Rmd script should have paths set up so that the code reads all data and scripts and generates figures seamlessly.

## Using the code
To reproduce all analyses in Budd & Gunn et al. (2020), download this repository and place in a desired home directory. This may be done on your local machine, but we recommend downloading to a high-performance computing cluster so that all code will run seamlessly in one environment, as long as Rstudio is installed and the GUI can be called on the cluster.

Once all directories are downloaded, create a new sub-directory within the home directory (same level as the seven analysis directories, .Rproj, README.md, etc.) called "raw_data". This is where you will store the raw genomic data and associated sample metadata (see <i><b>Data</i></b> section below).

## Data
Raw 16S rRNA sequence reads and accompanying metadata are available on NCBI: <a href="url">https://www.ncbi.nlm.nih.gov/bioproject/PRJNA587772/</a>

Metagenomic data files (.fastq.gz) must be downloaded individually from the NBI database. Once downloaded, place all .fastq.gz files into a desired working directory to be accessed during bioinformatic processing in QIIME2.

Code for processing raw data files in QIIME2 are provided in the first analysis folder ("01_bioinformatic_analysis") in this repository.

You will need only the "raw_data" table outputs from QIIME2 to reproduce analyses 2 through [number].

If you have any questions or issues with data and/or code, please don't hesitate to contact me: jcgunn@uvm.edu

## Analyses

Figure 1 in the final ms is a three panel map showing the full native ranges of the two African elephant species on the African continent (Panel A), and specific study sites for African savanna elephants (<i>Loxidonta africana</i>; Panel B) and African forest elephants (<i>L. cyclotis</i>; Panel C). Maps were generated by K. Budd using ArcMap in ArcGIS (Esri).

### Analysis 1: Metagenomic data processing
In this analysis, we use the bioinformatic software pipeline QIIME2 to process all raw metagenomic data files (fastq.gz), i.e., to calculate absolute OTU read abundance across African elephant fecal samples.

#### Run the code: `01_bioinformatic_analysis/aegm_bioinformatic_analysis.Rmd`

### Analysis 2: Taxonomic classification and metadata summary analysis
In this analysis, we read in, filter, and clean the raw metadata (i.e., information about species, diet, and habitat) and taxonomic classification for microbial operational taxonomic units (OTUs) for our African elephant samples. After we prepare data, we summarize the taxonomic classification data, specifically to determine the proportion of bacterial OTUs classified to each taxonomic group and the confidence of classification. We then give a wholistic and broken-down summary of our elephant fecal samples, including sample numbers within species, diet treatments, and habitats.

#### Run the code: `02_data_summary_analysis/aegm_data_summary_analysis.Rmd`

### Analysis 3: Global and core microbiome analysis
In this analysis, we characterize the core microbial composition of African elephant individuals within groups of interest (phylogeny, diet, and habitat) at the Order level (see Analysis 2): Orders that are present in 100% of samples within a given group. For each group of interest, we control for all other groups to identify these orders. 

#### Run the code: `03_core_microbiome_analysis/aegm_core_microbiome_analysis.Rmd`

### Analysis 4: Microbiome diversity and differential OTU abundnace analysis
In this analysis, we assess differential abundance in relatively highly represented OTUs (i.e., within the most highly represented phyla) across African elephant sample groups of interest (phylogeny, diet, and habitat) to determine whether any major taxonomic groups explain variation between groups. We then assess both within-group (alpha) microbial diversity and between-group (beta) microbial diversity to determine whether there are overarching differences in microbiome structure and composition among elephant groups.  

#### Run the code: `04_microbiome_diversity_analysis/aegm_microbiome_diversity_analysis.Rmd`

### Analysis 5: Functional microbiome analysis
In this analysis, we characterized the physiological importance (aka roles) of the African elephant gut microbiome by classifying OTUs into their corresponding, primary metabolic functions. To identify metabolic pathways, we used the Kyoto Encyclopedia of Genes and Genome (KEGG) Orthology (KO) (Kanehisa & Goto, 2000; Kanehisa et al., 2014) and clustered OTUs by the BRITE hierarchy. We ultimately assessed differences in metabolic pathway abundance among African elephant groups of interest to test for potential differences due to phylogeny, diet, or habitat.

#### Run the code: `05_functional_microbiome_analysis/aegm_functional_microbiome_analysis.Rmd`

