---
title: "Analysis 4: Microbiome diversity and differential OTU abundnace analysis"
author: "Joe Gunn"
date: "2024-02-07"
output: html_document
---

# Project: Assessing effects of phylogeny, diet, and habitat on gut microbial composition in African elephants 
We characterized the gut microbiome of African Savanna elephants (<i>Loxodonta africana</i>) and African Forest elephants (<i>Loxodonta cyclotis</i>). We assessed the relationship of gut microbial composition, including analyses of alpha and beta diversity, with host phylogeny (i.e., species) and habitat type (i.e., forest or savanna) for both species. We also assessed the relationship between microbial composition and diet (i.e., crop-raiding vs. non-crop-raiding) within <i>L. africana</i>. 

## Specific Aim: Assessing differential OTU abundance and diverity within (alpha) and between (beta) African elephant sample groups (phylogeny, diet, and habitat) 
In this analysis, we assess differential abundance in relatively highly represented OTUs (i.e., within the most highly represented phyla) across African elephant sample groups of interest (phylogeny, diet, and habitat) to determine whether any major taxonomic groups explain variation between groups. We then assess both within-group (alpha) microbial diversity and between-group (beta) microbial diversity to determine whether there are overarching differences in microbiome structure and composition among elephant groups.  

## Phases of analysis:
### Phase 1: Data read-in, identification of major OTU groups, and filtering
### Phase 2: Differential OTU abundance analysis
### Phase 3: OTU diveristy analysis

## Libraries needed for analysis
```{r}
library(readxl)
library(cowplot)
library(tidyverse)
library(phyloseq)
library(lme4)
library(nlme)
library(car)
library(multcomp)
library(devtools)
library(qiime2R)
library(vegan)
library(ape)
```

## PHASE 1: DIFFERENTIAL OTU ABUNDANCE ANALYSIS
In this phase of the analysis, we read in the full, clean Order-level dataset from Analysis 3 ("order_data.Rda") and identify the most relatively abundant taxonomic groups. We then filter the full dataset into separate datasets by taxonomic group for downstream analyses. We then assess differential abundance only in microbial orders that have an average relative abundance of 10% across samples.

### STEP 1: Identify taxa with the greatest relative abundance at the phylum level
In this step, we determine relative abundance (summing abundance of all individuals) across the full sample set at the phylum level. We then use these groups to assess differences in abundance in lower taxonomic groups across different elephant groups of interest in Step 2.

#### 1a: Identify phyla with the highest relative abundance across samples; run the Rmd chunk below.

##### Obtain most abundant phyla in the dataset:
```{r}
# Load order dataset
load("../03_core_microbiome_analysis/data/order_data.Rda")

# Set scipen options to remove scientific notation
options(scipen = 999)

# Calculate total microbial abundance across samples
total_abundance <- sum(order_data$abundance)

# Calculate abundance and relative abundance across samples for each phylum represented in the dataset
phylum_abundance <- order_data %>%
  group_by(phylum) %>%
  summarize(abundance = sum(abundance),
            relative_abundance = abundance/total_abundance) %>%
  arrange(desc(relative_abundance))

# Get dataset with only most abundant phyla
phyla <- order_data %>%
  filter(phylum == "Firmicutes" | 
           phylum == "Proteobacteria" |
           phylum == "Bacteroidetes")

# Save phylum dataset for downstream analyses
save(phyla, file = "data/phyla.Rda")
```

<b> Data filtering results </b>: <br>
Phyla with greater than 10% relative abundance: Firmicutes, Bacteroidetes, Proteobacteria

### STEP 2: Assess differential OTU abundance at lower taxonomic levels within top phyla.
In this step, we zoom in on the three most abundant phyla ascertained in STEP 1 above (Firmucutes, Proteobacteria, and Bacteroidetes), and we assess differential abundance in microbial OTUs at the order and genus levels for African elephant groups of interest (i.e., phylogeny, diet, and habitat). We examine all orders and genera within each of the three top phyla successively; we assess all groups in Firmicutes in Step 2a, all groups in Proteobacteria in Step 2b, and all groups in Bacteroidetes in Step 2c. 

#### 2a: Assess differential OTU abundance within Phylum Firmicutes.

##### 2a.1. Identify most abundant orders within Phylum Firmicutes.
In this step, we identify orders within Firmicutes that have a relative abundance greater than 10%, and we assess differential abundance (Step 2a.2) between African elephant species (2a.2.1) and groups (2a.2.2) for these orders.

##### Get top orders in phylum Firmicutes:
```{r}
# Load phylum dataset
load("data/phyla.Rda")

# Get dataset with only phylum firmicutes
firmicutes <- phyla %>%
  filter(phylum == "Firmicutes")

# Get total abundance across samples
total_abundance <- sum(firmicutes$abundance)

# Calculate abundance and relative abundance across samples for each order represented in the dataset
order_abundance <- firmicutes %>%
  group_by(order) %>%
  summarize(abundance = sum(abundance),
            relative_abundance = abundance/total_abundance) %>%
  arrange(desc(relative_abundance))

# Get dataset with only top orders
firmicutes_orders <- firmicutes %>%
  filter(order == "Clostridiales" | 
           order == "Bacillales")

# Save order dataset for downstream analyses
save(firmicutes_orders, file = "data/firmicutes_orders.Rda")
```

<b> Data filtering results </b>: <br>
Orders with greater than 10% relative abundance: Clostridiales, Bacillales

##### 2a.2. Assess differential OTU abundance at the most abundant orders in Phylum Firmicutes; run the Rmd chunk below.

###### 2a.2.1. Assess differential OTU abundance between African elephant species (<i>L. africana</i>, <i>L. cyclotis</i>), controlling for diet and habitat.

##### Run general linear mixed models.
```{r}
# Load Firmicutes data
load("data/firmicutes_orders.Rda")

# Get total abundance per individual per order and spread the dataset to prepare for general linear mixed models.
firmicutes_orders <- firmicutes_orders %>%
  dplyr::select(ind_id, group, phylogeny, diet, habitat, sex, order, abundance) %>%
  group_by(ind_id, group, phylogeny, diet, habitat, sex, order) %>%
  summarize(abundance = sum(abundance)) %>%
  spread(order, abundance) 

# Set scipen to avoid scientific notation
options(scipen = 999)

# Filter data to include only forest non-crop-raiders ("f_ncr")
firmicutes_orders <- firmicutes_orders %>%
  filter(habitat == "Forest" & diet == "Nonraider")

# Create table to hold p values and F values from general linear mixed models 
stats <- data.frame(order = factor(colnames(firmicutes_orders[,7:ncol(firmicutes_orders)])),
                    p_value = numeric(length = ncol(firmicutes_orders[,7:ncol(firmicutes_orders)])),
                    f_value = numeric(length = ncol(firmicutes_orders[,7:ncol(firmicutes_orders)])))

# Run general linear mixed models on order columns
for (ii in 7:ncol(firmicutes_orders)) {
    
  ff <- as.formula(paste0(colnames(firmicutes_orders)[ii], "~ phylogeny + (1|sex)")) # define response 
  glmer <- glmer.nb(ff,  firmicutes_orders) # define linear model 
  p_value <- summary(glmer)$coefficients[2,4] # extract p value
  f_value <- anova(glmer)$`F value` # extract F value
  stats[ii-6,2] <- p_value # Put p value in table
  stats[ii-6,3] <- f_value # Put F value in table
  
}

# Append p values corrected for false discovery rate to the stats data table
stats$p_corrected <- p.adjust(stats$p_value, method = "fdr")
```

<b> Model results </b>: <br>
Bacillales: p_corrected < 0.001, F = 30.623
Clostridiales: p_corrected = 0.546, F = 0.3653

###### 2a.2.2. Assess differential OTU abundance between African elephant groups (diet and habitat), controlling for species.

##### Run general linear mixed models.
```{r}
# Load Firmicutes data
load("data/firmicutes_orders.Rda")

# Get total abundance per individual per order and spread the dataset to prepare for general linear mixed models.
firmicutes_orders <- firmicutes_orders %>%
  dplyr::select(ind_id, group, phylogeny, diet, habitat, sex, order, abundance) %>%
  group_by(ind_id, group, phylogeny, diet, habitat, sex, order) %>%
  summarize(abundance = sum(abundance)) %>%
  spread(order, abundance) 

# Set scipen to avoid scientific notation
options(scipen = 999)

# Filter data to include only forest non-crop-raiders ("f_ncr")
firmicutes_orders <- firmicutes_orders %>%
  filter(phylogeny == "africana")

# Create table to hold p values and F values from general linear mixed models 
stats <- data.frame(order = factor(colnames(firmicutes_orders[,7:ncol(firmicutes_orders)])),
                    p_value = numeric(length = ncol(firmicutes_orders[,7:ncol(firmicutes_orders)])),
                    f_value = numeric(length = ncol(firmicutes_orders[,7:ncol(firmicutes_orders)])))

# Run general linear mixed models on order columns
for (ii in 7:ncol(firmicutes_orders)) {
    
  ff <- as.formula(paste0(colnames(firmicutes_orders)[ii], "~ group + (1|sex)")) # define response 
  glmer <- glmer.nb(ff,  firmicutes_orders) # define linear model 
  p_value <- summary(glmer)$coefficients[2,4] # extract p value
  f_value <- anova(glmer)$`F value` # extract F value
  stats[ii-6,2] <- p_value # Put p value in table
  stats[ii-6,3] <- f_value # Put F value in table
  
}

# Append p values corrected for false discovery rate to the stats data table
stats$p_corrected <- p.adjust(stats$p_value, method = "fdr")
```

<b> Model results </b>: <br>
Bacillales: p_corrected = 0.109, F = 2.367
Clostridiales: p_corrected = 0.712, F = 2.275

##### 2a.3. Identify most abundant genera within significant orders.
In this step, we identify genera within Order Bacilalles, which was the only order with significant differences between species, that have a relative abundance greater than 10%. We then assess differential abundance (Step 2b.4) between African elephant species (2b.4.1; there were no significant differences between groups in Order Bacillales).

##### Get top genera in Order Bacillales:
```{r}
# Load firmicutes orders dataset
load("data/firmicutes_orders.Rda")

# Get dataset with only Order Bacillales
bacillales <- firmicutes_orders %>%
  filter(order == "Bacillales") %>%
  drop_na(genus)

# Get total abundance across samples
total_abundance <- sum(bacillales$abundance)

# Calculate abundance and relative abundance across samples for each order represented in the dataset
genera_abundance <- bacillales %>%
  group_by(genus) %>%
  summarize(abundance = sum(abundance),
            relative_abundance = abundance/total_abundance) %>%
  arrange(desc(relative_abundance))

# Get dataset with only top orders
bacillales_genera <- bacillales %>%
  filter(genus == "Solibacillus" | 
           genus == "Bacillus")

# Save order dataset for downstream analyses
save(bacillales_genera, file = "data/bacillales_genera.Rda")
```

<b> Data filtering results </b>: <br>
Genera with greater than 10% relative abundance: Solibacillus, Bacillus

##### 2a.4. Assess differential OTU abundance between African elephant species (<i>L. africana</i>, <i>L. cyclotis</i>), controlling for diet and habitat; run the Rmd chunk below.

##### Run general linear mixed models.
```{r}
# Load genus data
load("data/bacillales_genera.Rda")

# Get total abundance per individual per order and spread the dataset to prepare for general linear mixed models.
bacillales_genera <- bacillales_genera %>%
  dplyr::select(ind_id, group, phylogeny, diet, habitat, sex, genus, abundance) %>%
  group_by(ind_id, group, phylogeny, diet, habitat, sex, genus) %>%
  summarize(abundance = sum(abundance)) %>%
  spread(genus, abundance) 

# Set scipen to avoid scientific notation
options(scipen = 999)

# Filter data to include only forest non-crop-raiders ("f_ncr")
bacillales_genera <- bacillales_genera %>%
  filter(habitat == "Forest" & diet == "Nonraider")

# Create table to hold p values and F values from general linear mixed models 
stats <- data.frame(genus = factor(colnames(bacillales_genera[,7:ncol(bacillales_genera)])),
                    p_value = numeric(length = ncol(bacillales_genera[,7:ncol(bacillales_genera)])),
                    f_value = numeric(length = ncol(bacillales_genera[,7:ncol(bacillales_genera)])))

# Run general linear mixed models on order columns
for (ii in 7:ncol(bacillales_genera)) {
    
  ff <- as.formula(paste0(colnames(bacillales_genera)[ii], "~ phylogeny + (1|sex)")) # define response 
  glmer <- glmer.nb(ff,  bacillales_genera) # define linear model 
  p_value <- summary(glmer)$coefficients[2,4] # extract p value
  f_value <- anova(glmer)$`F value` # extract F value
  stats[ii-6,2] <- p_value # Put p value in table
  stats[ii-6,3] <- f_value # Put F value in table
  
}

# Append p values corrected for false discovery rate to the stats data table
stats$p_corrected <- p.adjust(stats$p_value, method = "fdr")
```

<b> Model results </b>: <br>
Bacillus: p_corrected = 0.654, F = 0.201
Solibacillus: p_corrected < 0.001, F = 42.547

#### 2b: Assess differential OTU abundance within Phylum Bacteroidetes.

##### 2b.1. Identify most abundant orders within Phylum Bacteroidetes
In this step, we identify orders within Bacteroidetes that have a relative abundance greater than 10%, and we assess differential abundance (Step 2b.2) between African elephant species (2b.2.1) and groups (2b.2.2) for these orders.

##### Get top orders in phylum Bacteroidetes:
```{r}
# Load phylum dataset
load("data/phyla.Rda")

# Get dataset with only phylum bacteroidetes
bacteroidetes <- phyla %>%
  filter(phylum == "Bacteroidetes")

# Get total abundance across samples
total_abundance <- sum(bacteroidetes$abundance)

# Calculate abundance and relative abundance across samples for each order represented in the dataset
order_abundance <- bacteroidetes %>%
  group_by(order) %>%
  summarize(abundance = sum(abundance),
            relative_abundance = abundance/total_abundance) %>%
  arrange(desc(relative_abundance))

# Get dataset with only top orders
bacteroidetes_orders <- bacteroidetes %>%
  filter(order == "Bacteroidales" | 
           order == "Flavobacteriales")

# Save order dataset for downstream analyses
save(bacteroidetes_orders, file = "data/bacteroidetes_orders.Rda")
```

<b> Data filtering results </b>: <br>
Orders with greater than 10% relative abundance: Bacteroidales, Flavobacteriales

##### 2b.2. Assess differential OTU abundance at the most abundant orders in Phylum Bacteroidetes.

###### 2b.2.1. Assess differential OTU abundance between African elephant species (<i>L. africana</i>, <i>L. cyclotis</i>), controlling for diet and habitat; run the Rmd chunk below.

##### Run general linear mixed models.
```{r}
# Load Bacteroidetes data
load("data/bacteroidetes_orders.Rda")

# Get total abundance per individual per order and spread the dataset to prepare for general linear mixed models.
bacteroidetes_orders <- bacteroidetes_orders %>%
  dplyr::select(ind_id, group, phylogeny, diet, habitat, sex, order, abundance) %>%
  group_by(ind_id, group, phylogeny, diet, habitat, sex, order) %>%
  summarize(abundance = sum(abundance)) %>%
  spread(order, abundance) 

# Set scipen to avoid scientific notation
options(scipen = 999)

# Filter data to include only forest non-crop-raiders ("f_ncr")
bacteroidetes_orders <- bacteroidetes_orders %>%
  filter(habitat == "Forest" & diet == "Nonraider")

# Create table to hold p values and F values from general linear mixed models 
stats <- data.frame(order = factor(colnames(bacteroidetes_orders[,7:ncol(bacteroidetes_orders)])),
                    p_value = numeric(length = ncol(bacteroidetes_orders[,7:ncol(bacteroidetes_orders)])),
                    f_value = numeric(length = ncol(bacteroidetes_orders[,7:ncol(bacteroidetes_orders)])))

# Run general linear mixed models on order columns
for (ii in 7:ncol(bacteroidetes_orders)) {
    
  ff <- as.formula(paste0(colnames(bacteroidetes_orders)[ii], "~ phylogeny + (1|sex)")) # define response 
  glmer <- glmer.nb(ff,  bacteroidetes_orders) # define linear model 
  p_value <- summary(glmer)$coefficients[2,4] # extract p value
  f_value <- anova(glmer)$`F value` # extract F value
  stats[ii-6,2] <- p_value # Put p value in table
  stats[ii-6,3] <- f_value # Put F value in table
  
}

# Append p values corrected for false discovery rate to the stats data table
stats$p_corrected <- p.adjust(stats$p_value, method = "fdr")
```

<b> Model results </b>: <br>
Bacteroidales: p_corrected 0.469, F = 0.525
Flavobacteriales: p_corrected = 0.139, F = 3.295

###### 2b.2.2. Assess differential OTU abundance between African elephant groups (diet and habitat), controlling for species.

##### Run general linear mixed models.
```{r}
# Load Bacteroidetes data
load("data/bacteroidetes_orders.Rda")

# Get total abundance per individual per order and spread the dataset to prepare for general linear mixed models.
bacteroidetes_orders <- bacteroidetes_orders %>%
  dplyr::select(ind_id, group, phylogeny, diet, habitat, sex, order, abundance) %>%
  group_by(ind_id, group, phylogeny, diet, habitat, sex, order) %>%
  summarize(abundance = sum(abundance)) %>%
  spread(order, abundance) 

# Set scipen to avoid scientific notation
options(scipen = 999)

# Filter data to include only forest non-crop-raiders ("f_ncr")
bacteroidetes_orders <- bacteroidetes_orders %>%
  filter(phylogeny == "africana")

# Create table to hold p values and F values from general linear mixed models 
stats <- data.frame(order = factor(colnames(bacteroidetes_orders[,7:ncol(bacteroidetes_orders)])),
                    p_value = numeric(length = ncol(bacteroidetes_orders[,7:ncol(bacteroidetes_orders)])),
                    f_value = numeric(length = ncol(bacteroidetes_orders[,7:ncol(bacteroidetes_orders)])))

# We attempted to run general linear mixed models in a for loop as above (for Phylum Firmicutes), but we found that the glmer.nb could not be computed for order Flavobacteriales, likely due to complete separation of the response variable. Thus, we needed to run each linear model separately ("mod_1" for Bacteroidales and "mod_2" for Flavobacteriales), and for mod_2, we implemented recommended model parameters (Bolker) to handle this issue. Resulting model output statistics were still loaded into the "stats" table and corrected for false discovery rate.

# Run general linear mixed models and extract statistics for Bacteroidales column
mod_1 <- glmer.nb(Bacteroidales ~ group + (1|sex), bacteroidetes_orders)
mod_1_p <- summary(mod_1)$coefficients[2,4] # extract p value
mod_1_f <- anova(mod_1)$`F value` # extract F value

# Run general linear mixed models and extract statistics for Flavobacteriales column
mod_2 <- glmer.nb(Flavobacteriales ~ group + (1|sex), bacteroidetes_orders,
                  control=glmerControl(optimizer="bobyqa", 
                                       tolPwrss=1e-3, 
                                       optCtrl = list(maxfun = 100000)))
mod_2_p <- summary(mod_2)$coefficients[2,4] # extract p value
mod_2_f <- anova(mod_2)$`F value` # extract F value

# Fill stats table
stats[1,2] <- mod_1_p
stats[1,3] <- mod_1_f
stats[2,2] <- mod_2_p
stats[2,3] <- mod_2_f

# Append p values corrected for false discovery rate to the stats data table
stats$p_corrected <- p.adjust(stats$p_value, method = "fdr")

# Run post-hoc Tukey-test on Flavobacteriales given that it was significant for group
posthoc <- glht(mod_2, mcp(group = "Tukey"))

# Get p-values for pairwise comparisons
summary(posthoc)
```

<b> Model results </b>: <br>
Bacteroidales: p_corrected = 0.167, F = 1.591991
Flavobacteriales: p_corrected = 0.002, F = 8.268724

<b> Post-hoc model results for Flavobacteriales </b>: <br>
s_cr - f_cr: p_value = 0.005 (habitat difference; significant)
f_ncr - f_cr: p_value = 0.49166 
s_ncr - f_cr: p_value = 0.56731 
f_ncr - s_cr: p_value < 0.001 (diet:habitat interaction; significant)
s_ncr - s_cr: p_value = 0.014 (diet difference; significant)
s_ncr - f_ncr: p_value = 0.12808

##### 2c.3. Identify most abundant genera within significant orders.
In this step, we identify genera within Order Flavobacteriales, which was the only order with significant differences among groups, that have a relative abundance greater than 10%. We then assess differential abundance (Step 2c.4) among African elephant groups (2b.4.1; there were no significant differences between species in Order Flavobacteriales).

##### Get top genera in Order Flavobacteriales:
```{r}
# Load bacteroidetes orders dataset
load("data/bacteroidetes_orders.Rda")

# Get dataset with only order Flavobacteriales
flavobacteriales <- bacteroidetes_orders %>%
  filter(order == "Flavobacteriales") %>%
  drop_na(genus)

# Get total abundance across samples
total_abundance <- sum(flavobacteriales$abundance)

# Calculate abundance and relative abundance across samples for each order represented in the dataset
genera_abundance <- flavobacteriales %>%
  group_by(genus) %>%
  summarize(abundance = sum(abundance),
            relative_abundance = abundance/total_abundance) %>%
  arrange(desc(relative_abundance))

# Get dataset with only top orders
flavobacteriales_genera <- flavobacteriales %>%
  filter(genus == "Wautersiella")

# Save order dataset for downstream analyses
save(flavobacteriales_genera, file = "data/flavobacteriales_genera.Rda")
```

<b> Data filtering results </b>: <br>
Genera with greater than 10% relative abundance: Wautersiella 

##### 2b.4. Assess differential OTU abundance between African elephant group, controlling for species; run the Rmd chunk below.

##### Run general linear mixed models.
```{r}
# Load genus data
load("data/flavobacteriales_genera.Rda")

# Get total abundance per individual per order and spread the dataset to prepare for general linear mixed models.
flavobacteriales_genera <- flavobacteriales_genera %>%
  dplyr::select(ind_id, group, phylogeny, diet, habitat, sex, genus, abundance) %>%
  group_by(ind_id, group, phylogeny, diet, habitat, sex, genus) %>%
  summarize(abundance = sum(abundance)) %>%
  spread(genus, abundance) 

# Set scipen to avoid scientific notation
options(scipen = 999)

# Filter data to include only L. africana 
flavobacteriales_genera <- flavobacteriales_genera %>%
  filter(phylogeny == "africana")

# Create table to hold p values and F values from general linear mixed models 
stats <- data.frame(genus = factor(colnames(flavobacteriales_genera[,7:ncol(flavobacteriales_genera)])),
                    p_value = numeric(length = ncol(flavobacteriales_genera[,7:ncol(flavobacteriales_genera)])),
                    f_value = numeric(length = ncol(flavobacteriales_genera[,7:ncol(flavobacteriales_genera)])))

# Run general linear mixed models on order columns
for (ii in 7:ncol(flavobacteriales_genera)) {
    
  ff <- as.formula(paste0(colnames(flavobacteriales_genera)[ii], "~ group + (1|sex)")) # define response 
  glmer <- glmer.nb(ff,  flavobacteriales_genera) # define linear model 
  p_value <- summary(glmer)$coefficients[2,4] # extract p value
  f_value <- anova(glmer)$`F value` # extract F value
  stats[ii-6,2] <- p_value # Put p value in table
  stats[ii-6,3] <- f_value # Put F value in table
  
}

# Append p values corrected for false discovery rate to the stats data table
stats$p_corrected <- p.adjust(stats$p_value, method = "fdr")

# Run post-hoc Tukey-test on Wautersiella given that it was significant for group
posthoc <- glht(glmer, mcp(group = "Tukey"))

# Get p-values for pairwise comparisons
summary(posthoc)
```

<b> Model results </b>: <br>
Wautersiella: p_corrected = < 0.001, F = 6.976

<b> Post-hoc model results for Flavobacteriales </b>: <br>
s_cr - f_cr: p_value = 0.003 (habitat difference; significant)
f_ncr - f_cr: p_value = 0.983   
s_ncr - f_cr: p_value = 0.424   
f_ncr - s_cr: p_value = 0.002 (diet:habitat interaction; significant)
s_ncr - s_cr: p_value = 0.016 (diet difference; significant) 
s_ncr - f_ncr: p_value = 0.336

#### 2c: Assess differential OTU abundance within Phylum Proteobacteria.

##### 2c.1. Identify most abundant orders within Phylum Proteobacteria.
In this step, we identify orders within Proteobacteria that have a relative abundance greater than 10%, and we assess differential abundance (Step 2c.2) between African elephant species (2c.2.1) and groups (2c.2.2) for these orders.

##### Get top orders in phylum Proteobacteria:
```{r}
# Load phylum dataset
load("data/phyla.Rda")

# Get dataset with only phylum proteobacteria
proteobacteria <- phyla %>%
  filter(phylum == "Proteobacteria")

# Get total abundance across samples
total_abundance <- sum(proteobacteria$abundance)

# Calculate abundance and relative abundance across samples for each order represented in the dataset
order_abundance <- proteobacteria %>%
  group_by(order) %>%
  summarize(abundance = sum(abundance),
            relative_abundance = abundance/total_abundance) %>%
  arrange(desc(relative_abundance))

# Get dataset with only top orders
proteobacteria_orders <- proteobacteria %>%
  filter(order == "Pseudomonadales" | 
           order == "Enterobacteriales")

# Save order dataset for downstream analyses
save(proteobacteria_orders, file = "data/proteobacteria_orders.Rda")
```

<b> Data filtering results </b>: <br>
Orders with greater than 10% relative abundance: Pseudomonadales, Enterobacteriales

##### 2c.2. Assess differential OTU abundance at the most abundant orders in Phylum Proteobacteria.

###### 2c.2.1. Assess differential OTU abundance between African elephant species (<i>L. africana</i>, <i>L. cyclotis</i>), controlling for diet and habitat; run the Rmd chunk below.

##### Run general linear mixed models.
```{r}
# Load Proteobacteria data
load("data/proteobacteria_orders.Rda")

# Get total abundance per individual per order and spread the dataset to prepare for general linear mixed models.
proteobacteria_orders <- proteobacteria_orders %>%
  dplyr::select(ind_id, group, phylogeny, diet, habitat, sex, order, abundance) %>%
  group_by(ind_id, group, phylogeny, diet, habitat, sex, order) %>%
  summarize(abundance = sum(abundance)) %>%
  spread(order, abundance) 

# Set scipen to avoid scientific notation
options(scipen = 999)

# Filter data to include only forest non-crop-raiders ("f_ncr")
proteobacteria_orders <- proteobacteria_orders %>%
  filter(habitat == "Forest" & diet == "Nonraider")

# Create table to hold p values and F values from general linear mixed models 
stats <- data.frame(order = factor(colnames(proteobacteria_orders[,7:ncol(proteobacteria_orders)])),
                    p_value = numeric(length = ncol(proteobacteria_orders[,7:ncol(proteobacteria_orders)])),
                    f_value = numeric(length = ncol(proteobacteria_orders[,7:ncol(proteobacteria_orders)])))

# Run general linear mixed models on order columns
for (ii in 7:ncol(proteobacteria_orders)) {
    
  ff <- as.formula(paste0(colnames(proteobacteria_orders)[ii], "~ phylogeny + (1|sex)")) # define response 
  glmer <- glmer.nb(ff,  proteobacteria_orders) # define linear model 
  p_value <- summary(glmer)$coefficients[2,4] # extract p value
  f_value <- anova(glmer)$`F value` # extract F value
  stats[ii-6,2] <- p_value # Put p value in table
  stats[ii-6,3] <- f_value # Put F value in table
  
}

# Append p values corrected for false discovery rate to the stats data table
stats$p_corrected <- p.adjust(stats$p_value, method = "fdr")
```

<b> Model results </b>: <br>
Enterobacteriales: p_corrected 0.253, F = 1.307
Pseudomonadales: p_corrected = 0.086, F = 4.093

###### 2c.2.2. Assess differential OTU abundance between African elephant groups (diet and habitat), controlling for species.

##### Run general linear mixed models.
```{r}
# Load Proteobacteria data
load("data/proteobacteria_orders.Rda")

# Get total abundance per individual per order and spread the dataset to prepare for general linear mixed models.
proteobacteria_orders <- proteobacteria_orders %>%
  dplyr::select(ind_id, group, phylogeny, diet, habitat, sex, order, abundance) %>%
  group_by(ind_id, group, phylogeny, diet, habitat, sex, order) %>%
  summarize(abundance = sum(abundance)) %>%
  spread(order, abundance) 

# Set scipen to avoid scientific notation
options(scipen = 999)

# Filter data to include only L. africana
proteobacteria_orders <- proteobacteria_orders %>%
  filter(phylogeny == "africana")

# Create table to hold p values and F values from general linear mixed models 
stats <- data.frame(order = factor(colnames(proteobacteria_orders[,7:ncol(proteobacteria_orders)])),
                    p_value = numeric(length = ncol(proteobacteria_orders[,7:ncol(proteobacteria_orders)])),
                    f_value = numeric(length = ncol(proteobacteria_orders[,7:ncol(proteobacteria_orders)])))

# Run general linear mixed models on order columns
for (ii in 7:ncol(proteobacteria_orders)) {
    
  ff <- as.formula(paste0(colnames(proteobacteria_orders)[ii], "~ group + (1|sex)")) # define response 
  glmer <- glmer.nb(ff,  proteobacteria_orders) # define linear model 
  p_value <- summary(glmer)$coefficients[2,4] # extract p value
  f_value <- anova(glmer)$`F value` # extract F value
  stats[ii-6,2] <- p_value # Put p value in table
  stats[ii-6,3] <- f_value # Put F value in table
  
}

# Append p values corrected for false discovery rate to the stats data table
stats$p_corrected <- p.adjust(stats$p_value, method = "fdr")

# Get model just for significant order (Pseudomonadales)
glmer <- glmer.nb(Pseudomonadales ~ group + (1|sex),  proteobacteria_orders)

# Run post-hoc Tukey-test on Flavobacteriales given that it was significant for group
posthoc <- glht(glmer, mcp(group = "Tukey"))

# Get p-values for pairwise comparisons
summary(posthoc)
```

<b> Model results </b>: <br>
Enterobacteriales: p_corrected = 0.057, F = 1.876
Pseudomonadales: p_corrected = 0.001, F = 4.77

<b> Post-hoc model results for Flavobacteriales </b>: <br>
s_cr - f_cr: p_value = 0.003 (habitat difference; significant)
f_ncr - f_cr: p_value = 0.971
s_ncr - f_cr: p_value = 0.999
f_ncr - s_cr: p_value = 0.002 (diet:habitat interaction; significant)
s_ncr - s_cr: p_value = 0.002 (diet difference; significant)
s_ncr - f_ncr: p_value = 0.988

##### 2c.3. Identify most abundant genera within significant orders.
In this step, we identify genera within Order Pseudomonadales, which was the only order with significant differences among groups, that have a relative abundance greater than 10%. We then assess differential abundance (Step 2c.4) among African elephant groups (2c.4.1; there were no significant differences between species in Order Pseudomonadales).

##### Get top genera in Order Pseudomonadales:
```{r}
# Load proteobacteria orders dataset
load("data/proteobacteria_orders.Rda")

# Get dataset with only order Pseudomonadales
pseudomonadales <- proteobacteria_orders %>%
  filter(order == "Pseudomonadales") %>%
  drop_na(genus)

# Get total abundance across samples
total_abundance <- sum(pseudomonadales$abundance)

# Calculate abundance and relative abundance across samples for each order represented in the dataset
genera_abundance <- pseudomonadales %>%
  group_by(genus) %>%
  summarize(abundance = sum(abundance),
            relative_abundance = abundance/total_abundance) %>%
  arrange(desc(relative_abundance))

# Get dataset with only top orders
pseudomonadales_genera <- pseudomonadales %>%
  filter(genus == "Acinetobacter")

# Save order dataset for downstream analyses
save(pseudomonadales_genera, file = "data/pseudomonadales_genera.Rda")
```

<b> Data filtering results </b>: <br>
Genera with greater than 10% relative abundance: Acinetobacter 

##### 2c.4. Assess differential OTU abundance between African elephant group, controlling for species; run the Rmd chunk below.

##### Run general linear mixed models.
```{r}
# Load genus data
load("data/pseudomonadales_genera.Rda")

# Get total abundance per individual per order and spread the dataset to prepare for general linear mixed models.
pseudomonadales_genera <- pseudomonadales_genera %>%
  dplyr::select(ind_id, group, phylogeny, diet, habitat, sex, genus, abundance) %>%
  group_by(ind_id, group, phylogeny, diet, habitat, sex, genus) %>%
  summarize(abundance = sum(abundance)) %>%
  spread(genus, abundance) 

# Set scipen to avoid scientific notation
options(scipen = 999)

# Filter data to include only L. africana 
pseudomonadales_genera <- pseudomonadales_genera %>%
  filter(phylogeny == "africana")

# Create table to hold p values and F values from general linear mixed models 
stats <- data.frame(genus = factor(colnames(pseudomonadales_genera[,7:ncol(pseudomonadales_genera)])),
                    p_value = numeric(length = ncol(pseudomonadales_genera[,7:ncol(pseudomonadales_genera)])),
                    f_value = numeric(length = ncol(pseudomonadales_genera[,7:ncol(pseudomonadales_genera)])))

# Run general linear mixed models on order columns
for (ii in 7:ncol(pseudomonadales_genera)) {
    
  ff <- as.formula(paste0(colnames(pseudomonadales_genera)[ii], "~ group + (1|sex)")) # define response 
  glmer <- glmer.nb(ff,  pseudomonadales_genera) # define linear model 
  p_value <- summary(glmer)$coefficients[2,4] # extract p value
  f_value <- anova(glmer)$`F value` # extract F value
  stats[ii-6,2] <- p_value # Put p value in table
  stats[ii-6,3] <- f_value # Put F value in table
  
}

# Append p values corrected for false discovery rate to the stats data table
stats$p_corrected <- p.adjust(stats$p_value, method = "fdr")

# Run post-hoc Tukey-test on Wautersiella given that it was significant for group
posthoc <- glht(glmer, mcp(group = "Tukey"))

# Get p-values for pairwise comparisons
summary(posthoc)
```

<b> Model results </b>: <br>
Acinetobacter: p_corrected = < 0.001, F = 4.760

<b> Post-hoc model results for Flavobacteriales </b>: <br>
s_cr - f_cr: p_value = 0.003 (habitat difference; significant)
f_ncr - f_cr: p_value = 0.972
s_ncr - f_cr: p_value = 0.999
f_ncr - s_cr: p_value = 0.002 (diet:habitat interaction; significant)
s_ncr - s_cr: p_value = 0.003 (diet difference; significant) 
s_ncr - f_ncr: p_value = 0.989   

### STEP 3: Plot differential abundance for all groups in each Phylum.
In this step, we generate box plots to visualize differential OTU abundance at the Order and Genus level within the top three phyla: Firmicutes, Bacteroidetes, and Proteobacteria. To better visualize abundance values, we square root transform all data points.

#### 3a: Plot differential abundance for orders and genera in Bacteroidetes; run the Rmd chunk below.

##### Plot differential abundance within Bacteroidetes: `figures/bacteroidetes.pdf`
```{r}
# Load Bacteroidetes orders data
load("data/bacteroidetes_orders.Rda")

# Get total abundance per individual per order, square root transform data for ease of visualization, and spread the dataset.
bacteroidetes_orders <- bacteroidetes_orders %>%
  dplyr::select(ind_id, group, phylogeny, diet, habitat, sex, order, abundance) %>%
  group_by(ind_id, group, phylogeny, diet, habitat, sex, order) %>%
  summarize(abundance = sum(abundance)) %>%
  mutate(sqrt_abundance = sqrt(abundance)) %>%
  dplyr::select(-abundance) %>%
  spread(order, sqrt_abundance) 

# Rename groups for easier plot visualization
bacteroidetes_orders$group_code <- factor(with(bacteroidetes_orders, ifelse(group=="L.africana_Crop_raider_Savanna", "Laf:S+CR",
                                                                            ifelse(group=="L.africana_Crop_raider_Forest", "Laf:F+CR",
                                                                                   ifelse(group=="L.africana_Savanna", "Laf:S",
                                                                                          ifelse(group=="L.africana_Forest", "Laf:F",
                                                                                                 ifelse(group=="L.cyclotis_Forest", "Lcy:F","done")))))))

# Reorder group codes to be more sensible when plotting
bacteroidetes_orders <- bacteroidetes_orders %>%
  mutate(group_code = fct_relevel(group_code, c("Lcy:F","Laf:F","Laf:F+CR","Laf:S","Laf:S+CR")))

# Plot Order Flavobacteriales
flavobacteriales <- ggplot(bacteroidetes_orders, aes(x = group_code, y = Flavobacteriales)) +  
  geom_boxplot(outlier.shape = NA, show.legend = T) + 
  geom_point(size = 3, color = "black", fill = "grey") +
  theme_set(theme_cowplot(12)) +
  labs(title = "Flavobacteriales", x = "Group", y = "Abundance") + 
  theme(plot.title = element_text(hjust = 0.5)) +
  theme(axis.title.x = element_text(size = 15)) + 
  theme(axis.title.y = element_text(size = 15)) +
  theme(axis.text.x = element_text(size = 15)) + 
  theme(axis.text.y = element_text(size = 15)) +
  theme(legend.position = c(0.65,0.7)) + 
  theme(axis.text.x = element_text(face = "italic")) +   
  theme(panel.border = element_rect(colour = "black", fill = NA)) +
  theme(plot.margin = margin(15,15,15,15))

# Plot Order Bacteroidales
bacteroidales <- ggplot(bacteroidetes_orders, aes(x = group_code, y = Bacteroidales)) +  
  geom_boxplot(outlier.shape = NA, show.legend = T) + 
  geom_point(size = 3, color = "black", fill = "grey") +
  theme_set(theme_cowplot(12)) +
  labs(title = "Bacteroidales", x = "Group", y = "Abundance") + 
  theme(plot.title = element_text(hjust = 0.5)) +
  theme(axis.title.x = element_text(size = 15)) + 
  theme(axis.title.y = element_text(size = 15)) +
  theme(axis.text.x = element_text(size = 15)) + 
  theme(axis.text.y = element_text(size = 15)) +
  theme(legend.position = c(0.65,0.7)) + 
  theme(axis.text.x = element_text(face = "italic")) +   
  theme(panel.border = element_rect(colour = "black", fill = NA)) +
  theme(plot.margin = margin(15,15,15,15))

# Load genus data
load("data/flavobacteriales_genera.Rda")

# Get total abundance per individual per order and spread the dataset to prepare for general linear mixed models.
flavobacteriales_genera <- flavobacteriales_genera %>%
  dplyr::select(ind_id, group, phylogeny, diet, habitat, sex, genus, abundance) %>%
  group_by(ind_id, group, phylogeny, diet, habitat, sex, genus) %>%
  summarize(abundance = sum(abundance)) %>%
  mutate(sqrt_abundance = sqrt(abundance)) %>%
  dplyr::select(-abundance) %>%
  spread(genus, sqrt_abundance) %>%
  filter(phylogeny == "africana")

# Rename groups for easier plot visualization
flavobacteriales_genera$group_code <- factor(with(flavobacteriales_genera, ifelse(group=="L.africana_Crop_raider_Savanna", "Laf:S+CR",
                                                                                  ifelse(group=="L.africana_Crop_raider_Forest", "Laf:F+CR",
                                                                                         ifelse(group=="L.africana_Savanna", "Laf:S",
                                                                                                ifelse(group=="L.africana_Forest", "Laf:F", "done"))))))

# Reorder group codes (group) to be more sensible when plotting
flavobacteriales_genera <- flavobacteriales_genera %>%
  mutate(group_code = fct_relevel(group_code, c("Laf:F","Laf:F+CR","Laf:S","Laf:S+CR")))

# Plot Genus Wautersiella by African elephant species
wautersiella <- ggplot(flavobacteriales_genera, aes(x = group_code, y = Wautersiella)) +  
  geom_boxplot(outlier.shape = NA, show.legend = T) + 
  geom_point(size = 3, color = "black", fill = "grey") +
  theme_set(theme_cowplot(12)) +
  labs(title = "Wautersiella", x = "Group", y = "Abundance") + 
  theme(plot.title = element_text(hjust = 0.5)) +
  theme(axis.title.x = element_text(size = 15)) + 
  theme(axis.title.y = element_text(size = 15)) +
  theme(axis.text.x = element_text(size = 15)) + 
  theme(axis.text.y = element_text(size = 15)) +
  theme(legend.position = c(0.65,0.7)) + 
  theme(axis.text.x = element_text(face = "italic")) +   
  theme(panel.border = element_rect(colour = "black", fill = NA)) +
  theme(plot.margin = margin(15,15,15,15))

# Plot all Bacteroidetes 
pdf("figures/bacteroidetes.pdf", width = 18, height = 6)

plot_grid(flavobacteriales,
          bacteroidales,
          wautersiella,
          ncol = 3)

dev.off()
```

These plots are included as the basis for Figure 2 (a-c) in the final ms.

#### 3b: Plot differential abundance for orders and genera in Firmicutes; run the Rmd chunk below.

##### Plot differential abundance within Firmicutes: `figures/firmicutes.pdf`
```{r}
# Load Firmicutes orders data
load("data/firmicutes_orders.Rda")

# Get total abundance per individual per order, square root transform data for ease of visualization, and spread the dataset.
firmicutes_orders <- firmicutes_orders %>%
  dplyr::select(ind_id, group, phylogeny, diet, habitat, sex, order, abundance) %>%
  group_by(ind_id, group, phylogeny, diet, habitat, sex, order) %>%
  summarize(abundance = sum(abundance)) %>%
  mutate(sqrt_abundance = sqrt(abundance)) %>%
  dplyr::select(-abundance) %>%
  spread(order, sqrt_abundance) 

# Rename groups for easier plot visualization
firmicutes_orders$group_code <- factor(with(firmicutes_orders, ifelse(group=="L.africana_Crop_raider_Savanna", "Laf:S+CR",
                                                                      ifelse(group=="L.africana_Crop_raider_Forest", "Laf:F+CR",
                                                                             ifelse(group=="L.africana_Savanna", "Laf:S",
                                                                                    ifelse(group=="L.africana_Forest", "Laf:F",
                                                                                           ifelse(group=="L.cyclotis_Forest", "Lcy:F","done")))))))

# Reorder group codes to be more sensible when plotting
firmicutes_orders <- firmicutes_orders %>%
  mutate(group_code = fct_relevel(group_code, c("Lcy:F","Laf:F","Laf:F+CR","Laf:S","Laf:S+CR")))

# Plot Order Bacillales
bacillales <- ggplot(firmicutes_orders, aes(x = group_code, y = Bacillales)) +  
  geom_boxplot(outlier.shape = NA, show.legend = T) + 
  geom_point(size = 3, color = "black", fill = "grey") +
  theme_set(theme_cowplot(12)) +
  labs(title = "Bacillales", x = "Group", y = "Abundance") + 
  theme(plot.title = element_text(hjust = 0.5)) +
  theme(axis.title.x = element_text(size = 15)) + 
  theme(axis.title.y = element_text(size = 15)) +
  theme(axis.text.x = element_text(size = 15)) + 
  theme(axis.text.y = element_text(size = 15)) +
  theme(legend.position = c(0.65,0.7)) + 
  theme(axis.text.x = element_text(face = "italic")) +   
  theme(panel.border = element_rect(colour = "black", fill = NA)) +
  theme(plot.margin = margin(15,15,15,15))

# Plot Order Clostridiales
clostridiales <- ggplot(firmicutes_orders, aes(x = group_code, y = Clostridiales)) +  
  geom_boxplot(outlier.shape = NA, show.legend = T) + 
  geom_point(size = 3, color = "black", fill = "grey") +
  theme_set(theme_cowplot(12)) +
  labs(title = "Clostridiales", x = "Group", y = "Abundance") + 
  theme(plot.title = element_text(hjust = 0.5)) +
  theme(axis.title.x = element_text(size = 15)) + 
  theme(axis.title.y = element_text(size = 15)) +
  theme(axis.text.x = element_text(size = 15)) + 
  theme(axis.text.y = element_text(size = 15)) +
  theme(legend.position = c(0.65,0.7)) + 
  theme(axis.text.x = element_text(face = "italic")) +   
  theme(panel.border = element_rect(colour = "black", fill = NA)) +
  theme(plot.margin = margin(15,15,15,15))

# Load genus data
load("data/bacillales_genera.Rda")

# Get total abundance per individual per order and spread the dataset to prepare for general linear mixed models.
bacillales_genera <- bacillales_genera %>%
  dplyr::select(ind_id, group, phylogeny, diet, habitat, sex, genus, abundance) %>%
  group_by(ind_id, group, phylogeny, diet, habitat, sex, genus) %>%
  summarize(abundance = sum(abundance)) %>%
  mutate(sqrt_abundance = sqrt(abundance)) %>%
  dplyr::select(-abundance) %>%
  spread(genus, sqrt_abundance) %>%
  filter(diet == "Nonraider" & habitat == "Forest")

# Rename groups for easier plot visualization
bacillales_genera$group_code <- factor(with(bacillales_genera, ifelse(group=="L.africana_Forest", "Laf:F",
                                                                      ifelse(group=="L.cyclotis_Forest", "Lcy:F","done"))))

# Reorder group codes (species) to be more sensible when plotting
bacillales_genera <- bacillales_genera %>%
  mutate(group_code = fct_relevel(group_code, c("Lcy:F","Laf:F")))

# Plot Genus Bacillus by African elephant species
bacillus <- ggplot(bacillales_genera, aes(x = group_code, y = Bacillus)) +  
  geom_boxplot(outlier.shape = NA, show.legend = T) + 
  geom_point(size = 3, color = "black", fill = "grey") +
  theme_set(theme_cowplot(12)) +
  labs(title = "Bacillus", x = "Group", y = "Abundance") + 
  theme(plot.title = element_text(hjust = 0.5)) +
  theme(axis.title.x = element_text(size = 15)) + 
  theme(axis.title.y = element_text(size = 15)) +
  theme(axis.text.x = element_text(size = 15)) + 
  theme(axis.text.y = element_text(size = 15)) +
  theme(legend.position = c(0.65,0.7)) + 
  theme(axis.text.x = element_text(face = "italic")) +   
  theme(panel.border = element_rect(colour = "black", fill = NA)) +
  theme(plot.margin = margin(15,15,15,15))

# Plot Genus Solibacillus by African elephant species
solibacillus <- ggplot(bacillales_genera, aes(x = group_code, y = Solibacillus)) +  
  geom_boxplot(outlier.shape = NA, show.legend = T) + 
  geom_point(size = 3, color = "black", fill = "grey") +
  theme_set(theme_cowplot(12)) +
  labs(title = "Solibacillus", x = "Group", y = "Abundance") + 
  theme(plot.title = element_text(hjust = 0.5)) +
  theme(axis.title.x = element_text(size = 15)) + 
  theme(axis.title.y = element_text(size = 15)) +
  theme(axis.text.x = element_text(size = 15)) + 
  theme(axis.text.y = element_text(size = 15)) +
  theme(legend.position = c(0.65,0.7)) + 
  theme(axis.text.x = element_text(face = "italic")) +   
  theme(panel.border = element_rect(colour = "black", fill = NA)) +
  theme(plot.margin = margin(15,15,15,15))

# Plot all Firmicutes 
pdf("figures/firmicutes.pdf", width = 18, height = 6)

plot_grid(bacillales,
          clostridiales,
          bacillus,
          solibacillus,
          ncol = 4)

dev.off()
```

These plots are included as the basis for Figure 2 (d-g) in the final ms.

#### 3c: Plot differential abundance for orders and genera in Proteobacteria; run the Rmd chunk below.

##### Plot differential abundance within Proteobacteria: `figures/proteobacteria.pdf`
```{r}
# Load Proteobacteria orders data
load("data/proteobacteria_orders.Rda")

# Get total abundance per individual per order, square root transform data for ease of visualization, and spread the dataset.
proteobacteria_orders <- proteobacteria_orders %>%
  dplyr::select(ind_id, group, phylogeny, diet, habitat, sex, order, abundance) %>%
  group_by(ind_id, group, phylogeny, diet, habitat, sex, order) %>%
  summarize(abundance = sum(abundance)) %>%
  mutate(sqrt_abundance = sqrt(abundance)) %>%
  dplyr::select(-abundance) %>%
  spread(order, sqrt_abundance) 

# Rename groups for easier plot visualization
proteobacteria_orders$group_code <- factor(with(proteobacteria_orders, ifelse(group=="L.africana_Crop_raider_Savanna", "Laf:S+CR",
                                                                              ifelse(group=="L.africana_Crop_raider_Forest", "Laf:F+CR",
                                                                                     ifelse(group=="L.africana_Savanna", "Laf:S",
                                                                                            ifelse(group=="L.africana_Forest", "Laf:F",
                                                                                                   ifelse(group=="L.cyclotis_Forest", "Lcy:F","done")))))))

# Reorder group codes to be more sensible when plotting
proteobacteria_orders <- proteobacteria_orders %>%
  mutate(group_code = fct_relevel(group_code, c("Lcy:F","Laf:F","Laf:F+CR","Laf:S","Laf:S+CR")))

# Plot Order Enterobacteriales
enterobacteriales <- ggplot(proteobacteria_orders, aes(x = group_code, y = Enterobacteriales)) +  
  geom_boxplot(outlier.shape = NA, show.legend = T) + 
  geom_point(size = 3, color = "black", fill = "grey") +
  theme_set(theme_cowplot(12)) +
  labs(title = "Enterobacteriales", x = "Group", y = "Abundance") + 
  theme(plot.title = element_text(hjust = 0.5)) +
  theme(axis.title.x = element_text(size = 15)) + 
  theme(axis.title.y = element_text(size = 15)) +
  theme(axis.text.x = element_text(size = 15)) + 
  theme(axis.text.y = element_text(size = 15)) +
  theme(legend.position = c(0.65,0.7)) + 
  theme(axis.text.x = element_text(face = "italic")) +   
  theme(panel.border = element_rect(colour = "black", fill = NA)) +
  theme(plot.margin = margin(15,15,15,15))

# Plot Order Pseudomonadales
pseudomonadales <- ggplot(proteobacteria_orders, aes(x = group_code, y = Pseudomonadales)) +  
  geom_boxplot(outlier.shape = NA, show.legend = T) + 
  geom_point(size = 3, color = "black", fill = "grey") +
  theme_set(theme_cowplot(12)) +
  labs(title = "Pseudomonadales", x = "Group", y = "Abundance") + 
  theme(plot.title = element_text(hjust = 0.5)) +
  theme(axis.title.x = element_text(size = 15)) + 
  theme(axis.title.y = element_text(size = 15)) +
  theme(axis.text.x = element_text(size = 15)) + 
  theme(axis.text.y = element_text(size = 15)) +
  theme(legend.position = c(0.65,0.7)) + 
  theme(axis.text.x = element_text(face = "italic")) +   
  theme(panel.border = element_rect(colour = "black", fill = NA)) +
  theme(plot.margin = margin(15,15,15,15))

# Load genus data
load("data/pseudomonadales_genera.Rda")

# Get total abundance per individual per order and spread the dataset to prepare for general linear mixed models.
pseudomonadales_genera <- pseudomonadales_genera %>%
  dplyr::select(ind_id, group, phylogeny, diet, habitat, sex, genus, abundance) %>%
  group_by(ind_id, group, phylogeny, diet, habitat, sex, genus) %>%
  summarize(abundance = sum(abundance)) %>%
  mutate(sqrt_abundance = sqrt(abundance)) %>%
  dplyr::select(-abundance) %>%
  spread(genus, sqrt_abundance) %>%
  filter(phylogeny == "africana")

# Rename groups for easier plot visualization
pseudomonadales_genera$group_code <- factor(with(pseudomonadales_genera, ifelse(group=="L.africana_Crop_raider_Savanna", "Laf:S+CR",
                                                                              ifelse(group=="L.africana_Crop_raider_Forest", "Laf:F+CR",
                                                                                     ifelse(group=="L.africana_Savanna", "Laf:S",
                                                                                            ifelse(group=="L.africana_Forest", "Laf:F", "done"))))))

# Reorder group codes (species) to be more sensible when plotting
pseudomonadales_genera <- pseudomonadales_genera %>%
  mutate(group_code = fct_relevel(group_code, c("Laf:F","Laf:F+CR","Laf:S","Laf:S+CR")))

# Plot Genus Acinetobacter by African elephant species
acinetobacter <- ggplot(pseudomonadales_genera, aes(x = group_code, y = Acinetobacter)) +  
  geom_boxplot(outlier.shape = NA, show.legend = T) + 
  geom_point(size = 3, color = "black", fill = "grey") +
  theme_set(theme_cowplot(12)) +
  labs(title = "Acinetobacter", x = "Group", y = "Abundance") + 
  theme(plot.title = element_text(hjust = 0.5)) +
  theme(axis.title.x = element_text(size = 15)) + 
  theme(axis.title.y = element_text(size = 15)) +
  theme(axis.text.x = element_text(size = 15)) + 
  theme(axis.text.y = element_text(size = 15)) +
  theme(legend.position = c(0.65,0.7)) + 
  theme(axis.text.x = element_text(face = "italic")) +   
  theme(panel.border = element_rect(colour = "black", fill = NA)) +
  theme(plot.margin = margin(15,15,15,15))

# Plot all Proteobacteria 
pdf("figures/proteobacteria.pdf", width = 18, height = 6)

plot_grid(enterobacteriales,
          pseudomonadales,
          acinetobacter,
          ncol = 3)

dev.off()
```

These plots are included as the basis for Figure 2 (h-j) in the final ms.

### ----------------------- END OF PHASE 2: DIFFERENTIAL OTU ABUNDANCE ANALYSIS  ----------------------- ###

## PHASE 3: OTU DIVERSITY ANALYSIS
In this phase of the analysis, we assess differences in microbiome composition, diversity, and structure among African elephant groups using a multivariate framework. Importantly, we use the FULL microbial dataset (with all, raw OTUs and their associated abundances included in the full dataset) for these analyses. We first calculate intrinsic diversity within each elephant group of interest (alpha diversity) and compare metrics among groups using linear mixed effect models. We then assess differences microbial community structure and composition among groups (beta diversity) by considering all microbial OTU abundances as response variables in PCOA. Finally, we use beta dispersion to assess differences in within-group variances around mean microbial diversity.

### STEP 1: Read in and prepare data for all downstream analyses, and compute alpha diversity for each sample.
In this step, we convert the full order dataset into long format, such that individual OTUs and their corresponding abundances within African elephant groups are columns in a matrix, with metadata as appended columns at the lefthand side of the table. This order matrix is then pared down to include only the groups of interest for downstream analyes.

#### 1a: Read in order data and convert to long format; run the Rmd chunk below.

##### Read in order data and convert to long format:
```{r}
# Load full dataset
load("../03_core_microbiome_analysis/data/full_data.Rda")

# Select only columns needed for analysis and spread so that unique OTUs are columns
otu_matrix <- full_data %>%
  dplyr::select(ind_id, otu_id, phylogeny, diet, habitat, age, sex, group, abundance) %>%
  spread(otu_id, abundance) %>%
  column_to_rownames("ind_id")
  
# save otu matrix for downstream analyses
save(otu_matrix, file = "data/otu_matrix.Rda")
```

### STEP 2: Assess alpha diversity within African elephant groups of interest.
In this step, we calculate alpha diversity as the Shannon Diversity Index within African elephant phylogenetic groups (i.e., species: <i>L. africana</i> and <i>L. cyclotis</i>), diets (crop-raiders and non-crop-raiders), and habitats (forest and savanna). We compare alpha diversity among groups using linear mixed effect models, including sex as a random effect in each model.

#### 2a: Compute alpha diversity across samples; run the Rmd chunk below:

##### Compute alpha diversity across samples:
```{r}
# Load otu matrix
load("data/otu_matrix.Rda")

# Load sample metadata
load("../02_data_summary_analysis/data/metadata.Rda")

# calculate Shannon diversity for each individual in the dataset
shannon <- otu_matrix %>%
  mutate(shannon_diversity = diversity(otu_matrix[7:ncol(otu_matrix)], index = "shannon")) %>%
  dplyr::select(phylogeny, diet, habitat, sex, age, shannon_diversity)

# Summarize mean, standard deviation (sd), and range (minimum - maximum) of alpha diversity across samples
shannon %>%
  summarize(mean = mean(shannon_diversity),
            sd = sd(shannon_diversity),
            min = min(shannon_diversity),
            max = max(shannon_diversity))

# Save shannon diversity data for downstream analyses
save(shannon, file = "data/shannon.Rda")
```

<b> Alpha diversity metrics</b>: <br>
mean = 4.964 (1.209)
range = 2.047 - 6.507

#### 2b: Calculate alpha diversity wihtin and between species and compare; run the Rmd chunk below:

##### Calculate alpha diversity within species and compare:
```{r}
# Load shannon diversity data
load("data/shannon.Rda")

# Get only forest non-crop-raiders to assess species differences
shannon <- shannon %>%
  filter(diet == "Nonraider" & habitat == "Forest")

# Summarize alpha diversity metrics by species
shannon %>%
  group_by(phylogeny) %>%
  summarize(mean = mean(shannon_diversity),
            sd = sd(shannon_diversity),
            min = min(shannon_diversity),
            max = max(shannon_diversity))
  

# Run linear mixed effect model by species
lme <- lme(shannon_diversity ~ phylogeny, random = ~1|sex, data = shannon)

# Get statistics
anova(lme)
```

<b> Alpha diversity metrics by species </b>: <br>
range = 2.047 - 6.324
mean<sub><i>L. africana</i> = 4.681 (0.996)
mean<sub><i>L. cyclotis</i> = 4.186 (0.149)

<b> Comparison of alpha diversity by species </b>: <br>
p_value = 0.449, F = 0.604

#### 2c: Calculate alpha diversity wihtin and between diet and habitat and compare; run the Rmd chunk below:

##### Calculate alpha diversity within diet and habitat and compare:
```{r}
# Load shannon diversity data
load("data/shannon.Rda")

# Get only forest non-crop-raiders to assess species differences
shannon <- shannon %>%
  filter(phylogeny == "africana")

# Summarize alpha diversity metrics by diet
shannon %>%
  group_by(diet) %>%
  summarize(mean = mean(shannon_diversity),
            sd = sd(shannon_diversity),
            min = min(shannon_diversity),
            max = max(shannon_diversity))
  
# Summarize alpha diversity metrics by habitat
shannon %>%
  group_by(habitat) %>%
  summarize(mean = mean(shannon_diversity),
            sd = sd(shannon_diversity),
            min = min(shannon_diversity),
            max = max(shannon_diversity))

# Run linear mixed effect model by diet and habitat, including an interaction for habitat and diet
lme <- lme(shannon_diversity ~ diet*habitat, random = ~1|sex, data = shannon)

# Get statistics
Anova(lme)
```

<b> Alpha diversity metrics by diet </b>: <br>
mean<sub><i>crop-raider</i> = 5.463 (0.681)
mean<sub><i>non-crop-raider</i> = 5.034 (1.164)

<b> Alpha diversity metrics by habitat </b>: <br>
mean<sub><i>savanna</i> = 5.336 (1.128)
mean<sub><i>forest</i> = 5.151 (0.884)

<b> Comparison of alpha diversity by diet and habitat </b>: <br>
diet: p_value = 0.207, F = 1.662
habitat: p_value = 2.216, F = 1.596
diet*habitat: p_value = 0.757, F = 0.098

#### 2d: Plot alpha diversity for each group of interest; run the Rmd chunk below.
Here, we plot alpha diversity as box plots for only major groups of interest (phylogeny, diet, and habitat).

##### Plot alpha diversity in major groups of interest: `figures/alpha.pdf`
```{r}
# Load shannon diversity data
load("data/shannon.Rda")

# Rename groups for easier plot visualization
shannon$species <- factor(with(shannon, ifelse(phylogeny=="africana", "L. africana",
                                               ifelse(phylogeny=="cyclotis", "L. cyclotis", "done"))))

# Get only forest non-crop-raiders for phylogeny plot
shannon_phylogeny <- shannon %>%
  filter(diet == "Nonraider" & habitat == "Forest")

# Plot alpha diversity by species
phylogeny <- ggplot(shannon_phylogeny, aes(x = species, y = shannon_diversity, fill = species)) +  
  geom_boxplot(outlier.shape = NA, show.legend = F, alpha = 0.8) + 
  geom_point(size = 3, color = "black", fill = "grey") +
  theme_set(theme_cowplot(12)) +
  labs(x = "Species", y = "Shannon Diversity") + 
  scale_fill_manual(values = c("red", "blue")) +
  theme(plot.title = element_text(hjust = 0.5)) +
  theme(axis.title.x = element_text(size = 15)) + 
  theme(axis.title.y = element_text(size = 15)) +
  theme(axis.text.x = element_text(size = 15)) + 
  theme(axis.text.y = element_text(size = 15)) +
  theme(legend.position = c(0.65,0.7)) + 
  theme(axis.text.x = element_text(face = "italic")) +   
  theme(panel.border = element_rect(colour = "black", fill = NA)) +
  theme(plot.margin = margin(15,15,15,15))

# Get only L. africana for diet and habitat plots
shannon_diet_habitat <- shannon %>%
  filter(phylogeny == "africana")

# Plot alpha diversity by diet
diet <- ggplot(shannon_diet_habitat, aes(x = diet, y = shannon_diversity, fill = diet)) +  
  geom_boxplot(outlier.shape = NA, show.legend = F, alpha = 0.8) + 
  geom_point(size = 3, color = "black", fill = "grey") +
  theme_set(theme_cowplot(12)) +
  labs(x = "Diet", y = "Shannon Diversity") + 
  scale_fill_manual(values = c("midnightblue", "deeppink1")) +
  theme(plot.title = element_text(hjust = 0.5)) +
  theme(axis.title.x = element_text(size = 15)) + 
  theme(axis.title.y = element_text(size = 15)) +
  theme(axis.text.x = element_text(size = 15)) + 
  theme(axis.text.y = element_text(size = 15)) +
  theme(legend.position = c(0.65,0.7)) + 
  theme(panel.border = element_rect(colour = "black", fill = NA)) +
  theme(plot.margin = margin(15,15,15,15))

# Plot alpha diversity by habitat
habitat <- ggplot(shannon_diet_habitat, aes(x = habitat, y = shannon_diversity, fill = habitat)) +  
  geom_boxplot(outlier.shape = NA, show.legend = F, alpha = 0.8) + 
  geom_point(size = 3, color = "black", fill = "grey") +
  theme_set(theme_cowplot(12)) +
  labs(x = "Habitat", y = "Shannon Diversity") + 
  scale_fill_manual(values = c("forestgreen", "coral1")) +
  theme(plot.title = element_text(hjust = 0.5)) +
  theme(axis.title.x = element_text(size = 15)) + 
  theme(axis.title.y = element_text(size = 15)) +
  theme(axis.text.x = element_text(size = 15)) + 
  theme(axis.text.y = element_text(size = 15)) +
  theme(legend.position = c(0.65,0.7)) + 
  theme(panel.border = element_rect(colour = "black", fill = NA)) +
  theme(plot.margin = margin(15,15,15,15))

# Plot all Alpha diversity 
pdf("figures/alpha.pdf", width = 6, height = 13)

plot_grid(phylogeny,
          diet,
          habitat,
          ncol = 1)

dev.off()
```

This figure was used as the basis for Figure S2 in the final ms.

#### 2e: Calculate alpha diversity wihtin and between age groups; run the Rmd chunk below:
In this step, we test whether there are differences in alpha diversity between age groups to determine whether age should be included in any other analyses of differential OTU abundance or beta diversity.

##### Calculate alpha diversity within diet and habitat and compare:
```{r}
# Load shannon diversity data
load("data/shannon.Rda")

# Get only L. africana individuals, because all individuals in L. cyclotis are adults
shannon <- shannon %>%
  filter(phylogeny == "africana")

# Summarize alpha diversity metrics by age
shannon %>%
  group_by(age) %>%
  summarize(mean = mean(shannon_diversity),
            sd = sd(shannon_diversity),
            min = min(shannon_diversity),
            max = max(shannon_diversity))
  
# Run linear mixed effect model by diet and habitat, including an interaction for habitat and diet
lme <- lme(shannon_diversity ~ age, random = ~1|sex, data = shannon)

# Get statistics
anova(lme)
```

<b> Alpha diversity metrics by age </b>: <br>
mean<sub><i>adult</i> = 5.214 (1.178)
mean<sub><i>sub-adult</i> = 5.182 (0.865)
mean<sub><i>juvenile</i> = 5.462 (0.205)

<b> Comparison of alpha diversity by age </b>: <br>
diet: p_value = 0.885, F = 0.123 (not significant - age therefore was not included in any previous or subsequent analyses)

### STEP 3: Assess beta diversity within African elephant groups of interest.
In this step, we calculate beta diversity among African elephant phylogenetic groups (i.e., species: <i>L. africana</i> and <i>L. cyclotis</i>), diets (crop-raiders and non-crop-raiders), and habitats (forest and savanna) using PCOA. We test for differences in microbiome structure and composition using PERMANOVA models, including sex as a random stratum in each model.

#### 3a: Calculate standardized abundance within and Bray-curtis distances among each individual to prepare data for beta dispersion and PERMANOVA; run the Rmd chunk below:

##### Standardize abundance and calculate Bray-Curtis distance among individual samples in the data:
```{r}
# Load OTU matrix
load("data/otu_matrix.Rda")

# Standardize abundance data by dividing each number by the total number of samples (individuals in the dataset)
std_abundance <- data.frame(decostand(otu_matrix[7:ncol(otu_matrix)], "total"))

# Column bind otu_matrix with standradized abundance
otu_matrix <- cbind(otu_matrix[1:6],
                    std_abundance)

# Get only L. africana elephants for diet and habitat analysis
phylogeny_matrix <- otu_matrix %>%
  filter(diet == "Nonraider" & habitat == "Forest")

# Get only L. africana elephants for diet and habitat analysis
diet_habitat_matrix <- otu_matrix %>%
  filter(phylogeny == "africana")

# Calculate Bray-curtis distances between species
bray_curtis_phylogeny <- vegdist(phylogeny_matrix[7:ncol(phylogeny_matrix)], "bray") 

# Calculate Bray-curtis distances between diets and habitats
bray_curtis_diet_habitat <- vegdist(diet_habitat_matrix[7:ncol(diet_habitat_matrix)], "bray") 

# Save phylogeny matrix for downstream analyses
save(phylogeny_matrix, file = "data/phylogeny_matrix.Rda")

# Save diet and habitat matrix for downstream analyses
save(diet_habitat_matrix, file = "data/diet_habitat_matrix.Rda")

# Save bray-curtis phylogeny data for downstream analyses
save(bray_curtis_phylogeny, file = "data/bray_curtis_phylogeny.Rda")

# Save bray-curtis diet and habitat data for downstream analyses
save(bray_curtis_diet_habitat, file = "data/bray_curtis_diet_habitat.Rda")
```

#### 3b: Run beta dispersion and PERMANOVA by species; run the Rmd chunk below:

##### Run beta dispersion and PERMANOVA by African elephant species:
```{r}
# Load bray-curtis phylogeny data
load("data/bray_curtis_phylogeny.Rda")

# Load phylogeny matrix
load("data/phylogeny_matrix.Rda")

# Run beta dispersion for  species
beta <- betadisper(bray_curtis_phylogeny, factor(phylogeny_matrix[,1]), type = c("centroid"))

# Get statistics for beta dispersion
anova(beta)

# Run PERMANOVA
permanova <- adonis2(bray_curtis_phylogeny ~ phylogeny, 
                     phylogeny_matrix, 
                     Strata = sex, 
                     distance = "bray", 
                     permutations = 9999)
```

<b> Beta dispersion by species </b>: <br>
p_value = 0.435, F = 0.641

<b> PERMANOVA by species </b>: <br>
p-value = 0.001, F = 3.527

#### 3c: Run beta dispersion and PERMANOVA by diet and habitat; run the Rmd chunk below:

##### Run beta dispersion and PERMANOVA by African elephant diet and habitat:
```{r}
# Load bray-curtis diet and habitat data
load("data/bray_curtis_diet_habitat.Rda")

# Load diet and habitat matrix
load("data/diet_habitat_matrix.Rda")

# Run beta dispersion for diet
beta_diet <- betadisper(bray_curtis_diet_habitat, factor(diet_habitat_matrix[,2]), type = c("centroid"))

# Run beta dispersion for habitat
beta_habitat <- betadisper(bray_curtis_diet_habitat, factor(diet_habitat_matrix[,3]), type = c("centroid"))

# Get statistics for beta dispersion by diet
anova(beta_diet)

# Get statistics for beta dispersion by habitat
anova(beta_habitat)

# Run PERMANOVA by diet
permanova_diet <- adonis2(bray_curtis_diet_habitat ~ diet, 
                          diet_habitat_matrix, 
                          Strata = sex, 
                          distance = "bray", 
                          permutations = 9999)

# Run PERMANOVA by habitat
permanova_habitat <- adonis2(bray_curtis_diet_habitat ~ habitat, 
                             diet_habitat_matrix, 
                             Strata = sex, 
                             distance = "bray", 
                             permutations = 9999)
```

<b> Beta dispersion by diet </b>: <br>
p_value = 0.122, F = 2.514

<b> Beta dispersion by diet </b>: <br>
p_value = 0.102, F = 2.827

<b> PERMANOVA by diet </b>: <br>
p-value = 0.012, F = 1.861

<b> PERMANOVA by habitat </b>: <br>
p-value < 0.001, F = 2.874

#### 3c: Plot beta diversity results using NMDS; run the Rmd chunk below:
In this step, we use non-parametric multidimensional scaling to plot beta diversity among African elephant groups.

##### Plot beta diversity in major groups of interest: `figures/beta.pdf`
```{r}
# Load bray-curtis phylogeny data
load("data/bray_curtis_phylogeny.Rda")

# Load phylogeny matrix
load("data/phylogeny_matrix.Rda")

# Load bray-curtis diet and habitat data
load("data/bray_curtis_diet_habitat.Rda")

# Load diet and habitat matrix
load("data/diet_habitat_matrix.Rda")

# Get multidimentional scaling for phylogeny
mds_phylogeny <- metaMDS(bray_curtis_phylogeny,
                         trace = FALSE, 
                         autotransform = F)

# Get NMDS scores for phylogeny
scores_phylogeny <- data.frame(scores(mds_phylogeny))

# Column bind NMDS scores to phylogeny metadata
scores_phylogeny <- cbind(phylogeny_matrix[1:6],
                          scores_phylogeny)

# Rename groups for easier plot visualization
scores_phylogeny$species <- factor(with(scores_phylogeny, ifelse(phylogeny=="africana", "L. africana",
                                                                 ifelse(phylogeny=="cyclotis", "L. cyclotis", "done"))))

# Get NMDS centroids for phylogeny
centroids_phylogeny <- aggregate(cbind(NMDS1, NMDS2) ~ species, 
                                 scores_phylogeny, 
                                 mean)

# Get variance on centroids
gg_phylogeny <- merge(scores_phylogeny, 
                      aggregate(cbind(mean.x = NMDS1, 
                                      mean.y = NMDS2) ~ species, 
                      scores_phylogeny, 
                      mean), 
                      by = "species")

# Get phylogeny beta diversity NMDS
phylogeny <- ggplot(gg_phylogeny, aes(NMDS1, NMDS2, color = species)) + 
  geom_point(size = 2, show.legend = T) + 
  geom_point(aes(x = mean.x, y = mean.y), size = 2, show.legend = T) + 
  geom_segment(aes(x = mean.x, y = mean.y, xend = NMDS1, yend = NMDS2), show.legend = T) + 
  theme_set(theme_cowplot(12)) +
  theme(legend.position=c(0.01,0.85)) + 
  scale_color_manual(values = c("red2", "blue2")) + 
  labs(color = "Elephant Species", y = "NMDS 2") +  
  theme(axis.title.x = element_blank()) + theme(legend.text = element_text(face = "italic")) +
  theme(plot.title = element_text(size = 20, hjust = 0.5, face = "bold")) +
  theme(panel.border = element_rect(colour = "black", fill = NA)) +
  theme(legend.position=c(0.01,0.85)) +
  theme(plot.margin = margin(15,15,15,15))

# Get multidimentional scaling for diet
mds_diet <- metaMDS(bray_curtis_diet_habitat,
                    trace = FALSE, 
                    autotransform = F)

# Get NMDS scores for diet
scores_diet <- data.frame(scores(mds_diet))

# Column bind NMDS scores to phylogeny metadata
scores_diet <- cbind(diet_habitat_matrix[1:6],
                     scores_diet)

# Get NMDS centroids for phylogeny
centroids_diet <- aggregate(cbind(NMDS1, NMDS2) ~ diet, 
                            scores_diet, 
                            mean)

# Get variance on centroids
gg_diet <- merge(scores_diet, 
                 aggregate(cbind(mean.x = NMDS1, 
                                 mean.y = NMDS2) ~ diet, 
                           scores_diet, 
                           mean), 
                 by = "diet")

# Get phylogeny beta diversity NMDS
diet <- ggplot(gg_diet, aes(NMDS1, NMDS2, color = diet)) + 
  geom_point(size = 2, show.legend = T) + 
  geom_point(aes(x = mean.x, y = mean.y), size = 2, show.legend = T) + 
  geom_segment(aes(x = mean.x, y = mean.y, xend = NMDS1, yend = NMDS2), show.legend = T) + 
  theme_set(theme_cowplot(12)) +
  theme(legend.position=c(0.01,0.85)) + 
  scale_color_manual(values = c("midnightblue", "deeppink1")) + 
  labs(color = "Elephant Diet", y = "NMDS 2") +  
  theme(plot.title = element_text(size = 20, hjust = 0.5, face = "bold")) +
  theme(panel.border = element_rect(colour = "black", fill = NA)) +
  theme(legend.position = c(0.01,0.85)) + 
  theme(plot.margin = margin(15,15,15,15))

# Get multidimentional scaling for habitat
mds_habitat <- metaMDS(bray_curtis_diet_habitat,
                       trace = FALSE, 
                       autotransform = F)

# Get NMDS scores for habitat
scores_habitat <- data.frame(scores(mds_habitat))

# Column bind NMDS scores to phylogeny metadata
scores_habitat <- cbind(diet_habitat_matrix[1:6],
                        scores_habitat)

# Get NMDS centroids for phylogeny
centroids_habitat <- aggregate(cbind(NMDS1, NMDS2) ~ habitat, 
                            scores_habitat, 
                            mean)

# Get variance on centroids
gg_habitat <- merge(scores_habitat, 
                 aggregate(cbind(mean.x = NMDS1, 
                                 mean.y = NMDS2) ~ habitat, 
                           scores_habitat, 
                           mean), 
                 by = "habitat")

# Get phylogeny beta diversity NMDS
habitat <- ggplot(gg_habitat, aes(NMDS1, NMDS2, color = habitat)) + 
  geom_point(size = 2, show.legend = T) + 
  geom_point(aes(x = mean.x, y = mean.y), size = 2, show.legend = T) + 
  geom_segment(aes(x = mean.x, y = mean.y, xend = NMDS1, yend = NMDS2), show.legend = T) + 
  theme_set(theme_cowplot(12)) +
  theme(legend.position=c(0.01,0.85)) + 
  scale_color_manual(values = c("forestgreen", "coral1")) + 
  labs(color = "Elephant Habitat", y = "NMDS 2") +  
  theme(plot.title = element_text(size = 20, hjust = 0.5, face = "bold")) +
  theme(panel.border = element_rect(colour = "black", fill = NA)) +
  theme(legend.position = c(0.01,0.85)) + 
  theme(plot.margin = margin(15,15,15,15))

# Plot all figures
pdf("figures/beta.pdf", width = 6, height = 12) 

plot_grid(phylogeny, 
          diet, 
          habitat, 
          ncol = 1)

dev.off()
```

This figure is the basis for Figure 3 in the final ms.

### ----------------------- END OF PHASE 3: OTU DIVERSITY ANALYSIS ----------------------- ###

### ----------------------- END OF ANALYSIS 4: MICROBIOME DIVERSITY AND DIFFERENTIAL OTU ABUNDANCE ANALYSIS  ----------------------- ###