---
title: "Analysis 4: Microbiome diversity and differential OTU abundnace analysis"
author: "Joe Gunn"
date: "2024-02-07"
output: html_document
---

# Project: Assessing effects of phylogeny, diet, and habitat on gut microbial composition in African elephants 
We characterized the gut microbiome of African Savanna elephants (<i>Loxodonta africana</i>) and African Forest elephants (<i>Loxodonta cyclotis</i>). We assessed the relationship of gut microbial composition, including analyses of alpha and beta diversity, with host phylogeny (i.e., species) and habitat type (i.e., forest or savanna) for both species. We also assessed the relationship between microbial composition and diet (i.e., crop-raiding vs. non-crop-raiding) within <i>L. africana</i>. 

## Specific Aim: Assessing differential OTU abundance and diverity within (alpha) and between (beta) African elephant sample groups (phylogeny, diet, and habitat) 
In this analysis, we assess differential abundance in relatively highly represented OTUs (i.e., within the most highly represented phyla) across African elephant sample groups of interest (phylogeny, diet, and habitat) to determine whether any major taxonomic groups explain variation between groups. We then assess both within-group (alpha) microbial diversity and between-group (beta) microbial diversity to determine whether there are overarching differences in microbiome structure and composition among elephant groups.  

## Phases of analysis:
### Phase 1: Data read-in, identification of major OTU groups, and filtering
### Phase 2: Differential OTU abundance analysis
### Phase 3: OTU diveristy analysis

## Libraries needed for analysis
```{r}
library(readxl)
library(cowplot)
library(tidyverse)
library(phyloseq)
library(lme4)
library(car)
library(multcomp)
library(devtools)
library(qiime2R)
library(vegan)
library(ape)
```

## PHASE 1: DATA READ-IN, IDENTIFICATION OF MAJOR OTU GROUPS, AND FILTERING
In this phase of the analysis, we read in the full, clean Order-level dataset from Analysis 3 ("order_data.Rda") and identify the most relatively abundant taxonomic groups. We then filter the full dataset into separate datasets by taxonomic group for downstream analyses. In Phase 2, we then assess differential abundance only in microbial orders that have an average relative abundance of 10% across samples.

### STEP 1: Identify phyla with the highest relative abundance for the full microbial OTU dataset and filter the full dataset to include only top phyla.

#### 1a: Determine relative abundance at the phylum level; run the Rmd chunk below.

##### Obtain most abundant phyla in the dataset:
```{r}
# Load order dataset
load("../03_core_microbiome_analysis/data/order_data.Rda")

# Set scipen options to remove scientific notation
options(scipen = 999)

# Calculate total microbial abundance across samples
total_abundance <- sum(order_data$abundance)

# Calculate abundance and relative abundance across samples for each phylum represented in the dataset
phylum_abundance <- order_data %>%
  group_by(phylum) %>%
  summarize(abundance = sum(abundance),
            relative_abundance = abundance/total_abundance) %>%
  arrange(desc(relative_abundance))
```

<b> Data filtering results </b>: <br>
Phyla with greater than 10% relative abundance: Firmicutes, Proteobacteria, Bacteroidetes

#### 1b: Filter the full dataset to include only high-proportiopn phyla; run the Rmd chunk below.

### Filter full data into separate datasets for each major phylum:
```{r}
# Get dataset with only phylum firmicutes
firmicutes <- order_data %>%
  filter(phylum == "Firmicutes")

# Get dataset with only phylum proteobacteria
proteobacteria <- order_data %>%
  filter(phylum == "Proteobacteria")

# Get dataset with only phylum bacteroidetes
bacteroidetes <- order_data %>%
  filter(phylum == "Bacteroidetes")

# Save phylum datasets for downstream analyses
save(firmicutes, file = "data/firmicutes.Rda")
save(proteobacteria, file = "data/proteobacteria.Rda")
save(bacteroidetes, file = "data/bacteroidetes.Rda")
```

### STEP 2: Determine relative abundance at the order level within phyla and filter the datasets to include only high-proportion orders.

#### 2a. Determine relative abundance and filter orders in phylum Firmicutes.

##### 2a.1. Determine relative abundance of orders within phylum Firmicutes; run the Rmd chunk below.

##### Get top orders in phylum Firmicutes:
```{r}
# Load Firmicutes data
load("data/firmicutes.Rda")

# Get total abundance across samples
total_abundance <- sum(firmicutes$abundance)

# Calculate abundance and relative abundance across samples for each order represented in the dataset
order_abundance <- firmicutes %>%
  group_by(order) %>%
  summarize(abundance = sum(abundance),
            relative_abundance = abundance/total_abundance) %>%
  arrange(desc(relative_abundance))
```

<b> Data filtering results </b>: <br>
Orders with greater than 10% relative abundance: Clostridiales, Bacillales

##### 2a.2. Filter the full dataset to include only high-proportiopn orders; run the Rmd chunk below.

### Filter data to only include major orders:
```{r}
# Load Firmicutes data
load("data/firmicutes.Rda")

# Get dataset with only top orders
top_firmicutes <- firmicutes %>%
  filter(order == "Clostridiales" | 
           order == "Bacillales")

# Save order dataset for downstream analyses
save(top_firmicutes, file = "data/top_firmicutes.Rda")
```

#### 2b. Determine relative abundance and filter orders in phylum Proteobacteria.

##### 2b.1. Determine relative abundance of orders within phylum Proteobacteria; run the Rmd chunk below.

##### Get top orders in phylum Proteobacteria:
```{r}
# Load Proteobacteria data
load("data/proteobacteria.Rda")

# Get total abundance across samples
total_abundance <- sum(proteobacteria$abundance)

# Calculate abundance and relative abundance across samples for each order represented in the dataset
order_abundance <- proteobacteria %>%
  group_by(order) %>%
  summarize(abundance = sum(abundance),
            relative_abundance = abundance/total_abundance) %>%
  arrange(desc(relative_abundance))
```

<b> Data filtering results </b>: <br>
Orders with greater than 10% relative abundance: Pseudomonadales, Enterobacteriales

##### 2b.2. Filter the full dataset to include only high-proportiopn orders; run the Rmd chunk below.

### Filter data to only include major orders:
```{r}
# Load Proteobacteria data
load("data/proteobacteria.Rda")

# Get dataset with top Proteobacteria orders 
top_proteobacteria <- proteobacteria %>%
  filter(order == "Pseudomonadales" |
           order == "Enterobacteriales")

# Save order dataset for downstream analyses
save(top_proteobacteria, file = "data/top_proteobacteria.Rda")
```

#### 2c. Determine relative abundance and filter orders in phylum Bacteroidetes

##### 2c.1. Determine relative abundance of orders within phylum Bacteroidetes; run the Rmd chunk below.

##### Get top orders in phylum Bacteroidetes:
```{r}
# Load Bacteroidetes data
load("data/bacteroidetes.Rda")

# Get total abundance across samples
total_abundance <- sum(bacteroidetes$abundance)

# Calculate abundance and relative abundance across samples for each order represented in the dataset
order_abundance <- bacteroidetes %>%
  group_by(order) %>%
  summarize(abundance = sum(abundance),
            relative_abundance = abundance/total_abundance) %>%
  arrange(desc(relative_abundance))
```

<b> Data filtering results </b>: <br>
Orders with greater than 10% relative abundance: Bacteroidales, Flavobacteriales

##### 2c.2. Filter the full dataset to include only high-proportiopn orders; run the Rmd chunk below.

### Filter data to only include major orders:
```{r}
# Load Bacteroidetes data
load("data/bacteroidetes.Rda")

# Get dataset with top Bacteroidetes orders
top_bacteroidetes <- order_data %>%
  filter(order == "Bacteroidales" | 
           order == "Flavobacteriales")

# Save order dataset for downstream analyses
save(top_bacteroidetes, file = "data/top_bacteroidetes.Rda")
```

### ----------------------- END OF PHASE 1: DATA READ-IN, IDENTIFICATION OF MAJOR OTU GROUPS, AND FILTERING ----------------------- ###

## PHASE 2: DIFFERENTIAL OTU ABUNDANCE ANALYSIS
In this phase of our analysis, we assess differences in microbial OTU abundance between African elephant groups of interest (i.e., phylogeny, diet, and habitat) at the Order level. Where possible and appropriate, we also assess differential abundance at the species level. We use a general linear mixed model (using individual sex, i.e., male or female, as a random effect) framework, assuming a negative-binomial (i.e., zero-inflated) distribution. We use post-hoc Tukey tests and p-value correction (false discovery rate) to conduct pairwise comparisons of each elephant group within taxonomic levels.

### STEP 1: Assess differential abundance in phylum Firmutes
In this step, we assess differential abundance among orders and genera in phylum Firmicutes.

#### 1a: Modify top firmicutes dataset for running general linear mixed models; run the Rmd chunk below.

##### Modify top firmicutes dataset for downstream general linear mixed models.
```{r}
# Load top Firmicutes data
load("data/top_firmicutes.Rda")

# Get total abundance per individual per order, square-root transform abundance, and spread the dataset to prepare for general linear mixed models.
firmicutes_orders <- top_firmicutes %>%
  dplyr::select(ind_id, group, sex, order, abundance) %>%
  group_by(ind_id, group, sex, order) %>%
  summarize(abundance = sum(abundance)) %>%
  mutate(sqrt_abundance = sqrt(abundance)) %>%
  dplyr::select(-abundance) %>%
  spread(order, sqrt_abundance) 

# Save order dataset for downstream analysis
save(firmicutes_orders, file = "data/firmicutes_orders.Rda")
```

#### 1b: Run general linear mixed models for each order to test for differential abundance among African elephant groups at the order level; run the Rmd chunk below:
In this step, we run general linear mixed models for the relationship of elephant group and order abundance (square-root transformed), with sex as a random effect and assuming a negative-binomial distribution, in a for loop construct, calculating p-values for each order column in the data.

##### Assess differential abundance across elephant groups within orders in phylum Firmicutes: 
```{r}
# Load firmicutes order data
load("data/firmicutes_orders.Rda")

# Generate table to hold p-values for each analysis
p <- data.frame(order = factor(c("Bacillales", "Clostridiales")), p_value = numeric(length = 2))

# Run general linear mixed model in for loop
for (ii in 4:ncol(firmicutes_orders)) {
      
  #col = firmicutes_orders[,ii] # iterate through order columns (i.e., 4:ncol)
  ff <- as.formula(paste0(colnames(firmicutes_orders)[ii]," ~ group + (1|sex)")) # specify model
  glmer <- glmer.nb(ff, firmicutes_orders) # run linear model           
  p_value <- summary(glmer)$coefficients[2,4] #extract p value
  p[ii-3,2] <- p_value # put p value in table
      
  }

# Adjust p-values for false discovery rate
p_fdr <- cbind(p, 
               data.frame(fdr_p = p.adjust(p$p_value, method = "fdr")), 
               data.frame(result = with(p_fdr, ifelse(fdr_p<0.05, "significant", "not-significant"))))
```

### STEP 2: Assess differential abundance in phylum Proteobacteria
In this step, we assess differential abundance among orders and genera in phylum Proteobacteria

#### 2a: Modify top proteobacteria dataset for running general linear mixed models; run the Rmd chunk below.

##### Modify top proteobacteria dataset for downstream general linear mixed models.
```{r}
# Load top Proteobacteria data
load("data/top_proteobacteria.Rda")

# Get total abundance per individual per order, square-root transform abundance, and spread the dataset to prepare for general linear mixed models.
proteobacteria_orders <- top_proteobacteria %>%
  dplyr::select(ind_id, group, sex, order, abundance) %>%
  group_by(ind_id, group, sex, order) %>%
  summarize(abundance = sum(abundance)) %>%
  mutate(sqrt_abundance = sqrt(abundance)) %>%
  dplyr::select(-abundance) %>%
  spread(order, sqrt_abundance) 

# Save order dataset for downstream analysis
save(proteobacteria_orders, file = "data/proteobacteria_orders.Rda")
```

#### 2b: Run general linear mixed models for each order to test for differential abundance among African elephant groups at the order level; run the Rmd chunk below:
In this step, we run general linear mixed models for the relationship of elephant group and order abundance (square-root transformed), with sex as a random effect and assuming a negative-binomial distribution, in a for loop construct, calculating p-values for each order column in the data.

##### Assess differential abundance across elephant groups within orders in phylum Proteobacteria: 
```{r}
# Load proteobacteria order data
load("data/proteobacteria_orders.Rda")

# Generate table to hold p-values for each analysis
p <- data.frame(order = factor(c("Enterobacteriales", "Pseudomonadales")), p_value = numeric(length = 2))

# Run general linear mixed model in for loop
for (ii in 4:ncol(proteobacteria_orders)) {
      
  #col = firmicutes_orders[,ii] # iterate through order columns (i.e., 4:ncol)
  ff <- as.formula(paste0(colnames(proteobacteria_orders)[ii]," ~ group + (1|sex)")) # specify model
  glmer <- glmer.nb(ff, proteobacteria_orders) # run linear model           
  p_value <- summary(glmer)$coefficients[2,4] #extract p value
  p[ii-3,2] <- p_value # put p value in table
      
  }

# Adjust p-values for false discovery rate
p_fdr <- cbind(p, 
               data.frame(fdr_p = p.adjust(p$p_value, method = "fdr")), 
               data.frame(result = with(p_fdr, ifelse(fdr_p<0.05, "significant", "not-significant"))))
```

### STEP 3: Assess differential abundance in phylum Proteobacteria
In this step, we assess differential abundance among orders and genera in phylum Bacteroidetes.

#### 3a: Modify top bacteroidetes dataset for running general linear mixed models; run the Rmd chunk below.

##### Modify top bacteroidetes dataset for downstream general linear mixed models.
```{r}
# Load top Bacteroidetes data
load("data/top_bacteroidetes.Rda")

# Get total abundance per individual per order, square-root transform abundance, and spread the dataset to prepare for general linear mixed models.
bacteroidetes_orders <- top_bacteroidetes %>%
  dplyr::select(ind_id, group, sex, order, abundance) %>%
  group_by(ind_id, group, sex, order) %>%
  summarize(abundance = sum(abundance)) %>%
  mutate(sqrt_abundance = sqrt(abundance)) %>%
  dplyr::select(-abundance) %>%
  spread(order, sqrt_abundance) 

# Save order dataset for downstream analysis
save(bacteroidetes_orders, file = "data/bacteroidetes_orders.Rda")
```

#### 3b: Run general linear mixed models for each order to test for differential abundance among African elephant groups at the order level; run the Rmd chunk below:
In this step, we run general linear mixed models for the relationship of elephant group and order abundance (square-root transformed), with sex as a random effect and assuming a negative-binomial distribution, in a for loop construct, calculating p-values for each order column in the data.

##### Assess differential abundance across elephant groups within orders in phylum Bacteroidetes: 
```{r}
# Load bacteroidetes order data
load("data/bacteroidetes_orders.Rda")

# Generate table to hold p-values for each analysis
p <- data.frame(order = factor(c("Bacteroidales", "Flavobacteriales")), p_value = numeric(length = 2))

# Run general linear mixed model in for loop
for (ii in 4:ncol(bacteroidetes_orders)) {
      
  ff <- as.formula(paste0(colnames(bacteroidetes_orders)[ii]," ~ group + (1|sex)")) # specify model
  glmer <- glmer.nb(ff, bacteroidetes_orders) # run linear model           
  p_value <- summary(glmer)$coefficients[2,4] #extract p value
  p[ii-3,2] <- p_value # put p value in table
      
  }

# Adjust p-values for false discovery rate
p_fdr <- cbind(p, 
               data.frame(fdr_p = p.adjust(p$p_value, method = "fdr")), 
               data.frame(result = with(p_fdr, ifelse(fdr_p<0.05, "significant", "not-significant"))))
```

# OTU ABUNDANCE - ORDER LEVEL

## Rarefied OTU abundance data 
```{r, echo = FALSE}
options(scipen = 999) #gets rid of scientific notatation, useful for reading p-values

#read in taxonomic designations and OTU abundance
aem_physeq <- qza_to_phyloseq(features = "table.qza", taxonomy = "taxonomy.qza",  tree = "rooted-tree.qza", metadata = "METADATA.tsv") 

#omit sample OB182 due to low read count. We don't want to rarefy to such a low number of reads across the whole dataset
aem_physeq <- subset_samples(aem_physeq, Elephant != "OB182") 

#Rarefy and clean the data set for use in downstream analysis
aem_physeq <- rarefy_even_depth(aem_physeq, rngsee = 5) #must set seed in order to maintain the same rarefied number of otus

    #Number of OTUS: 8248
    #Number of reads per sample: 11460 
      
#Extract data of all types (meta data, all taxonomy data)
aem_meta <- as.data.frame(sample_data(aem_physeq))
aem_tax <- as.data.frame(tax_table(aem_physeq))
aem_tax <- aem_tax %>% rownames_to_column("otu")
aem_data <- as.data.frame(otu_table(aem_physeq))
aem_data <- aem_data %>% rownames_to_column("otu")
aem_tax_data <- merge(aem_tax, aem_data, by = "otu")
```

## Most Abundant Phyla
```{r}
#Subset the data by the three most abundant phyla: Bacteroidetes, firmicutes, proteobacteria
aem_bacteroidetes <- aem_tax_data %>% filter(Phylum == "p__Bacteroidetes")
aem_firmicutes <- aem_tax_data %>% filter(Phylum == "p__Firmicutes")
aem_proteobacteria <- aem_tax_data %>% filter(Phylum == "p__Proteobacteria")
```

## Phylum and Order Proportions per African Elephant Group
```{r, echo = FALSE}

#Proportions of each order contributing to each african elephant group
aem_order_phylum_data <- aem_tax_data[,-c(1:2,4,6:8)]
aem_order_phylum_data <- as.tibble(aem_order_phylum_data)
aem_order_phylum_data <- aem_order_phylum_data %>% group_by(Phylum, Order) %>% summarize_all(funs(sum))
```

## Core Microbiome - Order Level
```{r}
#Summarize Rarefied OTU abundance by ALL Orders to figure out the core microbiome 
aem_order_data <- aem_tax_data[,-c(1:4,6:8)]
aem_order_data <- as.data.frame(aem_order_data)
aem_order_data <- aem_order_data %>% group_by(Order) %>% summarize_all(funs(sum))
aem_order_data <- aem_order_data %>% drop_na()
aem_order_data <- column_to_rownames(aem_order_data, "Order")
aem_order_data_t <- t(aem_order_data)
aem_order_data_t <- as.data.frame(aem_order_data_t)
aem_order_data_t <- aem_order_data_t %>% rownames_to_column("sample_id")
aem_order <- merge(metadata, aem_order_data_t, by = "sample_id")

###IMPORTANT NOTE: At this point, aem_orders, which provides abundance across all orders for all samples, was copied and entered into excel in order to determine the core microbiome and any other shared orders among african elephant groups. Figrue 1 in the ms is produced from the data within this excel file.
```

## Order Abundance within Phylum Bacteroidetes, Firmicutes, and Proteobacteria
```{r}
#Summarize Rarefied OTU abundance by Orders WITHIN BACTERIOIDETES
aem_bact_orders <- aem_bacteroidetes[,-c(1:4,6:8)]
aem_bact_orders <- as.data.frame(aem_bact_orders)
aem_bact_orders <- aem_bact_orders %>% group_by(Order) %>% summarize_all(funs(sum))
aem_bact_orders <- aem_bact_orders %>% drop_na()
aem_bact_orders <- column_to_rownames(aem_bact_orders, "Order")
aem_bact_orders_t <- t(aem_bact_orders)
aem_bact_orders_t <- as.data.frame(aem_bact_orders_t)
aem_bact_orders_t <- aem_bact_orders_t %>% rownames_to_column("sample_id")
aem_bact_orders_sums <- merge(metadata, aem_bact_orders_t, by = "sample_id")

#Summarize Rarefied OTU abundance by Orders WITHIN FIRMICUTES
aem_firm_orders <- aem_firmicutes[,-c(1:4,6:8)]
aem_firm_orders <- as.data.frame(aem_firm_orders)
aem_firm_orders <- aem_firm_orders %>% group_by(Order) %>% summarize_all(funs(sum))
aem_firm_orders <- aem_firm_orders %>% drop_na()
aem_firm_orders <- column_to_rownames(aem_firm_orders, "Order")
aem_firm_orders_t <- t(aem_firm_orders)
aem_firm_orders_t <- as.data.frame(aem_firm_orders_t)
aem_firm_orders_t <- aem_firm_orders_t %>% rownames_to_column("sample_id")
aem_firm_orders_sums <- merge(metadata, aem_firm_orders_t, by = "sample_id")

#Summarize Rarefied OTU abundance by Orders WITHIN PROTEOBACTERIA
aem_proteo_orders <- aem_proteobacteria[,-c(1:4,6:8)]
aem_proteo_orders <- as.data.frame(aem_proteo_orders)
aem_proteo_orders <- aem_proteo_orders %>% group_by(Order) %>% summarize_all(funs(sum))
aem_proteo_orders <- aem_proteo_orders %>% drop_na()
aem_proteo_orders <- column_to_rownames(aem_proteo_orders, "Order")
aem_proteo_orders_t <- t(aem_proteo_orders)
aem_proteo_orders_t <- as.data.frame(aem_proteo_orders_t)
aem_proteo_orders_t <- aem_proteo_orders_t %>% rownames_to_column("sample_id")
aem_proteo_orders_sums <- merge(metadata, aem_proteo_orders_t, by = "sample_id")

#Compile sums for most abundant phyla and subset the data for downstream analyses. One dataset for Africana (n = 35), one dataset for Forest non-crop-raiders (n = 19)

#Africana - diet and habitat
aem_bact_orders_sums_africana <- aem_bact_orders_sums %>% filter(Species == "africana")
aem_firm_orders_sums_africana <- aem_firm_orders_sums %>% filter(Species == "africana")
aem_proteo_orders_sums_africana <- aem_proteo_orders_sums %>% filter(Species == "africana")

#Forest, non-crop-raiders - species
aem_bact_orders_sums_forest_ncr <- aem_bact_orders_sums %>% filter(Habitat == "Forest") %>% filter(Raider == "No")
aem_firm_orders_sums_forest_ncr <- aem_firm_orders_sums %>% filter(Habitat == "Forest") %>% filter(Raider == "No")
aem_proteo_orders_sums_forest_ncr <- aem_proteo_orders_sums %>% filter(Habitat == "Forest") %>% filter(Raider == "No")


#Analyze OTU sum data by African elephant species
aem_bact_species <- aem_bact_orders_sums_forest_ncr[,-c(1:3,5:6,9:11)]
colnames(aem_bact_species) <- c("Species", "Age", "Sex", "Saprospirales", "Bacteroidales", "Cytophagales", "Flavobacteriales", "Sphingobacteriales")

aem_firm_species <- aem_firm_orders_sums_forest_ncr[,-c(1:3,5:6,9:11)]
colnames(aem_firm_species) <- c("Species", "Age", "Sex", "Bacillales", "Clostridiales", "Erysipelotrichales", "Lactobacillales")


aem_proteo_species <- aem_proteo_orders_sums_forest_ncr[,-c(1:3,5:6,9:11)]
colnames(aem_proteo_species) <- c("Species", "Age", "Sex", "Aeromonadales", "Alteromonadales", "Bdellovibrionales", "Burkholderiales", "Caulobacterales", "Desulfovibrionales", "Enterobacteriales", "gmd14h09", "Kiloniellales", "Neisseriales", "Oceanospirillales", "Pasteurellales", "phoshd29", "Pseudomonadales", "rf32", "Rhizobiales", "Rhodobacterales", "Rickettsiales", "Sphingomonadales", "Spirobacillales", "Tremblayales", "Xanthomonadales")

#Analyze OTU sum data by African elephant Groups
aem_bact_groups <- aem_bact_orders_sums_africana[,-c(1:6,9,11)]
colnames(aem_bact_groups) <- c("Age", "Sex", "Group" , "Saprospirales", "Bacteroidales", "Cytophagales", "Flavobacteriales", "Sphingobacteriales")

aem_firm_groups <- aem_firm_orders_sums_africana[,-c(1:6,9,11)]
colnames(aem_firm_groups) <- c("Age", "Sex", "Group", "Bacillales", "Clostridiales", "Erysipelotrichales", "Lactobacillales")

aem_proteo_groups <- aem_proteo_orders_sums_africana[,-c(1:6,9,11)]
colnames(aem_proteo_groups) <- c("Age", "Sex", "Group","Aeromonadales", "Alteromonadales", "Bdellovibrionales", "Burkholderiales", "Caulobacterales", "Desulfovibrionales", "Enterobacteriales", "gmd14h09", "Kiloniellales", "Neisseriales", "Oceanospirillales", "Pasteurellales", "phoshd29", "Pseudomonadales", "rf32", "Rhizobiales", "Rhodobacterales", "Rickettsiales", "Sphingomonadales", "Spirobacillales", "Tremblayales", "Xanthomonadales")
```

## Testing for Normality in all orders
```{r}

bact_norm1 <- ggqqplot(aem_bact_species$Bacteroidales)
      sb1 <- shapiro.test(aem_bact_species$Bacteroidales) #NORMAL
bact_norm2 <- ggqqplot(aem_bact_species$Cytophagales)
      sb2 <- shapiro.test(aem_bact_species$Cytophagales) #NON-NORMAL
bact_norm3 <- ggqqplot(aem_bact_species$Saprospirales)
      sb3 <- shapiro.test(aem_bact_species$Saprospirales) #NON-NORMAL
bact_norm4 <- ggqqplot(aem_bact_species$Flavobacteriales)
      sb4 <- shapiro.test(aem_bact_species$Flavobacteriales) #NON-NORMAL
bact_norm5 <- ggqqplot(aem_bact_species$Sphingobacteriales)
      sb5 <- shapiro.test(aem_bact_species$Sphingobacteriales) #NON-NORMAL


firm_norm1 <- ggqqplot(aem_firm_species$Bacillales)
      sf1 <- shapiro.test(aem_firm_species$Bacillales) #NON-NORMAL
firm_norm2 <- ggqqplot(aem_firm_species$Clostridiales)
      sf2 <- shapiro.test(aem_firm_species$Clostridiales) #NORMAL
firm_norm3 <- ggqqplot(aem_firm_species$Erysipelotrichales)
      sf3 <- shapiro.test(aem_firm_species$Erysipelotrichales) #NON-NORMAL
firm_norm4 <- ggqqplot(aem_firm_species$Lactobacillales)
      sf4 <- shapiro.test(aem_firm_species$Lactobacillales) #NON-NORMAL
      
#ALL NOT NORMAL EXCEPT PSUEDOMONADALES
proteo_norm1 <- ggqqplot(aem_proteo_species$Aeromonadales)
      sp1 <- shapiro.test(aem_proteo_species$Aeromonadales)
proteo_norm2 <- ggqqplot(aem_proteo_species$Alteromonadales)
      sp2 <- shapiro.test(aem_proteo_species$Alteromonadales)
proteo_norm3 <- ggqqplot(aem_proteo_species$Bdellovibrionales)
      sp3 <- shapiro.test(aem_proteo_species$Bdellovibrionales)
proteo_norm4 <- ggqqplot(aem_proteo_species$Burkholderiales)
      sp4 <- shapiro.test(aem_proteo_species$Burkholderiales)
proteo_norm5 <- ggqqplot(aem_proteo_species$Caulobacterales)
      sp5 <- shapiro.test(aem_proteo_species$Caulobacterales)
proteo_norm6 <- ggqqplot(aem_proteo_species$Desulfovibrionales)
      sp6 <- shapiro.test(aem_proteo_species$Desulfovibrionales)
proteo_norm7 <- ggqqplot(aem_proteo_species$Enterobacteriales)
      sp7 <- shapiro.test(aem_proteo_species$Enterobacteriales)
proteo_norm8 <- ggqqplot(aem_proteo_species$gmd14h09)
      sp8 <- shapiro.test(aem_proteo_species$gmd14h09)
proteo_norm9 <- ggqqplot(aem_proteo_species$Kiloniellales)
      sp9 <- shapiro.test(aem_proteo_species$Kiloniellales)
proteo_norm10 <- ggqqplot(aem_proteo_species$Neisseriales)
      sp10 <- shapiro.test(aem_proteo_species$Neisseriales)
proteo_norm11 <- ggqqplot(aem_proteo_species$Oceanospirillales)
      sp11 <- shapiro.test(aem_proteo_species$Oceanospirillales)
proteo_norm12 <- ggqqplot(aem_proteo_species$Pasteurellales)
      sp12 <- shapiro.test(aem_proteo_species$Pasteurellales)
proteo_norm13 <- ggqqplot(aem_proteo_species$phoshd29)
      sp13 <- shapiro.test(aem_proteo_species$phoshd29)
proteo_norm14 <- ggqqplot(aem_proteo_species$Pseudomonadales) 
      sp14 <- shapiro.test(aem_proteo_species$Pseudomonadales) #NORMAL
proteo_norm15 <- ggqqplot(aem_proteo_species$rf32)
      sp15 <- shapiro.test(aem_proteo_species$rf32)
proteo_norm16 <- ggqqplot(aem_proteo_species$Rhizobiales)
      sp16 <- shapiro.test(aem_proteo_species$Rhizobiales)
proteo_norm17 <- ggqqplot(aem_proteo_species$Rhodobacterales)
      sp17 <- shapiro.test(aem_proteo_species$Rhodobacterales)
proteo_norm18 <- ggqqplot(aem_proteo_species$Rickettsiales)
      sp18 <- shapiro.test(aem_proteo_species$Rickettsiales)
proteo_norm19 <- ggqqplot(aem_proteo_species$Sphingomonadales)
      sp19 <- shapiro.test(aem_proteo_species$Sphingomonadales)
proteo_norm20 <- ggqqplot(aem_proteo_species$Tremblayales)
      sp20 <- shapiro.test(aem_proteo_species$Tremblayales)
proteo_norm21 <- ggqqplot(aem_proteo_species$Xanthomonadales)
      sp21 <- shapiro.test(aem_proteo_species$Xanthomonadales)
```

## Statistical Significance of Bacterial Orders By African Elephant Species
```{r}
aem_bact_species_rel0.1 <- aem_bact_species[,-c(4,6,8)]
aem_firm_species_rel0.1 <- aem_firm_species[,-c(6,7)]
aem_proteo_species_rel0.1 <- aem_proteo_species[,-c(4:9,11:16,18:25)]

#PHYLUM Bacteroidetes
p_vals_bact_species <- data.frame(numeric(length = 2))

    for (ii in 4:ncol(aem_bact_species_rel0.1)) {
  
      col = aem_bact_species_rel0.1[,ii] #this tells the for loop that it should iterate through columns in the data set
      
      ff <- as.formula(paste0(colnames(aem_bact_species_rel0.1)[ii]," ~ Species + (1|Sex)"))
      
      glmer_temp <- glmer.nb(ff,  aem_bact_species_rel0.1) #tells the for loop to run linear model  for each column           specified above
      p_value <- summary(glmer_temp)$coefficients[2,4] #extracts p value
      p_vals_bact_species[ii-3,] <- p_value
    }

#PHYLUM Firmicutes
p_vals_firm_species <- data.frame(numeric(length = 2))

    for (ii in 4:ncol(aem_firm_species_rel0.1)) {
  
      col = aem_firm_species_rel0.1[,ii] #this tells the for loop that it should iterate through columns in the data set
      ff <- as.formula(paste0(colnames(aem_firm_species_rel0.1)[ii]," ~ Species + (1|Sex)"))
      
      glmer_temp <- glmer.nb(ff, data = aem_firm_species_rel0.1) #tells the for loop to run linear model  for each column           specified above
      p_value <- summary(glmer_temp)$coefficients[2,4] #extracts p value
      p_vals_firm_species[ii-3,] <- p_value
    }

#PHYLUM Proteobacteria
aem_proteo_species <- aem_proteo_species[,-c(23)] #SPIROBACILLES REMOVED BECAUSE IT WAS ALL ZEROS
p_vals_proteo_species <- data.frame(numeric(length = 2))

    for (ii in 4:ncol(aem_proteo_species_rel0.1)) {
  
      col = aem_proteo_species_rel0.1[,ii] #this tells the for loop that it should iterate through columns in the data set
      ff <- as.formula(paste0(colnames(aem_proteo_species_rel0.1)[ii]," ~ Species + (1|Sex)"))
      
      glmer_temp <- glmer.nb(ff, data = aem_proteo_species_rel0.1) #tells the for loop to run linear model  for each column         specified above
      p_value <- summary(glmer_temp)$coefficients[2,4] #extracts p value
      p_vals_proteo_species[ii-3,] <- p_value
    }

#Convert dataframes with p_values for each order within each phylum into tables we can work with
    
#PHYLUM Bacteroidetes orders p-values
p_vals_bact_species <- p_vals_bact_species %>% drop_na()
colnames(p_vals_bact_species) <- c("pvals")
p_vals_bact_species <- as.data.frame(p_vals_bact_species)

#PHYLUM Firmicutes orders p-values
p_vals_firm_species <- p_vals_firm_species %>% drop_na()
colnames(p_vals_firm_species) <- c("pvals")
p_vals_firm_species <- as.data.frame(p_vals_firm_species)

#PHYLUM Proteobacteria orders p-values
p_vals_proteo_species <- p_vals_proteo_species %>% drop_na()
colnames(p_vals_proteo_species) <- c("pvals")
p_vals_proteo_species <- as.data.frame(p_vals_proteo_species)

    
#Correct p-values for multiple tests using false discovery rate
p_vals_bact_species_fdr <- p.adjust(p_vals_bact_species$pvals, method = "fdr")
p_vals_firm_species_fdr <- p.adjust(p_vals_firm_species$pvals, method = "fdr")
p_vals_proteo_species_fdr <- p.adjust(p_vals_proteo_species$pvals, method = "fdr")

#p_vals_bact_species_fdr #Nothing is significant
#p_vals_firm_species_fdr #Bacillales and Lactobacillales are significant
#p_vals_proteo_species_fdr #nothing significant
```

## Make Box Plots for ALL ORDERS
```{r}
#Log-transform y axis so that all values will show up

#PHYLUM Bacteroidetes
species_bacteroidetes_order_abun <- read_excel("species_bacteroidetes_order_abun.xlsx")
species_bacteroidetes_order_abun <- species_bacteroidetes_order_abun %>% mutate(log_abundance = log10(abundance), sqrt_abundance = sqrt(abundance))

#PHYLUM Firmicutes
species_firmicutes_order_abun <- read_excel("species_firmicutes_order_abun.xlsx")
species_firmicutes_order_abun <- species_firmicutes_order_abun %>% mutate(log_abundance = log10(abundance), sqrt_abundance = sqrt(abundance))

#PHYLUM Proteobacteria
species_proteobacteria_order_abun <- read_excel("species_proteobacteria_order_abun.xlsx")
species_proteobacteria_order_abun <- species_proteobacteria_order_abun %>% mutate(log_abundance = log10(abundance), sqrt_abundance = sqrt(abundance))
 
#Make box-plots of Orders for African Elephant Species

#Bacteroidetes
species_bacteroidetes_box_plots <- ggplot(species_bacteroidetes_order_abun, aes(x = order, y = sqrt_abundance, fill = sex)) + geom_boxplot(show.legend = T, outlier.shape = NA) + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + labs(fill = "Elephant Sex", title = "Orders by Elephant Species", y = "Bacteroidetes Abundance") + scale_fill_manual(values = c("red", "blue")) + theme(axis.title.x = element_blank()) + theme(legend.position = c(0.65,0.7)) + theme(legend.text = element_text(face = "italic")) + theme(axis.text.x = element_text(face = "italic")) 

#Firmicutes
species_firmicutes_box_plots <- ggplot(species_firmicutes_order_abun, aes(x = order, y = sqrt_abundance, fill = species)) + geom_boxplot(show.legend = F, outlier.shape = NA) + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + labs(y = "Firmicutes Abundance") + scale_fill_manual(values = c("red", "blue")) + theme(axis.title.x = element_blank()) + theme(axis.text.x = element_text(face = "italic"))


#Proteobacteria
species_proteobacteria_box_plots <- ggplot(species_proteobacteria_order_abun, aes(x = order, y = sqrt_abundance, fill = Species)) + geom_boxplot(show.legend = F, outlier.shape = NA) + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + labs(y = "Proteobacteria Abundance") + scale_fill_manual(values = c("red", "blue")) + theme(axis.title.x = element_blank())+ theme(axis.text.x = element_text(face = "italic"))
    

jpeg("/Users/joegunn/Desktop/Grad_School_Stuff/Projects/Elephant_Microbiome/NEW_STATS/AEMB_FINAL_STATS/African_Elephant_Microbiome/phylum_species_plots.jpeg", width=2000, height=3000, units='px', res=250) 

plot_grid(species_bacteroidetes_box_plots, species_firmicutes_box_plots, species_proteobacteria_box_plots, nrow = 3, labels = c("A", "B", "C"))

dev.off()


#For combining figures
phy_species_plot <- plot_grid(species_bacteroidetes_box_plots, species_firmicutes_box_plots, species_proteobacteria_box_plots, nrow = 3, labels = c("A", "B", "C"))
```

## Make Box Plots for FILTERED ORDERS (AVERAGE RELATIVE ABUNDANCE = 0.1) by Species
```{r}
#Log-transform y axis so that all values will show up

#PHYLUM Bacteroidetes
species_bacteroidetes_order_abun_rel0.1 <- read_excel("species_bacteroidetes_order_abun_rel0.1.xlsx")
species_bacteroidetes_order_abun_rel0.1 <- species_bacteroidetes_order_abun_rel0.1 %>% mutate(log_abundance = log10(abundance), sqrt_abundance = sqrt(abundance))

#PHYLUM Firmicutes
species_firmicutes_order_abun_rel0.1 <- read_excel("species_firmicutes_order_abun_rel0.1.xlsx")
species_firmicutes_order_abun_rel0.1 <- species_firmicutes_order_abun_rel0.1 %>% mutate(log_abundance = log10(abundance), sqrt_abundance = sqrt(abundance))

#PHYLUM Proteobacteria
species_proteobacteria_order_abun_rel0.1 <- read_excel("species_proteobacteria_order_abun_rel0.1.xlsx")
species_proteobacteria_order_abun_rel0.1 <- species_proteobacteria_order_abun_rel0.1 %>% mutate(log_abundance = log10(abundance), sqrt_abundance = sqrt(abundance))
 
#Make box-plots of Orders for African Elephant Species

#Bacteroidetes
species_bacteroidetes_box_plots_rel0.1 <- ggplot(species_bacteroidetes_order_abun_rel0.1, aes(x = order, y = sqrt_abundance, fill = species)) + geom_boxplot(show.legend = T, outlier.shape = NA) + theme(axis.text = element_text(size = 15)) + labs(fill = "Elephant Species", title = "Orders by Elephant Species", y = "Bacteroidetes Abundance") + theme(plot.title = element_text(size = 20, hjust = 0.5, face = "bold")) + scale_fill_manual(values = c("red", "blue")) + theme(axis.title.x = element_blank()) + theme(legend.position = c(0.85,0.8)) + theme(axis.title.y = element_text(size = 20)) + theme(legend.text = element_text(size = 20, face = "italic")) + theme(axis.text.x = element_text(face = "italic", size = 20)) + theme(plot.background = element_blank()) + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),panel.background = element_blank(), axis.line = element_line(colour = "black")) + theme(legend.title = element_text(size = 20))

#Firmicutes
species_firmicutes_box_plots_rel0.1 <- ggplot(species_firmicutes_order_abun_rel0.1, aes(x = order, y = sqrt_abundance, fill = species)) + geom_boxplot(show.legend = F, outlier.shape = NA) + theme(axis.text = element_text(size = 15)) + labs(y = "Firmicutes Abundance") + scale_fill_manual(values = c("red", "blue")) + theme(axis.title.x = element_blank()) + theme(axis.text.x = element_text(face = "italic", size = 20))  + theme(plot.background = element_blank())  + theme(axis.title.y = element_text(size = 20)) + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),panel.background = element_blank(), axis.line = element_line(colour = "black")) + theme(legend.box.background = element_blank())

#Proteobacteria
species_proteobacteria_box_plots_rel0.1 <- ggplot(species_proteobacteria_order_abun_rel0.1, aes(x = order, y = sqrt_abundance, fill = Species)) + geom_boxplot(show.legend = F, outlier.shape = NA) + theme(axis.text = element_text(size = 15)) + labs(y = "Proteobacteria Abundance") + scale_fill_manual(values = c("red", "blue")) + theme(axis.title.x = element_blank())+ theme(axis.text.x = element_text(face = "italic", size = 20))  + theme(axis.title.y = element_text(size = 20)) + theme(plot.background = element_blank())  + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),panel.background = element_blank(), axis.line = element_line(colour = "black")) + theme(legend.box.background = element_blank())
    

jpeg("/Users/joegunn/Desktop/Grad_School_Stuff/Projects/Elephant_Microbiome/NEW_STATS/AEMB_FINAL_STATS/African_Elephant_Microbiome/phylum_species_plots_rel0.1.jpeg", width=2000, height=3000, units='px', res=250) 

plot_grid(species_bacteroidetes_box_plots_rel0.1, species_firmicutes_box_plots_rel0.1, species_proteobacteria_box_plots_rel0.1, nrow = 3, labels = c("A", "B", "C"), label_size = 20)

dev.off()


#For combining figures
phy_species_plot_rel0.1 <- plot_grid(species_bacteroidetes_box_plots_rel0.1, species_firmicutes_box_plots_rel0.1, species_proteobacteria_box_plots_rel0.1, nrow = 3, labels = c("A", "B", "C"), label_size = 20)
```

## Statistical Signifiance of Orders By African Elephant Groups
```{r}
aem_bact_groups_rel0.1 <- aem_bact_groups[,-c(4,6,8)]
aem_firm_groups_rel0.1 <- aem_firm_groups[,-c(6,7)]
aem_proteo_groups_rel0.1 <- aem_proteo_groups[,-c(4:9,11:16,18:25)]

#PHYLUM Bacteroidetes
gbact <- glmer.nb(Bacteroidales ~ Group + (1|Sex), data = aem_bact_groups_rel0.1)
gbact_p <- Anova(gbact)$`Pr(>Chisq)` #P = 0.1889
gflav <- glmer.nb(Flavobacteriales ~ Group + (1|Sex), data = aem_bact_groups_rel0.1, control=glmerControl(optimizer="bobyqa", tolPwrss=1e-3, optCtrl = list(maxfun = 100000)))
gflav_p <- Anova(gflav)$`Pr(>Chisq)` #P = 0.001998

p_vals_bact_groups <- as.data.frame(c(gbact_p, gflav_p))
colnames(p_vals_bact_groups) <- c("pval")

#Post-hoc Tukey-test
post_flavo <- glht(gflav, mcp(Group = "Tukey"))
summary(post_flavo)

#PHYLUM Firmicutes
gbaci <- glmer.nb(Bacillales ~ Group + (1|Sex), data = aem_firm_groups_rel0.1)
gbaci_p <- Anova(gbaci)$`Pr(>Chisq)`
gclos <- glmer.nb(Clostridiales ~ Group + (1|Sex), data = aem_firm_groups_rel0.1)
gclos_p <- Anova(gclos)$`Pr(>Chisq)`

p_vals_firm_groups <- as.data.frame(c(gbaci_p, gclos_p))
colnames(p_vals_firm_groups) <- c("pval")

#Post-hoc Tukey-test
post_bacillales <- glht(gbaci, mcp(Group="Tukey"))
summary(post_bacillales)

#PHYLUM Proteobacteria
genter <- glmer.nb(Enterobacteriales ~ Group + (1|Sex), data = aem_proteo_groups_rel0.1)
genter_p <- Anova(genter)$`Pr(>Chisq)`
gpseudo <- glmer.nb(Pseudomonadales ~ Group + (1|Sex), data = aem_proteo_groups_rel0.1)
gpseudo_p <- Anova(gpseudo)$`Pr(>Chisq)`

p_vals_proteo_groups <- as.data.frame(c(genter_p, gpseudo_p))
colnames(p_vals_proteo_groups) <- c("pval")

#Post-hoc Tukey-Test
post_pseudo <- glht(gpseudo, mcp(Group = "Tukey"))
summary(post_pseudo)


#For loops for ALL ORDERS

#Bacteroides
p_vals_bact_groups <- data.frame(numeric(length = 2)) #This for loop had trouble knitting to html, so I included hashtags to force it to finish. WE DID USE THIS CODE TO PRODUCE RESULTS FOR THE ms

   # for (ii in 4:ncol(aem_bact_groups_rel0.1)) {
      
    #  ff <- as.formula(paste0(colnames(aem_bact_groups_rel0.1)[ii]," ~ Group + (1|Sex)"))
    #  glmer_temp <- glmer.nb(ff, data = aem_bact_groups_rel0.1, control=glmerControl(optimizer="bobyqa", tolPwrss=1e-3, optCtrl = list(maxfun = 100000))) #tells the for loop to run linear model  for each column specified above
    #  p_value <- Anova(glmer_temp)$`Pr(>Chisq)` #extracts f statistics
    #  p_vals_bact_groups[ii-3,] <- p_value
    #}

#PHYLUM Firmicutes
p_vals_firm_groups <- data.frame(numeric(length = 2))

    for (ii in 4:ncol(aem_firm_groups_rel0.1)) {
  
      col = aem_firm_groups_rel0.1[,ii] #this tells the for loop that it should iterate through columns in the data set
      lm_temp <- lm(col ~ Group, data = aem_firm_groups_rel0.1) #tells the for loop to run linear model for each column specified above
      fstat <- summary(lm_temp)$fstatistic
      p_value <- pf(fstat[1], fstat[2], fstat[3], lower.tail = F) #extracts p value
      p_vals_firm_groups[ii-3,] <- p_value
    }

#PHYLUM Proteobacteria
p_vals_proteo_groups <- data.frame(numeric(length = 2))

    for (ii in 1:ncol(aem_proteo_groups_rel0.1)) {
  
      col = aem_proteo_groups_rel0.1[,ii] #this tells the for loop that it should iterate through columns in the data set
      lm_temp <- lm(col ~ Group, data = aem_proteo_groups_rel0.1) #tells the for loop to run linear model  for each column specified above
      fstat <- summary(lm_temp)$fstatistic
      p_value <- pf(fstat[1], fstat[2], fstat[3], lower.tail = F) #extracts p value
      p_vals_proteo_groups[ii,] <- p_value
    }

#Convert dataframes with p_values into tables we can work with

#PHYLUM Bacterioidetes
p_vals_bact_groups <- p_vals_bact_groups %>% drop_na()
colnames(p_vals_bact_groups) <- c("pval")
p_vals_bact_groups <- as.data.frame(p_vals_bact_groups)

#PHYLUM Firmicutes
p_vals_firm_groups <- p_vals_firm_groups %>% drop_na()
colnames(p_vals_firm_groups) <- c("pval")
p_vals_firm_groups <- as.data.frame(p_vals_firm_groups)

#PHYLUM Proteobacteria
p_vals_proteo_groups <- p_vals_proteo_groups %>% drop_na()
colnames(p_vals_proteo_groups) <- c("pval")
p_vals_proteo_groups <- as.data.frame(p_vals_proteo_groups)

#Correct p-values for multiple tests using false discovery rate
p_vals_bact_groups_fdr <- p.adjust(p_vals_bact_groups$pval, method = "fdr")
p_vals_firm_groups_fdr <- p.adjust(p_vals_firm_groups$pval, method = "fdr")
p_vals_proteo_groups_fdr <- p.adjust(p_vals_proteo_groups$pval, method = "fdr")

#p_vals_bact_groups_fdr
#p_vals_firm_groups_fdr
#p_vals_proteo_groups_fdr
```

## Make Box Plots for ALL ORDERS
```{r}
#Log-transform y axis so that all values will show up
groups_bacteroidetes_order_abun <- read_excel("groups_bacteroidetes_order_abun.xlsx")
groups_bacteroidetes_order_abun <- groups_bacteroidetes_order_abun %>% mutate(log_abundance = log10(abundance), sqrt_abundance = sqrt(abundance))

groups_firmicutes_order_abun <- read_excel("groups_firmicutes_order_abun.xlsx")
groups_firmicutes_order_abun <- groups_firmicutes_order_abun %>% mutate(log_abundance = log10(abundance), sqrt_abundance = sqrt(abundance))

groups_proteobacteria_order_abun <- read_excel("groups_proteobacteria_order_abun.xlsx")
groups_proteobacteria_order_abun  <- groups_proteobacteria_order_abun %>% mutate(log_abundance = log10(abundance), sqrt_abundance = sqrt(abundance))


#PHYLUM Bacterioidetes
bact_group_plots <- ggplot(groups_bacteroidetes_order_abun, aes(x = order, y = sqrt_abundance, fill = group)) + geom_boxplot(outlier.shape = NA, show.legend = T) + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + labs(title = "Orders by Elephant Group", fill = "Elephant Group") + scale_fill_manual(values = c("green", "purple", "red", "goldenrod")) + theme(axis.title.x = element_blank()) + theme(axis.title.y = element_blank()) + theme(legend.position = c(0.65,0.7)) + theme(axis.text.x = element_text(face = "italic"))  


#PHYLUM Firmicutes
firm_group_plots <- ggplot(groups_firmicutes_order_abun, aes(x = order, y = sqrt_abundance, fill = group)) + geom_boxplot(outlier.shape = NA, show.legend = F) + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + scale_fill_manual(values = c("green", "purple", "red", "goldenrod")) + theme(axis.title.x = element_blank()) + theme(axis.title.y = element_blank()) + theme(axis.text.x = element_text(face = "italic"))

#PHYLUM Proteobacteria
proteo_group_plots <- ggplot(groups_proteobacteria_order_abun, aes(x = order, y = sqrt_abundance, fill = group)) + geom_boxplot(outlier.shape = NA, show.legend = F) + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + scale_fill_manual(values = c("green", "purple", "red", "goldenrod")) + theme(axis.title.x = element_blank()) + theme(axis.title.y = element_blank()) + theme(axis.text.x = element_text(face = "italic"))

jpeg("/Users/joegunn/Desktop/Grad_School_Stuff/Projects/Elephant_Microbiome/NEW_STATS/AEMB_FINAL_STATS/African_Elephant_Microbiome/phylum_group_plots.jpeg", width=2000, height=3000, units='px', res=250) 

plot_grid(bact_group_plots, firm_group_plots, proteo_group_plots, nrow = 3, labels = c("A", "B", "C"))

dev.off()

#For combining figures

phy_group_plot <- plot_grid(bact_group_plots, firm_group_plots, proteo_group_plots, nrow = 3, labels = c("D", "E", "F"))

##WE did not look for genera within Saprospirales, because it is in such low abundance (only found in n = 8 samples)


###COMBINE PHYLUM PLOTS
pdf("/Users/joegunn/Desktop/Grad_School_Stuff/Projects/Elephant_Microbiome/NEW_STATS/AEMB_FINAL_STATS/African_Elephant_Microbiome/phy_combined_plots.pdf", width=15, height=20) 

plot_grid(phy_species_plot, phy_group_plot, ncol = 2)

dev.off()
```

## Make Box Plots for FILTERED ORDERS (AVERAGE RELATIVE ABUNDANCE = 0.1 or more)
```{r}
#Log-transform y axis so that all values will show up
groups_bacteroidetes_order_abun_rel0.1 <- read_excel("groups_bacteroidetes_order_abun_rel0.1.xlsx")
groups_bacteroidetes_order_abun_rel0.1 <- groups_bacteroidetes_order_abun_rel0.1 %>% mutate(log_abundance = log10(abundance), sqrt_abundance = sqrt(abundance))

groups_firmicutes_order_abun_rel0.1 <- read_excel("groups_firmicutes_order_abun_rel0.1.xlsx")
groups_firmicutes_order_abun_rel0.1 <- groups_firmicutes_order_abun %>% mutate(log_abundance = log10(abundance), sqrt_abundance = sqrt(abundance))

groups_proteobacteria_order_abun_rel0.1 <- read_excel("groups_proteobacteria_order_abun_rel0.1.xlsx")
groups_proteobacteria_order_abun_rel0.1  <- groups_proteobacteria_order_abun_rel0.1 %>% mutate(log_abundance = log10(abundance), sqrt_abundance = sqrt(abundance))


#PHYLUM Bacterioidetes
bact_group_plots_rel0.1 <- ggplot(groups_bacteroidetes_order_abun_rel0.1, aes(x = order, y = sqrt_abundance, fill = group)) + geom_boxplot(outlier.shape = NA, show.legend = T) + theme(plot.title = element_text(size = 20, hjust = 0.5, face = "bold")) + theme(axis.text = element_text(size = 20)) + labs(title = "Orders by Elephant Group", fill = "Elephant Group") + scale_fill_manual(values = c("green", "purple", "red", "goldenrod")) + theme(axis.title.x = element_blank()) + theme(axis.title.y = element_blank()) + theme(legend.position = c(0.85,0.8)) + theme(legend.text = element_text(size = 20)) + theme(axis.text.x = element_text(face = "italic", size = 20)) + theme(plot.background = element_blank())  + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),panel.background = element_blank(), axis.line = element_line(colour = "black")) + theme(legend.title = element_text(size = 20))


#PHYLUM Firmicutes
firm_group_plots_rel0.1 <- ggplot(groups_firmicutes_order_abun_rel0.1, aes(x = order, y = sqrt_abundance, fill = group)) + geom_boxplot(outlier.shape = NA, show.legend = F) + theme(axis.text = element_text(size = 15)) + scale_fill_manual(values = c("green", "purple", "red", "goldenrod")) + theme(axis.title.x = element_blank()) + theme(axis.title.y = element_blank()) + theme(axis.text.x = element_text(face = "italic", size = 20)) + theme(plot.background = element_blank())  + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), panel.background = element_blank(), axis.line = element_line(colour = "black")) + theme(legend.title = element_text(size = 20))

#PHYLUM Proteobacteria
proteo_group_plots_rel0.1 <- ggplot(groups_proteobacteria_order_abun_rel0.1, aes(x = order, y = sqrt_abundance, fill = group)) + geom_boxplot(outlier.shape = NA, show.legend = F) + theme(axis.text = element_text(size = 15)) + scale_fill_manual(values = c("green", "purple", "red", "goldenrod")) + theme(axis.title.x = element_blank()) + theme(axis.title.y = element_blank()) + theme(axis.text.x = element_text(face = "italic", size = 20)) + theme(plot.background = element_blank())  + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),panel.background = element_blank(), axis.line = element_line(colour = "black")) + theme(legend.title = element_text(size = 20))

jpeg("/Users/joegunn/Desktop/Grad_School_Stuff/Projects/Elephant_Microbiome/NEW_STATS/AEMB_FINAL_STATS/African_Elephant_Microbiome/phylum_group_plots_rel0.1.jpeg", width=2000, height=3000, units='px', res=250) 

plot_grid(bact_group_plots_rel0.1, firm_group_plots_rel0.1, proteo_group_plots_rel0.1, nrow = 3, labels = c("A", "B", "C"))

dev.off()


#For combining figures

phy_group_plot_rel0.1 <- plot_grid(bact_group_plots_rel0.1, firm_group_plots_rel0.1, proteo_group_plots_rel0.1, nrow = 3, labels = c("D", "E", "F"), label_size = 20)


##WE did not look for genera within Saprospirales, because it is in such low abundance (only found in n = 8 samples)


###COMBINE PHYLUM PLOTS
pdf("/Users/joegunn/Desktop/Grad_School_Stuff/Projects/Elephant_Microbiome/NEW_STATS/AEMB_FINAL_STATS/African_Elephant_Microbiome/phy_combined_plots_rel0.1.pdf", width=15, height=20)

plot_grid(phy_species_plot_rel0.1, phy_group_plot_rel0.1, ncol = 2)

dev.off()
```

# OTU ABUNDANCE - GENUS LEVEL

## Statistical Signifiance of Genera within Orders Bacillales and Lactobacillales By African Elephant Species
```{r}
#Now we want to find the genera within each of the significant orders (Bacillales and Lactobacillales within Firmicutes) and see which are significantly different between africana and cyclotis species
    
#Data frames with genera only within the significant orders for african elephant SPECIES
aem_firm_bacillales <- aem_firmicutes %>% filter(Order == "o__Bacillales")
aem_firm_lactobacillales <- aem_firmicutes %>% filter(Order == "o__Lactobacillales")

#Data frames with genera only within the significant orders for african elephant GROUPS
aem_bact_flavobacteriales <- aem_bacteroidetes %>% filter(Order == "o__Flavobacteriales")
aem_proteo_pseudomonadales <- aem_proteobacteria %>% filter(Order == "o__Pseudomonadales")

#Extract Genera from Bacillales and Lactobacillales within Firmicutes (SPECIES)

    #Bacillales
    aem_firm_bacillales_genera <- aem_firm_bacillales[,-c(1:6,8)]
    aem_firm_bacillales_genera <- as.data.frame(aem_firm_bacillales_genera)
    aem_firm_bacillales_genera <- aem_firm_bacillales_genera %>% group_by(Genus) %>% summarize_all(funs(sum))
    aem_firm_bacillales_genera <- aem_firm_bacillales_genera %>% drop_na()
    aem_firm_bacillales_genera <- column_to_rownames(aem_firm_bacillales_genera, "Genus")
    aem_firm_bacillales_genera_t <- t(aem_firm_bacillales_genera)
    aem_firm_bacillales_genera_t <- as.data.frame(aem_firm_bacillales_genera_t)
    aem_firm_bacillales_genera_t <- aem_firm_bacillales_genera_t %>% rownames_to_column("sample_id")
    aem_firm_bacillales_genera_sums <- merge(metadata, aem_firm_bacillales_genera_t, by = "sample_id")
    
    #Lactobacillales
    aem_firm_lactobacillales_genera <- aem_firm_lactobacillales[,-c(1:6,8)]
    aem_firm_lactobacillales_genera <- as.data.frame(aem_firm_lactobacillales_genera)
    aem_firm_lactobacillales_genera <- aem_firm_lactobacillales_genera %>% group_by(Genus) %>% summarize_all(funs(sum))
    aem_firm_lactobacillales_genera <- aem_firm_lactobacillales_genera %>% drop_na()
    aem_firm_lactobacillales_genera <- column_to_rownames(aem_firm_lactobacillales_genera, "Genus")
    aem_firm_lactobacillales_genera_t <- t(aem_firm_lactobacillales_genera)
    aem_firm_lactobacillales_genera_t <- as.data.frame(aem_firm_lactobacillales_genera_t)
    aem_firm_lactobacillales_genera_t <- aem_firm_lactobacillales_genera_t %>% rownames_to_column("sample_id")
    aem_firm_lactobacillales_genera_sums <- merge(metadata, aem_firm_lactobacillales_genera_t, by = "sample_id")


#Extract Genera from Flavobacteriales within Bacteroidetes (GROUPS)

    #Flavobacteriales
    aem_bact_flavobacteriales_genera <- aem_bact_flavobacteriales[,-c(1:6,8)]
    aem_bact_flavobacteriales_genera <- as.data.frame(aem_bact_flavobacteriales_genera)
    aem_bact_flavobacteriales_genera <- aem_bact_flavobacteriales_genera %>% group_by(Genus) %>% summarize_all(funs(sum))
    aem_bact_flavobacteriales_genera <- aem_bact_flavobacteriales_genera %>% drop_na()
    aem_bact_flavobacteriales_genera <- column_to_rownames(aem_bact_flavobacteriales_genera, "Genus")
    aem_bact_flavobacteriales_genera_t <- t(aem_bact_flavobacteriales_genera)
    aem_bact_flavobacteriales_genera_t <- as.data.frame(aem_bact_flavobacteriales_genera_t)
    aem_bact_flavobacteriales_genera_t <- aem_bact_flavobacteriales_genera_t %>% rownames_to_column("sample_id")
    aem_bact_flavobacteriales_genera_sums <- merge(metadata, aem_bact_flavobacteriales_genera_t, by = "sample_id")

    
#Extract Genera from Pseudomonadales within Proteobacteria (GROUPS)

    #Pseudomonadales
    aem_proteo_pseudomonadales_genera <- aem_proteo_pseudomonadales[,-c(1:6,8)]
    aem_proteo_pseudomonadales_genera <- as.data.frame(aem_proteo_pseudomonadales_genera)
    aem_proteo_pseudomonadales_genera <- aem_proteo_pseudomonadales_genera %>% group_by(Genus) %>% summarize_all(funs(sum))
    aem_proteo_pseudomonadales_genera <- aem_proteo_pseudomonadales_genera %>% drop_na()
    aem_proteo_pseudomonadales_genera <- column_to_rownames(aem_proteo_pseudomonadales_genera, "Genus")
    aem_proteo_pseudomonadales_genera_t <- t(aem_proteo_pseudomonadales_genera)
    aem_proteo_pseudomonadales_genera_t <- as.data.frame(aem_proteo_pseudomonadales_genera_t)
    aem_proteo_pseudomonadales_genera_t <- aem_proteo_pseudomonadales_genera_t %>% rownames_to_column("sample_id")
    aem_proteo_pseudomonadales_genera_sums <- merge(metadata, aem_proteo_pseudomonadales_genera_t, by = "sample_id")
    
  
#Separate genera into subsetted datasets
aem_firm_lactobacillales_genera_sums_forest_ncr <- aem_firm_lactobacillales_genera_sums %>% filter(Habitat == "Forest") %>% filter(Raider == "No")
aem_firm_bacillales_genera_sums_forest_ncr <- aem_firm_bacillales_genera_sums %>% filter(Habitat == "Forest") %>% filter(Raider == "No")

aem_bacillales_genera_by_species <- aem_firm_bacillales_genera_sums_forest_ncr[,-c(1:3,5,6,9:11)]
aem_lactobacillales_genera_by_species <- aem_firm_lactobacillales_genera_sums_forest_ncr[,-c(1:3,5,6,9:11)]


aem_bact_flavobacteriales_genera_sums_africana <- aem_bact_flavobacteriales_genera_sums %>% filter(Species == "africana")
aem_proteo_pseudomonadales_genera_sums_africana <- aem_proteo_pseudomonadales_genera_sums %>% filter(Species == "africana")

aem_flavobacteriales_genera_by_group <- aem_bact_flavobacteriales_genera_sums_africana[,-c(1:6,9,11)]
aem_pseudomonadales_genera_by_group <- aem_proteo_pseudomonadales_genera_sums_africana[,-c(1:6,9,11)]


colnames(aem_flavobacteriales_genera_by_group) <- c("Age", "Sex", "Group", "Chryseobacterium", "Elizabethkingia", "Flavobacterium", "Fluviicola", "Wautersiella")
colnames(aem_pseudomonadales_genera_by_group) <- c("Age", "Sex", "Group", "Acinetobacter", "Perlucidibaca", "Pseudomonas", "Serpens" )

#Calculate p-values for genera at group level and at species level

gwaut <- glmer.nb(Wautersiella ~ Group + (1|Sex), data = aem_flavobacteriales_genera_by_group)
gwaut_p <- Anova(gwaut)$`Pr(>Chisq)`

#Post-hoc Tukey_test
post_gwaut <- glht(gwaut, mcp(Group = "Tukey"))
summary(post_gwaut)

gacin <- glmer.nb(Acinetobacter ~ Group + (1|Sex), data = aem_pseudomonadales_genera_by_group)
gacin_p <- Anova(gacin)$`Pr(>Chisq)`

#Post-hoc Tukey_test
post_gacin <- glht(gacin, mcp(Group = "Tukey"))
summary(post_gacin)

gbacillus <- glmer.nb(g__Bacillus ~ Species + (1|Sex), data = aem_bacillales_genera_by_species)
gbacillus_p <- Anova(gbacillus)$`Pr(>Chisq)`

gsolibac <- glmer.nb(g__Solibacillus ~ Species + (1|Sex), data = aem_bacillales_genera_by_species)
gsolibac_p <- Anova(gsolibac)$`Pr(>Chisq)`

p_vals_bacillales_species <- as.data.frame(c(gbacillus_p, gsolibac_p))
colnames(p_vals_bacillales_species) <- c("pval")
p_vals_bacillales_species_fdr <- p.adjust(p_vals_bacillales_species$pval, method = "fdr")
```

## calculate p values for ALL GENERA
```{r}
#P-values

#Bacillales
p_vals_bacillales_genera_by_species <- data.frame(numeric(length = ncol(aem_bacillales_genera_by_species)))

    for (ii in 1:ncol(aem_bacillales_genera_by_species)) {
  
      col = aem_bacillales_genera_by_species[,ii] #this tells the for loop that it should iterate through columns in the data set
      lm_temp <- lm(col ~ Species, data = aem_bacillales_genera_by_species) #tells the for loop to run linear model  for each column           specified above
      p_value <- summary(lm_temp)$coefficients[2,4] #extracts p value
      p_vals_bacillales_genera_by_species[ii,] <- p_value
    }

#Lactobacillales
p_vals_lactobacillales_genera_by_species <- data.frame(numeric(length = ncol(aem_lactobacillales_genera_by_species)))

    for (ii in 1:ncol(aem_lactobacillales_genera_by_species)) {
  
      col = aem_lactobacillales_genera_by_species[,ii] #this tells the for loop that it should iterate through columns in the data set
      lm_temp <- lm(col ~ Species, data = aem_lactobacillales_genera_by_species) #tells the for loop to run linear model  for each            column specified above
      p_value <- summary(lm_temp)$coefficients[2,4] #extracts p value
      p_vals_lactobacillales_genera_by_species[ii,] <- p_value
    }

#Correct p-values for multiple tests using false discovery rate
p_vals_bacillales_genera_by_species <- p_vals_bacillales_genera_by_species %>% drop_na() %>% as.data.frame()
colnames(p_vals_bacillales_genera_by_species) <- c("pvals")

p_vals_bacillales_genera_by_species_fdr <- p.adjust(p_vals_bacillales_genera_by_species$pvals, method = "fdr")

p_vals_lactobacillales_genera_by_species <- p_vals_lactobacillales_genera_by_species %>% drop_na() %>% as.data.frame()
colnames(p_vals_lactobacillales_genera_by_species) <- c("pvals")
p_vals_lactobacillales_genera_by_species_fdr <- p.adjust(p_vals_lactobacillales_genera_by_species$pvals, method ="fdr")

```

## Make box plots for ALL GENERA
```{r}
#square root transform the y-axis
species_bacillales_genera_abun <- read_excel("species_bacillales_genera_abun.xlsx")
species_bacillales_genera_abun <- species_bacillales_genera_abun %>% mutate(log_abundance = log10(abundance), sqrt_abundance = sqrt(abundance))

species_lactobacillales_genera_abun <- read_excel("species_lactobacillales_genera_abun.xlsx")
species_lactobacillales_genera_abun <- species_lactobacillales_genera_abun %>% mutate(log_abundance = log10(abundance), sqrt_abundance = sqrt(abundance))


#Plots of genera within Firmicutes Orders by African Elephant Species
bacillales_plot <- ggplot(species_bacillales_genera_abun, aes(x = order, y = sqrt_abundance, fill = species)) + geom_boxplot(outlier.shape = NA, show.legend = T) + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + labs(title = "Genera of Order Bacillales", fill = "Elephant Species", x = "Genus", y = "Mean Abundance") + scale_fill_manual(values = c("red", "blue")) + theme(legend.text = element_text(face = "italic")) + theme(axis.text.x = element_text(face = "italic")) + theme(legend.position = c(0.05, 0.7))

lacto_plot <- ggplot(species_lactobacillales_genera_abun, aes(x = order, y = sqrt_abundance, fill = species)) + geom_boxplot(outlier.shape = NA, show.legend = F) + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + labs(title = "Genera of Order Lactobacillales", x = "Genus", y = "Mean Abundance") + scale_fill_manual(values = c("red", "blue")) + theme(axis.text.x = element_text(face = "italic"))


jpeg("/Users/joegunn/Desktop/Grad_School_Stuff/Projects/Elephant_Microbiome/NEW_STATS/AEMB_FINAL_STATS/African_Elephant_Microbiome/firmicutes_genera_byspecies.jpeg", width=3700, height=5000, units='px', res=600) 

plot_grid(bacillales_plot, lacto_plot, nrow = 2, labels = c("A", "B"))

dev.off()
```

## Make Box Plots for FILTERED GENERA (AVERAGE RELATIVE ABUNDANCE = 0.1 OR MORE)
```{r}
#BY SPECIES

#square root transform the y-axis
bacillales_genera_abun_rel0.1 <- read_excel("bacillales_genera_abun_rel0.1.xlsx")
bacillales_genera_abun_rel0.1 <- bacillales_genera_abun_rel0.1 %>% mutate(log_abundance = log10(abundance), sqrt_abundance = sqrt(abundance))

#Plots of genera within Firmicutes Orders by African Elephant Species
bacillales_plot_rel0.1 <- ggplot(bacillales_genera_abun_rel0.1, aes(x = Genus, y = sqrt_abundance, fill = Species)) + geom_boxplot(outlier.shape = NA, show.legend = T) + theme(axis.text = element_text(size = 15)) + theme(axis.text.x = element_text(size = 20)) + labs(title = "Genera of Phylum Firmicutes: Order Bacillales", fill = "Elephant Species", x = "Genus", y = "Mean Abundance") + theme(plot.title = element_text(size = 20, hjust = 0.5, face = "bold")) + scale_fill_manual(values = c("red", "blue")) + theme(legend.text = element_text(face = "italic")) + theme(axis.text.x = element_text(face = "italic")) + theme(legend.position = c(0.2, 0.7)) + theme(plot.background = element_blank())  + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),panel.background = element_blank(), axis.line = element_line(colour = "black")) + theme(legend.title = element_text(size = 20)) + theme(legend.text = element_text(size = 20)) + theme(axis.title = element_text(size = 20))


pdf("/Users/joegunn/Desktop/Grad_School_Stuff/Projects/Elephant_Microbiome/NEW_STATS/AEMB_FINAL_STATS/African_Elephant_Microbiome/firmicutes_genera_byspecies.pdf", width=12, height=10) 

bacillales_plot_rel0.1

dev.off()


#BY GROUPS


flavobacteriales_genera_abun_rel0.1 <- read_excel("flavobacteriales_genera_abun_rel0.1.xlsx")
flavobacteriales_genera_abun_rel0.1 <- flavobacteriales_genera_abun_rel0.1 %>% mutate(log_abundance = log10(abundance), sqrt_abundance = sqrt(abundance))

pseudomonadales_genera_abun_rel0.1 <- read_excel("pseudomonadales_genera_abun_rel0.1.xlsx")
pseudomonadales_genera_abun_rel0.1 <- pseudomonadales_genera_abun_rel0.1 %>% mutate(log_abundance = log10(abundance), sqrt_abundance = sqrt(abundance))

#Flavobacteriales Plot

flavobacteriales_plot_rel0.1 <- ggplot(flavobacteriales_genera_abun_rel0.1, aes(x = Genus, y = sqrt_abundance, fill = Group)) + geom_boxplot(outlier.shape = NA, show.legend = T) + theme(axis.text = element_text(size = 15)) + theme(axis.text.x = element_text(size = 20)) + labs(title = "Genus of Phylum Bacteroidetes: Order Flavobacteriales", fill = "Elephant Group", x = "Genus", y = "Mean Abundance") + theme(plot.title = element_text(size = 20, hjust = 0.5, face = "bold")) + scale_fill_manual(values = c("green", "purple", "red", "goldenrod")) + theme(legend.text = element_text(face = "italic")) + theme(axis.text.x = element_text(face = "italic")) + theme(legend.position = c(0.8, 0.8)) + theme(plot.background = element_blank())  + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),panel.background = element_blank(), axis.line = element_line(colour = "black")) + theme(legend.title = element_text(size = 20)) + theme(legend.text = element_text(size = 20)) + theme(axis.title = element_text(size = 20))

#Pseudomonadales Plot

pseudomonadales_plot_rel0.1 <- ggplot(pseudomonadales_genera_abun_rel0.1, aes(x = Genus, y = sqrt_abundance, fill = Group)) + geom_boxplot(outlier.shape = NA, show.legend = T) + theme(axis.text = element_text(size = 15)) + theme(axis.text.x = element_text(size = 20)) + labs(title = "Genus of Phylum Proteobacteria: Order Pseudomonadales", fill = "Elephant Group", x = "Genus", y = "Mean Abundance") + theme(plot.title = element_text(size = 20, hjust = 0.5, face = "bold")) + scale_fill_manual(values = c("green", "purple", "red", "goldenrod")) + theme(legend.text = element_text(face = "italic")) + theme(axis.text.x = element_text(face = "italic")) + theme(legend.position = c(0.8, 0.8)) + theme(plot.background = element_blank())  + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),panel.background = element_blank(), axis.line = element_line(colour = "black")) + theme(legend.title = element_text(size = 20)) + theme(legend.text = element_text(size = 20)) + theme(axis.title = element_text(size = 20))

#Combined Plot

pdf("/Users/joegunn/Desktop/Grad_School_Stuff/Projects/Elephant_Microbiome/NEW_STATS/AEMB_FINAL_STATS/African_Elephant_Microbiome/genera_combined_bygroup.pdf", width=10, height=15) 

plot_grid(flavobacteriales_plot_rel0.1, pseudomonadales_plot_rel0.1, nrow = 2, labels = c("A", "B"), label_size = 20)

dev.off()
```


### ----------------------- END OF ANALYSIS 4: MICROBIOME DIVERSITY AND DIFFERENTIAL OTU ABUNDANCE ANALYSIS  ----------------------- ###