---
title: "Analysis 3: OTU comparative abundance analysis "
author: "Joe Gunn"
date: "2024-02-07"
output: html_document
---

# Project: Assessing effects of phylogeny, diet, and habitat on gut microbial composition in African elephants 
We characterized the gut microbiome of African Savanna elephants (<i>Loxodonta africana</i>) and African Forest elephants (<i>Loxodonta cyclotis</i>). We assessed the relationship of gut microbial composition, including analyses of alpha and beta diversity, with host phylogeny (i.e., species) and habitat type (i.e., forest or savanna) for both species. We also assessed the relationship between microbial composition and diet (i.e., crop-raiding vs. non-crop-raiding) within <i>L. africana</i>. 

## Specific Aim: Comparing OTU abundance among African elephant sample groups (phylogeny, diet, and habitat) at multiple taxonomic levels
In this analysis, 

## Phases of analysis:
### Phase 1: Data read-in, filtering, rarefaction, and summarization
### Phase 2: Summary of OTU abundance data

## Libraries needed for analysis
```{r}
library(readxl)
library(cowplot)
library(tidyverse)
library(phyloseq)
library(qiime2R)
library(vegan)
library(ape)
```

## PHASE 1: DATA READ-IN, FILTERING, RAREFACTION, AND SUMMARIZATION
In this phase of the analysis, we read in raw data from bioinformatic analyses in QIIME, filter the raw OTU and taxonomic classification data, rarefy read counts, and summarize the full microbiome dataset.

### STEP 2: Summarize OTU abundance across samples and filter samples with low read-count for rarefaction
In this step, we summarize OTU abundance at the elephant sample level (i.e., how many individual sequence reads were detected per OTU within each elephant sample) to determine average read-count and to identify any outliers (i.e., either extremely high or low read counts). We use the raw phyloseq object with taxonomic classification, OTU abundance, phylogenetic relationship, and metadata. We calculate the sum of read counts across all OTUs within each sample. Ultimately, we need to rarefy the abundance to an even read depth per sample to limit observation bias; for this reason, we want to omit any samples with extremely low read-counts, because we do not want to rarefy the dataset to an extremely low read count.

#### 1a. Generate, clean, and summarize OTU abundance by elephant sample; run the Rmd chunk below.

##### Generate, clean, and summarize the OTU abundance data from phyloseq object:
```{r}
# Read in phyloseq object
load("../02_taxon_summary_analysis/data/phyloseq.Rda")

# Create a data object with otu abundnace data, clean, and gather to make tall table
otu_abundance <- as.data.frame(otu_table(phy_raw)) %>%
  rownames_to_column("otu_id") %>%
  mutate(otu_id = factor(otu_id)) %>%
  gather(MM69:M10, key = "ind_id", value = "abundance")

# Sum read counts for all individuals
reads <- otu_abundance %>%
  group_by(ind_id) %>%
  summarize(reads = sum(abundance)) %>%
  mutate(ind_id = factor(ind_id))

# Summarize mean (standard deviation) of read counts across the dataset
reads %>%
  summarize(mean = mean(reads),
            sd = sd(reads))

# Summarize minimum and maximum number of read counts across the dataset
reads %>%
  arrange(desc(reads))

# Save otu abundance summary table for downstream analysis 
save(reads, file = "data/reads.Rda")
```

### Data summary:

## Mean (standard deviation) of read counts for full dataset
<b><i>mean</i><sub>reads</sub> = 129,632.8 (463,334.5) </b><br>

## Minimum and maximum confidence for full dataset
<b><i>max</i><sub>reads</sub> = 3,268,723 </b><br>
<b><i>min</i><sub>reads</sub> = 163 </b><br>

#### 1b. Visualize OTU read counts per elephant sample; run the Rmd chunk below.

##### Plot OTU read counts per elephant sample: `figures/reads.pdf`
```{r}
# Load read count data
load("data/reads.Rda")

# Plot read counts per individual
pdf("figures/reads.pdf", width = 10, height = 4) 

ggplot(reads, aes(x = ind_id, y = reads)) + 
  geom_bar(stat = "identity", color = "black", fill = "grey") +
  theme_set(theme_cowplot(12)) +
  labs(x = "Elephant ID", y = "Read count") +
  theme(axis.title = element_text(size = 15)) + 
  theme(axis.text = element_text(size = 15)) + 
  theme(axis.text.x = element_text(angle = 90, hjust = 0.9, vjust = 0.3)) + 
  theme(panel.border = element_rect(colour = "black", fill = NA, size = 1))

dev.off()
```

#### 1c. Omit sample with extremely high read count relative to other samples (CR71), and re-visualize read counts to identify low-read individuals; run the Rmd chunk below.

##### Plot OTU read counts per elephant sample without individual CR71: `figures/reads_filtered.pdf`
```{r}
# Load read count data
load("data/reads.Rda")

# Omit individual CR71 from the data
reads <- reads %>%
  filter(ind_id != "CR71")

# Plot confidence and proportion of samples per taxonomic level
pdf("figures/reads_filtered.pdf", width = 10, height = 4) 

ggplot(reads, aes(x = ind_id, y = reads)) + 
  geom_bar(stat = "identity", color = "black", fill = "grey") +
  theme_set(theme_cowplot(12)) +
  labs(x = "Elephant ID", y = "Read count") +
  theme(axis.title = element_text(size = 15)) + 
  theme(axis.text = element_text(size = 15)) + 
  theme(axis.text.x = element_text(angle = 90, hjust = 0.9, vjust = 0.3)) + 
  theme(panel.border = element_rect(colour = "black", fill = NA, size = 1))

dev.off()
```

<b> Data summary results </b>: <br>

After data summarization, we observed that individual OB182 had an order of magnitude lower read count (163) than the next most abundant individual (11,460; see Step 1a above). For this reason, we opted to drop OB182 from the final dataset for downstream analyses, and the OTU raw abundance data were rarefied to 11,460 reads (Continue on to Step 1d).

#### 1d. Subset out sample with extremely low read count relative to other samples (OB182), and generate new phyloseq object for downstream analyses; run the Rmd chunk below.

##### Subset sample with low read count and generate new phyloseq object:
```{r}
# Load read count data
load("../02_taxon_summary_analysis/data/phyloseq.Rda")

# Remove individual OB182 from the data
phyloseq <- subset_samples(phyloseq, ind_id != "OB182") 

# Save filtered otu abundance phyloseq object
save(phyloseq, file = "data/phyloseq.Rda")
```

### STEP 2: Rarefy OTU abundance data
In this step, we rarefy the OTU abundance dataset to the lowest number of reads captured in the full dataset for an individual (CR3, <i>N</i>=11,460). Rarefaction is a necessary step to randomly sample OTU abundance data to an even number of "samples", aka reads, to prevent sampling bias in downstream analyses. 

#### 2a. Rarefy the OTU abundance data; run the Rmd chunk below.

##### Rarefy abundance data:
```{r}
# Load filtered phyloseq object
load("data/phyloseq.Rda")

# Read in, combine, and convert QIIME output and metadata to phyloseq format
phyloseq_rarefied <- rarefy_even_depth(phyloseq, 
                                       rngsee = 5)

# Save rarefied OTU abundance data for downstream analyses
save(phyloseq_rarefied, file = "data/phyloseq_rarefied.Rda")
```

<b> Data filtering results </b>: <br>
<b><i>read depth</i> = 11,460 </b><br>

### STEP 3: Summarize OTU abundance across taxonomic levels (i.e., kingdom, phylum, class, and order)
In this step, we summarize OTU abundance data across four different taxonomic levels (i.e., kingdom, phylum, class, order, etc.) based on prior analysis of taxonomic classification success and confidence; we ultimately summarize OTUs and the core microbiome of elephant samples at the Order level (see Analysis 2).

#### 3a. Summarize OTU abundance by kingdom, phylum, class, and order; run the Rmd chunk below.
In this step, we summarize OTU abundance at the kingdom level.

##### Rarefy abundance data:
```{r}
# Load rarefied phyloseq object
load("data/phyloseq_rarefied.Rda")

# Create a data object with otu abundnace data and clean
otu_abundance <- as.data.frame(t(otu_table(phyloseq_rarefied)))


vegan <- decostand(otu_abundance, "total") #standardize abundance data by dividing each number by the total number of samples (individuals in the dataset)
vegan <- vegdist(vegan, "bray") #calculate Bray-curtis distances between samples
pcoa <- pcoa(vegan) #calculates principal components

pcoa$vectors


pcs <- pcoa$vectors
pcs <- as.data.frame(pcs)
pcs <- pcs %>% rownames_to_column("sample_id")
ggplot(pcs, aes(x = Axis.1, y = Axis.2, label = sample_id)) +
  geom_point() +
  geom_text()


pcs <- merge(otu_abundance, pcs, by = "otu_id")
aem_physeq_ado1 <- column_to_rownames(aem_physeq_forest_ncr, "sample_id")





# Create a data object with taxonomic classification data and clean
taxa <- tax_table(phyloseq_rarefied) %>%
  as.data.frame() %>%
  rownames_to_column("otu_id") %>%
  mutate(otu_id = factor(otu_id), 
         kingdom = factor(Kingdom), 
         phylum = factor(Phylum), 
         class = factor(Class), 
         order = factor(Order), 
         family = factor(Family), 
         genus = factor(Genus), 
         species = factor(Species)) %>%
  select(otu_id, kingdom, phylum, class, order, family, genus, species)

# Create a data object with otu abundnace data and clean
otu_abundance <- as.data.frame(otu_table(phyloseq_rarefied)) %>%
  rownames_to_column("otu_id") %>%
  mutate(otu_id = factor(otu_id))

# Merge taxonomic classifications with confidence data
otu_abundance <- merge(otu_abundance, 
                       taxa, 
                       by = "otu_id")











# Create tall data that can be sorted by taxonomic level
 otu_abundance <- otu_abundance %>%
  gather(MM69:M10, key = "ind_id", value ="abundance")

# Sum read counts for kingdoms
otu_abundance %>%
  group_by(kingdom) %>%
  summarize(reads = sum(abundance))

# Sum read counts for phyla
otu_abundance %>%
  group_by(kingdom, phylum) %>%
  summarize(reads = sum(abundance))

# Sum read counts for classes
otu_abundance %>%
  group_by(class) %>%
  summarize(reads = sum(abundance))

# Sum read counts for orders
otu_abundance %>%
  group_by(order) %>%
  summarize(reads = sum(abundance))
```
## Data summary at Kingdom level:
<b><i>reads</i><sub>archae</sub> = 4,818 </b><br>
<b><i>reads</i><sub>bacteria</sub> = 533,802 </b><br>

## Data summary at Phylum level:
<b><i>reads</i><sub>euryarchaeota (archaea)</sub> = 4,818</b><br>
<b><i>reads</i><sub>actinobacteria (bacteria)</sub> = 1649</b><br>
<b><i>reads</i><sub>armatimonadetes (bacteria)</sub> = 318</b><br>
<b><i>reads</i><sub>bacteroidetes (bacteria)</sub> = 103,575</b><br>
<b><i>reads</i><sub>chloroflexi (bacteria)</sub> = 954 </b><br>
<b><i>reads</i><sub>cyanobacteria (bacteria)</sub> = 2,059</b><br>
<b><i>reads</i><sub>elusimicrobia (bacteria)</sub> = 145</b><br>
<b><i>reads</i><sub>fibrobacteres (bacteria)</sub> = 423</b><br>
<b><i>reads</i><sub>firmicutes (bacteria)</sub> = 202,891</b><br>
<b><i>reads</i><sub>ld1 (bacteria)</sub> = 13</b><br>
<b><i>reads</i><sub>lentisphaerae (bacteria)</sub> = 8853 </b><br>
<b><i>reads</i><sub>planctomycetes (bacteria)</sub> = 439 </b><br>
<b><i>reads</i><sub>proteoacteria (bacteria)</sub> = 150,111 </b><br>
<b><i>reads</i><sub>spirochaetes (bacteria)</sub> = 15,132 </b><br>
<b><i>reads</i><sub>sr1 (bacteria)</sub> = 92 </b><br>
<b><i>reads</i><sub>synergistetes (bacteria)</sub> = 942 </b><br>
<b><i>reads</i><sub>tenericutes (bacteria)</sub> = 6,590 </b><br>
<b><i>reads</i><sub>verrucomicrobia (bacteria)</sub> = 37,564 </b><br>
<b><i>reads</i><sub>wps-2 (bacteria)</sub> = 83 </b><br>

## Data summary at Class level:
<b><i>reads</i><sub>archae</sub> = 4,818 </b><br>
<b><i>reads</i><sub>bacteria</sub> = 533,802 </b><br>

## Data summary at Order level:
<b><i>reads</i><sub>archae</sub> = 4,818 </b><br>
<b><i>reads</i><sub>bacteria</sub> = 533,802 </b><br>


# OTU ABUNDANCE - ORDER LEVEL

## Rarefied OTU abundance data 
```{r, echo = FALSE}
options(scipen = 999) #gets rid of scientific notatation, useful for reading p-values

#read in taxonomic designations and OTU abundance
aem_physeq <- qza_to_phyloseq(features = "table.qza", taxonomy = "taxonomy.qza",  tree = "rooted-tree.qza", metadata = "METADATA.tsv") 

#omit sample OB182 due to low read count. We don't want to rarefy to such a low number of reads across the whole dataset
aem_physeq <- subset_samples(aem_physeq, Elephant != "OB182") 

#Rarefy and clean the data set for use in downstream analysis
aem_physeq <- rarefy_even_depth(aem_physeq, rngsee = 5) #must set seed in order to maintain the same rarefied number of otus

    #Number of OTUS: 8248
    #Number of reads per sample: 11460 
      
#Extract data of all types (meta data, all taxonomy data)
aem_meta <- as.data.frame(sample_data(aem_physeq))
aem_tax <- as.data.frame(tax_table(aem_physeq))
aem_tax <- aem_tax %>% rownames_to_column("otu")
aem_data <- as.data.frame(otu_table(aem_physeq))
aem_data <- aem_data %>% rownames_to_column("otu")
aem_tax_data <- merge(aem_tax, aem_data, by = "otu")
```

## Most Abundant Phyla
```{r}
#Subset the data by the three most abundant phyla: Bacteroidetes, firmicutes, proteobacteria
aem_bacteroidetes <- aem_tax_data %>% filter(Phylum == "p__Bacteroidetes")
aem_firmicutes <- aem_tax_data %>% filter(Phylum == "p__Firmicutes")
aem_proteobacteria <- aem_tax_data %>% filter(Phylum == "p__Proteobacteria")
```

## Phylum and Order Proportions per African Elephant Group
```{r, echo = FALSE}

#Proportions of each order contributing to each african elephant group
aem_order_phylum_data <- aem_tax_data[,-c(1:2,4,6:8)]
aem_order_phylum_data <- as.tibble(aem_order_phylum_data)
aem_order_phylum_data <- aem_order_phylum_data %>% group_by(Phylum, Order) %>% summarize_all(funs(sum))
```

## Core Microbiome - Order Level
```{r}
#Summarize Rarefied OTU abundance by ALL Orders to figure out the core microbiome 
aem_order_data <- aem_tax_data[,-c(1:4,6:8)]
aem_order_data <- as.data.frame(aem_order_data)
aem_order_data <- aem_order_data %>% group_by(Order) %>% summarize_all(funs(sum))
aem_order_data <- aem_order_data %>% drop_na()
aem_order_data <- column_to_rownames(aem_order_data, "Order")
aem_order_data_t <- t(aem_order_data)
aem_order_data_t <- as.data.frame(aem_order_data_t)
aem_order_data_t <- aem_order_data_t %>% rownames_to_column("sample_id")
aem_order <- merge(metadata, aem_order_data_t, by = "sample_id")

###IMPORTANT NOTE: At this point, aem_orders, which provides abundance across all orders for all samples, was copied and entered into excel in order to determine the core microbiome and any other shared orders among african elephant groups. Figrue 1 in the manuscript is produced from the data within this excel file.
```

## Order Abundance within Phylum Bacteroidetes, Firmicutes, and Proteobacteria
```{r}
#Summarize Rarefied OTU abundance by Orders WITHIN BACTERIOIDETES
aem_bact_orders <- aem_bacteroidetes[,-c(1:4,6:8)]
aem_bact_orders <- as.data.frame(aem_bact_orders)
aem_bact_orders <- aem_bact_orders %>% group_by(Order) %>% summarize_all(funs(sum))
aem_bact_orders <- aem_bact_orders %>% drop_na()
aem_bact_orders <- column_to_rownames(aem_bact_orders, "Order")
aem_bact_orders_t <- t(aem_bact_orders)
aem_bact_orders_t <- as.data.frame(aem_bact_orders_t)
aem_bact_orders_t <- aem_bact_orders_t %>% rownames_to_column("sample_id")
aem_bact_orders_sums <- merge(metadata, aem_bact_orders_t, by = "sample_id")

#Summarize Rarefied OTU abundance by Orders WITHIN FIRMICUTES
aem_firm_orders <- aem_firmicutes[,-c(1:4,6:8)]
aem_firm_orders <- as.data.frame(aem_firm_orders)
aem_firm_orders <- aem_firm_orders %>% group_by(Order) %>% summarize_all(funs(sum))
aem_firm_orders <- aem_firm_orders %>% drop_na()
aem_firm_orders <- column_to_rownames(aem_firm_orders, "Order")
aem_firm_orders_t <- t(aem_firm_orders)
aem_firm_orders_t <- as.data.frame(aem_firm_orders_t)
aem_firm_orders_t <- aem_firm_orders_t %>% rownames_to_column("sample_id")
aem_firm_orders_sums <- merge(metadata, aem_firm_orders_t, by = "sample_id")

#Summarize Rarefied OTU abundance by Orders WITHIN PROTEOBACTERIA
aem_proteo_orders <- aem_proteobacteria[,-c(1:4,6:8)]
aem_proteo_orders <- as.data.frame(aem_proteo_orders)
aem_proteo_orders <- aem_proteo_orders %>% group_by(Order) %>% summarize_all(funs(sum))
aem_proteo_orders <- aem_proteo_orders %>% drop_na()
aem_proteo_orders <- column_to_rownames(aem_proteo_orders, "Order")
aem_proteo_orders_t <- t(aem_proteo_orders)
aem_proteo_orders_t <- as.data.frame(aem_proteo_orders_t)
aem_proteo_orders_t <- aem_proteo_orders_t %>% rownames_to_column("sample_id")
aem_proteo_orders_sums <- merge(metadata, aem_proteo_orders_t, by = "sample_id")

#Compile sums for most abundant phyla and subset the data for downstream analyses. One dataset for Africana (n = 35), one dataset for Forest non-crop-raiders (n = 19)

#Africana - diet and habitat
aem_bact_orders_sums_africana <- aem_bact_orders_sums %>% filter(Species == "africana")
aem_firm_orders_sums_africana <- aem_firm_orders_sums %>% filter(Species == "africana")
aem_proteo_orders_sums_africana <- aem_proteo_orders_sums %>% filter(Species == "africana")

#Forest, non-crop-raiders - species
aem_bact_orders_sums_forest_ncr <- aem_bact_orders_sums %>% filter(Habitat == "Forest") %>% filter(Raider == "No")
aem_firm_orders_sums_forest_ncr <- aem_firm_orders_sums %>% filter(Habitat == "Forest") %>% filter(Raider == "No")
aem_proteo_orders_sums_forest_ncr <- aem_proteo_orders_sums %>% filter(Habitat == "Forest") %>% filter(Raider == "No")


#Analyze OTU sum data by African elephant species
aem_bact_species <- aem_bact_orders_sums_forest_ncr[,-c(1:3,5:6,9:11)]
colnames(aem_bact_species) <- c("Species", "Age", "Sex", "Saprospirales", "Bacteroidales", "Cytophagales", "Flavobacteriales", "Sphingobacteriales")

aem_firm_species <- aem_firm_orders_sums_forest_ncr[,-c(1:3,5:6,9:11)]
colnames(aem_firm_species) <- c("Species", "Age", "Sex", "Bacillales", "Clostridiales", "Erysipelotrichales", "Lactobacillales")


aem_proteo_species <- aem_proteo_orders_sums_forest_ncr[,-c(1:3,5:6,9:11)]
colnames(aem_proteo_species) <- c("Species", "Age", "Sex", "Aeromonadales", "Alteromonadales", "Bdellovibrionales", "Burkholderiales", "Caulobacterales", "Desulfovibrionales", "Enterobacteriales", "gmd14h09", "Kiloniellales", "Neisseriales", "Oceanospirillales", "Pasteurellales", "phoshd29", "Pseudomonadales", "rf32", "Rhizobiales", "Rhodobacterales", "Rickettsiales", "Sphingomonadales", "Spirobacillales", "Tremblayales", "Xanthomonadales")

#Analyze OTU sum data by African elephant Groups
aem_bact_groups <- aem_bact_orders_sums_africana[,-c(1:6,9,11)]
colnames(aem_bact_groups) <- c("Age", "Sex", "Group" , "Saprospirales", "Bacteroidales", "Cytophagales", "Flavobacteriales", "Sphingobacteriales")

aem_firm_groups <- aem_firm_orders_sums_africana[,-c(1:6,9,11)]
colnames(aem_firm_groups) <- c("Age", "Sex", "Group", "Bacillales", "Clostridiales", "Erysipelotrichales", "Lactobacillales")

aem_proteo_groups <- aem_proteo_orders_sums_africana[,-c(1:6,9,11)]
colnames(aem_proteo_groups) <- c("Age", "Sex", "Group","Aeromonadales", "Alteromonadales", "Bdellovibrionales", "Burkholderiales", "Caulobacterales", "Desulfovibrionales", "Enterobacteriales", "gmd14h09", "Kiloniellales", "Neisseriales", "Oceanospirillales", "Pasteurellales", "phoshd29", "Pseudomonadales", "rf32", "Rhizobiales", "Rhodobacterales", "Rickettsiales", "Sphingomonadales", "Spirobacillales", "Tremblayales", "Xanthomonadales")
```

## Testing for Normality in all orders
```{r}

bact_norm1 <- ggqqplot(aem_bact_species$Bacteroidales)
      sb1 <- shapiro.test(aem_bact_species$Bacteroidales) #NORMAL
bact_norm2 <- ggqqplot(aem_bact_species$Cytophagales)
      sb2 <- shapiro.test(aem_bact_species$Cytophagales) #NON-NORMAL
bact_norm3 <- ggqqplot(aem_bact_species$Saprospirales)
      sb3 <- shapiro.test(aem_bact_species$Saprospirales) #NON-NORMAL
bact_norm4 <- ggqqplot(aem_bact_species$Flavobacteriales)
      sb4 <- shapiro.test(aem_bact_species$Flavobacteriales) #NON-NORMAL
bact_norm5 <- ggqqplot(aem_bact_species$Sphingobacteriales)
      sb5 <- shapiro.test(aem_bact_species$Sphingobacteriales) #NON-NORMAL


firm_norm1 <- ggqqplot(aem_firm_species$Bacillales)
      sf1 <- shapiro.test(aem_firm_species$Bacillales) #NON-NORMAL
firm_norm2 <- ggqqplot(aem_firm_species$Clostridiales)
      sf2 <- shapiro.test(aem_firm_species$Clostridiales) #NORMAL
firm_norm3 <- ggqqplot(aem_firm_species$Erysipelotrichales)
      sf3 <- shapiro.test(aem_firm_species$Erysipelotrichales) #NON-NORMAL
firm_norm4 <- ggqqplot(aem_firm_species$Lactobacillales)
      sf4 <- shapiro.test(aem_firm_species$Lactobacillales) #NON-NORMAL
      
#ALL NOT NORMAL EXCEPT PSUEDOMONADALES
proteo_norm1 <- ggqqplot(aem_proteo_species$Aeromonadales)
      sp1 <- shapiro.test(aem_proteo_species$Aeromonadales)
proteo_norm2 <- ggqqplot(aem_proteo_species$Alteromonadales)
      sp2 <- shapiro.test(aem_proteo_species$Alteromonadales)
proteo_norm3 <- ggqqplot(aem_proteo_species$Bdellovibrionales)
      sp3 <- shapiro.test(aem_proteo_species$Bdellovibrionales)
proteo_norm4 <- ggqqplot(aem_proteo_species$Burkholderiales)
      sp4 <- shapiro.test(aem_proteo_species$Burkholderiales)
proteo_norm5 <- ggqqplot(aem_proteo_species$Caulobacterales)
      sp5 <- shapiro.test(aem_proteo_species$Caulobacterales)
proteo_norm6 <- ggqqplot(aem_proteo_species$Desulfovibrionales)
      sp6 <- shapiro.test(aem_proteo_species$Desulfovibrionales)
proteo_norm7 <- ggqqplot(aem_proteo_species$Enterobacteriales)
      sp7 <- shapiro.test(aem_proteo_species$Enterobacteriales)
proteo_norm8 <- ggqqplot(aem_proteo_species$gmd14h09)
      sp8 <- shapiro.test(aem_proteo_species$gmd14h09)
proteo_norm9 <- ggqqplot(aem_proteo_species$Kiloniellales)
      sp9 <- shapiro.test(aem_proteo_species$Kiloniellales)
proteo_norm10 <- ggqqplot(aem_proteo_species$Neisseriales)
      sp10 <- shapiro.test(aem_proteo_species$Neisseriales)
proteo_norm11 <- ggqqplot(aem_proteo_species$Oceanospirillales)
      sp11 <- shapiro.test(aem_proteo_species$Oceanospirillales)
proteo_norm12 <- ggqqplot(aem_proteo_species$Pasteurellales)
      sp12 <- shapiro.test(aem_proteo_species$Pasteurellales)
proteo_norm13 <- ggqqplot(aem_proteo_species$phoshd29)
      sp13 <- shapiro.test(aem_proteo_species$phoshd29)
proteo_norm14 <- ggqqplot(aem_proteo_species$Pseudomonadales) 
      sp14 <- shapiro.test(aem_proteo_species$Pseudomonadales) #NORMAL
proteo_norm15 <- ggqqplot(aem_proteo_species$rf32)
      sp15 <- shapiro.test(aem_proteo_species$rf32)
proteo_norm16 <- ggqqplot(aem_proteo_species$Rhizobiales)
      sp16 <- shapiro.test(aem_proteo_species$Rhizobiales)
proteo_norm17 <- ggqqplot(aem_proteo_species$Rhodobacterales)
      sp17 <- shapiro.test(aem_proteo_species$Rhodobacterales)
proteo_norm18 <- ggqqplot(aem_proteo_species$Rickettsiales)
      sp18 <- shapiro.test(aem_proteo_species$Rickettsiales)
proteo_norm19 <- ggqqplot(aem_proteo_species$Sphingomonadales)
      sp19 <- shapiro.test(aem_proteo_species$Sphingomonadales)
proteo_norm20 <- ggqqplot(aem_proteo_species$Tremblayales)
      sp20 <- shapiro.test(aem_proteo_species$Tremblayales)
proteo_norm21 <- ggqqplot(aem_proteo_species$Xanthomonadales)
      sp21 <- shapiro.test(aem_proteo_species$Xanthomonadales)
```

## Statistical Significance of Bacterial Orders By African Elephant Species
```{r}
aem_bact_species_rel0.1 <- aem_bact_species[,-c(4,6,8)]
aem_firm_species_rel0.1 <- aem_firm_species[,-c(6,7)]
aem_proteo_species_rel0.1 <- aem_proteo_species[,-c(4:9,11:16,18:25)]

#PHYLUM Bacteroidetes
p_vals_bact_species <- data.frame(numeric(length = 2))

    for (ii in 4:ncol(aem_bact_species_rel0.1)) {
  
      col = aem_bact_species_rel0.1[,ii] #this tells the for loop that it should iterate through columns in the data set
      
      ff <- as.formula(paste0(colnames(aem_bact_species_rel0.1)[ii]," ~ Species + (1|Sex)"))
      
      glmer_temp <- glmer.nb(ff,  aem_bact_species_rel0.1) #tells the for loop to run linear model  for each column           specified above
      p_value <- summary(glmer_temp)$coefficients[2,4] #extracts p value
      p_vals_bact_species[ii-3,] <- p_value
    }

#PHYLUM Firmicutes
p_vals_firm_species <- data.frame(numeric(length = 2))

    for (ii in 4:ncol(aem_firm_species_rel0.1)) {
  
      col = aem_firm_species_rel0.1[,ii] #this tells the for loop that it should iterate through columns in the data set
      ff <- as.formula(paste0(colnames(aem_firm_species_rel0.1)[ii]," ~ Species + (1|Sex)"))
      
      glmer_temp <- glmer.nb(ff, data = aem_firm_species_rel0.1) #tells the for loop to run linear model  for each column           specified above
      p_value <- summary(glmer_temp)$coefficients[2,4] #extracts p value
      p_vals_firm_species[ii-3,] <- p_value
    }

#PHYLUM Proteobacteria
aem_proteo_species <- aem_proteo_species[,-c(23)] #SPIROBACILLES REMOVED BECAUSE IT WAS ALL ZEROS
p_vals_proteo_species <- data.frame(numeric(length = 2))

    for (ii in 4:ncol(aem_proteo_species_rel0.1)) {
  
      col = aem_proteo_species_rel0.1[,ii] #this tells the for loop that it should iterate through columns in the data set
      ff <- as.formula(paste0(colnames(aem_proteo_species_rel0.1)[ii]," ~ Species + (1|Sex)"))
      
      glmer_temp <- glmer.nb(ff, data = aem_proteo_species_rel0.1) #tells the for loop to run linear model  for each column         specified above
      p_value <- summary(glmer_temp)$coefficients[2,4] #extracts p value
      p_vals_proteo_species[ii-3,] <- p_value
    }

#Convert dataframes with p_values for each order within each phylum into tables we can work with
    
#PHYLUM Bacteroidetes orders p-values
p_vals_bact_species <- p_vals_bact_species %>% drop_na()
colnames(p_vals_bact_species) <- c("pvals")
p_vals_bact_species <- as.data.frame(p_vals_bact_species)

#PHYLUM Firmicutes orders p-values
p_vals_firm_species <- p_vals_firm_species %>% drop_na()
colnames(p_vals_firm_species) <- c("pvals")
p_vals_firm_species <- as.data.frame(p_vals_firm_species)

#PHYLUM Proteobacteria orders p-values
p_vals_proteo_species <- p_vals_proteo_species %>% drop_na()
colnames(p_vals_proteo_species) <- c("pvals")
p_vals_proteo_species <- as.data.frame(p_vals_proteo_species)

    
#Correct p-values for multiple tests using false discovery rate
p_vals_bact_species_fdr <- p.adjust(p_vals_bact_species$pvals, method = "fdr")
p_vals_firm_species_fdr <- p.adjust(p_vals_firm_species$pvals, method = "fdr")
p_vals_proteo_species_fdr <- p.adjust(p_vals_proteo_species$pvals, method = "fdr")

#p_vals_bact_species_fdr #Nothing is significant
#p_vals_firm_species_fdr #Bacillales and Lactobacillales are significant
#p_vals_proteo_species_fdr #nothing significant
```

## Make Box Plots for ALL ORDERS
```{r}
#Log-transform y axis so that all values will show up

#PHYLUM Bacteroidetes
species_bacteroidetes_order_abun <- read_excel("species_bacteroidetes_order_abun.xlsx")
species_bacteroidetes_order_abun <- species_bacteroidetes_order_abun %>% mutate(log_abundance = log10(abundance), sqrt_abundance = sqrt(abundance))

#PHYLUM Firmicutes
species_firmicutes_order_abun <- read_excel("species_firmicutes_order_abun.xlsx")
species_firmicutes_order_abun <- species_firmicutes_order_abun %>% mutate(log_abundance = log10(abundance), sqrt_abundance = sqrt(abundance))

#PHYLUM Proteobacteria
species_proteobacteria_order_abun <- read_excel("species_proteobacteria_order_abun.xlsx")
species_proteobacteria_order_abun <- species_proteobacteria_order_abun %>% mutate(log_abundance = log10(abundance), sqrt_abundance = sqrt(abundance))
 
#Make box-plots of Orders for African Elephant Species

#Bacteroidetes
species_bacteroidetes_box_plots <- ggplot(species_bacteroidetes_order_abun, aes(x = order, y = sqrt_abundance, fill = sex)) + geom_boxplot(show.legend = T, outlier.shape = NA) + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + labs(fill = "Elephant Sex", title = "Orders by Elephant Species", y = "Bacteroidetes Abundance") + scale_fill_manual(values = c("red", "blue")) + theme(axis.title.x = element_blank()) + theme(legend.position = c(0.65,0.7)) + theme(legend.text = element_text(face = "italic")) + theme(axis.text.x = element_text(face = "italic")) 

#Firmicutes
species_firmicutes_box_plots <- ggplot(species_firmicutes_order_abun, aes(x = order, y = sqrt_abundance, fill = species)) + geom_boxplot(show.legend = F, outlier.shape = NA) + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + labs(y = "Firmicutes Abundance") + scale_fill_manual(values = c("red", "blue")) + theme(axis.title.x = element_blank()) + theme(axis.text.x = element_text(face = "italic"))


#Proteobacteria
species_proteobacteria_box_plots <- ggplot(species_proteobacteria_order_abun, aes(x = order, y = sqrt_abundance, fill = Species)) + geom_boxplot(show.legend = F, outlier.shape = NA) + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + labs(y = "Proteobacteria Abundance") + scale_fill_manual(values = c("red", "blue")) + theme(axis.title.x = element_blank())+ theme(axis.text.x = element_text(face = "italic"))
    

jpeg("/Users/joegunn/Desktop/Grad_School_Stuff/Projects/Elephant_Microbiome/NEW_STATS/AEMB_FINAL_STATS/African_Elephant_Microbiome/phylum_species_plots.jpeg", width=2000, height=3000, units='px', res=250) 

plot_grid(species_bacteroidetes_box_plots, species_firmicutes_box_plots, species_proteobacteria_box_plots, nrow = 3, labels = c("A", "B", "C"))

dev.off()


#For combining figures
phy_species_plot <- plot_grid(species_bacteroidetes_box_plots, species_firmicutes_box_plots, species_proteobacteria_box_plots, nrow = 3, labels = c("A", "B", "C"))
```

## Make Box Plots for FILTERED ORDERS (AVERAGE RELATIVE ABUNDANCE = 0.1) by Species
```{r}
#Log-transform y axis so that all values will show up

#PHYLUM Bacteroidetes
species_bacteroidetes_order_abun_rel0.1 <- read_excel("species_bacteroidetes_order_abun_rel0.1.xlsx")
species_bacteroidetes_order_abun_rel0.1 <- species_bacteroidetes_order_abun_rel0.1 %>% mutate(log_abundance = log10(abundance), sqrt_abundance = sqrt(abundance))

#PHYLUM Firmicutes
species_firmicutes_order_abun_rel0.1 <- read_excel("species_firmicutes_order_abun_rel0.1.xlsx")
species_firmicutes_order_abun_rel0.1 <- species_firmicutes_order_abun_rel0.1 %>% mutate(log_abundance = log10(abundance), sqrt_abundance = sqrt(abundance))

#PHYLUM Proteobacteria
species_proteobacteria_order_abun_rel0.1 <- read_excel("species_proteobacteria_order_abun_rel0.1.xlsx")
species_proteobacteria_order_abun_rel0.1 <- species_proteobacteria_order_abun_rel0.1 %>% mutate(log_abundance = log10(abundance), sqrt_abundance = sqrt(abundance))
 
#Make box-plots of Orders for African Elephant Species

#Bacteroidetes
species_bacteroidetes_box_plots_rel0.1 <- ggplot(species_bacteroidetes_order_abun_rel0.1, aes(x = order, y = sqrt_abundance, fill = species)) + geom_boxplot(show.legend = T, outlier.shape = NA) + theme(axis.text = element_text(size = 15)) + labs(fill = "Elephant Species", title = "Orders by Elephant Species", y = "Bacteroidetes Abundance") + theme(plot.title = element_text(size = 20, hjust = 0.5, face = "bold")) + scale_fill_manual(values = c("red", "blue")) + theme(axis.title.x = element_blank()) + theme(legend.position = c(0.85,0.8)) + theme(axis.title.y = element_text(size = 20)) + theme(legend.text = element_text(size = 20, face = "italic")) + theme(axis.text.x = element_text(face = "italic", size = 20)) + theme(plot.background = element_blank()) + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),panel.background = element_blank(), axis.line = element_line(colour = "black")) + theme(legend.title = element_text(size = 20))

#Firmicutes
species_firmicutes_box_plots_rel0.1 <- ggplot(species_firmicutes_order_abun_rel0.1, aes(x = order, y = sqrt_abundance, fill = species)) + geom_boxplot(show.legend = F, outlier.shape = NA) + theme(axis.text = element_text(size = 15)) + labs(y = "Firmicutes Abundance") + scale_fill_manual(values = c("red", "blue")) + theme(axis.title.x = element_blank()) + theme(axis.text.x = element_text(face = "italic", size = 20))  + theme(plot.background = element_blank())  + theme(axis.title.y = element_text(size = 20)) + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),panel.background = element_blank(), axis.line = element_line(colour = "black")) + theme(legend.box.background = element_blank())

#Proteobacteria
species_proteobacteria_box_plots_rel0.1 <- ggplot(species_proteobacteria_order_abun_rel0.1, aes(x = order, y = sqrt_abundance, fill = Species)) + geom_boxplot(show.legend = F, outlier.shape = NA) + theme(axis.text = element_text(size = 15)) + labs(y = "Proteobacteria Abundance") + scale_fill_manual(values = c("red", "blue")) + theme(axis.title.x = element_blank())+ theme(axis.text.x = element_text(face = "italic", size = 20))  + theme(axis.title.y = element_text(size = 20)) + theme(plot.background = element_blank())  + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),panel.background = element_blank(), axis.line = element_line(colour = "black")) + theme(legend.box.background = element_blank())
    

jpeg("/Users/joegunn/Desktop/Grad_School_Stuff/Projects/Elephant_Microbiome/NEW_STATS/AEMB_FINAL_STATS/African_Elephant_Microbiome/phylum_species_plots_rel0.1.jpeg", width=2000, height=3000, units='px', res=250) 

plot_grid(species_bacteroidetes_box_plots_rel0.1, species_firmicutes_box_plots_rel0.1, species_proteobacteria_box_plots_rel0.1, nrow = 3, labels = c("A", "B", "C"), label_size = 20)

dev.off()


#For combining figures
phy_species_plot_rel0.1 <- plot_grid(species_bacteroidetes_box_plots_rel0.1, species_firmicutes_box_plots_rel0.1, species_proteobacteria_box_plots_rel0.1, nrow = 3, labels = c("A", "B", "C"), label_size = 20)
```

## Statistical Signifiance of Orders By African Elephant Groups
```{r}
aem_bact_groups_rel0.1 <- aem_bact_groups[,-c(4,6,8)]
aem_firm_groups_rel0.1 <- aem_firm_groups[,-c(6,7)]
aem_proteo_groups_rel0.1 <- aem_proteo_groups[,-c(4:9,11:16,18:25)]

#PHYLUM Bacteroidetes
gbact <- glmer.nb(Bacteroidales ~ Group + (1|Sex), data = aem_bact_groups_rel0.1)
gbact_p <- Anova(gbact)$`Pr(>Chisq)` #P = 0.1889
gflav <- glmer.nb(Flavobacteriales ~ Group + (1|Sex), data = aem_bact_groups_rel0.1, control=glmerControl(optimizer="bobyqa", tolPwrss=1e-3, optCtrl = list(maxfun = 100000)))
gflav_p <- Anova(gflav)$`Pr(>Chisq)` #P = 0.001998

p_vals_bact_groups <- as.data.frame(c(gbact_p, gflav_p))
colnames(p_vals_bact_groups) <- c("pval")

#Post-hoc Tukey-test
post_flavo <- glht(gflav, mcp(Group = "Tukey"))
summary(post_flavo)

#PHYLUM Firmicutes
gbaci <- glmer.nb(Bacillales ~ Group + (1|Sex), data = aem_firm_groups_rel0.1)
gbaci_p <- Anova(gbaci)$`Pr(>Chisq)`
gclos <- glmer.nb(Clostridiales ~ Group + (1|Sex), data = aem_firm_groups_rel0.1)
gclos_p <- Anova(gclos)$`Pr(>Chisq)`

p_vals_firm_groups <- as.data.frame(c(gbaci_p, gclos_p))
colnames(p_vals_firm_groups) <- c("pval")

#Post-hoc Tukey-test
post_bacillales <- glht(gbaci, mcp(Group="Tukey"))
summary(post_bacillales)

#PHYLUM Proteobacteria
genter <- glmer.nb(Enterobacteriales ~ Group + (1|Sex), data = aem_proteo_groups_rel0.1)
genter_p <- Anova(genter)$`Pr(>Chisq)`
gpseudo <- glmer.nb(Pseudomonadales ~ Group + (1|Sex), data = aem_proteo_groups_rel0.1)
gpseudo_p <- Anova(gpseudo)$`Pr(>Chisq)`

p_vals_proteo_groups <- as.data.frame(c(genter_p, gpseudo_p))
colnames(p_vals_proteo_groups) <- c("pval")

#Post-hoc Tukey-Test
post_pseudo <- glht(gpseudo, mcp(Group = "Tukey"))
summary(post_pseudo)


#For loops for ALL ORDERS

#Bacteroides
p_vals_bact_groups <- data.frame(numeric(length = 2)) #This for loop had trouble knitting to html, so I included hashtags to force it to finish. WE DID USE THIS CODE TO PRODUCE RESULTS FOR THE MANUSCRIPT

   # for (ii in 4:ncol(aem_bact_groups_rel0.1)) {
      
    #  ff <- as.formula(paste0(colnames(aem_bact_groups_rel0.1)[ii]," ~ Group + (1|Sex)"))
    #  glmer_temp <- glmer.nb(ff, data = aem_bact_groups_rel0.1, control=glmerControl(optimizer="bobyqa", tolPwrss=1e-3, optCtrl = list(maxfun = 100000))) #tells the for loop to run linear model  for each column specified above
    #  p_value <- Anova(glmer_temp)$`Pr(>Chisq)` #extracts f statistics
    #  p_vals_bact_groups[ii-3,] <- p_value
    #}

#PHYLUM Firmicutes
p_vals_firm_groups <- data.frame(numeric(length = 2))

    for (ii in 4:ncol(aem_firm_groups_rel0.1)) {
  
      col = aem_firm_groups_rel0.1[,ii] #this tells the for loop that it should iterate through columns in the data set
      lm_temp <- lm(col ~ Group, data = aem_firm_groups_rel0.1) #tells the for loop to run linear model for each column specified above
      fstat <- summary(lm_temp)$fstatistic
      p_value <- pf(fstat[1], fstat[2], fstat[3], lower.tail = F) #extracts p value
      p_vals_firm_groups[ii-3,] <- p_value
    }

#PHYLUM Proteobacteria
p_vals_proteo_groups <- data.frame(numeric(length = 2))

    for (ii in 1:ncol(aem_proteo_groups_rel0.1)) {
  
      col = aem_proteo_groups_rel0.1[,ii] #this tells the for loop that it should iterate through columns in the data set
      lm_temp <- lm(col ~ Group, data = aem_proteo_groups_rel0.1) #tells the for loop to run linear model  for each column specified above
      fstat <- summary(lm_temp)$fstatistic
      p_value <- pf(fstat[1], fstat[2], fstat[3], lower.tail = F) #extracts p value
      p_vals_proteo_groups[ii,] <- p_value
    }

#Convert dataframes with p_values into tables we can work with

#PHYLUM Bacterioidetes
p_vals_bact_groups <- p_vals_bact_groups %>% drop_na()
colnames(p_vals_bact_groups) <- c("pval")
p_vals_bact_groups <- as.data.frame(p_vals_bact_groups)

#PHYLUM Firmicutes
p_vals_firm_groups <- p_vals_firm_groups %>% drop_na()
colnames(p_vals_firm_groups) <- c("pval")
p_vals_firm_groups <- as.data.frame(p_vals_firm_groups)

#PHYLUM Proteobacteria
p_vals_proteo_groups <- p_vals_proteo_groups %>% drop_na()
colnames(p_vals_proteo_groups) <- c("pval")
p_vals_proteo_groups <- as.data.frame(p_vals_proteo_groups)

#Correct p-values for multiple tests using false discovery rate
p_vals_bact_groups_fdr <- p.adjust(p_vals_bact_groups$pval, method = "fdr")
p_vals_firm_groups_fdr <- p.adjust(p_vals_firm_groups$pval, method = "fdr")
p_vals_proteo_groups_fdr <- p.adjust(p_vals_proteo_groups$pval, method = "fdr")

#p_vals_bact_groups_fdr
#p_vals_firm_groups_fdr
#p_vals_proteo_groups_fdr
```

## Make Box Plots for ALL ORDERS
```{r}
#Log-transform y axis so that all values will show up
groups_bacteroidetes_order_abun <- read_excel("groups_bacteroidetes_order_abun.xlsx")
groups_bacteroidetes_order_abun <- groups_bacteroidetes_order_abun %>% mutate(log_abundance = log10(abundance), sqrt_abundance = sqrt(abundance))

groups_firmicutes_order_abun <- read_excel("groups_firmicutes_order_abun.xlsx")
groups_firmicutes_order_abun <- groups_firmicutes_order_abun %>% mutate(log_abundance = log10(abundance), sqrt_abundance = sqrt(abundance))

groups_proteobacteria_order_abun <- read_excel("groups_proteobacteria_order_abun.xlsx")
groups_proteobacteria_order_abun  <- groups_proteobacteria_order_abun %>% mutate(log_abundance = log10(abundance), sqrt_abundance = sqrt(abundance))


#PHYLUM Bacterioidetes
bact_group_plots <- ggplot(groups_bacteroidetes_order_abun, aes(x = order, y = sqrt_abundance, fill = group)) + geom_boxplot(outlier.shape = NA, show.legend = T) + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + labs(title = "Orders by Elephant Group", fill = "Elephant Group") + scale_fill_manual(values = c("green", "purple", "red", "goldenrod")) + theme(axis.title.x = element_blank()) + theme(axis.title.y = element_blank()) + theme(legend.position = c(0.65,0.7)) + theme(axis.text.x = element_text(face = "italic"))  


#PHYLUM Firmicutes
firm_group_plots <- ggplot(groups_firmicutes_order_abun, aes(x = order, y = sqrt_abundance, fill = group)) + geom_boxplot(outlier.shape = NA, show.legend = F) + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + scale_fill_manual(values = c("green", "purple", "red", "goldenrod")) + theme(axis.title.x = element_blank()) + theme(axis.title.y = element_blank()) + theme(axis.text.x = element_text(face = "italic"))

#PHYLUM Proteobacteria
proteo_group_plots <- ggplot(groups_proteobacteria_order_abun, aes(x = order, y = sqrt_abundance, fill = group)) + geom_boxplot(outlier.shape = NA, show.legend = F) + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + scale_fill_manual(values = c("green", "purple", "red", "goldenrod")) + theme(axis.title.x = element_blank()) + theme(axis.title.y = element_blank()) + theme(axis.text.x = element_text(face = "italic"))

jpeg("/Users/joegunn/Desktop/Grad_School_Stuff/Projects/Elephant_Microbiome/NEW_STATS/AEMB_FINAL_STATS/African_Elephant_Microbiome/phylum_group_plots.jpeg", width=2000, height=3000, units='px', res=250) 

plot_grid(bact_group_plots, firm_group_plots, proteo_group_plots, nrow = 3, labels = c("A", "B", "C"))

dev.off()

#For combining figures

phy_group_plot <- plot_grid(bact_group_plots, firm_group_plots, proteo_group_plots, nrow = 3, labels = c("D", "E", "F"))

##WE did not look for genera within Saprospirales, because it is in such low abundance (only found in n = 8 samples)


###COMBINE PHYLUM PLOTS
pdf("/Users/joegunn/Desktop/Grad_School_Stuff/Projects/Elephant_Microbiome/NEW_STATS/AEMB_FINAL_STATS/African_Elephant_Microbiome/phy_combined_plots.pdf", width=15, height=20) 

plot_grid(phy_species_plot, phy_group_plot, ncol = 2)

dev.off()
```

## Make Box Plots for FILTERED ORDERS (AVERAGE RELATIVE ABUNDANCE = 0.1 or more)
```{r}
#Log-transform y axis so that all values will show up
groups_bacteroidetes_order_abun_rel0.1 <- read_excel("groups_bacteroidetes_order_abun_rel0.1.xlsx")
groups_bacteroidetes_order_abun_rel0.1 <- groups_bacteroidetes_order_abun_rel0.1 %>% mutate(log_abundance = log10(abundance), sqrt_abundance = sqrt(abundance))

groups_firmicutes_order_abun_rel0.1 <- read_excel("groups_firmicutes_order_abun_rel0.1.xlsx")
groups_firmicutes_order_abun_rel0.1 <- groups_firmicutes_order_abun %>% mutate(log_abundance = log10(abundance), sqrt_abundance = sqrt(abundance))

groups_proteobacteria_order_abun_rel0.1 <- read_excel("groups_proteobacteria_order_abun_rel0.1.xlsx")
groups_proteobacteria_order_abun_rel0.1  <- groups_proteobacteria_order_abun_rel0.1 %>% mutate(log_abundance = log10(abundance), sqrt_abundance = sqrt(abundance))


#PHYLUM Bacterioidetes
bact_group_plots_rel0.1 <- ggplot(groups_bacteroidetes_order_abun_rel0.1, aes(x = order, y = sqrt_abundance, fill = group)) + geom_boxplot(outlier.shape = NA, show.legend = T) + theme(plot.title = element_text(size = 20, hjust = 0.5, face = "bold")) + theme(axis.text = element_text(size = 20)) + labs(title = "Orders by Elephant Group", fill = "Elephant Group") + scale_fill_manual(values = c("green", "purple", "red", "goldenrod")) + theme(axis.title.x = element_blank()) + theme(axis.title.y = element_blank()) + theme(legend.position = c(0.85,0.8)) + theme(legend.text = element_text(size = 20)) + theme(axis.text.x = element_text(face = "italic", size = 20)) + theme(plot.background = element_blank())  + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),panel.background = element_blank(), axis.line = element_line(colour = "black")) + theme(legend.title = element_text(size = 20))


#PHYLUM Firmicutes
firm_group_plots_rel0.1 <- ggplot(groups_firmicutes_order_abun_rel0.1, aes(x = order, y = sqrt_abundance, fill = group)) + geom_boxplot(outlier.shape = NA, show.legend = F) + theme(axis.text = element_text(size = 15)) + scale_fill_manual(values = c("green", "purple", "red", "goldenrod")) + theme(axis.title.x = element_blank()) + theme(axis.title.y = element_blank()) + theme(axis.text.x = element_text(face = "italic", size = 20)) + theme(plot.background = element_blank())  + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), panel.background = element_blank(), axis.line = element_line(colour = "black")) + theme(legend.title = element_text(size = 20))

#PHYLUM Proteobacteria
proteo_group_plots_rel0.1 <- ggplot(groups_proteobacteria_order_abun_rel0.1, aes(x = order, y = sqrt_abundance, fill = group)) + geom_boxplot(outlier.shape = NA, show.legend = F) + theme(axis.text = element_text(size = 15)) + scale_fill_manual(values = c("green", "purple", "red", "goldenrod")) + theme(axis.title.x = element_blank()) + theme(axis.title.y = element_blank()) + theme(axis.text.x = element_text(face = "italic", size = 20)) + theme(plot.background = element_blank())  + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),panel.background = element_blank(), axis.line = element_line(colour = "black")) + theme(legend.title = element_text(size = 20))

jpeg("/Users/joegunn/Desktop/Grad_School_Stuff/Projects/Elephant_Microbiome/NEW_STATS/AEMB_FINAL_STATS/African_Elephant_Microbiome/phylum_group_plots_rel0.1.jpeg", width=2000, height=3000, units='px', res=250) 

plot_grid(bact_group_plots_rel0.1, firm_group_plots_rel0.1, proteo_group_plots_rel0.1, nrow = 3, labels = c("A", "B", "C"))

dev.off()


#For combining figures

phy_group_plot_rel0.1 <- plot_grid(bact_group_plots_rel0.1, firm_group_plots_rel0.1, proteo_group_plots_rel0.1, nrow = 3, labels = c("D", "E", "F"), label_size = 20)


##WE did not look for genera within Saprospirales, because it is in such low abundance (only found in n = 8 samples)


###COMBINE PHYLUM PLOTS
pdf("/Users/joegunn/Desktop/Grad_School_Stuff/Projects/Elephant_Microbiome/NEW_STATS/AEMB_FINAL_STATS/African_Elephant_Microbiome/phy_combined_plots_rel0.1.pdf", width=15, height=20)

plot_grid(phy_species_plot_rel0.1, phy_group_plot_rel0.1, ncol = 2)

dev.off()
```

# OTU ABUNDANCE - GENUS LEVEL

## Statistical Signifiance of Genera within Orders Bacillales and Lactobacillales By African Elephant Species
```{r}
#Now we want to find the genera within each of the significant orders (Bacillales and Lactobacillales within Firmicutes) and see which are significantly different between africana and cyclotis species
    
#Data frames with genera only within the significant orders for african elephant SPECIES
aem_firm_bacillales <- aem_firmicutes %>% filter(Order == "o__Bacillales")
aem_firm_lactobacillales <- aem_firmicutes %>% filter(Order == "o__Lactobacillales")

#Data frames with genera only within the significant orders for african elephant GROUPS
aem_bact_flavobacteriales <- aem_bacteroidetes %>% filter(Order == "o__Flavobacteriales")
aem_proteo_pseudomonadales <- aem_proteobacteria %>% filter(Order == "o__Pseudomonadales")

#Extract Genera from Bacillales and Lactobacillales within Firmicutes (SPECIES)

    #Bacillales
    aem_firm_bacillales_genera <- aem_firm_bacillales[,-c(1:6,8)]
    aem_firm_bacillales_genera <- as.data.frame(aem_firm_bacillales_genera)
    aem_firm_bacillales_genera <- aem_firm_bacillales_genera %>% group_by(Genus) %>% summarize_all(funs(sum))
    aem_firm_bacillales_genera <- aem_firm_bacillales_genera %>% drop_na()
    aem_firm_bacillales_genera <- column_to_rownames(aem_firm_bacillales_genera, "Genus")
    aem_firm_bacillales_genera_t <- t(aem_firm_bacillales_genera)
    aem_firm_bacillales_genera_t <- as.data.frame(aem_firm_bacillales_genera_t)
    aem_firm_bacillales_genera_t <- aem_firm_bacillales_genera_t %>% rownames_to_column("sample_id")
    aem_firm_bacillales_genera_sums <- merge(metadata, aem_firm_bacillales_genera_t, by = "sample_id")
    
    #Lactobacillales
    aem_firm_lactobacillales_genera <- aem_firm_lactobacillales[,-c(1:6,8)]
    aem_firm_lactobacillales_genera <- as.data.frame(aem_firm_lactobacillales_genera)
    aem_firm_lactobacillales_genera <- aem_firm_lactobacillales_genera %>% group_by(Genus) %>% summarize_all(funs(sum))
    aem_firm_lactobacillales_genera <- aem_firm_lactobacillales_genera %>% drop_na()
    aem_firm_lactobacillales_genera <- column_to_rownames(aem_firm_lactobacillales_genera, "Genus")
    aem_firm_lactobacillales_genera_t <- t(aem_firm_lactobacillales_genera)
    aem_firm_lactobacillales_genera_t <- as.data.frame(aem_firm_lactobacillales_genera_t)
    aem_firm_lactobacillales_genera_t <- aem_firm_lactobacillales_genera_t %>% rownames_to_column("sample_id")
    aem_firm_lactobacillales_genera_sums <- merge(metadata, aem_firm_lactobacillales_genera_t, by = "sample_id")


#Extract Genera from Flavobacteriales within Bacteroidetes (GROUPS)

    #Flavobacteriales
    aem_bact_flavobacteriales_genera <- aem_bact_flavobacteriales[,-c(1:6,8)]
    aem_bact_flavobacteriales_genera <- as.data.frame(aem_bact_flavobacteriales_genera)
    aem_bact_flavobacteriales_genera <- aem_bact_flavobacteriales_genera %>% group_by(Genus) %>% summarize_all(funs(sum))
    aem_bact_flavobacteriales_genera <- aem_bact_flavobacteriales_genera %>% drop_na()
    aem_bact_flavobacteriales_genera <- column_to_rownames(aem_bact_flavobacteriales_genera, "Genus")
    aem_bact_flavobacteriales_genera_t <- t(aem_bact_flavobacteriales_genera)
    aem_bact_flavobacteriales_genera_t <- as.data.frame(aem_bact_flavobacteriales_genera_t)
    aem_bact_flavobacteriales_genera_t <- aem_bact_flavobacteriales_genera_t %>% rownames_to_column("sample_id")
    aem_bact_flavobacteriales_genera_sums <- merge(metadata, aem_bact_flavobacteriales_genera_t, by = "sample_id")

    
#Extract Genera from Pseudomonadales within Proteobacteria (GROUPS)

    #Pseudomonadales
    aem_proteo_pseudomonadales_genera <- aem_proteo_pseudomonadales[,-c(1:6,8)]
    aem_proteo_pseudomonadales_genera <- as.data.frame(aem_proteo_pseudomonadales_genera)
    aem_proteo_pseudomonadales_genera <- aem_proteo_pseudomonadales_genera %>% group_by(Genus) %>% summarize_all(funs(sum))
    aem_proteo_pseudomonadales_genera <- aem_proteo_pseudomonadales_genera %>% drop_na()
    aem_proteo_pseudomonadales_genera <- column_to_rownames(aem_proteo_pseudomonadales_genera, "Genus")
    aem_proteo_pseudomonadales_genera_t <- t(aem_proteo_pseudomonadales_genera)
    aem_proteo_pseudomonadales_genera_t <- as.data.frame(aem_proteo_pseudomonadales_genera_t)
    aem_proteo_pseudomonadales_genera_t <- aem_proteo_pseudomonadales_genera_t %>% rownames_to_column("sample_id")
    aem_proteo_pseudomonadales_genera_sums <- merge(metadata, aem_proteo_pseudomonadales_genera_t, by = "sample_id")
    
  
#Separate genera into subsetted datasets
aem_firm_lactobacillales_genera_sums_forest_ncr <- aem_firm_lactobacillales_genera_sums %>% filter(Habitat == "Forest") %>% filter(Raider == "No")
aem_firm_bacillales_genera_sums_forest_ncr <- aem_firm_bacillales_genera_sums %>% filter(Habitat == "Forest") %>% filter(Raider == "No")

aem_bacillales_genera_by_species <- aem_firm_bacillales_genera_sums_forest_ncr[,-c(1:3,5,6,9:11)]
aem_lactobacillales_genera_by_species <- aem_firm_lactobacillales_genera_sums_forest_ncr[,-c(1:3,5,6,9:11)]


aem_bact_flavobacteriales_genera_sums_africana <- aem_bact_flavobacteriales_genera_sums %>% filter(Species == "africana")
aem_proteo_pseudomonadales_genera_sums_africana <- aem_proteo_pseudomonadales_genera_sums %>% filter(Species == "africana")

aem_flavobacteriales_genera_by_group <- aem_bact_flavobacteriales_genera_sums_africana[,-c(1:6,9,11)]
aem_pseudomonadales_genera_by_group <- aem_proteo_pseudomonadales_genera_sums_africana[,-c(1:6,9,11)]


colnames(aem_flavobacteriales_genera_by_group) <- c("Age", "Sex", "Group", "Chryseobacterium", "Elizabethkingia", "Flavobacterium", "Fluviicola", "Wautersiella")
colnames(aem_pseudomonadales_genera_by_group) <- c("Age", "Sex", "Group", "Acinetobacter", "Perlucidibaca", "Pseudomonas", "Serpens" )

#Calculate p-values for genera at group level and at species level

gwaut <- glmer.nb(Wautersiella ~ Group + (1|Sex), data = aem_flavobacteriales_genera_by_group)
gwaut_p <- Anova(gwaut)$`Pr(>Chisq)`

#Post-hoc Tukey_test
post_gwaut <- glht(gwaut, mcp(Group = "Tukey"))
summary(post_gwaut)

gacin <- glmer.nb(Acinetobacter ~ Group + (1|Sex), data = aem_pseudomonadales_genera_by_group)
gacin_p <- Anova(gacin)$`Pr(>Chisq)`

#Post-hoc Tukey_test
post_gacin <- glht(gacin, mcp(Group = "Tukey"))
summary(post_gacin)

gbacillus <- glmer.nb(g__Bacillus ~ Species + (1|Sex), data = aem_bacillales_genera_by_species)
gbacillus_p <- Anova(gbacillus)$`Pr(>Chisq)`

gsolibac <- glmer.nb(g__Solibacillus ~ Species + (1|Sex), data = aem_bacillales_genera_by_species)
gsolibac_p <- Anova(gsolibac)$`Pr(>Chisq)`

p_vals_bacillales_species <- as.data.frame(c(gbacillus_p, gsolibac_p))
colnames(p_vals_bacillales_species) <- c("pval")
p_vals_bacillales_species_fdr <- p.adjust(p_vals_bacillales_species$pval, method = "fdr")
```

## calculate p values for ALL GENERA
```{r}
#P-values

#Bacillales
p_vals_bacillales_genera_by_species <- data.frame(numeric(length = ncol(aem_bacillales_genera_by_species)))

    for (ii in 1:ncol(aem_bacillales_genera_by_species)) {
  
      col = aem_bacillales_genera_by_species[,ii] #this tells the for loop that it should iterate through columns in the data set
      lm_temp <- lm(col ~ Species, data = aem_bacillales_genera_by_species) #tells the for loop to run linear model  for each column           specified above
      p_value <- summary(lm_temp)$coefficients[2,4] #extracts p value
      p_vals_bacillales_genera_by_species[ii,] <- p_value
    }

#Lactobacillales
p_vals_lactobacillales_genera_by_species <- data.frame(numeric(length = ncol(aem_lactobacillales_genera_by_species)))

    for (ii in 1:ncol(aem_lactobacillales_genera_by_species)) {
  
      col = aem_lactobacillales_genera_by_species[,ii] #this tells the for loop that it should iterate through columns in the data set
      lm_temp <- lm(col ~ Species, data = aem_lactobacillales_genera_by_species) #tells the for loop to run linear model  for each            column specified above
      p_value <- summary(lm_temp)$coefficients[2,4] #extracts p value
      p_vals_lactobacillales_genera_by_species[ii,] <- p_value
    }

#Correct p-values for multiple tests using false discovery rate
p_vals_bacillales_genera_by_species <- p_vals_bacillales_genera_by_species %>% drop_na() %>% as.data.frame()
colnames(p_vals_bacillales_genera_by_species) <- c("pvals")

p_vals_bacillales_genera_by_species_fdr <- p.adjust(p_vals_bacillales_genera_by_species$pvals, method = "fdr")

p_vals_lactobacillales_genera_by_species <- p_vals_lactobacillales_genera_by_species %>% drop_na() %>% as.data.frame()
colnames(p_vals_lactobacillales_genera_by_species) <- c("pvals")
p_vals_lactobacillales_genera_by_species_fdr <- p.adjust(p_vals_lactobacillales_genera_by_species$pvals, method ="fdr")

```

## Make box plots for ALL GENERA
```{r}
#square root transform the y-axis
species_bacillales_genera_abun <- read_excel("species_bacillales_genera_abun.xlsx")
species_bacillales_genera_abun <- species_bacillales_genera_abun %>% mutate(log_abundance = log10(abundance), sqrt_abundance = sqrt(abundance))

species_lactobacillales_genera_abun <- read_excel("species_lactobacillales_genera_abun.xlsx")
species_lactobacillales_genera_abun <- species_lactobacillales_genera_abun %>% mutate(log_abundance = log10(abundance), sqrt_abundance = sqrt(abundance))


#Plots of genera within Firmicutes Orders by African Elephant Species
bacillales_plot <- ggplot(species_bacillales_genera_abun, aes(x = order, y = sqrt_abundance, fill = species)) + geom_boxplot(outlier.shape = NA, show.legend = T) + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + labs(title = "Genera of Order Bacillales", fill = "Elephant Species", x = "Genus", y = "Mean Abundance") + scale_fill_manual(values = c("red", "blue")) + theme(legend.text = element_text(face = "italic")) + theme(axis.text.x = element_text(face = "italic")) + theme(legend.position = c(0.05, 0.7))

lacto_plot <- ggplot(species_lactobacillales_genera_abun, aes(x = order, y = sqrt_abundance, fill = species)) + geom_boxplot(outlier.shape = NA, show.legend = F) + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + labs(title = "Genera of Order Lactobacillales", x = "Genus", y = "Mean Abundance") + scale_fill_manual(values = c("red", "blue")) + theme(axis.text.x = element_text(face = "italic"))


jpeg("/Users/joegunn/Desktop/Grad_School_Stuff/Projects/Elephant_Microbiome/NEW_STATS/AEMB_FINAL_STATS/African_Elephant_Microbiome/firmicutes_genera_byspecies.jpeg", width=3700, height=5000, units='px', res=600) 

plot_grid(bacillales_plot, lacto_plot, nrow = 2, labels = c("A", "B"))

dev.off()
```

## Make Box Plots for FILTERED GENERA (AVERAGE RELATIVE ABUNDANCE = 0.1 OR MORE)
```{r}
#BY SPECIES

#square root transform the y-axis
bacillales_genera_abun_rel0.1 <- read_excel("bacillales_genera_abun_rel0.1.xlsx")
bacillales_genera_abun_rel0.1 <- bacillales_genera_abun_rel0.1 %>% mutate(log_abundance = log10(abundance), sqrt_abundance = sqrt(abundance))

#Plots of genera within Firmicutes Orders by African Elephant Species
bacillales_plot_rel0.1 <- ggplot(bacillales_genera_abun_rel0.1, aes(x = Genus, y = sqrt_abundance, fill = Species)) + geom_boxplot(outlier.shape = NA, show.legend = T) + theme(axis.text = element_text(size = 15)) + theme(axis.text.x = element_text(size = 20)) + labs(title = "Genera of Phylum Firmicutes: Order Bacillales", fill = "Elephant Species", x = "Genus", y = "Mean Abundance") + theme(plot.title = element_text(size = 20, hjust = 0.5, face = "bold")) + scale_fill_manual(values = c("red", "blue")) + theme(legend.text = element_text(face = "italic")) + theme(axis.text.x = element_text(face = "italic")) + theme(legend.position = c(0.2, 0.7)) + theme(plot.background = element_blank())  + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),panel.background = element_blank(), axis.line = element_line(colour = "black")) + theme(legend.title = element_text(size = 20)) + theme(legend.text = element_text(size = 20)) + theme(axis.title = element_text(size = 20))


pdf("/Users/joegunn/Desktop/Grad_School_Stuff/Projects/Elephant_Microbiome/NEW_STATS/AEMB_FINAL_STATS/African_Elephant_Microbiome/firmicutes_genera_byspecies.pdf", width=12, height=10) 

bacillales_plot_rel0.1

dev.off()


#BY GROUPS


flavobacteriales_genera_abun_rel0.1 <- read_excel("flavobacteriales_genera_abun_rel0.1.xlsx")
flavobacteriales_genera_abun_rel0.1 <- flavobacteriales_genera_abun_rel0.1 %>% mutate(log_abundance = log10(abundance), sqrt_abundance = sqrt(abundance))

pseudomonadales_genera_abun_rel0.1 <- read_excel("pseudomonadales_genera_abun_rel0.1.xlsx")
pseudomonadales_genera_abun_rel0.1 <- pseudomonadales_genera_abun_rel0.1 %>% mutate(log_abundance = log10(abundance), sqrt_abundance = sqrt(abundance))

#Flavobacteriales Plot

flavobacteriales_plot_rel0.1 <- ggplot(flavobacteriales_genera_abun_rel0.1, aes(x = Genus, y = sqrt_abundance, fill = Group)) + geom_boxplot(outlier.shape = NA, show.legend = T) + theme(axis.text = element_text(size = 15)) + theme(axis.text.x = element_text(size = 20)) + labs(title = "Genus of Phylum Bacteroidetes: Order Flavobacteriales", fill = "Elephant Group", x = "Genus", y = "Mean Abundance") + theme(plot.title = element_text(size = 20, hjust = 0.5, face = "bold")) + scale_fill_manual(values = c("green", "purple", "red", "goldenrod")) + theme(legend.text = element_text(face = "italic")) + theme(axis.text.x = element_text(face = "italic")) + theme(legend.position = c(0.8, 0.8)) + theme(plot.background = element_blank())  + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),panel.background = element_blank(), axis.line = element_line(colour = "black")) + theme(legend.title = element_text(size = 20)) + theme(legend.text = element_text(size = 20)) + theme(axis.title = element_text(size = 20))

#Pseudomonadales Plot

pseudomonadales_plot_rel0.1 <- ggplot(pseudomonadales_genera_abun_rel0.1, aes(x = Genus, y = sqrt_abundance, fill = Group)) + geom_boxplot(outlier.shape = NA, show.legend = T) + theme(axis.text = element_text(size = 15)) + theme(axis.text.x = element_text(size = 20)) + labs(title = "Genus of Phylum Proteobacteria: Order Pseudomonadales", fill = "Elephant Group", x = "Genus", y = "Mean Abundance") + theme(plot.title = element_text(size = 20, hjust = 0.5, face = "bold")) + scale_fill_manual(values = c("green", "purple", "red", "goldenrod")) + theme(legend.text = element_text(face = "italic")) + theme(axis.text.x = element_text(face = "italic")) + theme(legend.position = c(0.8, 0.8)) + theme(plot.background = element_blank())  + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),panel.background = element_blank(), axis.line = element_line(colour = "black")) + theme(legend.title = element_text(size = 20)) + theme(legend.text = element_text(size = 20)) + theme(axis.title = element_text(size = 20))

#Combined Plot

pdf("/Users/joegunn/Desktop/Grad_School_Stuff/Projects/Elephant_Microbiome/NEW_STATS/AEMB_FINAL_STATS/African_Elephant_Microbiome/genera_combined_bygroup.pdf", width=10, height=15) 

plot_grid(flavobacteriales_plot_rel0.1, pseudomonadales_plot_rel0.1, nrow = 2, labels = c("A", "B"), label_size = 20)

dev.off()
```


