---
title: "elephants_ORDER_ABUNDANCE"
author: "Joe Gunn"
date: "8/12/2019"
output: html_document
---
# AFRICAN ELEPHANT MICROBIOME - ORDER ABUNDANCE

Purpose: Here we determine OTU abundance among African Elephant individuals across the entire dataset at the microbial ORDER level. We then assess differences in abundance among African Elephant species and "groups" using general linear models (with a negative binomial distribution on zero-inflated count data). Groups include all combinations of diet and habitat so that all combinations can be compared.

Data used:

      TABLE output from QIIME: all OTUs detected across African Elephant sampels, taxonomic level, and raw abundance for each individual <- 8,248 OTUs (after rarefaction) 

## Libraries 
```{r libraries, echo = FALSE}

library(readxl)
library(cowplot)
library(tidyverse)
library(qiime2R)
library(vegan)
library(devtools)
library(DescTools)
library(factoextra)
library(ade4)
library(lme4)
library(nlme)
library(ggrepel)
library(GGally)
library(lattice)
library(epiDisplay)
library(multcomp)
library(taRifx)
library(ape)
library(openxlsx)
library(multcomp)
library(phyloseq)
library(lmerTest)
library(pscl)
library(ggpubr)
library(car)
```

## Metadata
```{r Metadata, include = FALSE}
##Read in the metadata, which includes all labels and covariates associated with each individual in the data set. Metadata includes individual sample ID (index), barcode sequence, linker primer sequence, species (africana or cyclotis), diet (crop raider or non-crop raider), habitat (forest or savanna), age, sex, and group (a descriptor for each unique combination of species-diet-habitat)

metadata <- read_tsv("../data/metadata/METADATA.tsv") #metadata read in

metadata <- metadata %>% 
  as.data.frame() %>% 
  mutate(BarcodeSequence = factor(BarcodeSequence), LinkerPrimerSequence = factor(LinkerPrimerSequence), Species = factor(Species), Raider = factor(Raider), Habitat = factor(Habitat), Age = factor(Age), Sex = factor(Sex), Elephant = factor(Elephant), Group = factor(Group), Description = factor(Description)) #convert table to data frame and convert all characters, except for sample id, into factors

colnames(metadata)[colnames(metadata) == "Sample-id"] <- "sample_id" 
```

## Get Rarefied OTU abundance data 
```{r, echo = FALSE}
options(scipen = 999) #gets rid of scientific notatation, useful for reading p-values

#read in taxonomic designations and OTU abundance
aem_physeq <- qza_to_phyloseq(features = "../data/qiime_data/table.qza", taxonomy = "../data/qiime_data/taxonomy.qza",  tree = "../data/qiime_data/rooted-tree.qza", metadata = "../data/metadata/METADATA.tsv") 

#omit sample OB182 due to low read count. We don't want to rarefy to such a low number of reads across the whole dataset
aem_physeq <- subset_samples(aem_physeq, Elephant != "OB182") 

#Rarefy and clean the data set for use in downstream analysis
aem_physeq <- rarefy_even_depth(aem_physeq, rngsee = 5) #must set seed in order to maintain the same rarefied number of otus

    #Number of OTUS: 8248
    #Number of reads per sample: 11460 
      
#Extract data of all types (meta data, all taxonomy data)
aem_meta <- as.data.frame(sample_data(aem_physeq))
aem_tax <- as.data.frame(tax_table(aem_physeq))
aem_tax <- aem_tax %>% 
  rownames_to_column("otu")

aem_data <- as.data.frame(otu_table(aem_physeq))
aem_data <- aem_data %>% 
  rownames_to_column("otu")

aem_tax_data <- merge(aem_tax, aem_data, by = "otu")
```

## Get Most Abundant Phyla
```{r}
#Subset the data by the three most abundant phyla: Bacteroidetes, firmicutes, proteobacteria
aem_bacteroidetes <- aem_tax_data %>% 
  filter(Phylum == "p__Bacteroidetes")

aem_firmicutes <- aem_tax_data %>% 
  filter(Phylum == "p__Firmicutes")

aem_proteobacteria <- aem_tax_data %>% 
  filter(Phylum == "p__Proteobacteria")
```


## Phylum and Order Proportions per African Elephant Group
```{r, echo = FALSE}
#Proportions of each order contributing to each african elephant group
aem_order_phylum_data <- aem_tax_data[,-c(1:2,4,6:8)]
aem_order_phylum_data <- as.tibble(aem_order_phylum_data)
aem_order_phylum_data <- aem_order_phylum_data %>% 
  group_by(Phylum, Order) %>% 
  summarize_all(funs(sum))
```

## Core Microbiome - Order Level
```{r}
#Summarize Rarefied OTU abundance by ALL Orders to figure out the core microbiome 
aem_order_data <- aem_tax_data[,-c(1:4,6:8)]
aem_order_data <- as.data.frame(aem_order_data)
aem_order_data <- aem_order_data %>% 
  group_by(Order) %>% 
  summarize_all(funs(sum))

aem_order_data <- aem_order_data %>% 
  drop_na()
aem_order_data <- column_to_rownames(aem_order_data, "Order")
aem_order_data_t <- t(aem_order_data)
aem_order_data_t <- as.data.frame(aem_order_data_t)
aem_order_data_t <- aem_order_data_t %>% 
  rownames_to_column("sample_id")

aem_order <- merge(metadata, aem_order_data_t, by = "sample_id")

###IMPORTANT NOTE: At this point, aem_orders, which provides abundance across all orders for all samples, was copied and entered into excel in order to determine the core microbiome and any other shared orders among african elephant groups. Figrue 1 in the manuscript is produced from the data within this excel file.
```

## Order Abundance within Phylum Bacteroidetes, Firmicutes, and Proteobacteria
```{r}
#Summarize Rarefied OTU abundance by Orders WITHIN BACTERIOIDETES - get sums across orders
aem_bact_orders <- aem_bacteroidetes[,-c(1:4,6:8)]
aem_bact_orders <- as.data.frame(aem_bact_orders)
aem_bact_orders <- aem_bact_orders %>% 
  group_by(Order) %>% 
  summarize_all(funs(sum))

aem_bact_orders <- aem_bact_orders %>% 
  drop_na()
aem_bact_orders <- column_to_rownames(aem_bact_orders, "Order")
aem_bact_orders_t <- t(aem_bact_orders)
aem_bact_orders_t <- as.data.frame(aem_bact_orders_t)
aem_bact_orders_t <- aem_bact_orders_t %>% 
  rownames_to_column("sample_id")
aem_bact_orders_sums <- merge(metadata, aem_bact_orders_t, by = "sample_id")

#Summarize Rarefied OTU abundance by Orders WITHIN FIRMICUTES - get sums across orders
aem_firm_orders <- aem_firmicutes[,-c(1:4,6:8)]
aem_firm_orders <- as.data.frame(aem_firm_orders)
aem_firm_orders <- aem_firm_orders %>% 
  group_by(Order) %>% 
  summarize_all(funs(sum))

aem_firm_orders <- aem_firm_orders %>% 
  drop_na()
aem_firm_orders <- column_to_rownames(aem_firm_orders, "Order")
aem_firm_orders_t <- t(aem_firm_orders)
aem_firm_orders_t <- as.data.frame(aem_firm_orders_t)
aem_firm_orders_t <- aem_firm_orders_t %>% 
  rownames_to_column("sample_id")

aem_firm_orders_sums <- merge(metadata, aem_firm_orders_t, by = "sample_id")

#Summarize Rarefied OTU abundance by Orders WITHIN PROTEOBACTERIA - get sums across orders
aem_proteo_orders <- aem_proteobacteria[,-c(1:4,6:8)]
aem_proteo_orders <- as.data.frame(aem_proteo_orders)
aem_proteo_orders <- aem_proteo_orders %>% 
  group_by(Order) %>% 
  summarize_all(funs(sum))

aem_proteo_orders <- aem_proteo_orders %>% 
  drop_na()
aem_proteo_orders <- column_to_rownames(aem_proteo_orders, "Order")
aem_proteo_orders_t <- t(aem_proteo_orders)
aem_proteo_orders_t <- as.data.frame(aem_proteo_orders_t)
aem_proteo_orders_t <- aem_proteo_orders_t %>% 
  rownames_to_column("sample_id")

aem_proteo_orders_sums <- merge(metadata, aem_proteo_orders_t, by = "sample_id")
```

## Compile sums for most abundant phyla and subset the data for downstream analyses. One dataset for Africana (n = 35), one dataset for Forest non-crop-raiders (n = 19)
```{r}
#Africana - diet and habitat
aem_bact_orders_sums_africana <- aem_bact_orders_sums %>% 
  filter(Species == "africana")

aem_firm_orders_sums_africana <- aem_firm_orders_sums %>% 
  filter(Species == "africana")

aem_proteo_orders_sums_africana <- aem_proteo_orders_sums %>% 
  filter(Species == "africana")

#Forest, non-crop-raiders - species
aem_bact_orders_sums_forest_ncr <- aem_bact_orders_sums %>% 
  filter(Habitat == "Forest") %>% 
  filter(Raider == "No")

aem_firm_orders_sums_forest_ncr <- aem_firm_orders_sums %>% 
  filter(Habitat == "Forest") %>% 
  filter(Raider == "No")

aem_proteo_orders_sums_forest_ncr <- aem_proteo_orders_sums %>% 
  filter(Habitat == "Forest") %>% 
  filter(Raider == "No")


#Analyze OTU sum data by African elephant species
aem_bact_species <- aem_bact_orders_sums_forest_ncr[,-c(1:3,5:6,9:11)]
colnames(aem_bact_species) <- c("Species", "Age", "Sex", "Saprospirales", "Bacteroidales", "Cytophagales", "Flavobacteriales", "Sphingobacteriales")

aem_firm_species <- aem_firm_orders_sums_forest_ncr[,-c(1:3,5:6,9:11)]
colnames(aem_firm_species) <- c("Species", "Age", "Sex", "Bacillales", "Clostridiales", "Erysipelotrichales", "Lactobacillales")


aem_proteo_species <- aem_proteo_orders_sums_forest_ncr[,-c(1:3,5:6,9:11)]
colnames(aem_proteo_species) <- c("Species", "Age", "Sex", "Aeromonadales", "Alteromonadales", "Bdellovibrionales", "Burkholderiales", "Caulobacterales", "Desulfovibrionales", "Enterobacteriales", "gmd14h09", "Kiloniellales", "Neisseriales", "Oceanospirillales", "Pasteurellales", "phoshd29", "Pseudomonadales", "rf32", "Rhizobiales", "Rhodobacterales", "Rickettsiales", "Sphingomonadales", "Spirobacillales", "Tremblayales", "Xanthomonadales")

#Analyze OTU sum data by African elephant Groups
aem_bact_groups <- aem_bact_orders_sums_africana[,-c(1:6,9,11)]
colnames(aem_bact_groups) <- c("Age", "Sex", "Group" , "Saprospirales", "Bacteroidales", "Cytophagales", "Flavobacteriales", "Sphingobacteriales")

aem_firm_groups <- aem_firm_orders_sums_africana[,-c(1:6,9,11)]
colnames(aem_firm_groups) <- c("Age", "Sex", "Group", "Bacillales", "Clostridiales", "Erysipelotrichales", "Lactobacillales")

aem_proteo_groups <- aem_proteo_orders_sums_africana[,-c(1:6,9,11)]
colnames(aem_proteo_groups) <- c("Age", "Sex", "Group","Aeromonadales", "Alteromonadales", "Bdellovibrionales", "Burkholderiales", "Caulobacterales", "Desulfovibrionales", "Enterobacteriales", "gmd14h09", "Kiloniellales", "Neisseriales", "Oceanospirillales", "Pasteurellales", "phoshd29", "Pseudomonadales", "rf32", "Rhizobiales", "Rhodobacterales", "Rickettsiales", "Sphingomonadales", "Spirobacillales", "Tremblayales", "Xanthomonadales")
```

# ANALYZE OTU SUM DATA FOR STATISTICAL SIGNIFICANCE
## SPECIES 
### We only analyzed orders that had at least a relative abundance of 0.1
## Statistical Significance of Bacterial Orders By African Elephant Species
```{r}
aem_bact_species_rel0.1 <- aem_bact_species[,-c(4,6,8)]
aem_firm_species_rel0.1 <- aem_firm_species[,-c(6,7)]
aem_proteo_species_rel0.1 <- aem_proteo_species[,-c(4:9,11:16,18:25)]

#PHYLUM Bacteroidetes
p_vals_bact_species <- data.frame(numeric(length = 2))

    for (ii in 4:ncol(aem_bact_species_rel0.1)) {
  
      col = aem_bact_species_rel0.1[,ii] #this tells the for loop that it should iterate through columns in the data set
      
      ff <- as.formula(paste0(colnames(aem_bact_species_rel0.1)[ii]," ~ Species + (1|Sex)"))
      
      glmer_temp <- glmer.nb(ff,  aem_bact_species_rel0.1) #tells the for loop to run linear model  for each column           specified above
      p_value <- summary(glmer_temp)$coefficients[2,4] #extracts p value
      p_vals_bact_species[ii-3,] <- p_value
    }

#PHYLUM Firmicutes
p_vals_firm_species <- data.frame(numeric(length = 2))

    for (ii in 4:ncol(aem_firm_species_rel0.1)) {
  
      col = aem_firm_species_rel0.1[,ii] #this tells the for loop that it should iterate through columns in the data set
      ff <- as.formula(paste0(colnames(aem_firm_species_rel0.1)[ii]," ~ Species + (1|Sex)"))
      
      glmer_temp <- glmer.nb(ff, data = aem_firm_species_rel0.1) #tells the for loop to run linear model  for each column           specified above
      p_value <- summary(glmer_temp)$coefficients[2,4] #extracts p value
      p_vals_firm_species[ii-3,] <- p_value
    }

#PHYLUM Proteobacteria
aem_proteo_species <- aem_proteo_species[,-c(23)] #SPIROBACILLES REMOVED BECAUSE IT WAS ALL ZEROS
p_vals_proteo_species <- data.frame(numeric(length = 2))

    for (ii in 4:ncol(aem_proteo_species_rel0.1)) {
  
      col = aem_proteo_species_rel0.1[,ii] #this tells the for loop that it should iterate through columns in the data set
      ff <- as.formula(paste0(colnames(aem_proteo_species_rel0.1)[ii]," ~ Species + (1|Sex)"))
      
      glmer_temp <- glmer.nb(ff, data = aem_proteo_species_rel0.1) #tells the for loop to run linear model  for each column         specified above
      p_value <- summary(glmer_temp)$coefficients[2,4] #extracts p value
      p_vals_proteo_species[ii-3,] <- p_value
    }

#Convert dataframes with p_values for each order within each phylum into tables we can work with
    
#PHYLUM Bacteroidetes orders p-values
p_vals_bact_species <- p_vals_bact_species %>% drop_na()
colnames(p_vals_bact_species) <- c("pvals")
p_vals_bact_species <- as.data.frame(p_vals_bact_species)

#PHYLUM Firmicutes orders p-values
p_vals_firm_species <- p_vals_firm_species %>% drop_na()
colnames(p_vals_firm_species) <- c("pvals")
p_vals_firm_species <- as.data.frame(p_vals_firm_species)

#PHYLUM Proteobacteria orders p-values
p_vals_proteo_species <- p_vals_proteo_species %>% drop_na()
colnames(p_vals_proteo_species) <- c("pvals")
p_vals_proteo_species <- as.data.frame(p_vals_proteo_species)

    
#Correct p-values for multiple tests using false discovery rate
p_vals_bact_species_fdr <- p.adjust(p_vals_bact_species$pvals, method = "fdr")
p_vals_firm_species_fdr <- p.adjust(p_vals_firm_species$pvals, method = "fdr")
p_vals_proteo_species_fdr <- p.adjust(p_vals_proteo_species$pvals, method = "fdr")

p_vals_bact_species_fdr #Nothing is significant
p_vals_firm_species_fdr #Bacillales is significant
p_vals_proteo_species_fdr #nothing significant
```

## Make Box Plots for FILTERED ORDERS (AVERAGE RELATIVE ABUNDANCE = 0.1) by Species
```{r}
#Log-transform y axis so that all values will show up

#PHYLUM Bacteroidetes
species_bacteroidetes_order_abun_rel0.1 <- read_excel("../data/excel_data/abundance/species_bacteroidetes_order_abun_rel0.1.xlsx")
species_bacteroidetes_order_abun_rel0.1 <- species_bacteroidetes_order_abun_rel0.1 %>% 
  mutate(log_abundance = log10(abundance), sqrt_abundance = sqrt(abundance))

#PHYLUM Firmicutes
species_firmicutes_order_abun_rel0.1 <- read_excel("../data/excel_data/abundance/species_firmicutes_order_abun_rel0.1.xlsx")
species_firmicutes_order_abun_rel0.1 <- species_firmicutes_order_abun_rel0.1 %>% 
  mutate(log_abundance = log10(abundance), sqrt_abundance = sqrt(abundance))

#PHYLUM Proteobacteria
species_proteobacteria_order_abun_rel0.1 <- read_excel("../data/excel_data/abundance/species_proteobacteria_order_abun_rel0.1.xlsx")
species_proteobacteria_order_abun_rel0.1 <- species_proteobacteria_order_abun_rel0.1 %>% 
  mutate(log_abundance = log10(abundance), sqrt_abundance = sqrt(abundance))
 
#Make box-plots of Orders for African Elephant Species

#Bacteroidetes
species_bacteroidetes_box_plots_rel0.1 <- ggplot(species_bacteroidetes_order_abun_rel0.1, aes(x = order, y = sqrt_abundance, fill = species)) +
  geom_boxplot(show.legend = T, outlier.shape = NA, alpha = 0.8) + 
  geom_point(position = position_dodge(width = 0.75), show.legend = F, size = 2) +
  theme_set(theme_cowplot(12)) + 
  theme(axis.text = element_text(size = 20)) + 
  labs(fill = "Elephant Species", title = "Orders by Elephant Species", y = "Bacteroidetes Abundance") + 
  theme(plot.title = element_text(size = 20, hjust = 0.5, face = "bold")) + 
  scale_fill_manual(values = c("red", "blue")) + 
  scale_color_manual(values = c("black", "black")) +
  theme(axis.title.x = element_blank()) + 
  theme(legend.position = c(0.75,0.8)) + 
  theme(axis.title.y = element_text(size = 20)) + 
  theme(legend.text = element_text(size = 20, face = "italic")) + 
  theme(axis.text.x = element_text(face = "italic", size = 20)) + 
  theme(plot.background = element_blank()) + 
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), panel.background = element_blank(), axis.line = element_line(colour = "black")) +
  theme(legend.title = element_text(size = 20))

#Firmicutes
species_firmicutes_box_plots_rel0.1 <- ggplot(species_firmicutes_order_abun_rel0.1, aes(x = order, y = sqrt_abundance, fill = species)) + 
  geom_boxplot(show.legend = F, outlier.shape = NA, alpha = 0.8) + 
  geom_point(position = position_dodge(width = 0.75), show.legend = F, size = 2) +
  theme_set(theme_cowplot(12)) +  
  theme(axis.text = element_text(size = 20)) + 
  labs(y = "Firmicutes Abundance") + 
  scale_fill_manual(values = c("red", "blue")) +
  scale_color_manual(values = c("black", "black")) +
  theme(axis.title.x = element_blank()) + 
  theme(axis.text.x = element_text(face = "italic", size = 20)) + 
  theme(plot.background = element_blank()) + 
  theme(axis.title.y = element_text(size = 20)) + 
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), panel.background = element_blank(), axis.line = element_line(colour = "black")) +
  theme(legend.box.background = element_blank())

#Proteobacteria
species_proteobacteria_box_plots_rel0.1 <- ggplot(species_proteobacteria_order_abun_rel0.1, aes(x = order, y = sqrt_abundance, fill = Species)) +
  geom_boxplot(show.legend = F, outlier.shape = NA, alpha = 0.8) + 
  geom_point(position = position_dodge(width = 0.75), show.legend = F, size = 2) +
  theme_set(theme_cowplot(12)) + 
  theme(axis.text = element_text(size = 20)) +
  labs(y = "Proteobacteria Abundance") + 
  scale_fill_manual(values = c("red", "blue")) + 
  scale_color_manual(values = c("black", "black")) +
  theme(axis.title.x = element_blank()) + 
  theme(axis.text.x = element_text(face = "italic", size = 20)) + 
  theme(axis.title.y = element_text(size = 20)) + 
  theme(plot.background = element_blank()) + 
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),panel.background = element_blank(), axis.line = element_line(colour = "black")) +
  theme(legend.box.background = element_blank())
    
#Visualize Analysis of Phylum Abundance
pdf("/Users/joegunn/Desktop/Grad_School_Stuff/Research/Projects/Elephant_Microbiome/Attempt_2/visualization/order_abundance_figures/phylum_species_plots_rel0.1.pdf", width=10, height=15) 

plot_grid(species_bacteroidetes_box_plots_rel0.1, species_firmicutes_box_plots_rel0.1, species_proteobacteria_box_plots_rel0.1, nrow = 3, labels = c("A", "B", "C"), label_size = 20)

dev.off()

#For combining figures
phy_species_plot_rel0.1 <- plot_grid(species_bacteroidetes_box_plots_rel0.1, species_firmicutes_box_plots_rel0.1, species_proteobacteria_box_plots_rel0.1, nrow = 3, labels = c("A", "B", "C"), label_size = 20)
```

## GROUPS
### We only analyzed orders that had at least a relative abundance of 0.1
## Statistical Signifiance of Orders By African Elephant Groups
```{r}
aem_bact_groups_rel0.1 <- aem_bact_groups[,-c(4,6,8)]
aem_firm_groups_rel0.1 <- aem_firm_groups[,-c(6,7)]
aem_proteo_groups_rel0.1 <- aem_proteo_groups[,-c(4:9,11:16,18:25)]

#PHYLUM Bacteroidetes
gbact <- glmer.nb(Bacteroidales ~ Group + (1|Sex), data = aem_bact_groups_rel0.1)
gbact_p <- Anova(gbact)$`Pr(>Chisq)` #P = 0.1889
gflav <- glmer.nb(Flavobacteriales ~ Group + (1|Sex), data = aem_bact_groups_rel0.1, control=glmerControl(optimizer="bobyqa", tolPwrss=1e-3, optCtrl = list(maxfun = 100000)))
gflav_p <- Anova(gflav)$`Pr(>Chisq)` #P = 0.001998

p_vals_bact_groups <- as.data.frame(c(gbact_p, gflav_p))
colnames(p_vals_bact_groups) <- c("pval")
p_vals_bact_groups

#Post-hoc Tukey-test
post_flavo <- glht(gflav, mcp(Group = "Tukey"))
summary(post_flavo)

#PHYLUM Firmicutes
gbaci <- glmer.nb(Bacillales ~ Group + (1|Sex), data = aem_firm_groups_rel0.1)
gbaci_p <- Anova(gbaci)$`Pr(>Chisq)`
gclos <- glmer.nb(Clostridiales ~ Group + (1|Sex), data = aem_firm_groups_rel0.1)
gclos_p <- Anova(gclos)$`Pr(>Chisq)`

p_vals_firm_groups <- as.data.frame(c(gbaci_p, gclos_p))
colnames(p_vals_firm_groups) <- c("pval")

#Post-hoc Tukey-test
post_bacillales <- glht(gbaci, mcp(Group="Tukey"))
summary(post_bacillales)

#PHYLUM Proteobacteria
aem_proteo_groups_rel0.1
genter <- glmer.nb(Enterobacteriales ~ Group + (1|Sex), data = aem_proteo_groups_rel0.1)
genter_p <- Anova(genter)$`Pr(>Chisq)`
gpseudo <- glmer.nb(Pseudomonadales ~ Group + (1|Sex), data = aem_proteo_groups_rel0.1)
gpseudo_p <- Anova(gpseudo)$`Pr(>Chisq)`

p_vals_proteo_groups <- as.data.frame(c(genter_p, gpseudo_p))
colnames(p_vals_proteo_groups) <- c("pval")

#Post-hoc Tukey-Test
post_pseudo <- glht(gpseudo, mcp(Group = "Tukey"))
summary(post_pseudo)
coefficients(post_pseudo)

#PHYLUM Bacterioidetes
p_vals_bact_groups <- p_vals_bact_groups %>% 
  drop_na()
colnames(p_vals_bact_groups) <- c("pval")
p_vals_bact_groups <- as.data.frame(p_vals_bact_groups)

#PHYLUM Firmicutes
p_vals_firm_groups <- p_vals_firm_groups %>% 
  drop_na()
colnames(p_vals_firm_groups) <- c("pval")
p_vals_firm_groups <- as.data.frame(p_vals_firm_groups)

#PHYLUM Proteobacteria
p_vals_proteo_groups <- p_vals_proteo_groups %>%
  drop_na()
colnames(p_vals_proteo_groups) <- c("pval")
p_vals_proteo_groups <- as.data.frame(p_vals_proteo_groups)

#Correct p-values for multiple tests using false discovery rate
p_vals_bact_groups_fdr <- p.adjust(p_vals_bact_groups$pval, method = "fdr")
p_vals_firm_groups_fdr <- p.adjust(p_vals_firm_groups$pval, method = "fdr")
p_vals_proteo_groups_fdr <- p.adjust(p_vals_proteo_groups$pval, method = "fdr")

p_vals_bact_groups_fdr
p_vals_firm_groups_fdr
p_vals_proteo_groups_fdr
```

## Make Box Plots for FILTERED ORDERS (AVERAGE RELATIVE ABUNDANCE = 0.1 or more)
```{r}
#Log-transform y axis so that all values will show up
groups_bacteroidetes_order_abun_rel0.1 <- read_excel("../data/excel_data/abundance/groups_bacteroidetes_order_abun_rel0.1.xlsx")
groups_bacteroidetes_order_abun_rel0.1 <- groups_bacteroidetes_order_abun_rel0.1 %>% 
  mutate(log_abundance = log10(abundance), sqrt_abundance = sqrt(abundance))

groups_firmicutes_order_abun_rel0.1 <- read_excel("../data/excel_data/abundance/groups_firmicutes_order_abun_rel0.1.xlsx")
groups_firmicutes_order_abun_rel0.1 <- groups_firmicutes_order_abun_rel0.1 %>% 
  mutate(log_abundance = log10(abundance), sqrt_abundance = sqrt(abundance))

groups_proteobacteria_order_abun_rel0.1 <- read_excel("../data/excel_data/abundance/groups_proteobacteria_order_abun_rel0.1.xlsx")
groups_proteobacteria_order_abun_rel0.1  <- groups_proteobacteria_order_abun_rel0.1 %>% 
  mutate(log_abundance = log10(abundance), sqrt_abundance = sqrt(abundance))

#PHYLUM Bacterioidetes
bact_group_plots_rel0.1 <- ggplot(groups_bacteroidetes_order_abun_rel0.1, aes(x = order, y = sqrt_abundance, fill = group)) + 
  geom_boxplot(outlier.shape = NA, show.legend = T, alpha = 0.8) + 
  geom_point(position = position_dodge(width = 0.75), show.legend = F, size = 2) +
  theme_set(theme_cowplot(12)) +
  theme(plot.title = element_text(size = 20, hjust = 0.5, face = "bold")) + 
  theme(axis.text = element_text(size = 20)) + 
  labs(title = "Orders by Elephant Group", fill = "Elephant Group") + 
  scale_fill_manual(values = c("green", "purple", "red", "goldenrod")) + 
  theme(axis.title.x = element_blank()) + 
  theme(axis.title.y = element_blank()) + 
  theme(legend.position = c(0.7,0.8)) + 
  theme(legend.text = element_text(size = 20)) + 
  theme(axis.text.x = element_text(face = "italic", size = 20)) + 
  theme(plot.background = element_blank()) + 
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),panel.background = element_blank(), axis.line = element_line(colour = "black")) +
  theme(legend.title = element_text(size = 20))

#PHYLUM Firmicutes
firm_group_plots_rel0.1 <- ggplot(groups_firmicutes_order_abun_rel0.1, aes(x = order, y = sqrt_abundance, fill = group)) +
  geom_boxplot(outlier.shape = NA, show.legend = F, alpha = 0.8) + 
  geom_point(position = position_dodge(width = 0.75), show.legend = F, size = 2) +
  theme_set(theme_cowplot(12)) +
  theme(axis.text = element_text(size = 20)) + 
  scale_fill_manual(values = c("green", "purple", "red", "goldenrod")) + 
  theme(axis.title.x = element_blank()) + 
  theme(axis.title.y = element_blank()) + 
  theme(axis.text.x = element_text(face = "italic", size = 20)) + 
  theme(plot.background = element_blank()) + 
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), panel.background = element_blank(), axis.line = element_line(colour = "black")) +
  theme(legend.title = element_text(size = 20))

#PHYLUM Proteobacteria
proteo_group_plots_rel0.1 <- ggplot(groups_proteobacteria_order_abun_rel0.1, aes(x = order, y = sqrt_abundance, fill = group)) + 
  geom_boxplot(outlier.shape = NA, show.legend = F, alpha = 0.8) + 
  geom_point(position = position_dodge(width = 0.75), show.legend = F, size = 2) +
  theme_set(theme_cowplot(12)) +
  theme(axis.text = element_text(size = 20)) + 
  scale_fill_manual(values = c("green", "purple", "red", "goldenrod")) + 
  theme(axis.title.x = element_blank()) + 
  theme(axis.title.y = element_blank()) + 
  theme(axis.text.x = element_text(face = "italic", size = 20)) + 
  theme(plot.background = element_blank()) + 
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),panel.background = element_blank(), axis.line = element_line(colour = "black")) +
  theme(legend.title = element_text(size = 20))

#Combine Phylum Group Plots
pdf("/Users/joegunn/Desktop/Grad_School_Stuff/Research/Projects/Elephant_Microbiome/Attempt_2/visualization/order_abundance_figures/phylum_group_plots_rel0.1.pdf", width=10, height=15) 

plot_grid(bact_group_plots_rel0.1, firm_group_plots_rel0.1, proteo_group_plots_rel0.1, nrow = 3, labels = c("A", "B", "C"))

dev.off()

#For combining figures
phy_group_plot_rel0.1 <- plot_grid(bact_group_plots_rel0.1, firm_group_plots_rel0.1, proteo_group_plots_rel0.1, nrow = 3, labels = c("D", "E", "F"), label_size = 20)

##WE did not look for genera within Saprospirales, because it is in such low abundance (only found in n = 8 samples)

###COMBINE ALL PHYLUM PLOTS
pdf("/Users/joegunn/Desktop/Grad_School_Stuff/Research/Projects/Elephant_Microbiome/Attempt_2/visualization/order_abundance_figures/phy_combined_plots_rel0.1.pdf", width=20, height=20) 

plot_grid(phy_species_plot_rel0.1, phy_group_plot_rel0.1, ncol = 2)

dev.off()
```
