---
title: "elephants_TAX_CONFIDENCE"
author: "Joe Gunn"
date: "8/12/2019"
output: html_document
---

# AFRICAN ELEPHANT MICROBIOME - TAXONOMIC CLASSIFICATION OF OTUS

Purpose: Here I am assessing the confidence with which all individual microbial OTUs were assigned to a given taxonomic level (DOMAIN, KINGDOM, PHYLUM, CLASS, ORDER, FAMILY, GENUS, SPECIES). We used this analysis to determine the most specific (lowest) taxonomic level with high confidence that would be appropriate to use to compare taxonomic composition between African Elephant species, diets, and habitats.

Data used:

      TAXONOMY table output from QIIME: all OTUs detected across African Elephant sampels and confidence with which they were assigned to each taxonomic level <- 9,066 OTUs

## Libraries Needed for Analysis
```{r libraries, echo = FALSE}

library(readxl)
library(cowplot)
library(tidyverse)
library(qiime2R)
library(vegan)
library(devtools)
library(DescTools)
library(factoextra)
library(ade4)
library(lme4)
library(nlme)
library(ggrepel)
library(GGally)
library(lattice)
library(epiDisplay)
library(multcomp)
library(taRifx)
library(ape)
library(openxlsx)
library(multcomp)
library(phyloseq)
```

## Metadata
```{r Metadata, include = FALSE}
##Read in the metadata, which includes all labels and covariates associated with each individual in the data set. Metadata includes individual sample ID (index), barcode sequence, linker primer sequence, species (africana or cyclotis), diet (crop raider or non-crop raider), habitat (forest or savanna), age, sex, and group (a descriptor for each unique combination of species-diet-habitat)
metadata <- read_tsv("../data/metadata/METADATA.tsv") #metadata read in

metadata <- metadata %>% 
  as.data.frame() %>% 
  mutate(BarcodeSequence = factor(BarcodeSequence), LinkerPrimerSequence = factor(LinkerPrimerSequence), Species = factor(Species), Raider = factor(Raider), Habitat = factor(Habitat), Age = factor(Age), Sex = factor(Sex), Elephant = factor(Elephant), Group = factor(Group), Description = factor(Description)) #convert table to data frame and convert all characters, except for sample id, into factors

colnames(metadata)[colnames(metadata) == "Sample-id"] <- "sample_id" 
```

## Taxonomy Table (OTUs present in the total data set) and Confidence Calculations
```{r All OTUs - no abundance data}

##Read in Taxonomy Table. This table contains all OTUs present in the total data set, but does not provide the relative abundance of each taxon per sample. It only contains the taxonomic level of each OTU present and the confidence at which that taxonomic unit was classified.
tax_all <- read_qza("../data/qiime_data/taxonomy.qza") #read in table

tax_all<- tax_all$data %>% 
  as.tibble() %>% 
  separate(Taxon, sep = "; ", c("Kingdom", "Phylum", "Class", "Order", "Family", "Genus", "Species")) #clean data and label columns by taxonomic level
tax_all_df <- as.data.frame(tax_all)
  colnames(tax_all_df) <- c("otu_names", "Kingdom", "Phylum", "Class", "Order", "Family", "Genus", "Species", "conf")

##Calculate mean and standard deviation of confidence
mean_conf <- mean(tax_all$Confidence) #mean confidence across all OTUs (mean = 0.9615)
sd_conf <- sd(tax_all$Confidence) #standard deviation of confidence across all OTUs (sd = 0.07256)

#change all unclassified taxa that aren't already called "NA" to "NA"
tax_all$Species[tax_all$Species == "s__"] <- NA
tax_all$Genus[tax_all$Genus == "g__"] <- NA
tax_all$Family[tax_all$Family == "f__"] <- NA
tax_all$Order[tax_all$Order == "o__"] <- NA
tax_all$Class[tax_all$Class == "c__"] <- NA
tax_all$Phylum[tax_all$Phylum == "p__"] <- NA
tax_all$Kingdom[tax_all$Kingdom == "k__"] <- NA

phy_order <- tax_all[,-c(1:2,4,6:9 )]

tax_all <- as.data.frame(tax_all) #change table to data frame 

#Create separate data frames for each taxon and confidence 

kingdom <- as.data.frame(cbind(tax_all$Kingdom, tax_all$Confidence)) #kingdom
  colnames(kingdom) <- c("Kingdom", "Confidence") #add column names

phylum <- as.data.frame(cbind(tax_all$Phylum, tax_all$Confidence)) #phylum
  colnames(phylum) <- c("Phylum", "Confidence")
  
class <- as.data.frame(cbind(tax_all$Class, tax_all$Confidence)) #class
  colnames(class) <- c("Class", "Confidence")

order <- as.data.frame(cbind(tax_all$Order, tax_all$Confidence)) #order
  colnames(order) <- c("Order", "Confidence")

family <- as.data.frame(cbind(tax_all$Family, tax_all$Confidence)) #family
  colnames(family) <- c("Family", "Confidence")

genus <- as.data.frame(cbind(tax_all$Genus, tax_all$Confidence)) #genus
  colnames(genus) <- c("Genus", "Confidence")

species <- as.data.frame(cbind(tax_all$Species, tax_all$Confidence)) #species
  colnames(species) <- c("Species", "Confidence")

##Create data frames for ONLY the individual OTUs that were successfully classified at each level. Kingdom had many individual OTUs successfully classified, while Species had substantially fewer OTUs successfully classified
  
known_species <- as.data.frame(species[complete.cases(species),]) #species
known_genus <-  as.data.frame(genus[complete.cases(genus),]) #genus
known_family <- as.data.frame(family[complete.cases(family),]) #family
known_order <- as.data.frame(order[complete.cases(order),]) #order 
known_class <- as.data.frame(class[complete.cases(class),]) #class 
known_phylum <- as.data.frame(phylum[complete.cases(phylum),]) #phylum
known_kingdom <- as.data.frame(kingdom[complete.cases(kingdom),]) #kingdom

##Raw number of unique OTUs per taxonomic level (classified successfully)
#nrow(known_species) #271
#nrow(known_genus) #2279
#nrow(known_family) #6138
#nrow(known_order) #8706
#nrow(known_class) #8840
#nrow(known_phylum) #8878
#nrow(known_kingdom) #9066

otu_prop_vector <- c(271/9066, 2279/9066, 6138/9066, 8706/9066, 8840/9066, 8878/9066, 9066/9066)
otu_prop_vector <- as.data.frame(otu_prop_vector)

##mean confidence at each taxonomic level

known_species <- known_species %>% mutate(Confidence = as.character(Confidence)) #need to change the confidence variable from a factor to a numeric. NOTE: you MUST first convert from factor to character, then from character to factor. Not sure why, but otherwise it doesn't work right

  known_species <- known_species %>% mutate(Confidence = as.numeric(Confidence))
  known_genus <- known_genus %>% mutate(Confidence = as.character(Confidence))
  known_genus <-  known_genus %>% mutate(Confidence = as.numeric(Confidence))
  known_family <- known_family %>% mutate(Confidence = as.character(Confidence))
  known_family <-  known_family %>% mutate(Confidence = as.numeric(Confidence))
  known_order <- known_order %>% mutate(Confidence = as.character(Confidence))
  known_order <-  known_order %>% mutate(Confidence = as.numeric(Confidence))
  known_class <- known_class %>% mutate(Confidence = as.character(Confidence))
  known_class <-  known_class %>% mutate(Confidence = as.numeric(Confidence))
  known_phylum <- known_phylum %>% mutate(Confidence = as.character(Confidence))
  known_phylum <-  known_phylum %>% mutate(Confidence = as.numeric(Confidence))
  known_kingdom <- known_kingdom%>% mutate(Confidence = as.character(Confidence))
  known_kingdom <-  known_kingdom %>% mutate(Confidence = as.numeric(Confidence))
```

## Calculate mean confidence of assigning an individual to a given taxon
```{r}
##Species Confidence
ms <- mean(known_species$Confidence) #0.9107
ss <- sd(known_species$Confidence) #0.0976

##Genus Confidence
mg <- mean(known_genus$Confidence) #0.9577
sg <- sd(known_genus$Confidence) #0.0772

##Family Confidence
mf <- mean(known_family$Confidence) #0.9605
sf <- sd(known_family$Confidence) #0.0734

##Order Confidence
mo <- mean(known_order$Confidence) #0.9631
so <- sd(known_order$Confidence) #0.0708

##Class confidence
mc <- mean(known_class$Confidence) #0.9623
sc <- sd(known_class$Confidence) #0.0715

##Phylum confidence
mp <- mean(known_phylum$Confidence) #0.9619
sp <- sd(known_phylum$Confidence) #0.0721

##Kingdom confidence
mk <- mean(known_kingdom$Confidence) #0.9615
sk <- sd(known_kingdom$Confidence) #0.0725
```

## Read in new Excel table with confidence intervals and means
```{r}
conf_by_tax <- read_excel("../data/excel_data/tax_confidence/confidence.xlsx") #made a new dataframe using mean and sd data calculated above. Reading in this data frame and cleaning the data below

conf_by_tax <- conf_by_tax %>% 
  mutate(taxon = factor(taxon))

conf_by_tax$taxon <- factor(conf_by_tax$taxon, levels = c("kingdom", "phylum", "class", "order", "family", "genus", "species")) #reorder levels - otherwise it will automatically list aplphabetically

conf_by_tax <- cbind(conf_by_tax, otu_prop_vector)


#Visualize ##Plot relative confidence of each taxonomic group classificaiton
pdf("/Users/joegunn/Desktop/Grad_School_Stuff/Research/Projects/Elephant_Microbiome/Attempt_2/visualization/tax_confidence_figures/tax_confidence.pdf", width=7, height=6)

ggplot(conf_by_tax, aes(x = taxon, y = mean)) + 
  geom_point(size = 2) +
  geom_errorbar(aes(ymin = mean - sd, ymax = mean + sd), width=.3) + 
  theme_set(theme_cowplot(12)) +
  scale_y_continuous(name = "Mean Confidence of Classification", sec.axis = sec_axis(~. /0.97, name = "Proportion of Classified OTUs")) +
  geom_line(mapping = aes(x = conf_by_tax$taxon, y = conf_by_tax$otu_prop_vector, group = 1), size = 2, color = "blue", alpha = 0.6) +
  theme(axis.text.x = element_text(angle = 60, hjust = 1)) + 
  labs(x = "Taxonomic Group")

dev.off()
```

