---
title: "African Elephant Microbiome Stats"
author: "Joe Gunn"
date: "2/21/2019"
output: html_document
---

###Libraries Needed for Analysis

```{r libraries, ECHO = FALSE}

library(readxl)
library(cowplot)
library(tidyverse)
library(qiime2R)
library(vegan)
library(devtools)
library(DescTools)
library(factoextra)
library(ade4)
library(lme4)
library(nlme)
library(ggrepel)
library(GGally)
library(lattice)
library(epiDisplay)
library(multcomp)
library(taRifx)
library(ape)
library(openxlsx)
library(multcomp)
library(phyloseq)
library(lmerTest)
```

###Metadata
```{r Metadata}
##Read in the metadata, which includes all labels and covariates associated with each individual in the data set. Metadata includes individual sample ID (index), barcode sequence, linker primer sequence, species (africana or cyclotis), diet (crop raider or non-crop raider), habitat (forest or savanna), age, sex, and group (a descriptor for each unique combination of species-diet-habitat)

metadata <- read_tsv("METADATA.tsv") #metadata read in

metadata <- metadata %>% as.data.frame() %>% mutate(BarcodeSequence = factor(BarcodeSequence), LinkerPrimerSequence = factor(LinkerPrimerSequence), Species = factor(Species), Raider = factor(Raider), Habitat = factor(Habitat), Age = factor(Age), Sex = factor(Sex), Elephant = factor(Elephant), Group = factor(Group), Description = factor(Description)) #convert table to data frame and convert all characters, except for sample id, into factors
colnames(metadata)[colnames(metadata) == "Sample-id"] <- "sample_id" 
```

#Rarefied OTU abundance data 
```{r}
options(scipen = 999) #gets rid of scientific notatation, useful for reading p-values

#read in taxonomic designations and OTU abundance
aem_physeq <- qza_to_phyloseq(features = "table.qza", taxonomy = "taxonomy.qza",  tree = "rooted-tree.qza", metadata = "METADATA.tsv") 

#omit sample OB182 due to low read count. We don't want to rarefy to such a low number of reads across the whole dataset
aem_physeq <- subset_samples(aem_physeq, Elephant != "OB182") 

#Rarefy and clean the data set for use in downstream analysis
aem_physeq <- rarefy_even_depth(aem_physeq, rngsee = 5) #must set seed in order to maintain the same rarefied number of otus

    #Number of OTUS: 8248
    #Number of reads per sample: 11460 
      
#Extract data of all types (meta data, all taxonomy data)
aem_meta <- as.data.frame(sample_data(aem_physeq))
aem_tax <- as.data.frame(tax_table(aem_physeq))
aem_tax <- aem_tax %>% rownames_to_column("otu")
aem_data <- as.data.frame(otu_table(aem_physeq))
aem_data <- aem_data %>% rownames_to_column("otu")
aem_tax_data <- merge(aem_tax, aem_data, by = "otu")

```
```{r}
#Subset the data by the three most abundant phyla: Bacteroidetes, firmicutes, proteobacteria
aem_bacteroidetes <- aem_tax_data %>% filter(Phylum == "p__Bacteroidetes")
aem_firmicutes <- aem_tax_data %>% filter(Phylum == "p__Firmicutes")
aem_proteobacteria <- aem_tax_data %>% filter(Phylum == "p__Proteobacteria")

```


#Statistical Signifiance of Genera within Orders Bacillales and Lactobacillales By African Elephant Species
```{r}
#Now we want to find the genera within each of the significant orders (Bacillales and Lactobacillales within Firmicutes) and see which are significantly different between africana and cyclotis species
    
#Data frames with genera only within the significant orders for african elephant SPECIES
aem_firm_bacillales <- aem_firmicutes %>% filter(Order == "o__Bacillales")
aem_firm_lactobacillales <- aem_firmicutes %>% filter(Order == "o__Lactobacillales")

#Extract Genera from Bacillales and Lactobacillales within Firmicutes

    #Bacillales
    aem_firm_bacillales_genera <- aem_firm_bacillales[,-c(1:6,8)]
    aem_firm_bacillales_genera <- as.data.frame(aem_firm_bacillales_genera)
    aem_firm_bacillales_genera <- aem_firm_bacillales_genera %>% group_by(Genus) %>%                summarize_all(funs(sum))
    aem_firm_bacillales_genera <- aem_firm_bacillales_genera %>% drop_na()
    aem_firm_bacillales_genera <- column_to_rownames(aem_firm_bacillales_genera, "Genus")
    aem_firm_bacillales_genera_t <- t(aem_firm_bacillales_genera)
    aem_firm_bacillales_genera_t <- as.data.frame(aem_firm_bacillales_genera_t)
    aem_firm_bacillales_genera_t <- aem_firm_bacillales_genera_t %>% rownames_to_column("sample_id")
    aem_firm_bacillales_genera_sums <- merge(metadata, aem_firm_bacillales_genera_t, by = "sample_id")
    
    #Lactobacillales
    aem_firm_lactobacillales_genera <- aem_firm_lactobacillales[,-c(1:6,8)]
    aem_firm_lactobacillales_genera <- as.data.frame(aem_firm_lactobacillales_genera)
    aem_firm_lactobacillales_genera <- aem_firm_lactobacillales_genera %>% group_by(Genus) %>% summarize_all(funs(sum))
    aem_firm_lactobacillales_genera <- aem_firm_lactobacillales_genera %>% drop_na()
    aem_firm_lactobacillales_genera <- column_to_rownames(aem_firm_lactobacillales_genera, "Genus")
    aem_firm_lactobacillales_genera_t <- t(aem_firm_lactobacillales_genera)
    aem_firm_lactobacillales_genera_t <- as.data.frame(aem_firm_lactobacillales_genera_t)
    aem_firm_lactobacillales_genera_t <- aem_firm_lactobacillales_genera_t %>% rownames_to_column("sample_id")
    aem_firm_lactobacillales_genera_sums <- merge(metadata, aem_firm_lactobacillales_genera_t, by = "sample_id")


#Separate genera into subsetted datasets
aem_firm_lactobacillales_genera_sums_forest_ncr <- aem_firm_lactobacillales_genera_sums %>% filter(Habitat == "Forest") %>% filter(Raider == "No")
aem_firm_bacillales_genera_sums_forest_ncr <- aem_firm_bacillales_genera_sums %>% filter(Habitat == "Forest") %>% filter(Raider == "No")

aem_bacillales_genera_by_species <- aem_firm_bacillales_genera_sums_forest_ncr[,-c(1:3,5:11)]
aem_lactobacillales_genera_by_species <- aem_firm_lactobacillales_genera_sums_forest_ncr[,-c(1:3,5:11)]


#P-values

#Bacillales
p_vals_bacillales_genera_by_species <- data.frame(numeric(length = ncol(aem_bacillales_genera_by_species)))

    for (ii in 1:ncol(aem_bacillales_genera_by_species)) {
  
      col = aem_bacillales_genera_by_species[,ii] #this tells the for loop that it should iterate through columns in the data set
      lm_temp <- lm(col ~ Species, data = aem_bacillales_genera_by_species) #tells the for loop to run linear model  for each column           specified above
      p_value <- summary(lm_temp)$coefficients[2,4] #extracts p value
      p_vals_bacillales_genera_by_species[ii,] <- p_value
    }

#Lactobacillales
p_vals_lactobacillales_genera_by_species <- data.frame(numeric(length = ncol(aem_lactobacillales_genera_by_species)))

    for (ii in 1:ncol(aem_lactobacillales_genera_by_species)) {
  
      col = aem_lactobacillales_genera_by_species[,ii] #this tells the for loop that it should iterate through columns in the data set
      lm_temp <- lm(col ~ Species, data = aem_lactobacillales_genera_by_species) #tells the for loop to run linear model  for each            column specified above
      p_value <- summary(lm_temp)$coefficients[2,4] #extracts p value
      p_vals_lactobacillales_genera_by_species[ii,] <- p_value
    }

#Correct p-values for multiple tests using false discovery rate
p_vals_bacillales_genera_by_species <- p_vals_bacillales_genera_by_species %>% drop_na() %>% as.data.frame()
colnames(p_vals_bacillales_genera_by_species) <- c("pvals")
p_vals_bacillales_genera_by_species
p_vals_bacillales_genera_by_species_fdr <- p.adjust(p_vals_bacillales_genera_by_species$pvals, method = "fdr")

p_vals_lactobacillales_genera_by_species <- p_vals_lactobacillales_genera_by_species %>% drop_na() %>% as.data.frame()
colnames(p_vals_lactobacillales_genera_by_species) <- c("pvals")
p_vals_lactobacillales_genera_by_species_fdr <- p.adjust(p_vals_lactobacillales_genera_by_species$pvals, method ="fdr")


########################################################################################################################################
#MAKE BOXPLOTS TO SHOW SIGNIFICANCE


#square root transform the y-axis
species_bacillales_genera_abun <- read_excel("species_bacillales_genera_abun.xlsx")
species_bacillales_genera_abun <- species_bacillales_genera_abun %>% mutate(log_abundance = log10(abundance), sqrt_abundance = sqrt(abundance))

species_lactobacillales_genera_abun <- read_excel("species_lactobacillales_genera_abun.xlsx")
species_lactobacillales_genera_abun <- species_lactobacillales_genera_abun %>% mutate(log_abundance = log10(abundance), sqrt_abundance = sqrt(abundance))



#Plots of genera within Firmicutes Orders by African Elephant Species
bacillales_plot <- ggplot(species_bacillales_genera_abun, aes(x = order, y = sqrt_abundance, fill = species)) + geom_boxplot(outlier.shape = NA, show.legend = T) + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + labs(title = "Genera of Order Bacillales", fill = "Elephant Species", x = "Genus", y = "Mean Abundance") + scale_fill_manual(values = c("red", "blue")) + theme(legend.text = element_text(face = "italic")) + theme(axis.text.x = element_text(face = "italic")) + theme(legend.position = c(0.05, 0.7))

lacto_plot <- ggplot(species_lactobacillales_genera_abun, aes(x = order, y = sqrt_abundance, fill = species)) + geom_boxplot(outlier.shape = NA, show.legend = F) + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + labs(title = "Genera of Order Lactobacillales", x = "Genus", y = "Mean Abundance") + scale_fill_manual(values = c("red", "blue")) + theme(axis.text.x = element_text(face = "italic"))


jpeg("/Users/joegunn/Desktop/Grad_School_Stuff/Projects/Elephant_Microbiome/NEW_STATS/AEMB_FINAL_STATS/African_Elephant_Microbiome/firmicutes_genera_byspecies.jpeg", width=3700, height=5000, units='px', res=600) 

plot_grid(bacillales_plot, lacto_plot, nrow = 2, labels = c("A", "B"))

dev.off()




```

#Statistical Signifiance of Orders By African Elephant Groups
```{r}
###GROUPS

#Bacteroidetes
p_vals_bact_groups <- data.frame(numeric(length = ncol(aem_bact_groups)))

    for (ii in 1:ncol(aem_bact_groups)) {
  
      col = aem_bact_groups[,ii] #this tells the for loop that it should iterate through columns in the data set
      lm_temp <- lm(col ~ Group, data = aem_bact_groups) #tells the for loop to run linear model  for each column            specified above
      fstat <- summary(lm_temp)$fstatistic #extracts f statistics
      p_value <- pf(fstat[1], fstat[2], fstat[3], lower.tail = F)
      p_vals_bact_groups[ii,] <- p_value
    }

#Firmicutes
p_vals_firm_groups <- data.frame(numeric(length = ncol(aem_firm_groups)))

    for (ii in 1:ncol(aem_firm_groups)) {
  
      col = aem_firm_groups[,ii] #this tells the for loop that it should iterate through columns in the data set
      lm_temp <- lm(col ~ Group, data = aem_firm_groups) #tells the for loop to run linear model  for each column            specified        above
      fstat <- summary(lm_temp)$fstatistic
      p_value <- pf(fstat[1], fstat[2], fstat[3], lower.tail = F) #extracts p value
      p_vals_firm_groups[ii,] <- p_value
    }

#Proteobacteria
p_vals_proteo_groups <- data.frame(numeric(length = ncol(aem_proteo_groups)))

    for (ii in 1:ncol(aem_proteo_groups)) {
  
      col = aem_proteo_groups[,ii] #this tells the for loop that it should iterate through columns in the data set
      lm_temp <- lm(col ~ Group, data = aem_proteo_groups) #tells the for loop to run linear model  for each column          specified above
      fstat <- summary(lm_temp)$fstatistic
      p_value <- pf(fstat[1], fstat[2], fstat[3], lower.tail = F) #extracts p value
      p_vals_proteo_groups[ii,] <- p_value
    }

#Convert dataframes with p_values into tables we can work with

#Bacterioidetes
p_vals_bact_groups <- p_vals_bact_groups %>% drop_na()
colnames(p_vals_bact_groups) <- c("pval")
p_vals_bact_groups <- as.data.frame(p_vals_bact_groups)

#Firmicutes
p_vals_firm_groups <- p_vals_firm_groups %>% drop_na()
colnames(p_vals_firm_groups) <- c("pval")
p_vals_firm_groups <- as.data.frame(p_vals_firm_groups)

#Proteobacteria
p_vals_proteo_groups <- p_vals_proteo_groups %>% drop_na()
colnames(p_vals_proteo_groups) <- c("pval")
p_vals_proteo_groups <- as.data.frame(p_vals_proteo_groups)

#Correct p-values for multiple tests using false discovery rate
p_vals_bact_groups_fdr <- p.adjust(p_vals_bact_groups$pval, method = "fdr")
p_vals_firm_groups_fdr <- p.adjust(p_vals_firm_groups$pval, method = "fdr")
p_vals_proteo_groups_fdr <- p.adjust(p_vals_proteo_groups$pval, method = "fdr")


########################################################################################################################################
#MAKE BOXPLOTS TO SHOW SIGNIFICANCE

#Log-transform y axis so that all values will show up
groups_bacteroidetes_order_abun <- read_excel("groups_bacteroidetes_order_abun.xlsx")
groups_bacteroidetes_order_abun <- groups_bacteroidetes_order_abun %>% mutate(log_abundance = log10(abundance), sqrt_abundance = sqrt(abundance))

groups_firmicutes_order_abun <- read_excel("groups_firmicutes_order_abun.xlsx")
groups_firmicutes_order_abun <- groups_firmicutes_order_abun %>% mutate(log_abundance = log10(abundance), sqrt_abundance = sqrt(abundance))

groups_proteobacteria_order_abun <- read_excel("groups_proteobacteria_order_abun.xlsx")
groups_proteobacteria_order_abun  <- groups_proteobacteria_order_abun %>% mutate(log_abundance = log10(abundance), sqrt_abundance = sqrt(abundance))


#Bacterioidetes
bact_group_plots <- ggplot(groups_bacteroidetes_order_abun, aes(x = order, y = sqrt_abundance, fill = group)) + geom_boxplot(outlier.shape = NA, show.legend = T) + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + labs(title = "Orders by Elephant Group", fill = "Elephant Group") + scale_fill_manual(values = c("green", "purple", "red", "goldenrod")) + theme(axis.title.x = element_blank()) + theme(axis.title.y = element_blank()) + theme(legend.position = c(0.65,0.7)) + theme(axis.text.x = element_text(face = "italic")) 


#Firmicutes
firm_group_plots <- ggplot(groups_firmicutes_order_abun, aes(x = order, y = sqrt_abundance, fill = group)) + geom_boxplot(outlier.shape = NA, show.legend = F) + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + scale_fill_manual(values = c("green", "purple", "red", "goldenrod")) + theme(axis.title.x = element_blank()) + theme(axis.title.y = element_blank()) + theme(axis.text.x = element_text(face = "italic"))

#Proteobacteria
proteo_group_plots <- ggplot(groups_proteobacteria_order_abun, aes(x = order, y = sqrt_abundance, fill = group)) + geom_boxplot(outlier.shape = NA, show.legend = F) + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + scale_fill_manual(values = c("green", "purple", "red", "goldenrod")) + theme(axis.title.x = element_blank()) + theme(axis.title.y = element_blank()) + theme(axis.text.x = element_text(face = "italic"))


jpeg("/Users/joegunn/Desktop/Grad_School_Stuff/Projects/Elephant_Microbiome/NEW_STATS/AEMB_FINAL_STATS/African_Elephant_Microbiome/phylum_group_plots.jpeg", width=2000, height=3000, units='px', res=250) 

plot_grid(bact_group_plots, firm_group_plots, proteo_group_plots, nrow = 3, labels = c("A", "B", "C"))

dev.off()


#For combining figures

phy_group_plot <- plot_grid(bact_group_plots, firm_group_plots, proteo_group_plots, nrow = 3, labels = c("D", "E", "F"))


##WE did not look for genera within Saprospirales, because it is in such low abundance (only found in n = 8 samples)



###COMBINE PHYLUM PLOTS

jpeg("/Users/joegunn/Desktop/Grad_School_Stuff/Projects/Elephant_Microbiome/NEW_STATS/AEMB_FINAL_STATS/African_Elephant_Microbiome/phy_combined_plots.jpeg", width=5500, height=6000, units='px', res=500) 

plot_grid(phy_species_plot, phy_group_plot, ncol = 2)

dev.off()



```


# BETA DIVERSITY #####################################


#Prepare and clean data for two datasets
```{r}
#Prepare data at the OTU level for analysis with both datasets (just forest non-cropraiders from both species, and just africana)
aem_physeq <- as.data.frame(otu_table(aem_physeq))
aem_physeq_t <- t(aem_physeq)
aem_physeq <- as.data.frame(aem_physeq_t)
aem_physeq <- aem_physeq %>% rownames_to_column("sample_id")
aem_physeq <- merge(metadata, aem_physeq, by = "sample_id")
aem_physeq_forest_ncr <- aem_physeq %>% filter(Habitat == "Forest") %>% filter(Raider == "No")
aem_physeq_africana <- aem_physeq %>% filter(Species == "africana")
```


#Run PERMANOVA analysis for African elephant Species, Diet, and Habitat
```{r}
###Prepare data for PERMANOVA FOR SPECIES
aem_veg_data <- aem_physeq_forest_ncr[,-c(2:11)]
aem_veg_data <- column_to_rownames(aem_veg_data, "sample_id")
###Analysis##################################
aem_veg <- decostand(aem_veg_data, "total") #standardize abundance data by dividing each number by the total number of samples (individuals in the dataset)
aem_veg <- vegdist(aem_veg, "bray") #calculate Bray-curtis distances between samples
aem_pcoa <- pcoa(aem_veg) #calculates principal components
#############################################

#Prepare PCOA (unused in manuscript)
aem_pcs <- aem_pcoa$vectors
aem_pcs <- as.data.frame(aem_pcs)
aem_pcs <- aem_pcs %>% rownames_to_column("sample_id")
aem_pcs <- merge(metadata, aem_pcs, by = "sample_id")
aem_physeq_ado1 <- column_to_rownames(aem_physeq_forest_ncr, "sample_id")

#Beta Dispersion Analysis for African Elephant species
species_beta <- factor(aem_physeq_ado1[,3])
beta_analysis <- betadisper(aem_veg, species_beta)
beta_vectors <- beta_analysis$vectors
beta_vectors <- as.data.frame(beta_vectors)
beta_vectors <- beta_vectors %>% rownames_to_column("sample_id")
beta_vectors <- merge(metadata, beta_vectors, by = "sample_id")
beta_graph <- ggplot(beta_vectors, aes(x = PCoA1, y = PCoA2, color = Species)) + geom_point()

#####################################################################################

###Prepare data for PERMANOVA FOR DIET
aem_veg2_data <- aem_physeq_africana[,-c(2:11)]
aem_veg2_data <- column_to_rownames(aem_veg2_data, "sample_id")

##Analysis###################################
aem_veg2 <- decostand(aem_veg2_data, "total")
aem_veg2 <- vegdist(aem_veg2, "bray")
aem_pcoa2 <- pcoa(aem_veg2)
#############################################

#Prepare PCOA
aem_pcs2 <- aem_pcoa2$vectors
aem_pcs2 <- as.data.frame(aem_pcs2)
aem_pcs2 <- aem_pcs2 %>% rownames_to_column("sample_id")
aem_pcs2 <- merge(metadata, aem_pcs2, by = "sample_id")

aem_physeq_ado2 <- column_to_rownames(aem_physeq_africana, "sample_id")

#Beta Dispersion Analysis for Diet and Habitat
diet_beta <- factor(aem_physeq_ado2[,4])
habitat_beta <- factor(aem_physeq_ado2[,5])
beta_analysis2 <- betadisper(aem_veg2, diet_beta)
plot(beta_analysis2, axes = c(1,2), cex = 0.7, col = NULL, lty = "solid", lwd = 1, hull = TRUE)
beta_analysis3 <- betadisper(aem_veg2, habitat_beta)
plot(beta_analysis3, axes = c(1,2), cex = 0.7, col = NULL, lty = "solid", lwd = 1, hull = TRUE)



#Run Adonis code: Permanova. We ran this code for many different linear combinations and for all variables.
ado1 <- adonis(aem_veg ~ Species, aem_physeq_ado1, distance = "bray", permutations = 9999) #Significant
ado2 <- adonis(aem_veg2 ~ Raider, aem_physeq_ado2, distance = "bray", permutations = 9999)
#When we analyze only by crop raider, permanova is significant; when we analyze with habitat as a stratum, it is still significant.
ado3 <- adonis(aem_veg2 ~ Habitat, aem_physeq_ado2, distance = "bray", permutations = 9999) #VERY significant if just modeled by habitat alone. Also very significant if raider is a stratum
ado4 <- adonis(aem_veg2 ~ Age, aem_physeq_ado2, distance = "bray", permutations = 9999) #NOT SIGNIFICANT
ado5 <- adonis(aem_veg2 ~ Sex, aem_physeq_ado2, distance = "bray", permutations = 9999) #NOT SIGNIFICANT
ado6 <- adonis(aem_veg2 ~ Raider*Habitat, aem_physeq_ado2, distance = "bray", permutations = 9999)
ado7 <- adonis(aem_veg2 ~ Raider*Sex, aem_physeq_ado2, distance = "bray", permutations = 9999)
ado8 <- adonis(aem_veg2 ~ Raider*Age, aem_physeq_ado2, distance = "bray", permutations = 9999)
ado9 <- adonis(aem_veg2 ~ Habitat*Sex, aem_physeq_ado2, distance = "bray", permutations = 9999)
ado10 <- adonis(aem_veg2 ~ Habitat*Age, aem_physeq_ado2, distance = "bray", permutations = 9999)
ado11 <- adonis(aem_veg2 ~ Raider, aem_physeq_ado2, Strata = Habitat, distance = "bray", permutations = 9999)
ado12 <- adonis(aem_veg2 ~ Habitat, aem_physeq_ado2, Strata = Raider, distance = "bray", permutations = 9999)
ado13 <- adonis(aem_veg2 ~ Raider, aem_physeq_ado2, Strata = Sex, distance = "bray", permutations = 9999)
ado14 <- adonis(aem_veg2 ~ Raider, aem_physeq_ado2, Strata = Age, distance = "bray", permutations = 9999)
ado15 <- adonis(aem_veg2 ~ Habitat, aem_physeq_ado2, Strata = Sex, distance = "bray", permutations = 9999)
ado16 <- adonis(aem_veg2 ~ Habitat, aem_physeq_ado2, Strata = Age, distance = "bray", permutations = 9999)

ado17 <- adonis(aem_veg ~ Sex, aem_physeq_ado1, distance = "bray", permutations = 9999)
ado18 <- adonis(aem_veg ~ Age, aem_physeq_ado1, distance = "bray", permutations = 9999)
ado19 <- adonis(aem_veg ~ Species, aem_physeq_ado1, Strata = Age, distance = "bray", permutations = 9999)

ado20 <- adonis(aem_veg ~ Age, aem_physeq_ado1, Strata = Species, distance = "bray", permutations = 9999)
ado21 <- adonis(aem_veg ~ Species*Age, aem_physeq_ado1, distance = "bray", permutations = 9999)

#one-way PERMANOVA models for L.africana and L.cyclotis (only random effects of age and sex, no interaction)
##################################################################################################################

ado_species_age <- adonis(aem_veg ~ Species, aem_physeq_ado1, Strata = Age, distance = "bray", permutations = 9999)

ado_species_sex <- adonis(aem_veg ~ Species, aem_physeq_ado1, Strata = Sex, distance = "bray", permutations = 9999)

###################################################################################################################

#two-way interaction models for L. africana, including random effects of age and sex
ado_diet_hab_age <- adonis(aem_veg2 ~ Raider + Habitat, aem_physeq_ado2, Strata = Age, distance = "bray", permutations = 9999)
    #diet = 0.007
    #habitat = 0.0001
    #interaction = 0.095

ado_diet_hab_age

ado_diet_hab_sex <- adonis(aem_veg2 ~ Raider + Habitat, aem_physeq_ado2, Strata = Sex, distance = "bray", permutations = 9999)
    #diet = 0.007
    #habitat = 0.0001
    #interaction = 0.099

ado_diet_hab_sex


#PERMANOVA INDIVIDUAL RESULTS

#ado1 #Species
#ado2 #Diet
#ado3 #Habitat
#ado4 #Age within L. africana
#ado5 #Sex within L. africana
#ado6 #Diet by habitat interaction within L.africana
#ado7 #Diet by Sex interaction within L. africana
#ado8 #Diet by Age interaction within L. africana
#ado9 #Habitat by Sex interaction within L. africana
#ado10 #Habitat by Age interaction within L. africana
#ado11 #Diet with Habitat as a random effect within L. africana
#ado12 #Habitat with Diet as a random effect within L. africana
#ado13 #Diet with Sex as a random effect within L. africana
#ado14 #Diet with Age as a random effect within L. africana
#ado15 #Habitat with Sex as a random effect within L. africana
#ado16 #Habitat with Age as a random effect within L. africana

#ado17 #Sex between Species
#ado18 #Age between Species
#ado19 #Species with Age as a random effect within species
#ado20 #Age with Species as a random effect within species
#ado21 #Species by Age interaction


#Put individual PERMANOVA results into a single table
ado_p_vals <- data.frame(numeric(16))
ado_p_vals[1,1] <- ado1$aov.tab[1,6]
ado_p_vals[2,1] <- ado2$aov.tab[1,6]
ado_p_vals[3,1] <- ado3$aov.tab[1,6]
ado_p_vals[4,1] <- ado4$aov.tab[1,6]
ado_p_vals[5,1] <- ado5$aov.tab[1,6]
ado_p_vals[6,1] <- ado6$aov.tab[1,6]
ado_p_vals[7,1] <- ado7$aov.tab[1,6]
ado_p_vals[8,1] <- ado8$aov.tab[1,6]
ado_p_vals[9,1] <- ado9$aov.tab[1,6]
ado_p_vals[10,1] <- ado10$aov.tab[1,6]
ado_p_vals[11,1] <- ado11$aov.tab[1,6]
ado_p_vals[12,1] <- ado12$aov.tab[1,6]
ado_p_vals[13,1] <- ado13$aov.tab[1,6]
ado_p_vals[14,1] <- ado14$aov.tab[1,6]
ado_p_vals[15,1] <- ado15$aov.tab[1,6]
ado_p_vals[16,1] <- ado16$aov.tab[1,6]
ado_p_vals <- as.data.frame(ado_p_vals)
ado_p_vals <- ado_p_vals %>% rownames_to_column("model")
colnames(ado_p_vals) <- c("model", "p_value")

```

### NMDS for Abundance data

```{r}

###Visualize Nonlinear Multi-Dimensional Scaling (NMDS)

###PLOT NMDS BY SPECIES

#####Run NMDS######################################################
vare.mds0 <- isoMDS(aem_veg, trace = 0)
vare.mds <- metaMDS(aem_veg_data, trace = FALSE, autotransform = F)
score <- scores(vare.mds)
score <- as.data.frame(score)
score <- score %>% rownames_to_column("sample_id")
merge <- merge(metadata, score, by = "sample_id")
##################################################################

levels(merge$Species) <- c("L. africana", "L. cyclotis")
centroids <- aggregate(cbind(NMDS1,NMDS2)~Species,merge,mean)
gg <- merge(merge,aggregate(cbind(mean.x=NMDS1,mean.y=NMDS2)~Species,merge,mean),by="Species")

species_perm <- ggplot(gg, aes(NMDS1, NMDS2, color = Species)) + geom_point(size=1) + geom_point(aes(x=mean.x,y=mean.y),size=1) + geom_segment(aes(x=mean.x, y=mean.y, xend=NMDS1, yend=NMDS2)) + theme(legend.position=c(0.01,0.85)) + scale_color_manual(values = c("red2", "blue2")) + labs(color = "Elephant Species", title = "Beta Diversity: OTU Abundance", y = "NMDS 2") + theme(axis.title.x = element_blank()) + theme(legend.text = element_text(face = "italic"))



#PLOT NMDS BY DIET

#####Run NMDS######################################################
vare.mds1 <- isoMDS(aem_veg2, trace = 0)
vare.mds2 <- metaMDS(aem_veg2_data, trace = FALSE, autotransform = F)
score_diet <- scores(vare.mds2)
score_diet <- as.data.frame(score_diet)
score_diet <- score_diet %>% rownames_to_column("sample_id")
merge_diet <- merge(metadata, score_diet, by = "sample_id")
##################################################################

#Plot
levels(merge_diet$Raider) <- c("Non-Crop-Raider", "Crop-Raider")
centroids_diet <- aggregate(cbind(NMDS1,NMDS2)~Raider,merge_diet,mean)

gg_diet <- merge(merge_diet,aggregate(cbind(mean.x=NMDS1,mean.y=NMDS2)~Raider,merge_diet,mean),by="Raider")

diet_perm <- ggplot(gg_diet, aes(NMDS1, NMDS2, color = Raider)) + geom_point(size=1)+
  geom_point(aes(x=mean.x,y=mean.y),size=1)+
  geom_segment(aes(x=mean.x, y=mean.y, xend=NMDS1, yend=NMDS2)) + theme(legend.position = c(0.01,0.85)) + scale_color_manual(values = c("midnightblue", "deeppink1")) + theme(axis.title.x = element_blank()) + labs(y = "NMDS 2", color = "Elephant Diet")


#######################################################################################################
#PLOT NMDS BY HABITAT

#####Run NMDS######################################################
vare.mds_hab <- isoMDS(aem_veg2, trace = 0)
vare.mds_hab2 <- metaMDS(aem_veg2_data, trace = FALSE, autotransform = F)
score_hab <- scores(vare.mds_hab2)
score_hab <- as.data.frame(score_hab)
score_hab <- score_hab %>% rownames_to_column("sample_id")
merge_hab <- merge(metadata, score_hab, by = "sample_id")
##################################################################

#Plot
centroids_hab <- aggregate(cbind(NMDS1,NMDS2)~Habitat,merge_hab,mean)

gg_hab <- merge(merge_hab,aggregate(cbind(mean.x=NMDS1,mean.y=NMDS2)~Habitat,merge_hab,mean),by="Habitat")

habitat_perm <- ggplot(gg_hab, aes(NMDS1, NMDS2, color = Habitat)) + geom_point(size=1)+
  geom_point(aes(x=mean.x,y=mean.y),size=1)+
  geom_segment(aes(x=mean.x, y=mean.y, xend=NMDS1, yend=NMDS2)) + theme(legend.position = c(0.01,0.85)) + scale_color_manual(values = c("forestgreen", "coral1")) + labs(color = "Elephant Habitat", x = "NMDS 1", y = "NMDS 2")


#######################################################################################
#PLOT EVERYTHING TOGETHER

jpeg("/Users/joegunn/Desktop/Grad_School_Stuff/Projects/Elephant_Microbiome/NEW_STATS/AEMB_FINAL_STATS/African_Elephant_Microbiome/beta_diversity.jpeg", width=1800, height=3700, units='px', res=400) 

plot_grid(species_perm, diet_perm, habitat_perm, nrow = 3, labels = c("A", "B", "C"))

dev.off()



plot_grid(species_perm, diet_perm, habitat_perm, nrow = 3, labels = c("A", "B", "C"))



```

###Alpha Diversity - Shannon Diversity Index
```{r}
###SPECIES 

#####Analysis##########################################
shan_div_species <- diversity(aem_veg_data, "shannon")
#######################################################

#Prepare data for linear models
shan_div_species <- as.data.frame(shan_div_species)
shan_div_species <- shan_div_species %>% rownames_to_column("sample_id")
colnames(shan_div_species) <- c("sample_id", "shannon")
shan_div_species <- merge(metadata, shan_div_species, by = "sample_id")

#Averages and standard deviations of Shannon Diversity for species, diet, and habitat
shan_div_diet_aves <- shan_div_diet %>% group_by(Raider) %>% summarize(mean = mean(shannon))
shan_div_diet_sd <- shan_div_diet %>% group_by(Raider) %>% summarize(sd = sd(shannon))
shan_div_habitat_aves <- shan_div_diet %>% group_by(Habitat) %>% summarize(mean = mean(shannon))
shan_div_habitat_sd <- shan_div_diet %>% group_by(Habitat) %>% summarize(sd = sd(shannon))
shan_div_species_aves <- shan_div_species %>% group_by(Species) %>% summarize(mean = mean(shannon))
shan_div_species_sd <- shan_div_species %>% group_by(Species) %>% summarize(sd = sd(shannon))

#Linear Models of alpha diversity with Species data set
lm1 <- lm(shannon ~ Species, data = shan_div_species) #linear model by species alone
lm2 <- lm(shannon ~ Species*Sex, data = shan_div_species) #linear model with sex and species interaction (two-way ANOVA)
lm3 <- lm(shannon ~ Species*Age, data = shan_div_species) #linear model with age and sex interaction


lm_sex_species <- lm(shannon ~ Sex, data = shan_div_species)  #p = 0.821
lm_age_species <- lm(shannon ~ Age, data = shan_div_species)  #p = 0.588



###DIET


#####Analyis#########################################
shan_div_diet <- diversity(aem_veg2_data, "shannon")
#######################################################

#Prepare data for linear models
shan_div_diet <- as.data.frame(shan_div_diet)
shan_div_diet <- shan_div_diet %>% rownames_to_column("sample_id")
colnames(shan_div_diet) <- c("sample_id", "shannon")
shan_div_diet <- merge(metadata, shan_div_diet, by = "sample_id")

#Linear models of alpha diversity with diet and habitat data
lmA <- lm(shannon ~ Raider, data = shan_div_diet) #NOT SIGNIFICANT 
lmB <- lm(shannon ~ Habitat, data = shan_div_diet) #NOT SIGNIFICANT
lmC <- lm(shannon ~ Raider*Habitat, data = shan_div_diet)


#Various linear mixed effect models with different random effects just for fun
lme_species_age <- lme(shannon ~ Species, random = ~1|Age, data = shan_div_species)
lme_species_sex <- lme(shannon ~ Species, random = ~1|Sex, data = shan_div_species)

lme_int_diet_hab_age <- lme(shannon ~ Raider*Habitat, random = ~1|Age, data = shan_div_diet)
lme_int_diet_hab_sex <- lme(shannon ~ Raider*Habitat, random = ~1|Sex, data = shan_div_diet)
  
  
lme1 <- lme(shannon ~ Raider, random = ~1|Habitat, data = shan_div_diet) #$tTable
lme2 <- lme(shannon ~ Habitat, random = ~1|Raider, data = shan_div_diet)
lme3 <- lme(shannon ~ Sex, random = ~1|Raider, data = shan_div_diet)
lme4 <- lme(shannon ~ Sex, random = ~1|Habitat, data = shan_div_diet)


lm_sex <- lm(shannon ~ Sex, data = shan_div_diet)
lm_age <- lm(shannon ~ Age, data = shan_div_diet)


#Visualize Alpha Diversity. We can add a lot of graphs here, and we can compare all kinds of groups within groups.

levels(shan_div_species$Species) <- c("L. africana", "L cyclotis")

alpha_species <- ggplot(shan_div_species, aes(x = Species, y = shannon, fill = Species)) + geom_boxplot(outlier.shape = NA, show.legend = F) + labs(x = "Species", y = "Shannon Diversity") + scale_fill_manual(values = c("red", "blue")) + theme(axis.text.x = element_text(face = "italic"))

levels(shan_div_diet$Raider) <- c("Non-Crop-Raider", "Crop-Raider")

alpha_diet <- ggplot(shan_div_diet, aes(x = Raider, y = shannon, fill = Raider)) + geom_boxplot(outlier.shape = NA, show.legend = F) + labs(x = "Diet", y = "Shannon Diversity") + scale_fill_manual(values = c("midnightblue", "deeppink1"))

alpha_habitat <- ggplot(shan_div_diet, aes(x = Habitat, y = shannon, fill = Habitat)) + geom_boxplot(outlier.shape = NA, show.legend = F) + labs(x = "Habitat", y = "Shannon Diversity") + scale_fill_manual(values = c("forestgreen", "coral1"))
#Print the Figure

jpeg("/Users/joegunn/Desktop/Grad_School_Stuff/Projects/Elephant_Microbiome/NEW_STATS/AEMB_FINAL_STATS/African_Elephant_Microbiome/alpha_diversity.jpeg", width=1800, height=3700, units='px', res=400) 

plot_grid(alpha_species, alpha_diet, alpha_habitat, nrow = 3, labels = c("A", "B", "C"))

dev.off()


#Include Sex in Habitat and Diet Alpha diversity graphs, becasue sex is significant in some of the variables. This will be a supplementary figure.

#Make the Figures

alpha_diet_sex <- ggplot(shan_div_diet, aes(x = Raider, y = shannon, color = Sex)) + geom_boxplot(outlier.shape = NA) + labs(x = "Diet", y = "Shannon Diversity") + scale_color_manual(values = c("magenta", "mediumseagreen"))

alpha_habitat_sex <- ggplot(shan_div_diet, aes(x = Habitat, y = shannon, color = Sex)) + geom_boxplot(outlier.shape = NA) + labs(x = "Habitat", y = "Shannon Diversity") + scale_color_manual(values = c("magenta", "mediumseagreen"))

#Print the Figures

jpeg("/Users/joegunn/Desktop/Grad_School_Stuff/Projects/Elephant_Microbiome/NEW_STATS/AEMB_FINAL_STATS/African_Elephant_Microbiome/alpha_by_sex.jpeg", width=3700, height=3700, units='px', res=500) 

plot_grid(alpha_diet_sex, alpha_habitat_sex, nrow = 2, labels = c("A", "B"))

dev.off()

```


###KEGG PATHWAYS
```{r}

#Metadata by species (kegg_1_species) and by habitat and range (kegg_1_africana) for KEGG Pathways at level 1 metabolism 
kegg_1_all <- read_excel("kegg_1.xlsx")



kegg_1_meta <- kegg_1_all[,-1]
kegg_1_meta <- kegg_1_meta %>% group_by(Metabolism_1) %>% summarize_all(funs(sum))
kegg_1_meta <- as.data.frame(kegg_1_meta)
kegg_1_meta <- column_to_rownames(kegg_1_meta, "Metabolism_1")
kegg_1_meta <- t(kegg_1_meta)
kegg_1_meta <- as.data.frame(kegg_1_meta)
kegg_1_meta <- rownames_to_column(kegg_1_meta, "sample_id")
kegg_1_meta <- merge(metadata, kegg_1_meta, by = "sample_id")
kegg_1_species <- kegg_1_meta %>% filter(Habitat == "Forest") %>% filter(Raider == "No")
kegg_1_africana <- kegg_1_meta %>% filter(Species == "africana")
kegg_1_species <- column_to_rownames(kegg_1_species, "sample_id")

#Preparing Vegan matrix for both datasets
kegg_1 <- kegg_1_all[,-1] 
kegg_1 <- kegg_1 %>% group_by(Metabolism_1) %>% summarize_all(funs(sum))
kegg_1 <- as.data.frame(kegg_1)
kegg_1 <- column_to_rownames(kegg_1, "Metabolism_1")
kegg_1 <- t(kegg_1)
kegg_1 <- as.data.frame(kegg_1)
kegg_1 <- rownames_to_column(kegg_1, "sample_id")
kegg_1 <- merge(metadata, kegg_1, by = "sample_id")
kegg_1_by_species <- kegg_1 %>% filter(Habitat == "Forest") %>% filter(Raider == "No")
kegg_1_by_species <- kegg_1_by_species[,-c(2:11)]
kegg_1_by_species <- column_to_rownames(kegg_1_by_species, "sample_id")
kegg_1_by_africana <- kegg_1 %>% filter(Species == "africana")
kegg_1_by_africana <- kegg_1_by_africana[,-c(2:11)]
kegg_1_by_africana <- column_to_rownames(kegg_1_by_africana, "sample_id")

#####BETA DIVERSITY########################################################

#Run Vegan
kegg_1_veg_species_stand <- decostand(kegg_1_by_species, "total")
kegg_1_veg_species <- vegdist(kegg_1_veg_species_stand, "bray")

kegg_1_veg_africana_stand <- decostand(kegg_1_by_africana, "total")
kegg_1_veg_africana <- vegdist(kegg_1_veg_africana_stand, "bray")

kegg_1_species
#Beta Diversity with PERMANOVA
ado_kegg_species_age <- adonis(kegg_1_veg_species ~ Species, kegg_1_species, Strata = Age, distance = "bray", permutations = 9999)
ado_kegg_species_sex <- adonis(kegg_1_veg_species ~ Species, kegg_1_species, Strata = Sex, distance = "bray", permutations = 9999)
ado_kegg_species_age
ado_kegg_diet_hab_age <- adonis(kegg_1_veg_africana ~ Raider*Habitat, kegg_1_africana, Strata = Age, distance = "bray", permutations = 9999)
ado_kegg_diet_hab_sex <- adonis(kegg_1_veg_africana ~ Raider*Habitat, kegg_1_africana, Strata = Sex, distance = "bray", permutations = 9999)

############################################################################

```


```{r}
kegg_1_props_species <- read_excel("kegg_1_props_species.xlsx")
kegg_1_props_species <- as.data.frame(kegg_1_props_species)
kegg_1_props_species <- kegg_1_props_species %>% mutate(Species = factor(Species), Raider = factor(Raider), Habitat = factor(Habitat), Age = factor(Age), Sex = factor(Sex), Elephant = factor(Elephant), Group = factor(Group), Description = factor(Description))

kegg_1_props_africana <- read_excel("kegg_1_props_africana.xlsx")
kegg_1_props_africana <- kegg_1_props_africana %>% as.data.frame() %>% mutate(Species = factor(Species), Raider = factor(Raider), Habitat = factor(Habitat), Age = factor(Age), Sex = factor(Sex), Elephant = factor(Elephant), Group = factor(Group), Description = factor(Description))

kegg_1_props_africana
#By Species
p_vals_species <- data.frame("A" = numeric(1),  "B" = numeric(1),  "C" = numeric(1), 
                              "D" = numeric(1),  "E" = numeric(1),  "F" = numeric(1),
                              "G" = numeric(1),  "H" = numeric(1),  "I" = numeric(1), 
                              "J" = numeric(1),  "K" = numeric(1))

#iterate by column index of the data frame

options(scipen = 999) #this removes scientific notation from p-values, makes them easier to read

    for (ii in 12:ncol(kegg_1_props_species)) {
  
      col = kegg_1_props_species[,ii] #this tells the for loop that it should iterate through       columns in the data set
  
      lm_temp <- lm(col ~ Species, data = kegg_1_props_species) #tells the for loop to run         linear model  for each column specified above
  
      p_value <- summary(lm_temp)$coefficients[2,4] #extracts p value
  
      p_vals_species[,ii-11] <- p_value #stores p value in data frame constructed above. Need to       subtract 11 from the iterator so that the loop            doesn't fill starting at row 12
    }


#By habitat
p_vals_habitat <- data.frame("A" = numeric(1),  "B" = numeric(1),  "C" = numeric(1), 
                              "D" = numeric(1),  "E" = numeric(1),  "F" = numeric(1),
                              "G" = numeric(1),  "H" = numeric(1),  "I" = numeric(1), 
                              "J" = numeric(1),  "K" = numeric(1))

#iterate by column index of the data frame

options(scipen = 999) #this removes scientific notation from p-values, makes them easier to read

    for (ii in 12:ncol(kegg_1_props_africana)) {
  
      col = kegg_1_props_africana[,ii] #this tells the for loop that it should iterate through       columns in the data set
  
      lm_temp <- lm(col ~ Habitat, data = kegg_1_props_africana) #tells the for loop to run         linear model  for each column specified above
  
      p_value <- summary(lm_temp)$coefficients[2,4] #extracts p value
  
      p_vals_habitat[,ii-11] <- p_value #stores p value in data frame constructed above. Need to       subtract 11 from the iterator so that the loop            doesn't fill starting at row 12
    }

p_vals_habitat_fdr <- p.adjust(p_vals_habitat, method = "fdr")


#By Diet
p_vals_diet <- data.frame("A" = numeric(1),  "B" = numeric(1),  "C" = numeric(1), 
                              "D" = numeric(1),  "E" = numeric(1),  "F" = numeric(1),
                              "G" = numeric(1),  "H" = numeric(1),  "I" = numeric(1), 
                              "J" = numeric(1),  "K" = numeric(1))

options(scipen = 999) #this removes scientific notation from p-values, makes them easier to read

    for (ii in 12:ncol(kegg_1_props_africana)) {
  
      col = kegg_1_props_africana[,ii] #this tells the for loop that it should iterate through       columns in the data set
  
      lm_temp <- lm(col ~ Raider, data = kegg_1_props_africana) #tells the for loop to run         linear model  for each column specified above
  
      p_value <- summary(lm_temp)$coefficients[2,4] #extracts p value
  
      p_vals_diet[,ii-11] <- p_value #stores p value in data frame constructed above. Need to       subtract 11 from the iterator so that the loop            doesn't fill starting at row 12
    }

p_vals_diet_fdr <- p.adjust(p_vals_diet, method = "fdr")

##Species was the only factor that contained any signficant pathways, so we chose to only plot the metabolic pathways for this analysis. 

p_vals_species_fdr <- p.adjust(p_vals_species, method ="fdr")
p_vals_species_fdr <- as.data.frame(p_vals_species_fdr)
p_vals_species_fdr <- t(p_vals_species_fdr)
p_vals_species_fdr <- t(p_vals_species_fdr)
p_vals_species_fdr <- as.data.frame(p_vals_species_fdr)
p_vals_species_fdr <- rownames_to_column(p_vals_species_fdr, "code")

kegg_codes <- read_excel("kegg_codes.xlsx")
p_vals_species_fdr <- merge(kegg_codes, p_vals_species_fdr, by = "code")


```
```{r}
#Plot metabolic pathways at KEGG Level 1 for species

#NOTE: After we compiled data individual metabolic proportions across species, we exported the data into Excel and prepared a file organizing proportion by species and by metabolic pathway to save time (and to avoid long R code). We then used this re-organized data to caclucate averages and standard deviations between species for all pathways in R, and then exported this data again to prepare it for plotting in R.

kegg_species_met1 <- read_excel("kegg_1_species_prop_graph.xlsx")
kegg_species_met1 <- kegg_species_met1 %>% mutate(species = factor(species))
levels(kegg_species_met1$species) <- c("L. africana", "L. cyclotis")

jpeg("/Users/joegunn/Desktop/Grad_School_Stuff/Projects/Elephant_Microbiome/NEW_STATS/AEMB_FINAL_STATS/African_Elephant_Microbiome/kegg_level1_species.jpeg", width=6000, height=4000, units='px', res=300) 


ggplot(kegg_species_met1, aes(x = pathway, y = mean, fill = species)) + geom_bar(position=position_dodge(), stat="identity", show.legend = T) + geom_errorbar(aes(ymin=mean-sd, ymax=mean+sd), width=.4, position=position_dodge(.9)) + scale_fill_manual(values = c("red2", "blue2")) + theme(axis.text.x = element_text(angle = 0, hjust = 1)) + theme(axis.text = element_text(size = 20)) + theme(axis.title = element_text(size = 30)) + theme(legend.position = c(0.6,0.9)) + labs(fill = "Elephant Species", x = "Metabolic Pathway", y = "Mean Metabolic Contribution") + theme(legend.text = element_text(size = 30)) + theme(legend.title = element_text(size = 30)) + theme(legend.text = element_text(size = 30)) + theme(legend.text = element_text(face = "italic")) + coord_flip() + theme(axis.text.y = element_text(size = 30)) + theme(axis.title = element_text(size = 30))


dev.off()
```


## PRINCIPAL COORDINATE ANALYSIS for multi-taxa comparison ##########################

```{r}

#Re-calculating raw abundance values from proportion data
seqs <- read_excel("ley_seqs.xlsx")
gut_props <- read_excel("Species_Gut_Microbiome_Comparison.xlsx")
gut_all <- merge(seqs, gut_props, by = "common_name")

gut_abun <- read_excel("gut_abun.xlsx")
gut_abun <- as.data.frame(gut_abun)
gut_meta <- as.data.frame(gut_abun[,c(1:11)])
gut_abun <- gut_abun[,c(1,12:21)]
gut_abun <- column_to_rownames(gut_abun, "common_name")
gut_abun <- as.matrix(gut_abun)

veg_standard <- decostand(gut_abun, method = "total")
veg_gut <- vegdist(gut_abun, "bray")

pcoa_gut <- pcoa(veg_gut)

gut_pcs <- pcoa_gut$vectors
gut_pcs <- as.data.frame(gut_pcs)
gut_pcs <- gut_pcs %>% rownames_to_column("common_name")
gut_pcs <- merge(gut_meta, gut_pcs, by = "common_name")

ggplot(gut_pcs, aes(x = Axis.1, y = Axis.2, color = genus_species)) + geom_point()


#PCOA FOR DIGESTION (all data)

#Prepare data for Principal coordinate analysis
digestion <- read_excel("digestionmodels.xlsx")
digestion <- as.data.frame(digestion)
digestion_meta <- as.data.frame(digestion[,c(1:4)])
digestion <- digestion[,c(1,5:15)]
digestion <- column_to_rownames(digestion, "taxa")
digestion <- as.matrix(digestion)

####Analysis###############################
veg_dig <- vegdist(digestion, "bray")
pcoa_dig <- pcoa(veg_dig)
###########################################

#Prepare data for visualizing
dig_pcs <- pcoa_dig$vectors
dig_pcs <- as.data.frame(dig_pcs)
dig_pcs <- dig_pcs %>% rownames_to_column("taxa")
dig_pcs <- merge(digestion_meta, dig_pcs, by = "taxa")


#NMDS FOR DIGESTION (all data)

####Run NMDS##############################################
dig_nmds <- isoMDS(veg_dig, trace = 0)
dig_nmds_use <- metaMDS(digestion)
score_dig <- scores(dig_nmds_use)
score_dig<- as.data.frame(score_dig)
score_dig <- score_dig %>% rownames_to_column("taxa")
merge_dig <- merge(digestion_meta, score_dig, by = "taxa")
############################################################


#PCOA FOR DIGESTION (just top 3 taxa)

####Analysis##############################################
digestion_top3 <- digestion[,c(1:3)]
veg_top3 <- vegdist(digestion_top3, "bray")
pcoa_top3 <- pcoa(veg_top3)
top3_pcs <- pcoa_top3$vectors
top3_pcs <- as.data.frame(top3_pcs)
top3_pcs <- top3_pcs %>% rownames_to_column("taxa")
top3_pcs <- merge(digestion_meta, top3_pcs, by = "taxa")
############################################################

#Make Figures

jpeg("/Users/joegunn/Desktop/Grad_School_Stuff/Projects/Elephant_Microbiome/NEW_STATS/AEMB_FINAL_STATS/African_Elephant_Microbiome/dig_pcoa.jpeg", width=2000, height=1400, units='px', res=300) 

ggplot(dig_pcs, aes(x = Axis.1, y = Axis.2, color = digestion)) + geom_point(show.legend = T, size = 4) + labs(color = "Digestion Type") + labs(x = "PCoA Axis 1 (18.6 %)", y = "PCoA Axis 2 (13.4 %)") + theme(legend.position = c(0.1, 0.85)) + theme(axis.title = element_text(size = 20))

dev.off()


##################################################################################################################################
#59 Species

#PCOA FOR DIGESTION (all data)

#Prepare data for Principal coordinate analysis
multi_gut_top3 <- read_excel("species_gut_top3_no_car.xlsx")
multi_gut_top3 <- as.data.frame(multi_gut_top3)
multi_gut_top3_meta <- as.data.frame(multi_gut_top3[,c(1:10)])
multi_gut_top3 <- multi_gut_top3[,c(7,11:13)]
multi_gut_top3 <- column_to_rownames(multi_gut_top3, "Common_name")
multi_gut_top3 <- as.matrix(multi_gut_top3)


#Proteobacteira
ggplot(multi_gut, aes(x = Feeding_Strategy, y = Proteobacteria)) + geom_boxplot() + theme(axis.text.x = element_text(angle = 45, hjust = 1)) 
ggplot(multi_gut, aes(x = Diet, y = Proteobacteria)) + geom_boxplot() + theme(axis.text.x = element_text(angle = 45, hjust = 1)) 

#Firmicutes
ggplot(multi_gut, aes(x = Feeding_Strategy, y = Firmicutes)) + geom_boxplot() + theme(axis.text.x = element_text(angle = 45, hjust = 1)) 
ggplot(multi_gut, aes(x = Diet, y = Firmicutes)) + geom_boxplot() + theme(axis.text.x = element_text(angle = 45, hjust = 1)) 

#Bacteriodetes
ggplot(multi_gut, aes(x = Feeding_Strategy, y = Bacteriodetes)) + geom_boxplot() + theme(axis.text.x = element_text(angle = 45, hjust = 1)) 
ggplot(multi_gut, aes(x = Diet, y = Bacteriodetes)) + geom_boxplot() + theme(axis.text.x = element_text(angle = 45, hjust = 1)) 




####Analysis###############################
veg_multi_top3 <- vegdist(multi_gut_top3, "bray")
pcoa_multi_top3 <- pcoa(veg_multi_top3)
###########################################

options(scipen = 999)


#Prepare data for visualizing
multi_gut_top3_pcs <- pcoa_multi_top3$vectors
multi_gut_top3_pcs <- as.data.frame(multi_gut_top3_pcs)
multi_gut_top3_pcs <- multi_gut_top3_pcs %>% rownames_to_column("Common_name")
multi_gut_top3_pcs <- merge(multi_gut_top3_meta, multi_gut_top3_pcs, by = "Common_name")


gut_nmds <- isoMDS(veg_multi_top3, trace = T) #omitted Dairy Cow 2
gut_nmds_use <- metaMDS(multi_gut_top3)
score_gut <- scores(gut_nmds_use)
score_gut <- as.data.frame(score_gut)
score_gut <- score_gut %>% rownames_to_column("Common_name")
merge_gut <- merge(multi_gut_top3_meta, score_gut, by = "Common_name")


#Make Figures

jpeg("/Users/joegunn/Desktop/Grad_School_Stuff/Projects/Elephant_Microbiome/NEW_STATS/AEMB_FINAL_STATS/African_Elephant_Microbiome/dig_pcoa.jpeg", width=7000, height=4700, units='px', res=800) 

ggplot(multi_gut_top3_pcs, aes(x = Axis.1, y = Axis.2, color = Feeding_Strategy)) + geom_point(show.legend = T) + labs(x = "PCoA Axis 1 (4.6 %)", y = "PCoA Axis 2 (3.8 %)")

##geom_text(aes(label = Species), show.legend = T, angle = 30, hjust = 0, vjust = 0, nudge_y = -0.001, nudge_x = 0.01, check_overlap = F, size = 2.5) + labs(color = "Digestion Type") + 
dev.off()



```


