---
title: "Analysis 3: Global and core microbiome analysis"
author: "Joe Gunn"
date: "2024-02-07"
output: html_document
---

# Project: Assessing effects of phylogeny, diet, and habitat on gut microbial composition in African elephants 
We characterized the gut microbiome of African Savanna elephants (<i>Loxodonta africana</i>) and African Forest elephants (<i>Loxodonta cyclotis</i>). We assessed the relationship of gut microbial composition, including analyses of alpha and beta diversity, with host phylogeny (i.e., species) and habitat type (i.e., forest or savanna) for both species. We also assessed the relationship between microbial composition and diet (i.e., crop-raiding vs. non-crop-raiding) within <i>L. africana</i>. 

## Specific Aim: Assessing the global (full data) and core (shared) microbiome of African elephant sample groups (phylogeny, diet, and habitat) 
In this analysis, we characterize the core microbial composition of African elephant individuals within groups of interest (phylogeny, diet, and habitat) at the Order level (see Analysis 2): Orders that are present in 100% of samples within a given group. For each group of interest, we control for all other groups to identify these orders. 

## Phases of analysis:
### Phase 1: Data read-in, filtering, rarefaction, and summarization
### Phase 2: Summary of OTU abundance data and core microbiome

## Libraries needed for analysis
```{r}
library(readxl)
library(cowplot)
library(tidyverse)
library(phyloseq)
library(qiime2R)
```

## PHASE 1: DATA READ-IN, FILTERING, RAREFACTION, AND SUMMARIZATION
In this phase of the analysis, we read in raw data from bioinformatic analyses in QIIME, filter the raw OTU and taxonomic classification data, rarefy read counts, and summarize the full microbiome dataset.

### STEP 1: Summarize OTU abundance across samples and filter samples with low read-count for rarefaction
In this step, we summarize OTU abundance at the elephant sample level (i.e., how many individual sequence reads were detected per OTU within each elephant sample) to determine average read-count and to identify any outliers (i.e., either extremely high or low read counts). We use the raw phyloseq object with taxonomic classification, OTU abundance, phylogenetic relationship, and metadata. We calculate the sum of read counts across all OTUs within each sample. Ultimately, we need to rarefy the abundance to an even read depth per sample to limit observation bias; for this reason, we want to omit any samples with extremely low read-counts, because we do not want to rarefy the dataset to an extremely low read count.

#### 1a. Generate, clean, and summarize OTU abundance by elephant sample; run the Rmd chunk below.

##### Generate, clean, and summarize the OTU abundance data from phyloseq object:
```{r}
# Read in phyloseq object
load("../02_data_summary_analysis/data/phyloseq.Rda")

# Create a data object with otu abundnace data, clean, and gather to make tall table
otu_abundance <- as.data.frame(otu_table(phyloseq)) %>%
  rownames_to_column("otu_id") %>%
  mutate(otu_id = factor(otu_id)) %>%
  gather(MM69:M10, key = "ind_id", value = "abundance") %>% 
  mutate(ind_id = factor(ind_id))

# Sum read counts for all individuals
reads <- otu_abundance %>%
  group_by(ind_id) %>%
  summarize(sum(abundance))

# Summarize mean (standard deviation) of read counts across the dataset
reads %>%
  summarize(mean = mean(reads),
            sd = sd(reads))

# Summarize minimum and maximum number of read counts across the dataset
reads %>%
  arrange(desc(reads))

# Save otu abundance summary table for downstream analysis 
save(reads, file = "data/reads.Rda")
```

### Data summary:

## Mean (standard deviation) of read counts for full dataset
<b><i>mean</i><sub>reads</sub></b> = 129,632.8 (463,334.5) <br>

## Minimum and maximum confidence for full dataset
<b><i>max</i><sub>reads</sub></b> = 3,268,723 <br>
<b><i>min</i><sub>reads</sub></b> = 163 <br>

#### 1b. Visualize OTU read counts per elephant sample; run the Rmd chunk below.

##### Plot OTU read counts per elephant sample: `figures/reads.pdf`
```{r}
# Load read count data
load("data/reads.Rda")

# Plot read counts per individual
pdf("figures/reads.pdf", width = 10, height = 4) 

ggplot(reads, aes(x = ind_id, y = reads)) + 
  geom_bar(stat = "identity", color = "black", fill = "grey") +
  theme_set(theme_cowplot(12)) +
  labs(x = "Elephant ID", y = "Read count") +
  theme(axis.title = element_text(size = 15)) + 
  theme(axis.text = element_text(size = 15)) + 
  theme(axis.text.x = element_text(angle = 90, hjust = 0.9, vjust = 0.3)) + 
  theme(panel.border = element_rect(colour = "black", fill = NA, size = 1))

dev.off()
```

#### 1c. Omit sample with extremely high read count relative to other samples (CR71), and re-visualize read counts to identify low-read individuals; run the Rmd chunk below.

##### Plot OTU read counts per elephant sample without individual CR71: `figures/reads_filtered.pdf`
```{r}
# Load read count data
load("data/reads.Rda")

# Omit individual CR71 from the data
reads <- reads %>%
  filter(ind_id != "CR71")

# Plot confidence and proportion of samples per taxonomic level
pdf("figures/reads_filtered.pdf", width = 10, height = 4) 

ggplot(reads, aes(x = ind_id, y = reads)) + 
  geom_bar(stat = "identity", color = "black", fill = "grey") +
  theme_set(theme_cowplot(12)) +
  labs(x = "Elephant ID", y = "Read count") +
  theme(axis.title = element_text(size = 15)) + 
  theme(axis.text = element_text(size = 15)) + 
  theme(axis.text.x = element_text(angle = 90, hjust = 0.9, vjust = 0.3)) + 
  theme(panel.border = element_rect(colour = "black", fill = NA, size = 1))

dev.off()
```

<b> Data summary results </b>: <br>

After data summarization, we observed that individual OB182 had an order of magnitude lower read count (163) than the next most abundant individual (11,460; see Step 1a above). For this reason, we opted to drop OB182 from the final dataset for downstream analyses, and the OTU raw abundance data were rarefied to 11,460 reads (Continue on to Step 1d).

#### 1d. Subset out sample with extremely low read count relative to other samples (OB182), and generate new phyloseq object for downstream analyses; run the Rmd chunk below.

##### Subset sample with low read count and generate new phyloseq object:
```{r}
# Load read count data
load("../02_taxon_summary_analysis/data/phyloseq.Rda")

# Remove individual OB182 from the data
phyloseq <- subset_samples(phyloseq, ind_id != "OB182") 

# Save filtered otu abundance phyloseq object
save(phyloseq, file = "data/phyloseq.Rda")
```

### STEP 2: Rarefy OTU abundance data
In this step, we rarefy the OTU abundance dataset to the lowest number of reads captured in the full dataset for an individual (CR3, <i>N</i>=11,460). Rarefaction is a necessary step to randomly sample OTU abundance data to an even number of "samples", aka reads, to prevent sampling bias in downstream analyses. 

#### 2a. Rarefy the OTU abundance data; run the Rmd chunk below.

##### Rarefy abundance data:
```{r}
# Load filtered phyloseq object
load("data/phyloseq.Rda")

# R changed random number seed generation at R v.3.6.0, so this command is needed to revert to old seeds
RNGkind(sample.kind = "Rounding")

# Read in, combine, and convert QIIME output and metadata to phyloseq format
phyloseq_rarefied <- rarefy_even_depth(phyloseq, 
                                       rngseed = 5)

# Save rarefied OTU abundance data for downstream analyses
save(phyloseq_rarefied, file = "data/phyloseq_rarefied.Rda")
```

<b> Data filtering results </b>: <br>
<b><i>read depth</i> = 11,460 </b><br>
<b><i>N<sub>OTUs</sub></i> = 8,248 </b><br>

### STEP 3: Summarize OTU abundance across taxonomic levels (i.e., kingdom, phylum, class, and order)
In this step, we summarize OTU abundance data across four different taxonomic levels (i.e., kingdom, phylum, class, order) based on prior analysis of taxonomic classification success and confidence; we ultimately summarize OTUs and the core microbiome of elephant samples at the order level (see Analysis 2).

#### 3a. Generate the full, filtered, and clean dataset; run the Rmd chunk below.
In this step, we generate a full, filtered, and clean dataset, which merges taxonomic identification of all OTUs, rarefied abundance of OTUs for each African elephant sample, and metadata for each elephant sample. This dataset is saved for all downstream analyses.

##### Generate full, filtered, clean dataset for downstream analyses:
```{r}
# Load rarefied phyloseq object
load("data/phyloseq_rarefied.Rda")

# Load sample metadata
load("../02_data_summary_analysis/data/metadata.Rda")
  
# Create a data object with taxonomic classification data and clean
taxa <- tax_table(phyloseq_rarefied) %>%
  as.data.frame() %>%
  rownames_to_column("otu_id") %>%
  mutate(otu_id = factor(otu_id), 
         kingdom = factor(Kingdom), 
         phylum = factor(Phylum), 
         class = factor(Class), 
         order = factor(Order), 
         family = factor(Family), 
         genus = factor(Genus), 
         species = factor(Species)) %>%
  select(otu_id, kingdom, phylum, class, order, family, genus, species)

# Create a data object with otu abundance data and clean
otu_abundance <- as.data.frame(otu_table(phyloseq_rarefied)) %>%
  rownames_to_column("otu_id") %>%
  mutate(otu_id = factor(otu_id))

# Merge taxonomic classifications with abundance data
otu_abundance <- merge(otu_abundance, 
                       taxa, 
                       by = "otu_id")

# Create tall data that can be sorted by taxonomic level
otu_abundance <- otu_abundance %>%
  gather(MM69:M10, key = "ind_id", value ="abundance")

# Merge taxonomic classifications with elephant sample metadata
full_data <- merge(otu_abundance, 
                   metadata, 
                   by = "ind_id")

# Save full dataset for all downstream analyses
save(full_data, file = "data/full_data.Rda")
```

#### 3b. Generate complete dataset (no missing data) at the Order level; run the Rmd chunk below.
In this step, we filter the full dataset to only include OTUs which were identified to the taxonomic level of Order (see Analysis 2, Step 2). We exclude all OTUs that are not classified to the level of order.

##### Generate complete dataset at order level:
```{r}
# Load full data
load("data/full_data.Rda")

# Include only OTUs classified to order level
order_data <- full_data %>%
  drop_na(order)

# Save order dataset for all downstream analyses
save(order_data, file = "data/order_data.Rda")
```

#### 3c. Inspect individual microbial composition at the order level within groups of interest to control for outliers; run the Rmd chunk below.
In this step, we summarize OTU abundance at the order level (see Analysis 2, Step 2) for the fully filtered dataset. We then visually inspect microbial composition at the order level for each individual elephant sample within groups of interest (phylogeny, habitat, and diet) to control for (and eliminate) outliers.

##### 3c.1. Visually inspect order composition by elephant phylogeny; run the Rmd chunk below:

##### Visually inspect order composition by elephant phylogeny: `figures/orders_phylogeny.pdf`
```{r}
# Load order dataset
load("data/order_data.Rda")

# Get only L. africana samples
la <- order_data %>%
  filter(phylogeny == "africana")

# Get only L. cyclotis samples
lc <- order_data %>%
  filter(phylogeny == "cyclotis")

# Plot L. africana order composition
la_plot <- ggplot(la, aes(x = ind_id, y = abundance, fill = order)) + 
  geom_bar(position = "stack", stat = "identity", show.legend = F) +
  theme_set(theme_cowplot(12)) +
  labs(x = "Sample ID", y = "OTU abundance", title = "L. africana") +
  theme(plot.title = element_text(size = 15, face = c("bold.italic"))) +
  theme(axis.text = element_text(size = 15)) + 
  theme(axis.title = element_text(size = 15)) + 
  theme(axis.text.x = element_text(angle = 90)) +
  theme(axis.text.x = element_text(hjust = 0.95)) +
  theme(axis.text.x = element_text(vjust = 0.2)) +
  theme(axis.ticks = element_blank()) +
  theme(panel.border = element_rect(colour = "black", fill = NA, size = 1))

# Plot L. cyclotis order composition
lc_plot <- ggplot(lc, aes(x = ind_id, y = abundance, fill = order)) + 
  geom_bar(position = "stack", stat = "identity", show.legend = F) +
  theme_set(theme_cowplot(12)) +
  labs(x = "Sample ID", y = "OTU abundance", title = "L. cyclotis") +
  theme(plot.title = element_text(size = 15, face = "bold.italic")) +
  theme(axis.text = element_text(size = 15)) + 
  theme(axis.title = element_text(size = 15)) + 
  theme(axis.text.y = element_blank()) +
  theme(axis.title.y = element_blank()) +
  theme(axis.text.x = element_text(angle = 90)) +
  theme(axis.text.x = element_text(hjust = 0.95)) +
  theme(axis.text.x = element_text(vjust = 0.2)) +
  theme(axis.ticks = element_blank()) +
  theme(panel.border = element_rect(colour = "black", fill = NA, size = 1))

# Plot confidence and proportion of samples per taxonomic level
pdf("figures/orders_phylogeny.pdf", width = 10, height = 4) 

plot_grid(la_plot, 
          lc_plot,
          nrow = 1,
          ncol = 2)

dev.off()
```

This figure was only used for visual inspection during analysis and was not used in the final ms.

##### 3c.2. Visually inspect order composition by elephant diet; run the Rmd chunk below:

##### Visually inspect order composition by elephant diet: `figures/orders_diet.pdf`
```{r}
# Load order dataset
load("data/order_data.Rda")

# Get only L. africana samples
cr <- order_data %>%
  filter(diet == "Raider")

# Get only L. cyclotis samples
ncr <- order_data %>%
  filter(diet == "Nonraider")

# Plot crop raider order composition
cr_plot <- ggplot(cr, aes(x = ind_id, y = abundance, fill = order)) + 
  geom_bar(position = "stack", stat = "identity", show.legend = F) +
  theme_set(theme_cowplot(12)) +
  labs(x = "Sample ID", y = "OTU abundance", title = "Crop raiders") +
  theme(plot.title = element_text(size = 15, face = "bold")) +
  theme(axis.text = element_text(size = 15)) + 
  theme(axis.title = element_text(size = 15)) + 
  theme(axis.text.x = element_text(angle = 90)) +
  theme(axis.text.x = element_text(hjust = 0.95)) +
  theme(axis.text.x = element_text(vjust = 0.2)) +
  theme(axis.ticks = element_blank()) +
  theme(panel.border = element_rect(colour = "black", fill = NA, size = 1))

# Plot non-raider order composition
ncr_plot <- ggplot(ncr, aes(x = ind_id, y = abundance, fill = order)) + 
  geom_bar(position = "stack", stat = "identity", show.legend = F) +
  theme_set(theme_cowplot(12)) +
  labs(x = "Sample ID", y = "OTU abundance", title = "Non-crop raiders") +
  theme(plot.title = element_text(size = 15, face = "bold")) +
  theme(axis.text = element_text(size = 15)) + 
  theme(axis.title = element_text(size = 15)) + 
  theme(axis.text.y = element_blank()) +
  theme(axis.title.y = element_blank()) +
  theme(axis.text.x = element_text(angle = 90)) +
  theme(axis.text.x = element_text(hjust = 0.95)) +
  theme(axis.text.x = element_text(vjust = 0.2)) +
  theme(axis.ticks = element_blank()) +
  theme(panel.border = element_rect(colour = "black", fill = NA, size = 1))

# Plot both plots together
pdf("figures/orders_diet.pdf", width = 10, height = 4) 

plot_grid(cr_plot, 
          ncr_plot,
          nrow = 1,
          ncol = 2)

dev.off()
```

This figure was only used for visual inspection during analysis and was not used in the final ms.

##### 3c.2. Visually inspect order composition by elephant habitat; run the Rmd chunk below:

##### Visually inspect order composition by elephant habitat: `figures/orders_habitat.pdf`
```{r}
# Load order dataset
load("data/order_data.Rda")

# Get only L. africana samples
s <- order_data %>%
  filter(habitat == "Savanna")

# Get only L. cyclotis samples
f <- order_data %>%
  filter(habitat == "Forest")

# Plot savanna order composition
s_plot <- ggplot(s, aes(x = ind_id, y = abundance, fill = order)) + 
  geom_bar(position = "stack", stat = "identity", show.legend = F) +
  theme_set(theme_cowplot(12)) +
  labs(x = "Sample ID", y = "OTU abundance", title = "Savanna") +
  theme(plot.title = element_text(size = 15, face = "bold")) +
  theme(axis.text = element_text(size = 15)) + 
  theme(axis.title = element_text(size = 15)) + 
  theme(axis.text.x = element_text(angle = 90)) +
  theme(axis.text.x = element_text(hjust = 0.95)) +
  theme(axis.text.x = element_text(vjust = 0.2)) +
  theme(axis.ticks = element_blank()) +
  theme(panel.border = element_rect(colour = "black", fill = NA, size = 1))

# Plot forest order composition
f_plot <- ggplot(f, aes(x = ind_id, y = abundance, fill = order)) + 
  geom_bar(position = "stack", stat = "identity", show.legend = F) +
  theme_set(theme_cowplot(12)) +
  labs(x = "Sample ID", y = "OTU abundance", title = "Forest") +
  theme(plot.title = element_text(size = 15, face = "bold")) +
  theme(axis.text = element_text(size = 15)) + 
  theme(axis.title = element_text(size = 15)) + 
  theme(axis.text.y = element_blank()) +
  theme(axis.title.y = element_blank()) +
  theme(axis.text.x = element_text(angle = 90)) +
  theme(axis.text.x = element_text(hjust = 0.95)) +
  theme(axis.text.x = element_text(vjust = 0.2)) +
  theme(axis.ticks = element_blank()) +
  theme(panel.border = element_rect(colour = "black", fill = NA, size = 1))

# Plot both plots together
pdf("figures/orders_habitat.pdf", width = 10, height = 4) 

plot_grid(s_plot, 
          f_plot,
          nrow = 1,
          ncol = 2)

dev.off()
```

This figure was only used for visual inspection during analysis and was not used in the final ms.

We detected only one potential outlier sample (M10, <i>L. cyclotis</i>) driving extreme order differentation among groups of interest (i.e., phylogeny). We excluded this sample to identify the core microbiome of <i>L. cyclotis</i> but retained this individual for all downstream analyses.

### ----------------------- END OF PHASE 1: DATA READ-IN, FILTERING, RAREFACTION, and SUMMARIZATION  ----------------------- ###

## PHASE 2: SUMMARY OF OTU ABUNDANCE AND CORE MICROBIOME
In this phase of the analysis, we calculate the proportion of the total microbiome represented by each present taxonomic group within kingdoms, phyla, classes, and orders in the sample set. We then break these proportions down by phylogeny, diet, and habitat for direct comparison of microbiota among groups. 

We ultimately present proportions of microbial OTUs at the phylum and order levels, giving relative proportions of each phylum and order across the full microbiome dataset, as well as the relative proportions of each order within each phylum. These summary statistics are the basis for Table S1 in the final ms. 

### STEP 1: Summarize OTU abundance in groups of interest 
In this step, we summarize the proportion of OTU abundance represented by the four broadest microbial taxonomic levels (i.e., kingdom, phylum, class, order) within our elephant sample groups of interest. We do not conduct statistical tests to compare abundance across elephant groups, nor do we test for statistical differences of abundance among microbial taxonomic classifications. We provide these summary statistics simply as a description of the overall microbiome recovered in our study.

#### 1a. Summarize OTU abundance proportions within groups in <i>L. africana</i>.

##### 1a.1 Summarize OTU abundance across taxanomic groups in savanna-dwelling, non-crop-raiding <i>L. africana</i>; run the Rmd chunk below.

##### Summarize OTU abundance in savanna-dwelling, non-crop-raiding <i>L. africana</i>
```{r}
# Load order dataset
load("data/order_data.Rda")

# Set scipen options to remove scientific notation
options(scipen = 999)

# Get only L. africana savanna-dwelling samples
las <- order_data %>%
  filter(phylogeny == "africana" & habitat == "Savanna" & diet == "Nonraider")

# Calculate total microbial abundance across samples
total_abundance <- sum(las$abundance)

# Get kingdom proportions
kingdom <- las %>%
  group_by(kingdom) %>%
  summarize(kingdom_abundance = sum(abundance),
            kingdom_proportion = kingdom_abundance/total_abundance,
            kingdom_percent = kingdom_proportion*100)

# Get phylum proportions (within kingdoms)
phylum <- las %>%
  group_by(kingdom, phylum) %>%
  summarize(phylum_abundance = sum(abundance),
            phylum_proportion = phylum_abundance/total_abundance,
            phylum_percent = phylum_proportion*100)

# Get class proportions (within phyla)
class <- las %>%
  group_by(phylum, class) %>%
  summarize(class_abundance = sum(abundance),
            class_proportion = class_abundance/total_abundance,
            class_percent = class_proportion*100)

# Get order proportions (within phyla; we aren't interested in class at this point)
order <- las %>%
  group_by(phylum, order) %>%
  summarize(order_abundance = sum(abundance),
            order_proportion = order_abundance/total_abundance,
            order_percent = order_proportion*100)
  
# Join phylum and order proportions
full_abundance <- full_join(order, phylum, by = "phylum")

# Calculate proportion of each phylum constituted by each order 
full_abundance %>% 
  group_by(phylum, order) %>%
  summarize(order_percent = order_abundance/phylum_abundance*100)
```

##### 1a.2 Summarize OTU abundance across taxanomic groups in savanna-dwelling, crop-raiding <i>L. africana</i>; run the Rmd chunk below.

##### Summarize OTU abundance in savanna-dwelling, crop-raiding <i>L. africana</i>
```{r}
# Load order dataset
load("data/order_data.Rda")

# Set scipen options to remove scientific notation
options(scipen = 999)

# Get only L. africana savanna-dwelling, crop-raiding samples
lacr <- order_data %>%
  filter(phylogeny == "africana" & habitat == "Savanna" & diet == "Raider")

# Calculate total microbial abundance across samples
total_abundance <- sum(lacr$abundance)

# Get kingdom proportions
lacr %>%
  group_by(kingdom) %>%
  summarize(kingdom_abundance = sum(abundance),
            kingdom_proportion = kingdom_abundance/total_abundance,
            kingdom_percent = kingdom_proportion*100)

# Get phylum proportions (within kingdoms)
lacr %>%
  group_by(kingdom, phylum) %>%
  summarize(phylum_abundance = sum(abundance),
            phylum_proportion = phylum_abundance/total_abundance,
            phylum_percent = phylum_proportion*100)

# Get class proportions (within phyla)
lacr %>%
  group_by(phylum, class) %>%
  summarize(class_abundance = sum(abundance),
            class_proportion = class_abundance/total_abundance,
            class_percent = class_proportion*100)

# Get order proportions (within phyla; we aren't interested in class at this point)
lacr %>%
  group_by(phylum, order) %>%
  summarize(order_abundance = sum(abundance),
            order_proportion = order_abundance/total_abundance,
            order_percent = order_proportion*100)

# Join phylum and order proportions
full_abundance <- full_join(order, phylum, by = "phylum")

# Calculate proportion of each phylum constituted by each order 
full_abundance %>% 
  group_by(phylum, order) %>%
  summarize(order_percent = order_abundance/phylum_abundance*100)
```

##### 1a.3 Summarize OTU abundance across taxanomic groups in forest-dwelling, non-crop-raiding <i>L. africana</i>; run the Rmd chunk below.

##### Summarize OTU abundance in forest-dwelling, non-crop-raiding <i>L. africana</i>
```{r}
# Load order dataset
load("data/order_data.Rda")

# Set scipen options to remove scientific notation
options(scipen = 999)

# Get only L. africana forest-dwelling, non-crop-raiding samples
laf <- order_data %>%
  filter(phylogeny == "africana" & habitat == "Forest" & diet == "Nonraider")

# Calculate total microbial abundance across samples
total_abundance <- sum(laf$abundance)

# Get kingdom proportions
laf %>%
  group_by(kingdom) %>%
  summarize(kingdom_abundance = sum(abundance),
            kingdom_proportion = kingdom_abundance/total_abundance,
            kingdom_percent = kingdom_proportion*100)

# Get phylum proportions (within kingdoms)
laf %>%
  group_by(kingdom, phylum) %>%
  summarize(phylum_abundance = sum(abundance),
            phylum_proportion = phylum_abundance/total_abundance,
            phylum_percent = phylum_proportion*100)

# Get class proportions (within phyla)
laf %>%
  group_by(phylum, class) %>%
  summarize(class_abundance = sum(abundance),
            class_proportion = class_abundance/total_abundance,
            class_percent = class_proportion*100)

# Get order proportions (within phyla; we aren't interested in class at this point)
laf %>%
  group_by(phylum, order) %>%
  summarize(order_abundance = sum(abundance),
            order_proportion = order_abundance/total_abundance,
            order_percent = order_proportion*100)

# Join phylum and order proportions
full_abundance <- full_join(order, phylum, by = "phylum")

# Calculate proportion of each phylum constituted by each order 
full_abundance %>% 
  group_by(phylum, order) %>%
  summarize(order_percent = order_abundance/phylum_abundance*100)
```

##### 1a.4 Summarize OTU abundance across taxanomic groups in forest-dwelling, non-crop-raiding <i>L. africana</i>; run the Rmd chunk below.

##### Summarize OTU abundance in forest-dwelling, crop-raiding <i>L. africana</i>
```{r}
# Load order dataset
load("data/order_data.Rda")

# Set scipen options to remove scientific notation
options(scipen = 999)

# Get only L. africana forest-dwelling, crop-raiding samples
lafcr <- order_data %>%
  filter(phylogeny == "africana" & habitat == "Forest" & diet == "Raider")

# Calculate total microbial abundance across samples
total_abundance <- sum(lafcr$abundance)

# Get kingdom proportions
lafcr %>%
  group_by(kingdom) %>%
  summarize(kingdom_abundance = sum(abundance),
            kingdom_proportion = kingdom_abundance/total_abundance,
            kingdom_percent = kingdom_proportion*100)

# Get phylum proportions (within kingdoms)
lafcr %>%
  group_by(kingdom, phylum) %>%
  summarize(phylum_abundance = sum(abundance),
            phylum_proportion = phylum_abundance/total_abundance,
            phylum_percent = phylum_proportion*100)

# Get class proportions (within phyla)
lafcr %>%
  group_by(phylum, class) %>%
  summarize(class_abundance = sum(abundance),
            class_proportion = class_abundance/total_abundance,
            class_percent = class_proportion*100)

# Get order proportions (within phyla; we aren't interested in class at this point)
lafcr %>%
  group_by(phylum, order) %>%
  summarize(order_abundance = sum(abundance),
            order_proportion = order_abundance/total_abundance,
            order_percent = order_proportion*100)

# Join phylum and order proportions
full_abundance <- full_join(order, phylum, by = "phylum")

# Calculate proportion of each phylum constituted by each order 
full_abundance %>% 
  group_by(phylum, order) %>%
  summarize(order_percent = order_abundance/phylum_abundance*100)
```

#### 1b. Summarize OTU proportions within groups in <i>L. cyclotis</i>.

##### 1b.1 Summarize OTU abundance across taxanomic groups in <i>L. cyclotis</i>; run the Rmd chunk below.

##### Summarize OTU abundance in <i>L. cyclotis</i>
```{r}
# Load order dataset
load("data/order_data.Rda")

# Set scipen options to remove scientific notation
options(scipen = 999)

# Get only L. cyclotis samples
lc <- order_data %>%
  filter(phylogeny == "cyclotis")

# Calculate total microbial abundance across samples
total_abundance <- sum(lc$abundance)

# Get kingdom proportions
lc %>%
  group_by(kingdom) %>%
  summarize(kingdom_abundance = sum(abundance),
            kingdom_proportion = kingdom_abundance/total_abundance,
            kingdom_percent = kingdom_proportion*100)

# Get phylum proportions (within kingdoms)
lc %>%
  group_by(kingdom, phylum) %>%
  summarize(phylum_abundance = sum(abundance),
            phylum_proportion = phylum_abundance/total_abundance,
            phylum_percent = phylum_proportion*100)

# Get class proportions (within phyla)
lc %>%
  group_by(phylum, class) %>%
  summarize(class_abundance = sum(abundance),
            class_proportion = class_abundance/total_abundance,
            class_percent = class_proportion*100)

# Get order proportions (within phyla; we aren't interested in class at this point)
lc %>%
  group_by(phylum, order) %>%
  summarize(order_abundance = sum(abundance),
            order_proportion = order_abundance/total_abundance,
            order_percent = order_proportion*100)

# Join phylum and order proportions
full_abundance <- full_join(order, phylum, by = "phylum")

# Calculate proportion of each phylum constituted by each order 
full_abundance %>% 
  group_by(phylum, order) %>%
  summarize(order_percent = order_abundance/phylum_abundance*100)
```

### STEP 2. Summarize core microbiome by phylogeny, diet, and habitat
In this step, we use the full order dataset (all OTUs classified to the level of Order) to characterize the core microbiome of each group of interest.

#### 2a. Characterize the core microbiome by phylogeny (<i>L. africana</i> or <i>L. cyclotis</i>).
In this step, we characterize the core microbiome of each phylogenetic group in the data (African savanna elephants, African forest elephants). First, we identify the core microbiome of all individuals within each group (Steps 4a.1- 4a.3). Next, we directly compare the phylogenetic groups by controlling for diet and habitat. Since All Forest individuals are non-crop-raiding and live in forest habitat, we filter out only non-crop-raiding Savanna elephants found in forest habitat (Step 4a.4).

All core microbiome proportions are the basis for Table S2 in the final ms.

##### 2a.1. Characterize the core microbiome of all <i>L. africana</i>; run the Rmd chunk below.

##### Characterize core microbiome of <i>L. africana</i>:
```{r}
# Load order dataset
load("data/order_data.Rda")

# Filter only savanna elephants
la <- order_data %>%
  filter(phylogeny == "africana")

# Get the total number of individual savanna elephants
n_ind <- length(unique(la$ind_id))

# Filter out only savanna elephants, select individual id, order, and abundance, and summarize abundance by order within individuals
la <- order_data %>%
  filter(phylogeny == "africana") %>%
  select(ind_id, order, abundance) %>%
  group_by(order, ind_id) %>%
  summarize(abundance = sum(abundance)) %>%
  filter(abundance != "0") %>%
  count() %>%
  summarize(order_proportion = n/n_ind)

# Filter out only the core microbiome (orders with 100% presence across savanna elephant samples)
la_core <- la %>%
  filter(order_proportion == 1.00000000)
```

##### 2a.2. Characterize the core microbiome of all <i>L. cyclotis</i>; run the Rmd chunk below.

##### Characterize core microbiome of <i>L. cyclotis</i>:
```{r}
# Load order dataset
load("data/order_data.Rda")

# Filter only forest elephants
lc <- order_data %>%
  filter(phylogeny == "cyclotis")

# Get the total number of individual forest elephants
n_ind <- length(unique(lc$ind_id))

# Filter out only forest elephants, select individual id, order, and abundance, and summarize abundance by order within individuals
lc <- order_data %>%
  filter(phylogeny == "cyclotis") %>%
  select(ind_id, order, abundance) %>%
  group_by(order, ind_id) %>%
  summarize(abundance = sum(abundance)) %>%
  filter(abundance != "0") %>%
  count() %>%
  summarize(order_proportion = n/n_ind)

# Filter out only the core microbiome (orders with 100% presence across forest elephant samples)
lc_core <- forest %>%
  filter(order_proportion == 1.00000000)
```

##### 2a.3. Characterize the core microbiome of all <i>L. cyclotis</i> without individual M10; run the Rmd chunk below.

##### Characterize core microbiome of <i>L. cyclotis</i> without sample M10:
```{r}
# Load order dataset
load("data/order_data.Rda")

# Filter only forest elephants
lc_m10 <- order_data %>%
  filter(phylogeny == "cyclotis") %>%
  filter(ind_id != "M10")

# Get the total number of individual forest elephants
n_ind <- length(unique(lc_m10$ind_id))

# Filter out only forest elephants, select individual id, order, and abundance, and summarize abundance by order within individuals
lc_m10 <- order_data %>%
  filter(phylogeny == "cyclotis") %>%
  filter(ind_id != "M10") %>%
  select(ind_id, order, abundance) %>%
  group_by(order, ind_id) %>%
  summarize(abundance = sum(abundance)) %>%
  filter(abundance != "0") %>%
  count() %>%
  summarize(order_proportion = n/n_ind)

# Filter out only the core microbiome (orders with 100% presence across forest elephant samples)
lc_m10_core <- lc_m10 %>%
  filter(order_proportion == 1.00000000)
```

##### 2a.4. Characterize the core microbiome of only <i>L. africana</i> that were collected from forest habitat; run the Rmd chunk below.

##### Characterize core microbiome of <i>L. africana</i> from forest habitat:
```{r}
# Load order dataset
load("data/order_data.Rda")

# Filter only savanna forest elephants
laf <- order_data %>%
  filter(phylogeny == "africana" & habitat == "Forest" & diet == "Nonraider") 

# Get the total number of individual savanna forest elephants
n_ind <- length(unique(laf$ind_id))

# Filter out only savanna forest elephants, select individual id, order, and abundance, and summarize abundance by order within individuals
laf <- order_data %>%
  filter(phylogeny == "africana" & habitat == "Forest" & diet == "Nonraider") %>%
  select(ind_id, order, abundance) %>%
  group_by(order, ind_id) %>%
  summarize(abundance = sum(abundance)) %>%
  filter(abundance != "0") %>%
  count() %>%
  summarize(order_proportion = n/n_ind)

# Filter out only the core microbiome (orders with 100% presence across savanna forest elephant samples)
laf_core <- laf %>%
  filter(order_proportion == 1.00000000)
```

#### 2b. Characterize the core microbiome by diet (crop-raider or non-crop-raider).
In this step, we characterize the core microbiome of each diet group in the data (crop-radiers, non-crop-raiders). We assess diets only within African savanna elephants to control for phylogeny, and we identify the core microbiome of all individuals within each group. We do not control for habitat in order to maintain reasonable sample sizes for comparison.

All core microbiome proportions are the basis for Table S2 in the final ms.

##### 2b.1. Characterize the core microbiome of all <i>L. africana</i> crop-raiders; run the Rmd chunk below.

##### Characterize core microbiome of <i>L. africana</i> crop-raiders:
```{r}
# Load order dataset
load("data/order_data.Rda")

# Filter only savanna crop-raiding elephants
lacr <- order_data %>%
  filter(phylogeny == "africana" & diet == "Raider") 

# Get the total number of individual savanna crop-raiding elephants
n_ind <- length(unique(lacr$ind_id))

# Filter out only savanna crop-raiding elephants, select individual id, order, and abundance, and summarize abundance by order within individuals
lacr <- order_data %>%
  filter(phylogeny == "africana" & diet == "Raider") %>%
  select(ind_id, order, abundance) %>%
  group_by(order, ind_id) %>%
  summarize(abundance = sum(abundance)) %>%
  filter(abundance != "0") %>%
  count() %>%
  summarize(order_proportion = n/n_ind)

# Filter out only the core microbiome (orders with 100% presence across savanna crop-raiding elephant samples)
lacr_core <- lacr %>%
  filter(order_proportion == 1.00000000)
```

##### 2b.2. Characterize the core microbiome of all <i>L. africana</i> non-crop-raiders; run the Rmd chunk below.

##### Characterize core microbiome of <i>L. africana</i> non-crop-raiders:
```{r}
# Load order dataset
load("data/order_data.Rda")

# Filter only savanna non-crop-raider elephants
lancr <- order_data %>%
  filter(phylogeny == "africana" & diet == "Nonraider")

# Get the total number of individual savanna non-crop-raiding elephants
n_ind <- length(unique(lancr$ind_id))

# Filter out only savanna non-crop-raiding elephants, select individual id, order, and abundance, and summarize abundance by order within individuals
lancr <- order_data %>%
  filter(phylogeny == "africana" & diet == "Nonraider") %>%
  select(ind_id, order, abundance) %>%
  group_by(order, ind_id) %>%
  summarize(abundance = sum(abundance)) %>%
  filter(abundance != "0") %>%
  count() %>%
  summarize(order_proportion = n/n_ind)

# Filter out only the core microbiome (orders with 100% presence across savanna non-crop-raiding elephant samples)
lancr_core <- lancr %>%
  filter(order_proportion == 1.00000000)
```

#### 2c. Characterize the core microbiome by habitat (savanna or forest).
In this step, we characterize the core microbiome of each habitat group in the data (savanna, forest). We assess habitats only within African savanna elephants to control for phylogeny, and we identify the core microbiome of all individuals within each group. We do not control for diet in order to maintain reasonable sample sizes for comparison.

All core microbiome proportions are the basis for Table S2 in the final ms

##### 2c.1. Characterize the core microbiome of all <i>L. africana</i> savanna dwellers; run the Rmd chunk below.

##### Characterize core microbiome of <i>L. africana</i> savanna dwellers:
```{r}
# Load order dataset
load("data/order_data.Rda")

# Filter only savanna savanna-dwelling elephants
las <- order_data %>%
  filter(phylogeny == "africana" & habitat == "Savanna") 

# Get the total number of individual savanna savanna-dwelling elephants
n_ind <- length(unique(las$ind_id))

# Filter out only savanna savanna-dwelling  elephants, select individual id, order, and abundance, and summarize abundance by order within individuals
las <- order_data %>%
  filter(phylogeny == "africana" & habitat == "Savanna") %>%
  select(ind_id, order, abundance) %>%
  group_by(order, ind_id) %>%
  summarize(abundance = sum(abundance)) %>%
  filter(abundance != "0") %>%
  count() %>%
  summarize(order_proportion = n/n_ind)

# Filter out only the core microbiome (orders with 100% presence across savanna savanna-dwelling elephant samples)
las_core <- las %>%
  filter(order_proportion == 1.00000000)
```

##### 2b.2. Characterize the core microbiome of all <i>L. africana</i> forest dwellers; run the Rmd chunk below.

##### Characterize core microbiome of <i>L. africana</i> forest dwellers:
```{r}
# Load order dataset
load("data/order_data.Rda")

# Filter only savanna forest-dwelling elephants
laf <- order_data %>%
  filter(phylogeny == "africana" & habitat == "Forest")

# Get the total number of individual savanna forest-dwelling elephants
n_ind <- length(unique(laf$ind_id))

# Filter out only savanna forest-dwelling elephants, select individual id, order, and abundance, and summarize abundance by order within individuals
laf <- order_data %>%
  filter(phylogeny == "africana" & habitat == "Forest") %>%
  select(ind_id, order, abundance) %>%
  group_by(order, ind_id) %>%
  summarize(abundance = sum(abundance)) %>%
  filter(abundance != "0") %>%
  count() %>%
  summarize(order_proportion = n/n_ind)

# Filter out only the core microbiome (orders with 100% presence across savanna forest-dwelling elephant samples)
laf_core <- laf %>%
  filter(order_proportion == 1.00000000)
```

### ----------------------- END OF PHASE 2: SUMMARY OF OTU ABUNDANCE AND CORE MICROBIOME ----------------------- ###

### ----------------------- END OF ANALYSIS 3: GLOBAL AND CORE MICROBIOME ANALYSIS  ----------------------- ###