---
title: "Analysis 2: Taxonomic classification summary analysis"
author: "Joe Gunn"
date: "2024-02-06"
output: html_document
---

# Project: Assessing effects of phylogeny, diet, and habitat on gut microbial composition in African elephants 
We characterized the gut microbiome of African Savanna elephants (<i>Loxodonta africana</i>) and African Forest elephants (<i>Loxodonta cyclotis</i>). We assessed the relationship of gut microbial composition, including analyses of alpha and beta diversity, with host phylogeny (i.e., species) and habitat type (i.e., forest or savanna) for both species. We also assessed the relationship between microbial composition and diet (i.e., crop-raiding vs. non-crop-raiding) within <i>L. africana</i>. 

## Specific Aim: Data filtering, cleaning, and summarization
In this analysis, we read in, filter, and clean the raw metadata (i.e., information about species, diet, and habitat) and taxonomic classification for microbial operational taxonomic units (OTUs) for our African elephant samples. After we prepare data, we summarize the taxonomic classification data, specifically to determine the proportion of bacterial OTUs classified to each taxonomic group and the confidence of classification. We then give a wholistic and broken-down summary of our elephant fecal samples, including sample numbers within species, diet treatments, and habitats.

## Phases of analysis:
### Phase 1: Data read-in and preparation
### Phase 2: Summary of taxonomic classification data
### Phase 3: Summary of metadata

## Libraries needed for analysis
```{r}
library(readxl)
library(cowplot)
library(tidyverse)
library(qiime2R)
library(scales)
```

## PHASE 1: DATA READ-IN AND PREPARATION
In this phase of the analysis, we read in and prepare full metadata set for African elephant fecal samples as well as the full taxonomic classification dataset (based on the identification of microbial operational taxonomic units [OTUs] present in the samples from metagenomic analysis). We give a full description of columns in each dataset and prepare data objects for downstream analyses.

### STEP 1: Read in and clean metadata for elephant samples and taxonomic classification data for microbial OTUs.
In this step, we read in previously curated metadata for African savana elephants (<i>Loxidonta africana</i>) and African forest elephants (<i>L. cyclotis</i>) collected from across the species' native ranges (`../raw_data/metadata.xlsx`).

#### 1a. Read in and clean sample metadata.
In this step, we read in the full metadata for all elephants included in the study. Data include: 

   1. "sample_id": a unique, alphanumeric ID for each elephant sample
   2. "barcode": the sequence of nucleotides ligated to elephant metagenomic samples for downstream matching of OTUs to individual
   3. "linker": primer linker sequence
   4. "species": species of African elephant (L. africana or L. cyclotis)
   5. "raider": yes/no identifier indicating whether an individual was a known crop a raider (yes) or not a known crop raider (no)
   6. "habitat": habitat type of the individual (forest or savanna)
   7. "age": general description of the age cohort of the individual (adult, subadult, juvenile)
   8. "sex": sex of the individual (male or female)
   9. "ind_id": a unique alphanumerical identifier for an individual elephant
   10. "group": a combined identifier including species, diet, and habitat
   11. "description": a combined identifier including species, diet, habitat, age, and sex

##### 1a.1. Read in full metadata and convert characters to factors; run the Rmd chunk below.

##### Read in and clean full metadata:
```{r}
# Read in raw metadata file
metadata <- read_tsv("../raw_data/metadata.tsv")

# Convert characters to factors
metadata <- metadata %>%
  mutate(barcode = factor(barcode), 
         linker = factor(linker), 
         species = factor(species), 
         raider = factor(raider), 
         habitat = factor(habitat), 
         age = factor(age), 
         sex = factor(sex), 
         ind_id = factor(ind_id), 
         group = factor(group), 
         description = factor(description))

# Save metadata for downstream analyses
save(metadata, file = "data/metadata.Rda")
```

#### 1b. Read in and clean taxonomic classifcation data for metagenomic sequences of microbial OTUs.
In this step, we read in the full taxonomic classification data (`../raw_data/taxa.qza`) for metagenomic sequences of microbial OTUs, which is generated in QIIME (see QIIME pipeline code in Analysis 1). Data include: 

   1. "Feature.ID": a unique, alphanumeric ID for each OTU detected across the full dataset
   2. "Taxon": the taxonomic classificaiton for each OTU detected across the full dataset
   3. "Confidence": the probability (confidence) with which an OTU was accurately identified to the taxonomic classification indicated in the "Taxon" column
   
We also generate a "phyloseq" object, a formatted data object compatible with the phyloseq software library, which contains the taxonomic classificaiton, otu abundance, and phylogenetic tree relationship data, along with associated metadata. We use the phyloseq object for downstream analyses. We merge these two datasets to get a clean version of the taxonomic classification and confidence data in this analysis.

##### 1a.1. Read in full dataset, convert to datatable, restructure, and convert characters to facotrs; run the Rmd chunk below.

##### Read in and clean full taxonomy data:
```{r}
# Read in taxonomic classificaiton data alone to get confidence data
taxa <- read_qza("../raw_data/taxa.qza")

# Read in, combine, and convert QIIME output and metadata to phyloseq format
phyloseq <- qza_to_phyloseq(features = "../raw_data/otus.qza", 
                            taxonomy = "../raw_data/taxa.qza",  
                            tree = "../raw_data/tree.qza", 
                            metadata = "../raw_data/metadata.tsv")

# Convert the "data" module of the taxonomic classification dataset into a dataframe
otu_ids <- taxa$data %>% 
  as.data.frame() %>%
  mutate(otu_id = factor(`Feature.ID`),
         confidence = as.numeric(Confidence)) %>%
  dplyr::select(otu_id, confidence)

# Extract taxonomic classifcaitons from the phyloseq object
taxa <- tax_table(phyloseq) %>%
  as.data.frame() %>%
  rownames_to_column("otu_id") %>%
  mutate(otu_id = factor(otu_id), 
         kingdom = factor(Kingdom), 
         phylum = factor(Phylum), 
         class = factor(Class), 
         order = factor(Order), 
         family = factor(Family), 
         genus = factor(Genus), 
         species = factor(Species)) %>%
  select(otu_id, kingdom, phylum, class, order, family, genus, species)

# Merge taxonomic classifications with confidence data
taxa <- merge(otu_ids, 
              taxa, 
              by = "otu_id")

# Save taxonomic data for downstream analyses
save(taxa, file = "data/taxa.Rda")

# Save raw phyloseq object for downstream analyses
save(phyloseq, file = "data/phyloseq.Rda")
```

### ----------------------- END OF PHASE 1: DATA READ-IN AND PREPARATION ----------------------- ###

## PHASE 3: SUMMARY OF METADATA
In this phase of analysis, we summarize the full metadata set, giving numbers of samples per group of interest in downstream analyses.

### STEP 1: Summarize the full metadata.
In this step, we are summarizing confidence of classification to each taxonomic level for the full taxonomy dataset.

#### 1a. Read in and summarize metadata data; run the Rmd chunk below:

##### Summarize full metadata:
```{r}
# Load taxanomic data
load("data/metadata.Rda")

# Calculate number of elephant samples per species
metadata %>% 
  group_by(species) %>%
  count()

# Calculate number of elephant samples per diet (within species)
metadata %>% 
  group_by(species, raider) %>%
  count()

# Calculate number of elephant samples per diet (within species)
metadata %>% 
  group_by(species, habitat) %>%
  count()

# Calculate number of elephant samples per age cohort (within species)
metadata %>% 
  group_by(species, age) %>%
  count()

# Calculate number of elephant samples per age cohort (within species)
metadata %>% 
  group_by(species, sex) %>%
  count()

# Calculate number of elephant samples per age cohort (within species)
metadata %>% 
  group_by(species, habitat, raider) %>%
  count()
```

### Metadata summary:

## Total
<b><i>N</i><sub>total</sub></b> = 48 <br>

## By species (native range)
<b><i>N</i><sub>afr</sub></b> = 35 <br>
<b><i>N</i><sub>cyc</sub></b> = 15 <br>

## By diet (within species)
<b><i>N</i><sub>afr_raider</sub></b> = 16 <br>
<b><i>N</i><sub>afr_nonraider</sub></b> = 19 <br>

<b><i>N</i><sub>cyc_raider</sub></b> = 0 <br>
<b><i>N</i><sub>cyc_nonraider</sub></b> = 13 <br>

## By habitat (within species)
<b><i>N</i><sub>afr_savanna</sub></b> = 15 <br>
<b><i>N</i><sub>afr_forest</sub></b> = 20 <br>

<b><i>N</i><sub>cyc_savanna</sub></b> = 0 <br>
<b><i>N</i><sub>cyc_forest</sub></b> = 13 <br>

## By age cohort (within species)
<b><i>N</i><sub>afr_adult</sub></b> = 18 <br>
<b><i>N</i><sub>afr_subadult</sub></b> = 13 <br>
<b><i>N</i><sub>afr_juvenile</sub></b> = 4 <br>

<b><i>N</i><sub>cyc_adult</sub></b> = 13 <br>
<b><i>N</i><sub>cyc_subadult</sub></b> = 0 <br>
<b><i>N</i><sub>cyc_juvenile</sub></b> = 0 <br>

## By sex (within species)
<b><i>N</i><sub>afr_male</sub></b> = 17 <br>
<b><i>N</i><sub>afr_female</sub></b> = 18 <br>

<b><i>N</i><sub>cyc_male</sub></b> = 6 <br>
<b><i>N</i><sub>cyc_female</sub></b> = 7 <br>

## By species, habitat, and diet
<b><i>N</i><sub>afr_savanna_raider</sub></b> = 3 <br>
<b><i>N</i><sub>afr_savanna_nonraider</sub></b> = 12 <br>
<b><i>N</i><sub>afr_forest_raider</sub></b> = 13 <br>
<b><i>N</i><sub>afr_forest_nonraider</sub></b> = 7 <br>

<b><i>N</i><sub>cyc_savanna_raider</sub></b> = 0 <br>
<b><i>N</i><sub>cyc_savanna_nonraider</sub></b> = 0 <br>
<b><i>N</i><sub>cyc_forest_raider</sub></b> = 0 <br>
<b><i>N</i><sub>cyc_forest_nonraider</sub></b> = 13 <br>

This information is listed in the Results section of the final ms.

### ----------------------- END OF PHASE 2: SUMMARY OF METADATA ----------------------- ###

## PHASE 3: SUMMARY OF TAXONOMIC CLASSIFICATION DATA
In this phase of the analysis, we summarize the full taxonomic classification dataset, giving number of samples per group (taxon) of interest and calculating confidence of taxonomic classificaiton across OTUs.

### STEP 1: Summarize the full taxonomic classification dataset.
In this step, we are summarizing confidence of classification to each taxonomic level for the full taxonomy dataset.

#### 1a. Read in and summarize the full data; run the Rmd chunk below:

##### Summarize full taxonomic classification and confidence data:
```{r}
# Load taxanomic data
load("data/taxa.Rda")

# Calculate number of unique groups within each taxonomic level 
taxa %>%
  summarize(kingdoms = length(unique(kingdom)),
            phyla = length(unique(phylum)),
            classes = length(unique(class)),
            orders = length(unique(order)),
            families = length(unique(family)),
            genera = length(unique(genus)),
            species = length(unique(species)))

# Calculate overall mean and standard deviation of confidence for the full dataset
taxa %>% 
  summarize(mean_confidence = mean(confidence),
            sd_confidence = sd(confidence))

# Calculate mean (standard deviation) of confidence for each taxonomic level in addition to the number (and proportion) of samples successfully classified in each taxonomic group
confidence <- taxa %>%
  gather(kingdom:species, key = "taxonomic_level", value = "taxon") %>%
  mutate(taxonomic_level = factor(taxonomic_level)) %>%
  drop_na() %>%
  group_by(taxonomic_level) %>%
  summarize(mean_confidence = mean(confidence),
            sd_confidence = sd(confidence),
            n = as.numeric(n()),
            prop = n/nrow(taxa)) %>%
  mutate(taxonomic_level = fct_reorder(taxonomic_level, desc(prop)))

# Save confidence data for plotting
save(confidence, file = "data/confidence.Rda")
```

### Data summary:

## OTU Summary
<b><i>N</i><sub>OTUs</sub></b> = 9,066
<b><i>N</i><sub>kingdoms</sub></b> = 2 <br>
<b><i>N</i><sub>phyla</sub></b> = 21 <br>
<b><i>N</i><sub>classes</sub></b> = 40 <br>
<b><i>N</i><sub>orders</sub></b> = 61 <br>
<b><i>N</i><sub>families</sub></b> = 84 <br>
<b><i>N</i><sub>genera</sub></b> = 130 <br>
<b><i>N</i><sub>species</sub></b> = 66 <br>

## Mean (standard deviation) of confidence for full dataset
<b><i>mean</i><sub>all_taxa</sub> = 0.9615672 (0.07256966) </b><br>

## Mean (standard deviation) of confidence per taxonomic level
<b><i>mean</i><sub>kingdom</sub></b> = 0.9615672 (0.07256966) <br>
<b><i>mean</i><sub>phylum</sub></b> = 0.9619102 (0.07212261) <br>
<b><i>mean</i><sub>class</sub></b> = 0.9623321 (0.07155031) <br>
<b><i>mean</i><sub>order</sub></b> = 0.9631189 (0.07083155) <br>
<b><i>mean</i><sub>family</sub></b> = 0.9605259 (0.07348208) <br>
<b><i>mean</i><sub>genus</sub></b> = 0.9577107 (0.07726817) <br>
<b><i>mean</i><sub>species</sub></b> = 0.9107332 (0.09761776) <br>

## Number of samples (proportion) successfully classified per taxonomic level
<b><i>N</i><sub>kingdom</sub></b> = 9066 (1.0000000) <br>
<b><i>N</i><sub>phylum</sub></b> = 8878 (0.9792632) <br>
<b><i>N</i><sub>class</sub></b> = 8840 (0.9750717) <br>
<b><i>N</i><sub>order</sub></b> = 8706 (0.9602912) <br>
<b><i>N</i><sub>family</sub></b> = 6138 (0.6770351) <br>
<b><i>N</i><sub>genus</sub></b> = 2279 (0.2513788) <br>
<b><i>N</i><sub>species</sub></b> = 271 (0.0298919) <br>

These sample sizes were used as the basis for Table S1 in the final manuscript.

### STEP 2: Visualize confidence and proportion of taxonomic classification.

#### 2a. Generate plot for confidence and proportion of taxonomic classification, with these factors on dual y-axes; run the Rmd chunk below:

##### Plot confidence and proportion of taxonomic classification on dual y-axes: `figures/confidence.pdf`
```{r}
# Load confidence data
load("data/confidence.Rda")

# Plot confidence and proportion of samples per taxonomic level
pdf("figures/confidence.pdf", width = 5, height = 4) 

ggplot(confidence, aes(x = taxonomic_level, y = mean_confidence)) + 
  geom_point(size = 1) +
  geom_errorbar(aes(ymin = mean_confidence - sd_confidence, ymax = mean_confidence + sd_confidence), linewidth = .3) + 
  scale_y_continuous(name = "Mean confidence of classification (%)", sec.axis = sec_axis(~. /0.97, name = "Proportion of OTUs classfied (%)"), labels = number_format(accuracy = 0.1)) + 
  geom_line(mapping = aes(x = taxonomic_level, y = prop, group = 1), size = 2, color = "blue", alpha = 0.6) + 
  theme_set(theme_cowplot(12)) +
  labs(x = "Taxonomic level") +
  theme(axis.title = element_text(size = 15)) + 
  theme(axis.text = element_text(size = 15)) + 
  theme(panel.border = element_rect(colour = "black", fill = NA, size = 1))

dev.off()
```

This figure is the basis for Figure S1 in the final ms.

Given that Order was the lowest (finest scale) taxonomic level at which mean proportion of successful classification (0.9603) and mean confidence of classification (0.9613) were comparable and relatively high, we chose to summarize and assess differences in microbiome composition at the order level. We summarized OTU data at higher levels where appropriate.

### ----------------------- END OF PHASE 3: SUMMARY OF TAXONOMIC CLASSIFICATION DATA ----------------------- ###

### ----------------------- END OF ANALYSIS 2: DATA FILTERING AND TAXONOMIC SUMMARY ANALYSIS ----------------------- ###